---
title: "Pacific herring population genetic statistics"
author: "Joe McGirr"
date: '`r Sys.Date()`'
output:
  html_document:
    code_folding: hide
    collapsed: no
    df_print: paged
    number_sections: yes
    theme: paper
    toc: yes
    toc_depth: 5
    toc_float: yes
  html_notebook:
    toc: yes
    toc_depth: 5
editor_options: 
  chunk_output_type: console
---


```{css, echo=FALSE}
pre, code {white-space:pre !important; overflow-x:auto}
```

# Summary

This notebook generates scripts to calculate genome-wide summary statistics from a `.vcf` with ANGSD and visualize results. The markdown file can be found [here](https://github.com/joemcgirr/pac_herring/blob/master/Rmarkdown/population/Pi_TajD_Fst_Dxy.Rmd).

This notebook automates the tedious process of writing bash scripts for each job run on the computing cluster. Running this notebook will generate bash scripts using python and the R package reticulate. Bash scripts were made to run on the Farm cluster which uses SLURM scheduling. Each bash script has a SLURM header.

The vcf was generated with [these commands](https://github.com/joemcgirr/pac_herring/blob/master/Rmarkdown/fastq_to_vcf/Pacific_Herring_fastq_to_vcf.Rmd)

The data set is a filtered `.vcf` with 892 Pacific herring low coverage genomes:

minimum depth = 600  
maximum depth = 2000  
minimum base quality = 20  
minimum mapping quality = 30  
minor allele frequency > 5%  
genotyping rate > 50% for each population (grouped by samping location and year)

`ph_filtered_snps_minDP600_maxDP2000_minQ20_minMQ30_NS0.5_maf0.05.vcf`

Written by Joe McGirr, postdoc in Andrew Whitehead's lab.

# Load R libraries
```{r setup, include=TRUE, message=FALSE, warning=FALSE,}

# if (!requireNamespace("BiocManager", quietly = TRUE))
#     install.packages("BiocManager")
# BiocManager::install(version = "3.12")

knitr::opts_chunk$set(echo = TRUE)
library(ggplot2)
library(tidyverse)
library(reticulate)
library(ggpubr)
library(sf)
library(pophelper)
library(windowscanr)
library(data.table)

# color-blind friendly 
# Wong, B. Points of view: Color blindness. Nat Methods (2011).
bla <- "#000000"
blu <- "#0072b2" 
grb <- "#56b4e9"
lir <- "#cc79a7"
gre <- "#009e73"
red <- "#d55e00"
org <- "#e69f00"
yel <- "#f0e442"
gry<-  '#BBBBBB'

```

# Load python libraries
```{python, results='hide', eval = FALSE,class.source = 'fold-show'}
import io
import pandas as pd
import numpy as np

def sbatch_header(job,mem,tasks,hours):
    #sbatch submission script header
    script = 'script_' + job + '.sh'
    outfile = io.open(script,'w', newline='\n')    
    outfile.write('#!/bin/bash\n\n#SBATCH --job-name='+job+'\n')
    outfile.write('#SBATCH --mem='+mem+'G \n')
    outfile.write('#SBATCH --ntasks='+tasks+' \n')
    outfile.write('#SBATCH -e '+job+'_%A_%a.err \n')
    outfile.write('#SBATCH --time='+hours+':00:00  \n')
    outfile.write('#SBATCH --mail-user=jamcgirr@ucdavis.edu ##email you when job starts,ends,etc\n#SBATCH --mail-type=ALL\n')
    outfile.write('#SBATCH -p high \n\n')
    outfile.close()
    
def sbatch_header_loop(job,mem,tasks,hours,infile):
    #sbatch submission script header
    script = 'script_' + infile + job + '.sh'
    outfile = io.open(script,'w', newline='\n') 
    jobname= infile + job   
    outfile.write('#!/bin/bash\n\n#SBATCH --job-name='+jobname+'\n')
    outfile.write('#SBATCH --mem='+mem+'G \n')
    outfile.write('#SBATCH --ntasks='+tasks+' \n')
    outfile.write('#SBATCH -e '+jobname+'_%A_%a.err \n')
    outfile.write('#SBATCH --time='+hours+':00:00 \n')
    outfile.write('#SBATCH --mail-user=jamcgirr@ucdavis.edu ##email you when job starts,ends,etc\n#SBATCH --mail-type=ALL\n')
    outfile.write('#SBATCH -p high \n\n')
    outfile.close()

```

# Fst and Population Branch Statistic 

Generate site frequency spectrum using vcf sites.

## Subset master vcf into population vcfs
example script
```{bash,eval=FALSE,class.source = 'fold-show'}
#!/bin/bash

#SBATCH --job-name=BC17_subset_pops_vcfs
#SBATCH --mem=8G 
#SBATCH --ntasks=4 
#SBATCH -e BC17_subset_pops_vcfs_%A_%a.err 
#SBATCH --time=24:00:00 
#SBATCH --mail-user=jamcgirr@ucdavis.edu ##email you when job starts,ends,etc
#SBATCH --mail-type=ALL
#SBATCH -p high 

module load bcftools 
bcftools view -S /home/jamcgirr/ph/scripts/angsd/SFS/SFS_by_pop/BC17_plates_1_through_5_rm.txt --threads 4 /home/jamcgirr/ph/data/vcfs/ph_filtered_snps_minDP600_maxDP2000_minQ20_minMQ30_NS0.5_maf0.05.vcf.gz | bcftools +fill-tags -Oz -- -t all,'DP=sum(DP)' > /home/jamcgirr/ph/data/vcfs/split_pops/maf05/population_BC17_ph_filtered_snps_minDP600_maxDP2000_minQ20_minMQ30_NS0.5_maf0.05.vcf.gz 


#command to run: sbatch script_BC17_subset_pops_vcfs.sh

```

### Generate subset scripts

```{python,results='hide', eval = FALSE}

vcf_name = 'ph_filtered_snps_minDP600_maxDP2000_minQ20_minMQ30_NS0.5_maf0.05'
job_name = '_subset_pops_vcfs'

infiles = ["BC17","CA17","PWS07","PWS17","PWS91","PWS96","SS06","SS17","SS96","TB06","TB17","TB91","TB96","WA17"]

for infile in infiles:
    script = 'script_' + infile + job_name + '.sh'
    sbatch_header_loop(job_name,'8','4','24', infile)
    o = io.open(script,'a+', newline='\n')
    

    o.write('module load bcftools \n')
    o.write('bcftools view -S /home/jamcgirr/ph/scripts/angsd/SFS/SFS_by_pop/'+infile+'_plates_1_through_5_rm.txt --threads 4 /home/jamcgirr/ph/data/vcfs/'+vcf_name+'.vcf.gz | bcftools +fill-tags -Oz -- -t all,\'DP=sum(DP)\' > /home/jamcgirr/ph/data/vcfs/split_pops/maf05/population_'+infile+'_'+vcf_name+'.vcf.gz \n')
    
    
    o.write('\n\n#command to run: sbatch '+script)
    o.close()
    
#EST run time ~15min

```

## Use realSFS to make folded and unfolded SFS for each population

example script
```{bash,eval=FALSE,class.source = 'fold-show'}
#!/bin/bash

#SBATCH --job-name=BC17_SFS
#SBATCH --mem=8G 
#SBATCH --ntasks=4 
#SBATCH -e BC17_SFS_%A_%a.err 
#SBATCH --time=24:00:00 
#SBATCH --mail-user=jamcgirr@ucdavis.edu ##email you when job starts,ends,etc
#SBATCH --mail-type=ALL
#SBATCH -p high 

/home/jamcgirr/apps/angsd_sep_20/angsd/angsd -doSaf 1 -vcf-pl /home/jamcgirr/ph/data/vcfs/split_pops/maf05/population_BC17_ph_filtered_snps_minDP600_maxDP2000_minQ20_minMQ30_NS0.5_maf0.05.vcf.gz -out /home/jamcgirr/ph/data/angsd/SFS/vcf/maf05/BC17 -anc /home/jamcgirr/ph/data/c_harengus/c.harengus.fa 
/home/jamcgirr/apps/angsd_sep_20/angsd/misc/realSFS /home/jamcgirr/ph/data/angsd/SFS/vcf/maf05/BC17.saf.idx -P 4 -fold 1 > /home/jamcgirr/ph/data/angsd/SFS/vcf/maf05/BC17_folded.sfs 
/home/jamcgirr/apps/angsd_sep_20/angsd/misc/realSFS /home/jamcgirr/ph/data/angsd/SFS/vcf/maf05/BC17.saf.idx -P 4 > /home/jamcgirr/ph/data/angsd/SFS/vcf/maf05/BC17_unfolded.sfs 


#command to run: sbatch script_BC17_SFS.sh
```

### Generate SFS scripts

```{python,results='hide', eval = FALSE}

job_name = '_SFS'
infiles = ["BC17","CA17","PWS07","PWS17","PWS91","PWS96","SS06","SS17","SS96","TB06","TB17","TB91","TB96","WA17"]

for infile in infiles:
    script = 'script_' + infile + job_name + '.sh'
    sbatch_header_loop(job_name,'8','4','24', infile)
    o = io.open(script,'a+', newline='\n')
    
    vcf_name = 'ph_filtered_snps_minDP600_maxDP2000_minQ20_minMQ30_NS0.5_maf0.05'
    
    # make saf from vcf
    o.write('/home/jamcgirr/apps/angsd_sep_20/angsd/angsd -doSaf 1 -vcf-pl /home/jamcgirr/ph/data/vcfs/split_pops/maf05/population_'+infile+'_'+vcf_name+'.vcf.gz -out /home/jamcgirr/ph/data/angsd/SFS/vcf/maf05/'+infile+' -anc /home/jamcgirr/ph/data/c_harengus/c.harengus.fa \n')
    # make folded sfs
    o.write('/home/jamcgirr/apps/angsd_sep_20/angsd/misc/realSFS /home/jamcgirr/ph/data/angsd/SFS/vcf/maf05/'+infile+'.saf.idx -P 4 -fold 1 > /home/jamcgirr/ph/data/angsd/SFS/vcf/maf05/'+infile+'_folded.sfs \n')
    # make unfolded sfs
    o.write('/home/jamcgirr/apps/angsd_sep_20/angsd/misc/realSFS /home/jamcgirr/ph/data/angsd/SFS/vcf/maf05/'+infile+'.saf.idx -P 4 > /home/jamcgirr/ph/data/angsd/SFS/vcf/maf05/'+infile+'_unfolded.sfs \n')
    

    o.write('\n\n#command to run: sbatch '+script)
    o.close()

# EST run time ~15min
```


## Make 2D SFS for population pairs

example script
```{bash,eval=FALSE,class.source = 'fold-show'}
#!/bin/bash

#SBATCH --job-name=WA17_CA17_2d_sfs
#SBATCH --mem=8G 
#SBATCH --ntasks=4 
#SBATCH -e WA17_CA17_2d_sfs_%A_%a.err 
#SBATCH --time=24:00:00 
#SBATCH --mail-user=jamcgirr@ucdavis.edu ##email you when job starts,ends,etc
#SBATCH --mail-type=ALL
#SBATCH -p high 

/home/jamcgirr/apps/angsd/misc/realSFS /home/jamcgirr/ph/data/angsd/SFS/vcf/maf05/WA17.saf.idx /home/jamcgirr/ph/data/angsd/SFS/vcf/maf05/CA17.saf.idx -fold 1 -P 4 > /home/jamcgirr/ph/data/angsd/SFS/fst_pbs/maf05/folded_WA17_CA17.sfs 
/home/jamcgirr/apps/angsd/misc/realSFS /home/jamcgirr/ph/data/angsd/SFS/vcf/maf05/WA17.saf.idx /home/jamcgirr/ph/data/angsd/SFS/vcf/maf05/CA17.saf.idx -P 4 > /home/jamcgirr/ph/data/angsd/SFS/fst_pbs/maf05/unfolded_WA17_CA17.sfs 


#command to run: sbatch script_WA17_CA17_2d_sfs.sh

```

### Generate 2D SFS scripts

```{python,results='hide', eval = FALSE}
job_name = '_2d_sfs'

vcf_name = 'ph_filtered_snps_minDP600_maxDP2000_minQ20_minMQ30_NS0.5_maf0.05'
realSFS = '/home/jamcgirr/apps/angsd/misc/realSFS'
wk_dir = '/home/jamcgirr/ph/data/angsd/SFS/fst_pbs/maf05/'
saf_dir = '/home/jamcgirr/ph/data/angsd/SFS/vcf/maf05/'

infiles = ["BC17_CA17","BC17_WA17","PWS07_PWS17","PWS07_SS06","PWS17_BC17","PWS17_CA17","PWS17_SS17","PWS17_WA17","PWS91_PWS07","PWS91_PWS17","PWS91_PWS96","PWS96_PWS07","PWS96_PWS17","PWS96_SS96","SS06_SS17","SS17_BC17","SS17_CA17","SS17_WA17","SS96_SS06","SS96_SS17","TB06_PWS07","TB06_SS06","TB06_TB17","TB17_BC17","TB17_CA17","TB17_PWS17","TB17_SS17","TB17_WA17","TB91_TB06","TB91_TB17","TB91_TB96","TB96_PWS96","TB96_SS96","TB96_TB06","TB96_TB17","WA17_CA17"]

for infile in infiles:
    script = 'script_' + infile + job_name + '.sh'
    sbatch_header_loop(job_name,'8','4','24', infile)
    o = io.open(script,'a+', newline='\n')
    
    
    pops = ''.join(infile).split("_")
    o.write(realSFS+' '+saf_dir+pops[0]+'.saf.idx '+saf_dir+pops[1]+'.saf.idx -fold 1 -P 4 > '+wk_dir+'folded_'+infile+'.sfs \n')
    o.write(realSFS+' '+saf_dir+pops[0]+'.saf.idx '+saf_dir+pops[1]+'.saf.idx -P 4 > '+wk_dir+'unfolded_'+infile+'.sfs \n')

    
    o.write('\n\n#command to run: sbatch '+script)
    o.close()

```


## Calculate Fst and PBS in 50kb sliding windows

example script
```{bash,eval=FALSE,class.source = 'fold-show'}
#!/bin/bash

#SBATCH --job-name=SS96_SS06_SS17_fst_pbs_folded
#SBATCH --mem=8G 
#SBATCH --ntasks=1 
#SBATCH -e SS96_SS06_SS17_fst_pbs_folded_%A_%a.err 
#SBATCH --time=24:00:00 
#SBATCH --mail-user=jamcgirr@ucdavis.edu ##email you when job starts,ends,etc
#SBATCH --mail-type=ALL
#SBATCH -p high 

/home/jamcgirr/apps/angsd/misc/realSFS fst index /home/jamcgirr/ph/data/angsd/SFS/vcf/maf05/SS96.saf.idx /home/jamcgirr/ph/data/angsd/SFS/vcf/maf05/SS06.saf.idx /home/jamcgirr/ph/data/angsd/SFS/vcf/maf05/SS17.saf.idx -sfs /home/jamcgirr/ph/data/angsd/SFS/fst_pbs/maf05/folded_SS96_SS06.sfs -sfs /home/jamcgirr/ph/data/angsd/SFS/fst_pbs/maf05/folded_SS96_SS17.sfs -sfs /home/jamcgirr/ph/data/angsd/SFS/fst_pbs/maf05/folded_SS06_SS17.sfs -fold 1 -fstout /home/jamcgirr/ph/data/angsd/SFS/fst_pbs/maf05/folded_SS96_SS06_SS17_persite 
/home/jamcgirr/apps/angsd/misc/realSFS fst stats2 /home/jamcgirr/ph/data/angsd/SFS/fst_pbs/maf05/folded_SS96_SS06_SS17_persite.fst.idx -win 50000 -step 10000 > /home/jamcgirr/ph/data/angsd/SFS/fst_pbs/maf05/fst_pbs_50kb_win_10kb_step_folded_SS96_SS06_SS17.txt 


#command to run: sbatch script_SS96_SS06_SS17_fst_pbs_folded.sh
```

### Generate Fst PBS scripts

```{python,results='hide', eval = FALSE}
job_name = '_fst_pbs_folded'

vcf_name = 'ph_filtered_snps_minDP600_maxDP2000_minQ20_minMQ30_NS0.5_maf0.05'
realSFS = '/home/jamcgirr/apps/angsd/misc/realSFS'
wk_dir = '/home/jamcgirr/ph/data/angsd/SFS/fst_pbs/maf05/'
saf_dir = '/home/jamcgirr/ph/data/angsd/SFS/vcf/maf05/'

infiles = ["BC17_WA17_CA17","PWS17_SS17_BC17","PWS17_SS17_CA17","PWS17_SS17_WA17","TB17_PWS17_BC17","TB17_PWS17_CA17","TB17_PWS17_SS17","TB17_PWS17_WA17","TB06_PWS07_SS06","TB96_PWS96_SS96","PWS91_PWS07_PWS17","PWS91_PWS96_PWS07","PWS91_PWS96_PWS17","TB91_TB06_TB17","TB91_TB96_TB06","TB91_TB96_TB17","SS96_SS06_SS17"]

for infile in infiles:
    script = 'script_' + infile + job_name + '.sh'
    sbatch_header_loop(job_name,'8','1','24', infile)
    o = io.open(script,'a+', newline='\n')
    
    
    pops = ''.join(infile).split("_")
    #o.write(realSFS+' fst index '+saf_dir+'population_'+pops[0]+'_'+vcf_name+'.saf.idx '+saf_dir+'population_'+pops[1]+'_'+vcf_name+'.saf.idx -sfs '+wk_dir+'folded_'+pops[0]+'_'+pops[1]+'.sfs -fold 1 -fstout '+wk_dir+'folded_'+pops[0]+'_'+pops[1]+'persite \n')
    #o.write(realSFS+' fst index '+saf_dir+'population_'+pops[0]+'_'+vcf_name+'.saf.idx '+saf_dir+'population_'+pops[2]+'_'+vcf_name+'.saf.idx -sfs '+wk_dir+'folded_'+pops[0]+'_'+pops[2]+'.sfs -fold 1 -fstout '+wk_dir+'folded_'+pops[0]+'_'+pops[2]+'persite \n')
    #o.write(realSFS+' fst index '+saf_dir+'population_'+pops[1]+'_'+vcf_name+'.saf.idx '+saf_dir+'population_'+pops[2]+'_'+vcf_name+'.saf.idx -sfs '+wk_dir+'folded_'+pops[1]+'_'+pops[2]+'.sfs -fold 1 -fstout '+wk_dir+'folded_'+pops[1]+'_'+pops[2]+'persite \n\n')

    o.write(realSFS+' fst index '+saf_dir+pops[0]+'.saf.idx '+saf_dir+pops[1]+'.saf.idx '+saf_dir+pops[2]+'.saf.idx -sfs '+wk_dir+'folded_'+pops[0]+'_'+pops[1]+'.sfs -sfs '+wk_dir+'folded_'+pops[0]+'_'+pops[2]+'.sfs -sfs '+wk_dir+'folded_'+pops[1]+'_'+pops[2]+'.sfs -fold 1 -fstout '+wk_dir+'folded_'+pops[0]+'_'+pops[1]+'_'+pops[2]+'_persite \n')
    o.write(realSFS+' fst stats2 '+wk_dir+'folded_'+pops[0]+'_'+pops[1]+'_'+pops[2]+'_persite.fst.idx -win 50000 -step 10000 > '+wk_dir+'fst_pbs_50kb_win_10kb_step_folded_'+pops[0]+'_'+pops[1]+'_'+pops[2]+'.txt \n')

    
    o.write('\n\n#command to run: sbatch '+script)
    o.close()

```


## Plot Fst and PBS
### By Chromosome

```{r, message=FALSE, warning=FALSE, fig.width= 10, fig.height= 7,class.source = 'fold-show'}

pop_names = c("PWS91","PWS96","PWS07","PWS17","TB91","TB96","TB06","TB17","SS96","SS06","SS17","BC17","WA17","CA17")

pop0 <- "TB17"
pop1 <- "PWS17"
pop2 <- "SS17"

fsts <- list.files(path="C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/fst_pbs/maf05/", full.names=T)
f <- fsts[1]

#for (f in fsts){  

pops <- strsplit(f, "_")
pops <- c(pops[[1]][10], pops[[1]][11],gsub("\\.txt","",pops[[1]][12]))

fst <-read.delim(paste("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/fst_pbs/maf05/fst_pbs_50kb_win_10kb_step_folded_",pops[1],"_",pops[2],"_",pops[3],".txt", sep = ""))
#head(fst)
fst <- fst %>% mutate(chr = as.numeric(gsub("chr", "", chr)))
fst$chr_num <- factor(fst$chr, levels = c(1:26))

# Fst
p <- ggplot(fst, aes(x = midPos, y = Fst01)) + 
     geom_point(size = 1, color = gry,alpha = 0.4, shape = 1)+
     theme_minimal()+
     theme(axis.text.x=element_blank())+
     ylab("Fst\n")+ xlab("")+ 
     ggtitle(paste(pops[1],"vs. ", pops[2]))+
     geom_smooth(method = "loess", se = FALSE, span = 1/10, color = blu)+
     facet_wrap(~chr_num, ncol = 9)
print(p)

p <- ggplot(fst, aes(x = midPos, y = Fst02)) + 
     geom_point(size = 1, color = gry,alpha = 0.4, shape = 1)+
     theme_minimal()+
     theme(axis.text.x=element_blank())+
     ylab("Fst\n")+ xlab("")+ 
     ggtitle(paste(pops[1],"vs. ", pops[3]))+
     geom_smooth(method = "loess", se = FALSE, span = 1/10, color = blu)+
     facet_wrap(~chr_num, ncol = 9)
print(p)

p <- ggplot(fst, aes(x = midPos, y = Fst12)) + 
     geom_point(size = 1, color = gry,alpha = 0.4, shape = 1)+
     theme_minimal()+
     theme(axis.text.x=element_blank())+
     ylab("Fst\n")+ xlab("")+ 
     ggtitle(paste(pops[2],"vs. ", pops[3]))+
     geom_smooth(method = "loess", se = FALSE, span = 1/10, color = blu)+
     facet_wrap(~chr_num, ncol = 9)
print(p)

#PBS
p <- ggplot(fst, aes(x = midPos, y = PBS0)) + 
     geom_point(size = 1, color = gry,alpha = 0.4, shape = 1)+
     theme_minimal()+
     theme(axis.text.x=element_blank())+
     ylab("PBS\n")+ xlab("")+ 
     ggtitle(paste(pops[1]))+
     geom_smooth(method = "loess", se = FALSE, span = 1/10, color = red)+
     facet_wrap(~chr_num, ncol = 9)
print(p)

p <- ggplot(fst, aes(x = midPos, y = PBS1)) + 
     geom_point(size = 1, color = gry,alpha = 0.4, shape = 1)+
     theme_minimal()+
     theme(axis.text.x=element_blank())+
     ylab("PBS\n")+ xlab("")+ 
     ggtitle(paste(pops[2]))+
     geom_smooth(method = "loess", se = FALSE, span = 1/10, color = red)+
     facet_wrap(~chr_num, ncol = 9)
print(p)

p <- ggplot(fst, aes(x = midPos, y = PBS2)) + 
     geom_point(size = 1, color = gry,alpha = 0.4, shape = 1)+
     theme_minimal()+
     theme(axis.text.x=element_blank())+
     ylab("PBS\n")+ xlab("")+ 
     ggtitle(paste(pops[3]))+
     geom_smooth(method = "loess", se = FALSE, span = 1/10, color = red)+
     facet_wrap(~chr_num, ncol = 9)
print(p)

#}


```

### Genome-wide mean Fst 2017 samples

matrix 2017 samples
```{r, message=FALSE, warning=FALSE, fig.width= 10, fig.height= 7,class.source = 'fold-show'}

fsts <- list.files(path="C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/fst_pbs/maf05/", full.names=T)


# 2017 samples
fst <- read.delim("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/fst_pbs/maf05/fst_pbs_50kb_win_10kb_step_folded_TB17_PWS17_SS17.txt")
tbxpw <- round(mean(fst$Fst01),4)
tbxss <- round(mean(fst$Fst02),4)
pwxss <- round(mean(fst$Fst12),4)
fst <- read.delim("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/fst_pbs/maf05/fst_pbs_50kb_win_10kb_step_folded_TB17_PWS17_BC17.txt")
tbxbc <- round(mean(fst$Fst02),4)
pwxbc <- round(mean(fst$Fst12),4)
fst <- read.delim("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/fst_pbs/maf05/fst_pbs_50kb_win_10kb_step_folded_TB17_PWS17_WA17.txt")
tbxwa <- round(mean(fst$Fst02),4)
pwxwa <- round(mean(fst$Fst12),4)
fst <- read.delim("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/fst_pbs/maf05/fst_pbs_50kb_win_10kb_step_folded_TB17_PWS17_CA17.txt")
tbxca <- round(mean(fst$Fst02),4)
pwxca <- round(mean(fst$Fst12),4)

fst <- read.delim("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/fst_pbs/maf05/fst_pbs_50kb_win_10kb_step_folded_PWS17_SS17_BC17.txt")
ssxbc <- round(mean(fst$Fst12),4)
fst <- read.delim("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/fst_pbs/maf05/fst_pbs_50kb_win_10kb_step_folded_PWS17_SS17_WA17.txt")
ssxwa <- round(mean(fst$Fst12),4)
fst <- read.delim("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/fst_pbs/maf05/fst_pbs_50kb_win_10kb_step_folded_PWS17_SS17_CA17.txt")
ssxca <- round(mean(fst$Fst12),4)
fst <- read.delim("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/fst_pbs/maf05/fst_pbs_50kb_win_10kb_step_folded_BC17_WA17_CA17.txt")
bcxwa <- round(mean(fst$Fst01),4)
bcxca <- round(mean(fst$Fst02),4)
waxca <- round(mean(fst$Fst12),4)

fst_vec <- c(0,tbxpw,tbxss,tbxbc,tbxwa,tbxca,
             tbxpw,0,pwxss,pwxbc,pwxwa,pwxca,
             tbxss,pwxss,0,ssxbc,ssxwa,ssxca,
             tbxbc,pwxbc,ssxbc,0,bcxwa,bcxca,
             tbxwa,pwxwa,ssxwa,bcxwa,0,waxca,
             tbxca,pwxca,ssxca,bcxca,waxca,0)

fst_mat = matrix(fst_vec, nrow = 6, ncol = 6)
colnames(fst_mat) <- c("TB17","PWS17","SS17","BC17","WA17","CA17")
rownames(fst_mat) <- c("TB17","PWS17","SS17","BC17","WA17","CA17")
#print(fst_mat)

  # Get upper triangle of the correlation matrix
  get_upper_tri <- function(cormat){
    cormat[lower.tri(cormat)]<- NA
    return(cormat)
  }

    get_lower_tri<-function(cormat){
    cormat[upper.tri(cormat)] <- NA
    return(cormat)
  }
upper_tri <- get_upper_tri(fst_mat)

# Melt the correlation matrix
melted_cormat <- melt(upper_tri, na.rm = TRUE)
# Heatmap
ggplot(data = melted_cormat, aes(Var2, Var1, fill = value))+
 geom_tile(color = "white")+
 scale_fill_gradient(low = blu, high = red, 
    limit = c(min(c(fst_mat)[c(fst_mat) > 0]),max(fst_mat)), space = "Lab", 
   name="Fst") +
  theme_minimal()+ xlab("")+ylab("")+
 theme(axis.text.x = element_text(angle = 0, vjust = 0, 
    size = 12, hjust = 0.5))+
   theme(axis.text.y = element_text(size = 12))+
 coord_fixed()+
 geom_text(aes(Var2, Var1, label = value), color = "black", size = 5)


```

# Dxy
## Create .maf files

example script
```{bash,eval=FALSE,class.source = 'fold-show'}
#!/bin/bash

#SBATCH --job-name=BC17_maf
#SBATCH --mem=8G 
#SBATCH --ntasks=4 
#SBATCH -e BC17_maf_%A_%a.err 
#SBATCH --time=1:00:00 
#SBATCH --mail-user=jamcgirr@ucdavis.edu ##email you when job starts,ends,etc
#SBATCH --mail-type=ALL
#SBATCH -p high 

/home/jamcgirr/apps/angsd_sep_20/angsd/angsd -out /home/jamcgirr/ph/data/angsd/SFS/vcf/maf05/BC17 -fai /home/jamcgirr/ph/data/c_harengus/c.harengus.fa.fai -doGlf 2 -doMaf 3 -doMajorMinor 1 -doPost 1 -doGeno 2 -vcf-pl /home/jamcgirr/ph/data/vcfs/split_pops/maf05/population_BC17_ph_filtered_snps_minDP600_maxDP2000_minQ20_minMQ30_NS0.5_maf0.05.vcf.gz -P 4 
gzip -d /home/jamcgirr/ph/data/angsd/SFS/vcf/maf05/BC17.mafs.gz 


#command to run: sbatch script_BC17_maf.sh
```

### Generate maf scripts

```{python,results='hide', eval = FALSE}

job_name = '_maf'
vcf_name = 'ph_filtered_snps_minDP600_maxDP2000_minQ20_minMQ30_NS0.5_maf0.05'
infiles = ["BC17","CA17","PWS07","PWS17","PWS91","PWS96","SS06","SS17","SS96","TB06","TB17","TB91","TB96","WA17"]

for infile in infiles:
    script = 'script_' + infile + job_name + '.sh'
    sbatch_header_loop(job_name,'8','4','1', infile)
    o = io.open(script,'a+', newline='\n')
    
    
    # make mafs from vcf
    o.write('/home/jamcgirr/apps/angsd_sep_20/angsd/angsd -out /home/jamcgirr/ph/data/angsd/SFS/vcf/maf05/'+infile+' -fai /home/jamcgirr/ph/data/c_harengus/c.harengus.fa.fai -doGlf 2 -doMaf 3 -doMajorMinor 1 -doPost 1 -doGeno 2 -vcf-pl /home/jamcgirr/ph/data/vcfs/split_pops/maf05/population_'+infile+'_'+vcf_name+'.vcf.gz -P 4 \n')
    o.write('gzip -d /home/jamcgirr/ph/data/angsd/SFS/vcf/maf05/'+infile+'.mafs.gz \n')
    
    
    o.write('\n\n#command to run: sbatch '+script)
    o.close()

# EST run time ~ 5 minutes
```

## Calculate Dxy from maf files

```{r, results= "hide", eval=FALSE}

mafs_dir <- "C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/dxy/maf05/"
comps <- c("TB_PWS","TB_SS","TB_BC","TB_WA","TB_CA",
           "PWS_SS","PWS_BC","PWS_WA","PWS_CA",
           "SS_BC","SS_WA","SS_CA",
           "BC_WA","BC_CA",
           "WA_CA")
yr <- "17"
comp <- "TB_PWS"

for (comp in comps){

popA <- paste(strsplit(comp,"_")[[1]][1],yr, sep = "")
popB <- paste(strsplit(comp,"_")[[1]][2],yr, sep = "")

allfreqA <- read.table(paste(mafs_dir,popA,".mafs", sep = ""),sep='\t',row.names=NULL, header=T)
allfreqB <- read.table(paste(mafs_dir,popB,".mafs", sep = ""),sep='\t',row.names=NULL, header=T)

### Manipulating the table and print dxy table
allfreq <- merge(allfreqA, allfreqB, by=c("chromo","position"))
allfreq <- allfreq[order(allfreq$chromo, allfreq$position),]
# -> Actual dxy calculation
allfreq <- transform(allfreq, dxy=(knownEM.x*(1-knownEM.y))+(knownEM.y*(1-knownEM.x)))

#head(allfreq)

rol_win <- winScan(x = allfreq, 
                   groups = "chromo", 
                   position = "position", 
                   values = c("dxy"), 
                   win_size = 50000,
                   win_step = 10000,
                   funs = c("sum"))
rol_win$dxy <- rol_win$dxy_sum/50000

# write.table(rol_win, file=paste("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/dxy/maf05/windows/",popA,"_",popB,"_Dxy_50kb_win_10kb_step.txt",sep = ""),quote=FALSE, row.names=FALSE, sep='\t')
# print('Created Dxy_persite.txt')

### Print global dxy
# print(paste0(popA, popB,' Global dxy is: ',mean(rol_win$dxy)))
  
}

```

## Plot Dxy
### By chromosome

```{r, message=FALSE, warning=FALSE, fig.width= 10, fig.height= 7,class.source = 'fold-show'}

dxys <- list.files(path="C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/dxy/maf05/windows/", full.names=T)
d <- dxys[1]

#for (d in dxys){  

pops <- strsplit(d, "/")
pops <- strsplit(pops[[1]][10], "_")

pops <- c(pops[[1]][1], pops[[1]][2])
dxy <- read.delim(paste("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/dxy/maf05/windows/",pops[1],"_",pops[2],"_Dxy_50kb_win_10kb_step.txt", sep = ""))
#head(fst)
dxy <- dxy %>% mutate(chromo = as.numeric(gsub("chr", "", chromo)))
dxy$chr_num <- factor(dxy$chr, levels = c(1:26))


p <- ggplot(dxy, aes(x = win_mid, y = dxy)) + 
     geom_point(size = 1, color = gry,alpha = 0.4, shape = 1)+
     theme_minimal()+
     theme(axis.text.x=element_blank())+
     ylab("Dxy\n")+ xlab("")+ 
     ggtitle(paste(pops[1],"vs. ", pops[2]))+
     geom_smooth(method = "loess", se = FALSE, span = 1/10, color = org)+
     facet_wrap(~chr_num, ncol = 9)
print(p)

#}

```

### Genome-wide mean Dxy 2017 samples
 

```{r, message=FALSE, warning=FALSE, fig.width= 10, fig.height= 7,class.source = 'fold-show'}

dxys <- list.files(path="C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/dxy/maf05/windows/", full.names=T)
dxy_dir <- "C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/dxy/maf05/windows/"
# 2017 samples
tbxpw <- round(mean(read.delim(paste(dxy_dir,"TB17_PWS17_Dxy_50kb_win_10kb_step.txt", sep = ""))$dxy),7)
tbxss <- round(mean(read.delim(paste(dxy_dir,"TB17_SS17_Dxy_50kb_win_10kb_step.txt", sep = ""))$dxy),7)
pwxss <- round(mean(read.delim(paste(dxy_dir,"PWS17_SS17_Dxy_50kb_win_10kb_step.txt", sep = ""))$dxy),7)
tbxbc <- round(mean(read.delim(paste(dxy_dir,"TB17_BC17_Dxy_50kb_win_10kb_step.txt", sep = ""))$dxy),7)
pwxbc <- round(mean(read.delim(paste(dxy_dir,"PWS17_BC17_Dxy_50kb_win_10kb_step.txt", sep = ""))$dxy),7)
tbxwa <- round(mean(read.delim(paste(dxy_dir,"TB17_WA17_Dxy_50kb_win_10kb_step.txt", sep = ""))$dxy),7)
pwxwa <- round(mean(read.delim(paste(dxy_dir,"PWS17_WA17_Dxy_50kb_win_10kb_step.txt", sep = ""))$dxy),7)
tbxca <- round(mean(read.delim(paste(dxy_dir,"TB17_CA17_Dxy_50kb_win_10kb_step.txt", sep = ""))$dxy),7)
pwxca <- round(mean(read.delim(paste(dxy_dir,"PWS17_CA17_Dxy_50kb_win_10kb_step.txt", sep = ""))$dxy),7)

ssxbc <- round(mean(read.delim(paste(dxy_dir,"SS17_BC17_Dxy_50kb_win_10kb_step.txt", sep = ""))$dxy),7)
ssxwa <- round(mean(read.delim(paste(dxy_dir,"SS17_WA17_Dxy_50kb_win_10kb_step.txt", sep = ""))$dxy),7)
ssxca <- round(mean(read.delim(paste(dxy_dir,"SS17_CA17_Dxy_50kb_win_10kb_step.txt", sep = ""))$dxy),7)
bcxwa <- round(mean(read.delim(paste(dxy_dir,"BC17_WA17_Dxy_50kb_win_10kb_step.txt", sep = ""))$dxy),7)
bcxca <- round(mean(read.delim(paste(dxy_dir,"BC17_CA17_Dxy_50kb_win_10kb_step.txt", sep = ""))$dxy),7)
waxca <- round(mean(read.delim(paste(dxy_dir,"WA17_CA17_Dxy_50kb_win_10kb_step.txt", sep = ""))$dxy),7)

fst_vec <- c(0,tbxpw,tbxss,tbxbc,tbxwa,tbxca,
             tbxpw,0,pwxss,pwxbc,pwxwa,pwxca,
             tbxss,pwxss,0,ssxbc,ssxwa,ssxca,
             tbxbc,pwxbc,ssxbc,0,bcxwa,bcxca,
             tbxwa,pwxwa,ssxwa,bcxwa,0,waxca,
             tbxca,pwxca,ssxca,bcxca,waxca,0)

fst_mat = matrix(fst_vec, nrow = 6, ncol = 6)
colnames(fst_mat) <- c("TB17","PWS17","SS17","BC17","WA17","CA17")
rownames(fst_mat) <- c("TB17","PWS17","SS17","BC17","WA17","CA17")
#print(fst_mat)

  # Get upper triangle of the correlation matrix
  get_upper_tri <- function(cormat){
    cormat[lower.tri(cormat)]<- NA
    return(cormat)
  }

    get_lower_tri<-function(cormat){
    cormat[upper.tri(cormat)] <- NA
    return(cormat)
  }
upper_tri <- get_upper_tri(fst_mat)

# Melt the correlation matrix
melted_cormat <- melt(upper_tri, na.rm = TRUE)
# Heatmap
ggplot(data = melted_cormat, aes(Var2, Var1, fill = value))+
 geom_tile(color = "white")+
 scale_fill_gradient(low = blu, high = red, 
    limit = c(min(c(fst_mat)[c(fst_mat) > 0]),max(fst_mat)), space = "Lab", 
   name="Fst") +
  theme_minimal()+ xlab("")+ylab("")+
 theme(axis.text.x = element_text(angle = 0, vjust = 0, 
    size = 12, hjust = 0.5))+
   theme(axis.text.y = element_text(size = 12))+
 coord_fixed()+
 geom_text(aes(Var2, Var1, label = value), color = "black", size = 5)


```

# Tajimas D

## Estimate thetas using SFS from vcf sites 
example script
```{bash,eval=FALSE,class.source = 'fold-show'}
#!/bin/bash

#SBATCH --job-name=BC17_thetas_vcf
#SBATCH --mem=8G 
#SBATCH --ntasks=4 
#SBATCH -e BC17_thetas_vcf_%A_%a.err 
#SBATCH --time=24:00:00 
#SBATCH --mail-user=jamcgirr@ucdavis.edu ##email you when job starts,ends,etc
#SBATCH --mail-type=ALL
#SBATCH -p high 

/home/jamcgirr/apps/angsd_sep_20/angsd/misc/realSFS saf2theta /home/jamcgirr/ph/data/angsd/SFS/vcf/maf05/BC17.saf.idx -outname /home/jamcgirr/ph/data/angsd/SFS/vcf/maf05/thetas/BC17 -fold 1 -sfs /home/jamcgirr/ph/data/angsd/SFS/vcf/maf05/BC17_folded.sfs 
/home/jamcgirr/apps/angsd_sep_20/angsd/misc/thetaStat do_stat /home/jamcgirr/ph/data/angsd/SFS/vcf/maf05/thetas/BC17.thetas.idx 
/home/jamcgirr/apps/angsd_sep_20/angsd/misc/thetaStat do_stat /home/jamcgirr/ph/data/angsd/SFS/vcf/maf05/thetas/BC17.thetas.idx -win 50000 -step 10000 -outnames /home/jamcgirr/ph/data/angsd/SFS/vcf/maf05/thetas/BC17_50kb_win_10kb_step 


#run: sbatch script_BC17_thetas_vcf.sh
```

### Generate thetas scripts

```{python,results='hide', eval = FALSE}
job_name = '_thetas_vcf'
infiles = ["BC17","CA17","PWS07","PWS17","PWS91","PWS96","SS06","SS17","SS96","TB06","TB17","TB91","TB96","WA17"]

for infile in infiles:
    script = 'script_' + infile + job_name + '.sh'
    sbatch_header_loop(job_name,'8','4','24', infile)
    o = io.open(script,'a+', newline='\n')
    
    o.write('/home/jamcgirr/apps/angsd_sep_20/angsd/misc/realSFS saf2theta /home/jamcgirr/ph/data/angsd/SFS/vcf/maf05/'+infile+'.saf.idx -outname /home/jamcgirr/ph/data/angsd/SFS/vcf/maf05/thetas/'+infile+' -fold 1 -sfs /home/jamcgirr/ph/data/angsd/SFS/vcf/maf05/'+infile+'_folded.sfs \n')
    o.write('/home/jamcgirr/apps/angsd_sep_20/angsd/misc/thetaStat do_stat /home/jamcgirr/ph/data/angsd/SFS/vcf/maf05/thetas/'+infile+'.thetas.idx \n')
    o.write('/home/jamcgirr/apps/angsd_sep_20/angsd/misc/thetaStat do_stat /home/jamcgirr/ph/data/angsd/SFS/vcf/maf05/thetas/'+infile+'.thetas.idx -win 50000 -step 10000 -outnames /home/jamcgirr/ph/data/angsd/SFS/vcf/maf05/thetas/'+infile+'_50kb_win_10kb_step \n')    
    
    o.write('\n\n#run: sbatch '+script)
    o.close()

# EST run time ~ 1min

```


## Plot Tajimas D
### By population

```{r, message=FALSE, warning=FALSE, fig.width= 10, fig.height= 7,class.source = 'fold-show'}

pop_names <- c("BC17","CA17","PWS07","PWS17","PWS91","PWS96","SS06","SS17","SS96","TB06","TB17","TB91","TB96","WA17")
pop_names <- c("CA17","PWS07","PWS17","PWS91","PWS96","SS06","SS17","SS96","TB06","TB17","TB91","TB96","WA17")

pop <- "BC17"
thetas <-read.delim(paste('C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/thetas/vcf/maf05/',pop,'_50kb_win_10kb_step.pestPG',sep = ""))
#thetas <-read.delim(paste('C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/thetas/downsample_41/folded/',pop,'_minQ20_minMQ30_50kb_win_10kb_step.pestPG',sep = ""))
thetas <- thetas[thetas$nSites > 0,]
thetas <- thetas[c("Tajima")]
names(thetas)[names(thetas)=="Tajima"] <- pop

for (pop in pop_names){

thetas1 <-read.delim(paste('C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/thetas/vcf/maf05/',pop,'_50kb_win_10kb_step.pestPG',sep = ""))
#thetas1 <-read.delim(paste('C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/thetas/downsample_41/folded/',pop,'_minQ20_minMQ30_50kb_win_10kb_step.pestPG',sep = ""))
thetas1 <- thetas1[thetas1$nSites > 0,]
thetas1[c("Chr","WinCenter","Tajima")]
thetas$Tajima <- thetas1$Tajima
names(thetas)[names(thetas)=="Tajima"] <- pop 
  
}

my_cols <- c(grb,org,red,red,red,red,yel,yel,yel,blu,blu,blu,blu,lir)
theta_test <- thetas %>% 
      gather(key="MesureType", value="Val")
p1 <- thetas %>% 
      gather(key="MesureType", value="Val") %>%
      ggplot( aes(x=reorder(MesureType,Val, FUN = mean), y=Val, fill=MesureType)) +
      geom_violin() + xlab("\npopulation/year") + ylab("Tajima's D\n")+
      scale_fill_manual(name = "Population",values=my_cols,guide=FALSE)+
      #stat_summary(fun=mean, geom="line")+
      theme_minimal()
      #geom_jitter(height = 0, width = 0.1, alpha = 0.1)

#png("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/figs/population_structure/taj_d.png", height = 5, width = 7, units = 'in', res = 600)

p1

#dev.off()
```

### By chromosome

```{r, message=FALSE, warning=FALSE, fig.width= 10, fig.height= 7,class.source = 'fold-show'}

pop <- "BC17"
thetas <-read.delim(paste('C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/thetas/vcf/maf05/',pop,'_50kb_win_10kb_step.pestPG',sep = ""))
#thetas <-read.delim(paste('C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/thetas/downsample_41/folded/',pop,'_minQ20_minMQ30_50kb_win_10kb_step.pestPG',sep = ""))

thetas <- thetas %>% mutate(chr = as.numeric(gsub("chr", "", Chr)))
thetas$chr_num <- factor(thetas$chr, levels = c(1:26))

p <- ggplot(thetas, aes(x = WinCenter, y = Tajima)) + 
     geom_point(size = 1, color = gry,alpha = 0.6, shape = 1)+
     theme_minimal()+
     theme(axis.text.x=element_blank())+
     ylab("Tajima's D\n")+ xlab("")+ 
     ggtitle(pop)+
     geom_smooth(method = "loess", se = FALSE, span = 1/10, color = lir)+
     facet_wrap(~chr_num, ncol = 9)
print(p)

```


# Nucleotide diversity 
## Use ANGSD to estimate SFS using genotype likelihoods
The way ANGSD estimates pi changed with an update (from doThetas to safToThetas) and the estimates are way too high (https://github.com/ANGSD/angsd/issues/336). This issue has to do with invariant sites. In order to account for invariant sites, I run ANGSD directly on .bam files to create an .saf file for each population. Due to memory constraints, I randomly down sample the number of input .bam files to match the lowest population sample size (n = 41)

use the -nSites 100000000 flag due to memory limits
this breaks up genome into 7 chunks
this gives you seven lines in the output.sfs
need to sum columns to make final sfs for following step

example script
```{bash,eval=FALSE,class.source = 'fold-show'}
#!/bin/bash

#SBATCH --job-name=BC17_angsd_downsample_sfs
#SBATCH --mem=40G 
#SBATCH --ntasks=8 
#SBATCH -e BC17_angsd_downsample_sfs_%A_%a.err 
#SBATCH --time=48:00:00 
#SBATCH --mail-user=jamcgirr@ucdavis.edu ##email you when job starts,ends,etc
#SBATCH --mail-type=ALL
#SBATCH -p high 

shuf /home/jamcgirr/ph/data/angsd/SFS/bamlist_test/BC17_bams_p_1_5_rm.txt | head -41 > /home/jamcgirr/ph/data/angsd/SFS/downsample/downsample_bams_BC17.txt 

/home/jamcgirr/apps/angsd_sep_20/angsd/angsd -bam /home/jamcgirr/ph/data/angsd/SFS/downsample/downsample_bams_BC17.txt -doSaf 1 -doMajorMinor 1 -doMaf 3 -doCounts 1 -doGlf 3 -anc /home/jamcgirr/ph/data/c_harengus/c.harengus.fa -ref /home/jamcgirr/ph/data/c_harengus/c.harengus.fa -minMapQ 30 -minQ 20 -GL 1 -P 8 -uniqueOnly 1 -remove_bads 1 -only_proper_pairs 1 -trim 0 -C 50 -minInd 10 -setMinDepth 10 -setMaxDepth 100 -out /home/jamcgirr/ph/data/angsd/SFS/downsample/BC17_minQ20_minMQ30 

/home/jamcgirr/apps/angsd_sep_20/angsd/misc/realSFS /home/jamcgirr/ph/data/angsd/SFS/downsample/BC17_minQ20_minMQ30.saf.idx -P 8 -fold 1 -nSites 100000000 > /home/jamcgirr/ph/data/angsd/SFS/downsample/BC17_minQ20_minMQ30_folded.sfs 
/home/jamcgirr/apps/angsd_sep_20/angsd/misc/realSFS /home/jamcgirr/ph/data/angsd/SFS/downsample/BC17_minQ20_minMQ30.saf.idx -P 8 -nSites 100000000 > /home/jamcgirr/ph/data/angsd/SFS/downsample/BC17_minQ20_minMQ30_unfolded.sfs 

#run: sbatch script_BC17_angsd_downsample_sfs.sh

```

### Generate sfs scripts

```{python,results='hide', eval = FALSE}

# lets try downsampling populations to the smallest n (SS06 = 41)
# run angsd direct on bam files

job_name = '_angsd_downsample_sfs'
wk_dir = "/home/jamcgirr/ph/data/angsd/SFS/downsample/"
infiles = ["BC17","CA17","PWS07","PWS17","PWS91","PWS96","SS06","SS17","SS96","TB06","TB17","TB91","TB96","WA17"]

for infile in infiles:
    script = 'script_' + infile + job_name + '.sh'
    sbatch_header_loop(job_name,'40','8','48', infile)
    o = io.open(script,'a+', newline='\n')
    
    
    o.write('shuf /home/jamcgirr/ph/data/angsd/SFS/bamlist_test/'+infile+'_bams_p_1_5_rm.txt | head -41 > '+wk_dir+'downsample_bams_'+infile+'.txt \n\n')
    # make saf from bams
    o.write('/home/jamcgirr/apps/angsd_sep_20/angsd/angsd -bam '+wk_dir+'downsample_bams_'+infile+'.txt -doSaf 1 -doMajorMinor 1 -doMaf 3 -anc /home/jamcgirr/ph/data/c_harengus/c.harengus.fa -minMapQ 30 -minQ 20 -GL 1 -P 8 -out '+wk_dir+infile+'_minQ20_minMQ30 \n\n')
    # make saf from bams using filters
    o.write('/home/jamcgirr/apps/angsd_sep_20/angsd/angsd -bam '+wk_dir+'downsample_bams_'+infile+'.txt -doSaf 1 -doMajorMinor 1 -doMaf 3 -doCounts 1 -doGlf 3 -anc /home/jamcgirr/ph/data/c_harengus/c.harengus.fa -ref /home/jamcgirr/ph/data/c_harengus/c.harengus.fa -minMapQ 30 -minQ 20 -GL 1 -P 8 -uniqueOnly 1 -remove_bads 1 -only_proper_pairs 1 -trim 0 -C 50 -minInd 10 -setMinDepth 10 -setMaxDepth 100 -out '+wk_dir+infile+'_minQ20_minMQ30 \n\n')
    # make folded sfs
    o.write('/home/jamcgirr/apps/angsd_sep_20/angsd/misc/realSFS '+wk_dir+infile+'_minQ20_minMQ30.saf.idx -P 8 -fold 1 -nSites 100000000 > '+wk_dir+infile+'_minQ20_minMQ30_folded.sfs \n')
    # make unfolded sfs
    o.write('/home/jamcgirr/apps/angsd_sep_20/angsd/misc/realSFS '+wk_dir+infile+'_minQ20_minMQ30.saf.idx -P 8 -nSites 100000000 > '+wk_dir+infile+'_minQ20_minMQ30_unfolded.sfs \n')

    o.write('\n\n#run: sbatch '+script)
    o.close()

# EST run time ~ 1 day

```


## Estimate thetas

example script
```{bash,eval=FALSE,class.source = 'fold-show'}
#!/bin/bash

#SBATCH --job-name=BC17_angsd_downsample_thetas
#SBATCH --mem=30G 
#SBATCH --ntasks=8 
#SBATCH -e BC17_angsd_downsample_thetas_%A_%a.err 
#SBATCH --time=24:00:00 
#SBATCH --mail-user=jamcgirr@ucdavis.edu ##email you when job starts,ends,etc
#SBATCH --mail-type=ALL
#SBATCH -p high 

/home/jamcgirr/apps/angsd_sep_20/angsd/misc/realSFS saf2theta /home/jamcgirr/ph/data/angsd/SFS/downsample/saf/BC17_minQ20_minMQ30.saf.idx -outname /home/jamcgirr/ph/data/angsd/SFS/downsample/thetas/folded/BC17_minQ20_minMQ30 -fold 1 -sfs /home/jamcgirr/ph/data/angsd/SFS/downsample/sfs/folded/BC17_minQ20_minMQ30_folded.sfs 
/home/jamcgirr/apps/angsd_sep_20/angsd/misc/thetaStat do_stat /home/jamcgirr/ph/data/angsd/SFS/downsample/thetas/folded/BC17_minQ20_minMQ30.thetas.idx 
/home/jamcgirr/apps/angsd_sep_20/angsd/misc/thetaStat do_stat /home/jamcgirr/ph/data/angsd/SFS/downsample/thetas/folded/BC17_minQ20_minMQ30.thetas.idx -win 50000 -step 10000 -outnames /home/jamcgirr/ph/data/angsd/SFS/downsample/thetas/folded/BC17_minQ20_minMQ30_50kb_win_10kb_step 

/home/jamcgirr/apps/angsd_sep_20/angsd/misc/realSFS saf2theta /home/jamcgirr/ph/data/angsd/SFS/downsample/saf/BC17_minQ20_minMQ30.saf.idx -outname /home/jamcgirr/ph/data/angsd/SFS/downsample/thetas/unfolded/BC17_minQ20_minMQ30 -sfs /home/jamcgirr/ph/data/angsd/SFS/downsample/sfs/folded/BC17_minQ20_minMQ30_folded.sfs 
/home/jamcgirr/apps/angsd_sep_20/angsd/misc/thetaStat do_stat /home/jamcgirr/ph/data/angsd/SFS/downsample/thetas/unfolded/BC17_minQ20_minMQ30.thetas.idx 
/home/jamcgirr/apps/angsd_sep_20/angsd/misc/thetaStat do_stat /home/jamcgirr/ph/data/angsd/SFS/downsample/thetas/unfolded/BC17_minQ20_minMQ30.thetas.idx -win 50000 -step 10000 -outnames /home/jamcgirr/ph/data/angsd/SFS/downsample/thetas/unfolded/BC17_minQ20_minMQ30_50kb_win_10kb_step 


#run: sbatch script_BC17_angsd_downsample_thetas.sh

```

### Generate thetas scripts

```{python,results='hide', eval = FALSE}

job_name = '_angsd_downsample_thetas'
saf_dir = "/home/jamcgirr/ph/data/angsd/SFS/downsample/saf/"
fold_dir = "/home/jamcgirr/ph/data/angsd/SFS/downsample/sfs/folded/"
unfold_dir = "/home/jamcgirr/ph/data/angsd/SFS/downsample/sfs/folded/"
thetas_dir = "/home/jamcgirr/ph/data/angsd/SFS/downsample/thetas/"
infiles = ["BC17","CA17","PWS07","PWS17","PWS91","PWS96","SS06","SS17","SS96","TB06","TB17","TB91","TB96","WA17"]

for infile in infiles:
    script = 'script_' + infile + job_name + '.sh'
    sbatch_header_loop(job_name,'30','8','24', infile)
    o = io.open(script,'a+', newline='\n')
    
    # folded
    o.write('/home/jamcgirr/apps/angsd_sep_20/angsd/misc/realSFS saf2theta '+saf_dir+infile+'_minQ20_minMQ30.saf.idx -outname '+thetas_dir+'folded/'+infile+'_minQ20_minMQ30 -fold 1 -sfs '+fold_dir+infile+'_minQ20_minMQ30_folded.sfs \n')
    o.write('/home/jamcgirr/apps/angsd_sep_20/angsd/misc/thetaStat do_stat '+thetas_dir+'folded/'+infile+'_minQ20_minMQ30.thetas.idx \n')
    o.write('/home/jamcgirr/apps/angsd_sep_20/angsd/misc/thetaStat do_stat '+thetas_dir+'folded/'+infile+'_minQ20_minMQ30.thetas.idx -win 50000 -step 10000 -outnames '+thetas_dir+'folded/'+infile+'_minQ20_minMQ30_50kb_win_10kb_step \n\n')
    # unfolded
    o.write('/home/jamcgirr/apps/angsd_sep_20/angsd/misc/realSFS saf2theta '+saf_dir+infile+'_minQ20_minMQ30.saf.idx -outname '+thetas_dir+'unfolded/'+infile+'_minQ20_minMQ30 -sfs '+fold_dir+infile+'_minQ20_minMQ30_folded.sfs \n')
    o.write('/home/jamcgirr/apps/angsd_sep_20/angsd/misc/thetaStat do_stat '+thetas_dir+'unfolded/'+infile+'_minQ20_minMQ30.thetas.idx \n')
    o.write('/home/jamcgirr/apps/angsd_sep_20/angsd/misc/thetaStat do_stat '+thetas_dir+'unfolded/'+infile+'_minQ20_minMQ30.thetas.idx -win 50000 -step 10000 -outnames '+thetas_dir+'unfolded/'+infile+'_minQ20_minMQ30_50kb_win_10kb_step \n')
    
    
    o.write('\n\n#run: sbatch '+script)
    o.close()

# EST run time ~4hr

```


## Plot folded SFS

```{r, message=FALSE, warning=FALSE, fig.width= 10, fig.height= 7,class.source = 'fold-show'}
# using SFS from angsd -dosaf -bam

pop_names <- c("BC17","CA17","PWS07","PWS17","PWS91","PWS96","SS06","SS17","SS96","TB06","TB17","TB91","TB96","WA17")

pop_name <- "BC17"

#for (pop_name in pop_names){
sfs <- read.table(paste("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/sfs/downsample_41/folded/",pop_name,"_minQ20_minMQ30_folded.sfs",sep = ""), header = FALSE, stringsAsFactors = FALSE)
sfs <- as.vector(colSums(sfs))
#sfs
barplot(sfs[-c(1,length(sfs))]) #plot variable sites 

# Use this to sum columns from the 'chunks' sfs calculated with nSites

#sfs <- as.vector(colSums(sfs))
#write.table(t(sfs),paste("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/downsample_41/sfs/folded/",pop_name,"_minQ20_minMQ30_folded.sfs",sep = ""), quote = FALSE, row.names = FALSE, col.names = FALSE)
#}

# https://github.com/ANGSD/angsd/issues/97


```

## Plot unfolded SFS

```{r, message=FALSE, warning=FALSE, fig.width= 10, fig.height= 7,class.source = 'fold-show'}
# using SFS from angsd -dosaf -bam

pop_names <- c("BC17","CA17","PWS07","PWS17","PWS91","PWS96","SS06","SS17","SS96","TB06","TB17","TB91","TB96","WA17")

pop_name <- "BC17"

#for (pop_name in pop_names){
sfs <- read.table(paste("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/sfs/downsample_41/unfolded/",pop_name,"_minQ20_minMQ30_unfolded.sfs",sep = ""), header = FALSE, stringsAsFactors = FALSE)
#sfs
sfs <- as.vector(colSums(sfs))
barplot(sfs[-c(1,length(sfs))]) #plot variable sites 

# Use this to make the final sfs
# sum columns from the 'chunks' sfs calculated with -nSites

#write.table(t(sfs),paste("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/downsample_41/sfs/folded/",pop_name,"_minQ20_minMQ30_folded.sfs",sep = ""), quote = FALSE, row.names = FALSE, col.names = FALSE)
#}


```

## Plot Pi
### By population

```{r, message=FALSE, warning=FALSE, fig.width= 10, fig.height= 7,class.source = 'fold-show'}


#https://github.com/ANGSD/angsd/issues/329#issuecomment-681005380
# pi

pop <- "BC17"
thetas <-read.delim(paste('C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/thetas/downsample_41/folded/',pop,'_minQ20_minMQ30_50kb_win_10kb_step.pestPG',sep = ""))
#thetas$pi<- (thetas$tP*thetas$nSites)/10000
thetas$pi<- thetas$tP/thetas$nSites
#thetas$pi<- thetas$tP/50000
thetas <- thetas[c("pi")]
names(thetas)[names(thetas)=="pi"] <- pop

for (pop in pop_names){

thetas1 <-read.delim(paste('C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/thetas/downsample_41/folded/',pop,'_minQ20_minMQ30_50kb_win_10kb_step.pestPG',sep = ""))
#thetas1$pi<- (thetas1$tP*thetas1$nSites)/10000
thetas1$pi<- thetas1$tP/thetas1$nSites
thetas1[c("Chr","WinCenter","pi")]
thetas$pi <- thetas1$pi
names(thetas)[names(thetas)=="pi"] <- pop 
  
}

my_cols <- c(grb,org,red,red,red,red,yel,yel,yel,blu,blu,blu,blu,lir)
thetas <- na.omit(thetas)
p1 <- thetas %>% 
      gather(key="MesureType", value="Val") %>%
      ggplot( aes(x=reorder(MesureType,Val, FUN = median), y=Val, fill=MesureType)) +
      geom_violin() + xlab("\npopulation/year") + ylab(expression(pi))+
      scale_fill_manual(name = "Population",values=my_cols,guide=FALSE)+
      theme_minimal()
      #geom_jitter(height = 0, width = 0.1, alpha = 0.1)

#png("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/figs/pi_angsd.png", height = 5, width = 7, units = 'in', res = 600)

p1

#dev.off()

```

### By chromosome

```{r, message=FALSE, warning=FALSE, fig.width= 10, fig.height= 7,class.source = 'fold-show'}

pop <- "BC17"
thetas <-read.delim(paste('C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/thetas/downsample_41/folded/',pop,'_minQ20_minMQ30_50kb_win_10kb_step.pestPG',sep = ""))
thetas$pi<- thetas$tP/thetas$nSites

thetas <- thetas %>% mutate(chr = as.numeric(gsub("chr", "", Chr)))
thetas$chr_num <- factor(thetas$chr, levels = c(1:26))

p <- ggplot(thetas, aes(x = WinCenter, y = pi)) + 
     geom_point(size = 1, color = gry,alpha = 0.6, shape = 1)+
     theme_minimal()+
     theme(axis.text.x=element_blank())+
     ylab(expression(pi))+ xlab("")+ 
     ggtitle(pop)+
     geom_smooth(method = "loess", se = FALSE, span = 1/10, color = grb)+
     facet_wrap(~chr_num, ncol = 9)
print(p)

```

```{r, include=FALSE, eval=FALSE}
### vcftools pi just to get a sense with hard calls since angsd values are stupid high

#pi_50kb_win_BC17_ph_filtered_snps_minDP600_maxDP2000_maf0.05_minQ20_minMQ30_maxmiss0.5_plates_1_5_rm.txt.windowed


#pop <- "BC17"
#thetas <-read.delim(paste('C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/thetas/vcftools/pi_50kb_win_',pop,'_ph_filtered_snps_minDP600_maxDP2000_maf0.05_minQ20_minMQ30_maxmiss0.5_plates_1_5_rm.txt.windowed.pi',sep = ""))
##thetas$pi<- (thetas$tP*thetas$nSites)/10000
##thetas$pi<- thetas$tP/thetas$nSites
##thetas$pi<- thetas$tP/50000
#thetas <- head(thetas, 14500)
#thetas <- thetas[c("PI")]
#names(thetas)[names(thetas)=="PI"] <- pop
#
#for (pop in pop_names){

#thetas1 <-read.delim(paste('C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/thetas/vcftools/pi_50kb_win_',pop,'_ph_filtered_snps_minDP600_maxDP2000_maf0.05_minQ20_minMQ30_maxmiss0.5_plates_1_5_rm.txt.windowed.pi',sep = ""))
##thetas1$pi<- (thetas1$tP*thetas1$nSites)/10000
##thetas1$pi<- thetas1$tP/thetas1$nSites
##thetas1[c("Chr","WinCenter","pi")]
#thetas1 <- head(thetas1, nrow(thetas))
#thetas$pi <- thetas1$PI
#names(thetas)[names(thetas)=="pi"] <- pop 
  
#}

#my_cols <- c(grb,org,red,red,red,red,yel,yel,yel,blu,blu,blu,blu,lir)
#p1 <- thetas %>% 
#      gather(key="MesureType", value="Val") %>%
#      ggplot( aes(x=reorder(MesureType,Val, FUN = median), y=Val, fill=MesureType)) +
#      geom_violin() + xlab("\npopulation/year") + ylab("pi\n")+
#      scale_fill_manual(name = "Population",values=my_cols,guide=FALSE)+
#      theme_minimal()
#      #geom_jitter(height = 0, width = 0.1, alpha = 0.1)

#png("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/figs/population_structure/pi_vcftools.png", height = 5, width = 7, units = 'in', res = 600)

#p1

#dev.off()
```

