---
title: "Pi_TajD_Fst_Dxy"
author: "Joe McGirr"
date: '`r Sys.Date()`'
output:
  html_document:
    code_folding: hide
    collapsed: no
    df_print: paged
    number_sections: yes
    theme: paper
    toc: yes
    toc_depth: 5
    toc_float: yes
  html_notebook:
    toc: yes
    toc_depth: 5
editor_options: 
  chunk_output_type: console
---


```{css, echo=FALSE}
pre, code {white-space:pre !important; overflow-x:auto}
```

# Summary

This notebook generates scripts to calculate genome-wide summary statistics from a `.vcf` with ANGSD and visualize results. The markdown file can be found [here](https://github.com/joemcgirr/pac_herring/blob/master/Rmarkdown/population/Pi_TajD_Fst_Dxy.Rmd)

The vcf was generated with [these commands](https://github.com/joemcgirr/pac_herring/blob/master/Rmarkdown/fastq_to_vcf/Pacific_Herring_fastq_to_vcf.Rmd)

* Load R libraries
```{r setup, include=TRUE}

# if (!requireNamespace("BiocManager", quietly = TRUE))
#     install.packages("BiocManager")
# BiocManager::install(version = "3.12")

knitr::opts_chunk$set(echo = TRUE)
library(ggplot2)
library(tidyverse)
library(reticulate)
library(ggpubr)
library(sf)
library(pophelper)

# color-blind friendly 
# Wong, B. Points of view: Color blindness. Nat Methods (2011).
bla <- "#000000"
blu <- "#0072b2"
grb <- "#56b4e9"
lir <- "#cc79a7"
gre <- "#009e73"
red <- "#d55e00"
org <- "#e69f00"
yel <- "#f0e442"
gry<-  '#BBBBBB'

```

* Load python libraries
```{python, results='hide', eval = FALSE,class.source = 'fold-show'}
import io
import pandas as pd
import numpy as np

def sbatch_header(job,mem,tasks,hours):
    #sbatch submission script header
    script = 'script_' + job + '.sh'
    outfile = io.open(script,'w', newline='\n')    
    outfile.write('#!/bin/bash\n\n#SBATCH --job-name='+job+'\n')
    outfile.write('#SBATCH --mem='+mem+'G \n')
    outfile.write('#SBATCH --ntasks='+tasks+' \n')
    outfile.write('#SBATCH -e '+job+'_%A_%a.err \n')
    outfile.write('#SBATCH --time='+hours+':00:00  \n')
    outfile.write('#SBATCH --mail-user=jamcgirr@ucdavis.edu ##email you when job starts,ends,etc\n#SBATCH --mail-type=ALL\n')
    outfile.write('#SBATCH -p high \n\n')
    outfile.close()
    
def sbatch_header_loop(job,mem,tasks,hours,infile):
    #sbatch submission script header
    script = 'script_' + infile + job + '.sh'
    outfile = io.open(script,'w', newline='\n') 
    jobname= infile + job   
    outfile.write('#!/bin/bash\n\n#SBATCH --job-name='+jobname+'\n')
    outfile.write('#SBATCH --mem='+mem+'G \n')
    outfile.write('#SBATCH --ntasks='+tasks+' \n')
    outfile.write('#SBATCH -e '+jobname+'_%A_%a.err \n')
    outfile.write('#SBATCH --time='+hours+':00:00 \n')
    outfile.write('#SBATCH --mail-user=jamcgirr@ucdavis.edu ##email you when job starts,ends,etc\n#SBATCH --mail-type=ALL\n')
    outfile.write('#SBATCH -p high \n\n')
    outfile.close()

```



# Pi and Tajimas D
## Use ANGSD to estimate SFS using genotype likelihoods
Randomly down sample the number of input .bam files to match the lowest population sample size (n = 41)

use the -nSites 100000000 flag due to memory limits
this breaks up genome into 7 chunks
this gives you seven lines in the output.sfs
need to sum columns to make final sfs for following step

example script
```{bash,eval=FALSE,class.source = 'fold-show'}
#!/bin/bash

#SBATCH --job-name=BC17_angsd_downsample_sfs
#SBATCH --mem=40G 
#SBATCH --ntasks=8 
#SBATCH -e BC17_angsd_downsample_sfs_%A_%a.err 
#SBATCH --time=48:00:00 
#SBATCH --mail-user=jamcgirr@ucdavis.edu ##email you when job starts,ends,etc
#SBATCH --mail-type=ALL
#SBATCH -p high 

shuf /home/jamcgirr/ph/data/angsd/SFS/bamlist_test/BC17_bams_p_1_5_rm.txt | head -41 > /home/jamcgirr/ph/data/angsd/SFS/downsample/downsample_bams_BC17.txt 

/home/jamcgirr/apps/angsd_sep_20/angsd/angsd -bam /home/jamcgirr/ph/data/angsd/SFS/downsample/downsample_bams_BC17.txt -doSaf 1 -doMajorMinor 1 -doMaf 3 -doCounts 1 -doGlf 3 -anc /home/jamcgirr/ph/data/c_harengus/c.harengus.fa -ref /home/jamcgirr/ph/data/c_harengus/c.harengus.fa -minMapQ 30 -minQ 20 -GL 1 -P 8 -uniqueOnly 1 -remove_bads 1 -only_proper_pairs 1 -trim 0 -C 50 -minInd 10 -setMinDepth 10 -setMaxDepth 100 -out /home/jamcgirr/ph/data/angsd/SFS/downsample/BC17_minQ20_minMQ30 

/home/jamcgirr/apps/angsd_sep_20/angsd/misc/realSFS /home/jamcgirr/ph/data/angsd/SFS/downsample/BC17_minQ20_minMQ30.saf.idx -P 8 -fold 1 -nSites 100000000 > /home/jamcgirr/ph/data/angsd/SFS/downsample/BC17_minQ20_minMQ30_folded.sfs 
/home/jamcgirr/apps/angsd_sep_20/angsd/misc/realSFS /home/jamcgirr/ph/data/angsd/SFS/downsample/BC17_minQ20_minMQ30.saf.idx -P 8 -nSites 100000000 > /home/jamcgirr/ph/data/angsd/SFS/downsample/BC17_minQ20_minMQ30_unfolded.sfs 

#run: sbatch script_BC17_angsd_downsample_sfs.sh

```

### Generate sfs scripts

```{python,results='hide', eval = FALSE}

# lets try downsampling populations to the smallest n (SS06 = 41)
# run angsd direct on bam files

job_name = '_angsd_downsample_sfs'
wk_dir = "/home/jamcgirr/ph/data/angsd/SFS/downsample/"
infiles = ["BC17","CA17","PWS07","PWS17","PWS91","PWS96","SS06","SS17","SS96","TB06","TB17","TB91","TB96","WA17"]

for infile in infiles:
    script = 'script_' + infile + job_name + '.sh'
    sbatch_header_loop(job_name,'40','8','48', infile)
    o = io.open(script,'a+', newline='\n')
    
    
    o.write('shuf /home/jamcgirr/ph/data/angsd/SFS/bamlist_test/'+infile+'_bams_p_1_5_rm.txt | head -41 > '+wk_dir+'downsample_bams_'+infile+'.txt \n\n')
    # make saf from bams
    o.write('/home/jamcgirr/apps/angsd_sep_20/angsd/angsd -bam '+wk_dir+'downsample_bams_'+infile+'.txt -doSaf 1 -doMajorMinor 1 -doMaf 3 -anc /home/jamcgirr/ph/data/c_harengus/c.harengus.fa -minMapQ 30 -minQ 20 -GL 1 -P 8 -out '+wk_dir+infile+'_minQ20_minMQ30 \n\n')
    # make saf from bams using filters
    o.write('/home/jamcgirr/apps/angsd_sep_20/angsd/angsd -bam '+wk_dir+'downsample_bams_'+infile+'.txt -doSaf 1 -doMajorMinor 1 -doMaf 3 -doCounts 1 -doGlf 3 -anc /home/jamcgirr/ph/data/c_harengus/c.harengus.fa -ref /home/jamcgirr/ph/data/c_harengus/c.harengus.fa -minMapQ 30 -minQ 20 -GL 1 -P 8 -uniqueOnly 1 -remove_bads 1 -only_proper_pairs 1 -trim 0 -C 50 -minInd 10 -setMinDepth 10 -setMaxDepth 100 -out '+wk_dir+infile+'_minQ20_minMQ30 \n\n')
    # make folded sfs
    o.write('/home/jamcgirr/apps/angsd_sep_20/angsd/misc/realSFS '+wk_dir+infile+'_minQ20_minMQ30.saf.idx -P 8 -fold 1 -nSites 100000000 > '+wk_dir+infile+'_minQ20_minMQ30_folded.sfs \n')
    # make unfolded sfs
    o.write('/home/jamcgirr/apps/angsd_sep_20/angsd/misc/realSFS '+wk_dir+infile+'_minQ20_minMQ30.saf.idx -P 8 -nSites 100000000 > '+wk_dir+infile+'_minQ20_minMQ30_unfolded.sfs \n')

    o.write('\n\n#run: sbatch '+script)
    o.close()

# EST run time ~ 1 day

```


## Estimate thetas

example script
```{bash,eval=FALSE,class.source = 'fold-show'}
#!/bin/bash

#SBATCH --job-name=BC17_angsd_downsample_thetas
#SBATCH --mem=30G 
#SBATCH --ntasks=8 
#SBATCH -e BC17_angsd_downsample_thetas_%A_%a.err 
#SBATCH --time=24:00:00 
#SBATCH --mail-user=jamcgirr@ucdavis.edu ##email you when job starts,ends,etc
#SBATCH --mail-type=ALL
#SBATCH -p high 

/home/jamcgirr/apps/angsd_sep_20/angsd/misc/realSFS saf2theta /home/jamcgirr/ph/data/angsd/SFS/downsample/saf/BC17_minQ20_minMQ30.saf.idx -outname /home/jamcgirr/ph/data/angsd/SFS/downsample/thetas/folded/BC17_minQ20_minMQ30 -fold 1 -sfs /home/jamcgirr/ph/data/angsd/SFS/downsample/sfs/folded/BC17_minQ20_minMQ30_folded.sfs 
/home/jamcgirr/apps/angsd_sep_20/angsd/misc/thetaStat do_stat /home/jamcgirr/ph/data/angsd/SFS/downsample/thetas/folded/BC17_minQ20_minMQ30.thetas.idx 
/home/jamcgirr/apps/angsd_sep_20/angsd/misc/thetaStat do_stat /home/jamcgirr/ph/data/angsd/SFS/downsample/thetas/folded/BC17_minQ20_minMQ30.thetas.idx -win 50000 -step 10000 -outnames /home/jamcgirr/ph/data/angsd/SFS/downsample/thetas/folded/BC17_minQ20_minMQ30_50kb_win_10kb_step 

/home/jamcgirr/apps/angsd_sep_20/angsd/misc/realSFS saf2theta /home/jamcgirr/ph/data/angsd/SFS/downsample/saf/BC17_minQ20_minMQ30.saf.idx -outname /home/jamcgirr/ph/data/angsd/SFS/downsample/thetas/unfolded/BC17_minQ20_minMQ30 -sfs /home/jamcgirr/ph/data/angsd/SFS/downsample/sfs/folded/BC17_minQ20_minMQ30_folded.sfs 
/home/jamcgirr/apps/angsd_sep_20/angsd/misc/thetaStat do_stat /home/jamcgirr/ph/data/angsd/SFS/downsample/thetas/unfolded/BC17_minQ20_minMQ30.thetas.idx 
/home/jamcgirr/apps/angsd_sep_20/angsd/misc/thetaStat do_stat /home/jamcgirr/ph/data/angsd/SFS/downsample/thetas/unfolded/BC17_minQ20_minMQ30.thetas.idx -win 50000 -step 10000 -outnames /home/jamcgirr/ph/data/angsd/SFS/downsample/thetas/unfolded/BC17_minQ20_minMQ30_50kb_win_10kb_step 


#run: sbatch script_BC17_angsd_downsample_thetas.sh

```

### Generate thetas scripts

```{python,results='hide', eval = FALSE}

job_name = '_angsd_downsample_thetas'
saf_dir = "/home/jamcgirr/ph/data/angsd/SFS/downsample/saf/"
fold_dir = "/home/jamcgirr/ph/data/angsd/SFS/downsample/sfs/folded/"
unfold_dir = "/home/jamcgirr/ph/data/angsd/SFS/downsample/sfs/folded/"
thetas_dir = "/home/jamcgirr/ph/data/angsd/SFS/downsample/thetas/"
infiles = ["BC17","CA17","PWS07","PWS17","PWS91","PWS96","SS06","SS17","SS96","TB06","TB17","TB91","TB96","WA17"]

for infile in infiles:
    script = 'script_' + infile + job_name + '.sh'
    sbatch_header_loop(job_name,'30','8','24', infile)
    o = io.open(script,'a+', newline='\n')
    
    # folded
    o.write('/home/jamcgirr/apps/angsd_sep_20/angsd/misc/realSFS saf2theta '+saf_dir+infile+'_minQ20_minMQ30.saf.idx -outname '+thetas_dir+'folded/'+infile+'_minQ20_minMQ30 -fold 1 -sfs '+fold_dir+infile+'_minQ20_minMQ30_folded.sfs \n')
    o.write('/home/jamcgirr/apps/angsd_sep_20/angsd/misc/thetaStat do_stat '+thetas_dir+'folded/'+infile+'_minQ20_minMQ30.thetas.idx \n')
    o.write('/home/jamcgirr/apps/angsd_sep_20/angsd/misc/thetaStat do_stat '+thetas_dir+'folded/'+infile+'_minQ20_minMQ30.thetas.idx -win 50000 -step 10000 -outnames '+thetas_dir+'folded/'+infile+'_minQ20_minMQ30_50kb_win_10kb_step \n\n')
    # unfolded
    o.write('/home/jamcgirr/apps/angsd_sep_20/angsd/misc/realSFS saf2theta '+saf_dir+infile+'_minQ20_minMQ30.saf.idx -outname '+thetas_dir+'unfolded/'+infile+'_minQ20_minMQ30 -sfs '+fold_dir+infile+'_minQ20_minMQ30_folded.sfs \n')
    o.write('/home/jamcgirr/apps/angsd_sep_20/angsd/misc/thetaStat do_stat '+thetas_dir+'unfolded/'+infile+'_minQ20_minMQ30.thetas.idx \n')
    o.write('/home/jamcgirr/apps/angsd_sep_20/angsd/misc/thetaStat do_stat '+thetas_dir+'unfolded/'+infile+'_minQ20_minMQ30.thetas.idx -win 50000 -step 10000 -outnames '+thetas_dir+'unfolded/'+infile+'_minQ20_minMQ30_50kb_win_10kb_step \n')
    
    
    o.write('\n\n#run: sbatch '+script)
    o.close()

# EST run time ~4hr

```


## Plot folded SFS

```{r}
# using SFS from angsd -dosaf -bam

pop_names <- c("BC17","CA17","PWS07","PWS17","PWS91","PWS96","SS06","SS17","SS96","TB06","TB17","TB91","TB96","WA17")

pop_name <- "BC17"

#for (pop_name in pop_names){
sfs <- read.table(paste("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/sfs/downsample_41/folded/",pop_name,"_minQ20_minMQ30_folded.sfs",sep = ""), header = FALSE, stringsAsFactors = FALSE)
sfs
barplot(sfs[-c(1,length(sfs))]) #plot variable sites 

# Use this to sum columns from the 'chunks' sfs calculated with nSites

sfs <- as.vector(colSums(sfs))
#write.table(t(sfs),paste("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/downsample_41/sfs/folded/",pop_name,"_minQ20_minMQ30_folded.sfs",sep = ""), quote = FALSE, row.names = FALSE, col.names = FALSE)
#}

# https://github.com/ANGSD/angsd/issues/97


```

## Plot folded SFS

```{r}
# using SFS from angsd -dosaf -bam

pop_names <- c("BC17","CA17","PWS07","PWS17","PWS91","PWS96","SS06","SS17","SS96","TB06","TB17","TB91","TB96","WA17")

pop_name <- "BC17"

#for (pop_name in pop_names){
sfs <- read.table(paste("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/sfs/downsample_41/unfolded/",pop_name,"_minQ20_minMQ30_unfolded.sfs",sep = ""), header = FALSE, stringsAsFactors = FALSE)
sfs
sfs <- as.vector(colSums(sfs))
barplot(sfs[-c(1,length(sfs))]) #plot variable sites 

# Use this to make the final sfs
# sum columns from the 'chunks' sfs calculated with -nSites

#write.table(t(sfs),paste("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/downsample_41/sfs/folded/",pop_name,"_minQ20_minMQ30_folded.sfs",sep = ""), quote = FALSE, row.names = FALSE, col.names = FALSE)
#}

# https://github.com/ANGSD/angsd/issues/97


```


## Plot Tajimas D

```{r, message=FALSE, warning=FALSE, fig.width= 10, fig.height= 7,class.source = 'fold-show'}

pop_names <- c("BC17","CA17","PWS07","PWS17","PWS91","PWS96","SS06","SS17","SS96","TB06","TB17","TB91","TB96","WA17")
pop_names <- c("CA17","PWS07","PWS17","PWS91","PWS96","SS06","SS17","SS96","TB06","TB17","TB91","TB96","WA17")

pop <- "BC17"
#thetas <-read.delim(paste('C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/thetas/50kb/population_',pop,'_ph_filtered_snps_minDP600_maxDP2000_maf0.05_minQ20_minMQ30_maxmiss0.5_theta.thetas50kbWindow.gz.pestPG',sep = ""))
thetas <-read.delim(paste('C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/thetas/downsample_41/folded/',pop,'_minQ20_minMQ30_50kb_win_10kb_step.pestPG',sep = ""))
thetas <- thetas[c("Tajima")]
names(thetas)[names(thetas)=="Tajima"] <- pop

for (pop in pop_names){

thetas1 <-read.delim(paste('C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/thetas/downsample_41/folded/',pop,'_minQ20_minMQ30_50kb_win_10kb_step.pestPG',sep = ""))
thetas1[c("Chr","WinCenter","Tajima")]
thetas$Tajima <- thetas1$Tajima
names(thetas)[names(thetas)=="Tajima"] <- pop 
  
}

my_cols <- c(grb,org,red,red,red,red,yel,yel,yel,blu,blu,blu,blu,lir)
theta_test <- thetas %>% 
      gather(key="MesureType", value="Val")
p1 <- thetas %>% 
      gather(key="MesureType", value="Val") %>%
      ggplot( aes(x=reorder(MesureType,Val, FUN = mean), y=Val, fill=MesureType)) +
      geom_violin() + xlab("\npopulation/year") + ylab("Tajima's D\n")+
      scale_fill_manual(name = "Population",values=my_cols,guide=FALSE)+
      #stat_summary(fun=mean, geom="line")+
      theme_minimal()
      #geom_jitter(height = 0, width = 0.1, alpha = 0.1)

#png("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/figs/population_structure/taj_d.png", height = 5, width = 7, units = 'in', res = 600)

p1

#dev.off()
```


## Plot Pi

```{r, message=FALSE, warning=FALSE, fig.width= 10, fig.height= 7,class.source = 'fold-show'}
# thetas <-read.delim('C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/thetas/population_BC17_ph_filtered_snps_minD# P600_maxDP2000_maf0.05_minQ20_minMQ30_maxmiss0.5_theta.thetas10kbWindow.gz.pestPG')
# #pi = tP/nsites
# #https://doi.org/10.1186/s12862-018-1209-y
# thetas$pi <- thetas$tP/thetas$nSites 
# chr1 <- thetas %>% filter(Chr == "chr1")
# range(thetas$Tajima)
# range(na.omit(thetas$pi))
# 
# 
# plot(chr1$WinCenter, chr1$Tajima)
# plot(chr1$WinCenter, chr1$pi)


#https://github.com/ANGSD/angsd/issues/329#issuecomment-681005380
# pi

pop <- "BC17"
thetas <-read.delim(paste('C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/thetas/downsample_41/folded/',pop,'_minQ20_minMQ30_50kb_win_10kb_step.pestPG',sep = ""))
#thetas$pi<- (thetas$tP*thetas$nSites)/10000
thetas$pi<- thetas$tP/thetas$nSites
#thetas$pi<- thetas$tP/50000
thetas <- thetas[c("pi")]
names(thetas)[names(thetas)=="pi"] <- pop

for (pop in pop_names){

thetas1 <-read.delim(paste('C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/thetas/downsample_41/folded/',pop,'_minQ20_minMQ30_50kb_win_10kb_step.pestPG',sep = ""))
#thetas1$pi<- (thetas1$tP*thetas1$nSites)/10000
thetas1$pi<- thetas1$tP/thetas1$nSites
thetas1[c("Chr","WinCenter","pi")]
thetas$pi <- thetas1$pi
names(thetas)[names(thetas)=="pi"] <- pop 
  
}

my_cols <- c(grb,org,red,red,red,red,yel,yel,yel,blu,blu,blu,blu,lir)
thetas <- na.omit(thetas)
p1 <- thetas %>% 
      gather(key="MesureType", value="Val") %>%
      ggplot( aes(x=reorder(MesureType,Val, FUN = median), y=Val, fill=MesureType)) +
      geom_violin() + xlab("\npopulation/year") + ylab(expression(pi))+
      scale_fill_manual(name = "Population",values=my_cols,guide=FALSE)+
      theme_minimal()
      #geom_jitter(height = 0, width = 0.1, alpha = 0.1)

#png("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/figs/population_structure/pi_angsd.png", height = 5, width = 7, units = 'in', res = 600)

p1

#dev.off()

### vcftools just to get a sense with hard calls since angsd values are stupid high
#pi_50kb_win_BC17_ph_filtered_snps_minDP600_maxDP2000_maf0.05_minQ20_minMQ30_maxmiss0.5_plates_1_5_rm.txt.windowed


#pop <- "BC17"
#thetas <-read.delim(paste('C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/thetas/vcftools/pi_50kb_win_',pop,'_ph_filtered_snps_minDP600_maxDP2000_maf0.05_minQ20_minMQ30_maxmiss0.5_plates_1_5_rm.txt.windowed.pi',sep = ""))
##thetas$pi<- (thetas$tP*thetas$nSites)/10000
##thetas$pi<- thetas$tP/thetas$nSites
##thetas$pi<- thetas$tP/50000
#thetas <- head(thetas, 14500)
#thetas <- thetas[c("PI")]
#names(thetas)[names(thetas)=="PI"] <- pop
#
#for (pop in pop_names){

#thetas1 <-read.delim(paste('C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/thetas/vcftools/pi_50kb_win_',pop,'_ph_filtered_snps_minDP600_maxDP2000_maf0.05_minQ20_minMQ30_maxmiss0.5_plates_1_5_rm.txt.windowed.pi',sep = ""))
##thetas1$pi<- (thetas1$tP*thetas1$nSites)/10000
##thetas1$pi<- thetas1$tP/thetas1$nSites
##thetas1[c("Chr","WinCenter","pi")]
#thetas1 <- head(thetas1, nrow(thetas))
#thetas$pi <- thetas1$PI
#names(thetas)[names(thetas)=="pi"] <- pop 
  
#}

#my_cols <- c(grb,org,red,red,red,red,yel,yel,yel,blu,blu,blu,blu,lir)
#p1 <- thetas %>% 
#      gather(key="MesureType", value="Val") %>%
#      ggplot( aes(x=reorder(MesureType,Val, FUN = median), y=Val, fill=MesureType)) +
#      geom_violin() + xlab("\npopulation/year") + ylab("pi\n")+
#      scale_fill_manual(name = "Population",values=my_cols,guide=FALSE)+
#      theme_minimal()
#      #geom_jitter(height = 0, width = 0.1, alpha = 0.1)

#png("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/figs/population_structure/pi_vcftools.png", height = 5, width = 7, units = 'in', res = 600)

#p1

#dev.off()



```

## Plot Pi population matrix







# Fst, Dxy, and Population Branch Statistic 

Generate site frequency spectrum using vcf sites

## Subset master vcf into population vcfs
example script
```{bash,eval=FALSE,class.source = 'fold-show'}
#!/bin/bash

#SBATCH --job-name=BC17_subset_pops_vcfs
#SBATCH --mem=8G 
#SBATCH --ntasks=4 
#SBATCH -e BC17_subset_pops_vcfs_%A_%a.err 
#SBATCH --time=24:00:00 
#SBATCH --mail-user=jamcgirr@ucdavis.edu ##email you when job starts,ends,etc
#SBATCH --mail-type=ALL
#SBATCH -p high 

module load bcftools 
bcftools view -S /home/jamcgirr/ph/scripts/angsd/SFS/SFS_by_pop/BC17_plates_1_through_5_rm.txt --threads 4 /home/jamcgirr/ph/data/vcfs/ph_filtered_snps_minDP600_maxDP2000_minQ20_minMQ30_NS0.5_maf0.05.vcf.gz | bcftools +fill-tags -Oz -- -t all,'DP=sum(DP)' > /home/jamcgirr/ph/data/vcfs/split_pops/maf05/population_BC17_ph_filtered_snps_minDP600_maxDP2000_minQ20_minMQ30_NS0.5_maf0.05.vcf.gz 


#command to run: sbatch script_BC17_subset_pops_vcfs.sh

```

### Generate subset scripts

```{python,results='hide', eval = FALSE}

vcf_name = 'ph_filtered_snps_minDP600_maxDP2000_minQ20_minMQ30_NS0.5_maf0.05'
job_name = '_subset_pops_vcfs'

infiles = ["BC17","CA17","PWS07","PWS17","PWS91","PWS96","SS06","SS17","SS96","TB06","TB17","TB91","TB96","WA17"]

for infile in infiles:
    script = 'script_' + infile + job_name + '.sh'
    sbatch_header_loop(job_name,'8','4','24', infile)
    o = io.open(script,'a+', newline='\n')
    

    o.write('module load bcftools \n')
    o.write('bcftools view -S /home/jamcgirr/ph/scripts/angsd/SFS/SFS_by_pop/'+infile+'_plates_1_through_5_rm.txt --threads 4 /home/jamcgirr/ph/data/vcfs/'+vcf_name+'.vcf.gz | bcftools +fill-tags -Oz -- -t all,\'DP=sum(DP)\' > /home/jamcgirr/ph/data/vcfs/split_pops/maf05/population_'+infile+'_'+vcf_name+'.vcf.gz \n')
    
    
    o.write('\n\n#command to run: sbatch '+script)
    o.close()
    
#EST run time ~15min

```

## Use realSFS to make folded and unfolded SFS for each population

example script
```{bash,eval=FALSE,class.source = 'fold-show'}
#!/bin/bash

#SBATCH --job-name=BC17_SFS
#SBATCH --mem=8G 
#SBATCH --ntasks=4 
#SBATCH -e BC17_SFS_%A_%a.err 
#SBATCH --time=24:00:00 
#SBATCH --mail-user=jamcgirr@ucdavis.edu ##email you when job starts,ends,etc
#SBATCH --mail-type=ALL
#SBATCH -p high 

/home/jamcgirr/apps/angsd_sep_20/angsd/angsd -doSaf 1 -vcf-pl /home/jamcgirr/ph/data/vcfs/split_pops/maf05/population_BC17_ph_filtered_snps_minDP600_maxDP2000_minQ20_minMQ30_NS0.5_maf0.05.vcf.gz -out /home/jamcgirr/ph/data/angsd/SFS/vcf/maf05/BC17 -anc /home/jamcgirr/ph/data/c_harengus/c.harengus.fa 
/home/jamcgirr/apps/angsd_sep_20/angsd/misc/realSFS /home/jamcgirr/ph/data/angsd/SFS/vcf/maf05/BC17.saf.idx -P 4 -fold 1 > /home/jamcgirr/ph/data/angsd/SFS/vcf/maf05/BC17_folded.sfs 
/home/jamcgirr/apps/angsd_sep_20/angsd/misc/realSFS /home/jamcgirr/ph/data/angsd/SFS/vcf/maf05/BC17.saf.idx -P 4 > /home/jamcgirr/ph/data/angsd/SFS/vcf/maf05/BC17_unfolded.sfs 


#command to run: sbatch script_BC17_SFS.sh
```

### Generate SFS scripts

```{python,results='hide', eval = FALSE}

job_name = '_SFS'
infiles = ["BC17","CA17","PWS07","PWS17","PWS91","PWS96","SS06","SS17","SS96","TB06","TB17","TB91","TB96","WA17"]

for infile in infiles:
    script = 'script_' + infile + job_name + '.sh'
    sbatch_header_loop(job_name,'8','4','24', infile)
    o = io.open(script,'a+', newline='\n')
    
    vcf_name = 'ph_filtered_snps_minDP600_maxDP2000_minQ20_minMQ30_NS0.5_maf0.05'
    
    # make saf from vcf
    o.write('/home/jamcgirr/apps/angsd_sep_20/angsd/angsd -doSaf 1 -vcf-pl /home/jamcgirr/ph/data/vcfs/split_pops/maf05/population_'+infile+'_'+vcf_name+'.vcf.gz -out /home/jamcgirr/ph/data/angsd/SFS/vcf/maf05/'+infile+' -anc /home/jamcgirr/ph/data/c_harengus/c.harengus.fa \n')
    # make folded sfs
    o.write('/home/jamcgirr/apps/angsd_sep_20/angsd/misc/realSFS /home/jamcgirr/ph/data/angsd/SFS/vcf/maf05/'+infile+'.saf.idx -P 4 -fold 1 > /home/jamcgirr/ph/data/angsd/SFS/vcf/maf05/'+infile+'_folded.sfs \n')
    # make unfolded sfs
    o.write('/home/jamcgirr/apps/angsd_sep_20/angsd/misc/realSFS /home/jamcgirr/ph/data/angsd/SFS/vcf/maf05/'+infile+'.saf.idx -P 4 > /home/jamcgirr/ph/data/angsd/SFS/vcf/maf05/'+infile+'_unfolded.sfs \n')
    

    o.write('\n\n#command to run: sbatch '+script)
    o.close()

# EST run time ~15min
```








## Plot Fst

```{r}

pop_names = c("PWS91","PWS96","PWS07","PWS17","TB91","TB96","TB06","TB17","SS96","SS06","SS17","BC17","WA17","CA17")

pop0 <- "TB17"
pop1 <- "PWS17"
pop2 <- "SS17"

fsts <- list.files(path="C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/fst/sliding_window/", full.names=T)
#fsts <- fsts[1]

for (f in fsts){  

pops <- strsplit(f, "_")
pops <- c(pops[[1]][9], pops[[1]][10],gsub("\\.txt","",pops[[1]][11]))

fst <-read.delim(paste("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/fst/sliding_window/fst_pbs_50kb_win_10kb_step_",pops[1],"_",pops[2],"_",pops[3],".txt", sep = ""))
#head(fst)
fst <- fst %>% mutate(chr = as.numeric(gsub("chr", "", chr)))
fst$chr_num <- factor(fst$chr, levels = c(1:26))

# Fst
p <- ggplot(fst, aes(x = midPos, y = Fst01)) + 
     geom_point(size = 1, color = gry,alpha = 0.4, shape = 1)+
     theme_minimal()+
     theme(axis.text.x=element_blank())+
     ylab("Fst\n")+ xlab("")+ 
     ggtitle(paste(pops[1],"vs. ", pops[2]))+
     geom_smooth(method = "loess", se = FALSE, span = 1/10, color = blu)+
     facet_wrap(~chr_num, ncol = 9)
print(p)

p <- ggplot(fst, aes(x = midPos, y = Fst02)) + 
     geom_point(size = 1, color = gry,alpha = 0.4, shape = 1)+
     theme_minimal()+
     theme(axis.text.x=element_blank())+
     ylab("Fst\n")+ xlab("")+ 
     ggtitle(paste(pops[1],"vs. ", pops[3]))+
     geom_smooth(method = "loess", se = FALSE, span = 1/10, color = blu)+
     facet_wrap(~chr_num, ncol = 9)
print(p)

p <- ggplot(fst, aes(x = midPos, y = Fst12)) + 
     geom_point(size = 1, color = gry,alpha = 0.4, shape = 1)+
     theme_minimal()+
     theme(axis.text.x=element_blank())+
     ylab("Fst\n")+ xlab("")+ 
     ggtitle(paste(pops[2],"vs. ", pops[3]))+
     geom_smooth(method = "loess", se = FALSE, span = 1/10, color = blu)+
     facet_wrap(~chr_num, ncol = 9)
print(p)

#PBS
p <- ggplot(fst, aes(x = midPos, y = PBS0)) + 
     geom_point(size = 1, color = gry,alpha = 0.4, shape = 1)+
     theme_minimal()+
     theme(axis.text.x=element_blank())+
     ylab("PBS\n")+ xlab("")+ 
     ggtitle(paste(pops[1]))+
     geom_smooth(method = "loess", se = FALSE, span = 1/10, color = red)+
     facet_wrap(~chr_num, ncol = 9)
print(p)

p <- ggplot(fst, aes(x = midPos, y = PBS1)) + 
     geom_point(size = 1, color = gry,alpha = 0.4, shape = 1)+
     theme_minimal()+
     theme(axis.text.x=element_blank())+
     ylab("PBS\n")+ xlab("")+ 
     ggtitle(paste(pops[2]))+
     geom_smooth(method = "loess", se = FALSE, span = 1/10, color = red)+
     facet_wrap(~chr_num, ncol = 9)
print(p)

p <- ggplot(fst, aes(x = midPos, y = PBS2)) + 
     geom_point(size = 1, color = gry,alpha = 0.4, shape = 1)+
     theme_minimal()+
     theme(axis.text.x=element_blank())+
     ylab("PBS\n")+ xlab("")+ 
     ggtitle(paste(pops[3]))+
     geom_smooth(method = "loess", se = FALSE, span = 1/10, color = red)+
     facet_wrap(~chr_num, ncol = 9)
print(p)

}

#dxy




# chr1 <- fst %>% filter(chr == "chr1")
# 
# plot(chr1$midPos, chr1$Fst01)
# plot(chr1$midPos, chr1$Fst02)
# plot(chr1$midPos, chr1$Fst12)
# 
# plot(chr1$midPos, chr1$PBS0)
# plot(chr1$midPos, chr1$PBS1)
# plot(chr1$midPos, chr1$PBS2)

```



```{r, echo = FALSE,results='hide'}
## Site Frequency Spectrum
# s<-scan('C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/thetas/population_BC17_ph_filtered_snps_minDP600_maxDP20# 00_maf0.05_minQ20_maxmiss0.5.sfs')
# s<-s[-c(1,length(s))]
# s<-s/sum(s)
# barplot(s,names=1:length(s),main='SFS')
```


## Dxy

```{r, results= "hide", eval=FALSE}

mafs_dir <- "C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/dxy/mafs/"
comps <- c("TB_PWS","TB_SS","TB_BC","TB_WA","TB_CA",
           "PWS_SS","PWS_BC","PWS_WA","PWS_CA",
           "SS_BC","SS_WA","SS_CA",
           "BC_WA","BC_CA",
           "WA_CA")
yr <- "17"
comp <- "TB_PWS"

for (comp in comps){

popA <- paste(strsplit(comp,"_")[[1]][1],yr, sep = "")
popB <- paste(strsplit(comp,"_")[[1]][2],yr, sep = "")

allfreqA <- read.table(paste(mafs_dir,popA,"_vcf.mafs", sep = ""),sep='\t',row.names=NULL, header=T)
allfreqB <- read.table(paste(mafs_dir,popB,"_vcf.mafs", sep = ""),sep='\t',row.names=NULL, header=T)

### Manipulating the table and print dxy table
allfreq <- merge(allfreqA, allfreqB, by=c("chromo","position"))
allfreq <- allfreq[order(allfreq$chromo, allfreq$position),]
# -> Actual dxy calculation
allfreq <- transform(allfreq, dxy=(knownEM.x*(1-knownEM.y))+(knownEM.y*(1-knownEM.x)))

#head(allfreq)

rol_win <- winScan(x = allfreq, 
                   groups = "chromo", 
                   position = "position", 
                   values = c("dxy"), 
                   win_size = 50000,
                   win_step = 10000,
                   funs = c("sum"))
rol_win$dxy <- rol_win$dxy_sum/50000

#write.table(rol_win, file=paste("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/dxy/windows/",popA,"_",popB,"_Dxy_50kb_win_10kb_step.txt",sep = ""),quote=FALSE, row.names=FALSE, sep='\t')
print('Created Dxy_persite.txt')

### Print global dxy
print(paste0(popA, popB,' Global dxy is: ',mean(rol_win$dxy)))
  
}


dxys <- list.files(path="C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/dxy/windows/", full.names=T)
#d <- dxys[1]

for (d in dxys){  

pops <- strsplit(d, "/")
pops <- strsplit(pops[[1]][11], "_")

pops <- c(pops[[1]][1], pops[[1]][2])
dxy <- read.delim(paste("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/dxy/windows/",pops[1],"_",pops[2],"_Dxy_50kb_win_10kb_step.txt", sep = ""))
#head(fst)
dxy <- dxy %>% mutate(chromo = as.numeric(gsub("chr", "", chromo)))
dxy$chr_num <- factor(dxy$chr, levels = c(1:26))


p <- ggplot(dxy, aes(x = win_mid, y = dxy)) + 
     geom_point(size = 1, color = gry,alpha = 0.4, shape = 1)+
     theme_minimal()+
     theme(axis.text.x=element_blank())+
     ylab("Dxy\n")+ xlab("")+ 
     ggtitle(paste(pops[1],"vs. ", pops[2]))+
     geom_smooth(method = "loess", se = FALSE, span = 1/10, color = org)+
     facet_wrap(~chr_num, ncol = 9)
print(p)

}




fst_vec <- c(0,tbxpw,tbxss,tbxbc,tbxwa,tbxca,
             tbxpw,0,pwxss,pwxbc,pwxwa,pwxca,
             tbxss,pwxss,0,ssxbc,ssxwa,ssxca,
             tbxbc,pwxbc,ssxbc,0,bcxwa,bcxca,
             tbxwa,pwxwa,ssxwa,bcxwa,0,waxca,
             tbxca,pwxca,ssxca,bcxca,waxca,0)

fst_mat = matrix(fst_vec, nrow = 6, ncol = 6)
colnames(fst_mat) <- c("TB17","PWS17","SS17","BC17","WA17","CA17")
rownames(fst_mat) <- c("TB17","PWS17","SS17","BC17","WA17","CA17")
print(fst_mat)

```

