---
title: "Pacific Herring Fst"
author: "Joe McGirr"
date: "8/28/2020"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(ggplot2)
library(tidyverse)
library(reticulate)
library(reshape2)
library(plyranges)
library(seqinr)
#BiocManager::install("visFuns.R")


# color-blind friendly 
# Wong, B. Points of view: Color blindness. Nat Methods (2011).
bla <- "#000000"
blu <- "#0072b2"
grb <- "#56b4e9"
lir <- "#cc79a7"
gre <- "#009e73"
red <- "#d55e00"
org <- "#e69f00"
yel <- "#f0e442"
gry<-  '#BBBBBB'

```


## Fst matrix 2017 samples
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.width= 10, fig.height= 7}

fsts <- list.files(path="C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/fst/sliding_window/", full.names=T)


# 2017 samples
fst <- read.delim("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/fst/sliding_window/fst_pbs_50kb_win_10kb_step_folded_TB17_PWS17_SS17.txt")
tbxpw <- round(mean(fst$Fst01),4)
tbxss <- round(mean(fst$Fst02),4)
pwxss <- round(mean(fst$Fst12),4)
fst <- read.delim("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/fst/sliding_window/fst_pbs_50kb_win_10kb_step_folded_TB17_PWS17_BC17.txt")
tbxbc <- round(mean(fst$Fst02),4)
pwxbc <- round(mean(fst$Fst12),4)
fst <- read.delim("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/fst/sliding_window/fst_pbs_50kb_win_10kb_step_folded_TB17_PWS17_WA17.txt")
tbxwa <- round(mean(fst$Fst02),4)
pwxwa <- round(mean(fst$Fst12),4)
fst <- read.delim("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/fst/sliding_window/fst_pbs_50kb_win_10kb_step_folded_TB17_PWS17_CA17.txt")
tbxca <- round(mean(fst$Fst02),4)
pwxca <- round(mean(fst$Fst12),4)

fst <- read.delim("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/fst/sliding_window/fst_pbs_50kb_win_10kb_step_folded_PWS17_SS17_BC17.txt")
ssxbc <- round(mean(fst$Fst12),4)
fst <- read.delim("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/fst/sliding_window/fst_pbs_50kb_win_10kb_step_folded_PWS17_SS17_WA17.txt")
ssxwa <- round(mean(fst$Fst12),4)
fst <- read.delim("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/fst/sliding_window/fst_pbs_50kb_win_10kb_step_folded_PWS17_SS17_CA17.txt")
ssxca <- round(mean(fst$Fst12),4)
fst <- read.delim("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/fst/sliding_window/fst_pbs_50kb_win_10kb_step_folded_BC17_WA17_CA17.txt")
bcxwa <- round(mean(fst$Fst01),4)
bcxca <- round(mean(fst$Fst02),4)
waxca <- round(mean(fst$Fst12),4)

fst_vec <- c(0,tbxpw,tbxss,tbxbc,tbxwa,tbxca,
             tbxpw,0,pwxss,pwxbc,pwxwa,pwxca,
             tbxss,pwxss,0,ssxbc,ssxwa,ssxca,
             tbxbc,pwxbc,ssxbc,0,bcxwa,bcxca,
             tbxwa,pwxwa,ssxwa,bcxwa,0,waxca,
             tbxca,pwxca,ssxca,bcxca,waxca,0)

fst_mat = matrix(fst_vec, nrow = 6, ncol = 6)
colnames(fst_mat) <- c("TB17","PWS17","SS17","BC17","WA17","CA17")
rownames(fst_mat) <- c("TB17","PWS17","SS17","BC17","WA17","CA17")
print(fst_mat)

  # Get upper triangle of the correlation matrix
  get_upper_tri <- function(cormat){
    cormat[lower.tri(cormat)]<- NA
    return(cormat)
  }

    get_lower_tri<-function(cormat){
    cormat[upper.tri(cormat)] <- NA
    return(cormat)
  }
upper_tri <- get_upper_tri(fst_mat)

# Melt the correlation matrix
melted_cormat <- melt(upper_tri, na.rm = TRUE)
# Heatmap
ggplot(data = melted_cormat, aes(Var2, Var1, fill = value))+
 geom_tile(color = "white")+
 scale_fill_gradient(low = blu, high = red, 
    limit = c(0,max(fst_mat)), space = "Lab", 
   name="Fst") +
  theme_minimal()+ xlab("")+ylab("")+
 theme(axis.text.x = element_text(angle = 0, vjust = 0, 
    size = 12, hjust = 0.5))+
   theme(axis.text.y = element_text(size = 12))+
 coord_fixed()+
 geom_text(aes(Var2, Var1, label = value), color = "black", size = 5)


```

## Fst matrix temporal samples (within population comparisons)

```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.width= 10, fig.height= 7}
# temporal samples PWS
fst <- read.delim("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/fst/sliding_window/fst_pbs_50kb_win_10kb_step_folded_PWS91_PWS96_PWS07.txt")
p91xp96 <- round(mean(fst$Fst01),4)
p91xp07 <- round(mean(fst$Fst02),4)
p96xp07 <- round(mean(fst$Fst12),4)
fst <- read.delim("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/fst/sliding_window/fst_pbs_50kb_win_10kb_step_folded_PWS91_PWS96_PWS17.txt")
p91xp17 <- round(mean(fst$Fst02),4)
p96xp17 <- round(mean(fst$Fst12),4)
fst <- read.delim("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/fst/sliding_window/fst_pbs_50kb_win_10kb_step_folded_PWS91_PWS07_PWS17.txt")
p07xp17 <- round(mean(fst$Fst12),4)

fst_vec <- c(0,p91xp96,p91xp07,p91xp17,
             p91xp96,0,p96xp07,p96xp17,
             p91xp07,p96xp07,0,p07xp17,
             p91xp17,p96xp17,p07xp17,0)

fst_mat = matrix(fst_vec, nrow = 4, ncol = 4)
colnames(fst_mat) <- c("PWS91","PWS96","PWS07","PWS17")
rownames(fst_mat) <- c("PWS91","PWS96","PWS07","PWS17")
print(fst_mat)
upper_tri <- get_upper_tri(fst_mat)

# Melt the correlation matrix
melted_cormat <- melt(upper_tri, na.rm = TRUE)
# Heatmap
ggplot(data = melted_cormat, aes(Var2, Var1, fill = value))+
 geom_tile(color = "white")+
 scale_fill_gradient(low = blu, high = red, 
    limit = c(0,0.015), space = "Lab", 
   name="Fst") +
  theme_minimal()+ xlab("")+ylab("")+
 theme(axis.text.x = element_text(angle = 0, vjust = 0, 
    size = 12, hjust = 0.5))+
   theme(axis.text.y = element_text(size = 12))+
 coord_fixed()+
 geom_text(aes(Var2, Var1, label = value), color = "black", size = 5)


# temporal samples TB
fst <- read.delim("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/fst/sliding_window/fst_pbs_50kb_win_10kb_step_folded_TB91_TB96_TB06.txt")
p91xp96 <- round(mean(fst$Fst01),4)
p91xp07 <- round(mean(fst$Fst02),4)
p96xp07 <- round(mean(fst$Fst12),4)
fst <- read.delim("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/fst/sliding_window/fst_pbs_50kb_win_10kb_step_folded_TB91_TB96_TB17.txt")
p91xp17 <- round(mean(fst$Fst02),4)
p96xp17 <- round(mean(fst$Fst12),4)
fst <- read.delim("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/fst/sliding_window/fst_pbs_50kb_win_10kb_step_folded_TB91_TB06_TB17.txt")
p07xp17 <- round(mean(fst$Fst12),4)

fst_vec <- c(0,p91xp96,p91xp07,p91xp17,
            p91xp96,0,p96xp07,p96xp17,
            p91xp07,p96xp07,0,p07xp17,
            p91xp17,p96xp17,p07xp17,0)

fst_mat = matrix(fst_vec, nrow = 4, ncol = 4)
colnames(fst_mat) <- c("TB91","TB96","TB06","TB17")
rownames(fst_mat) <- c("TB91","TB96","TB06","TB17")
print(fst_mat)
upper_tri <- get_upper_tri(fst_mat)

# Melt the correlation matrix
melted_cormat <- melt(upper_tri, na.rm = TRUE)
# Heatmap
ggplot(data = melted_cormat, aes(Var2, Var1, fill = value))+
geom_tile(color = "white")+
scale_fill_gradient(low = blu, high = red, 
   limit = c(0,0.015), space = "Lab", 
  name="Fst") +
 theme_minimal()+ xlab("")+ylab("")+
theme(axis.text.x = element_text(angle = 0, vjust = 0, 
   size = 12, hjust = 0.5))+
  theme(axis.text.y = element_text(size = 12))+
coord_fixed()+
geom_text(aes(Var2, Var1, label = value), color = "black", size = 5)


# temporal samples SS
fst <- read.delim("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/fst/sliding_window/fst_pbs_50kb_win_10kb_step_folded_SS96_SS06_SS17.txt")
p96xp07 <- round(mean(fst$Fst01),4)
p96xp17 <- round(mean(fst$Fst02),4)
p07xp17 <- round(mean(fst$Fst12),4)


fst_vec <- c(0,p96xp07,p96xp17,
            p96xp07,0,p07xp17,
            p96xp17,p07xp17,0)

fst_mat = matrix(fst_vec, nrow = 3, ncol = 3)
colnames(fst_mat) <- c("SS96","SS06","SS17")
rownames(fst_mat) <- c("SS96","SS06","SS17")
print(fst_mat)
upper_tri <- get_upper_tri(fst_mat)

# Melt the correlation matrix
melted_cormat <- melt(upper_tri, na.rm = TRUE)
# Heatmap
ggplot(data = melted_cormat, aes(Var2, Var1, fill = value))+
geom_tile(color = "white")+
scale_fill_gradient(low = blu, high = red, 
   limit = c(0,0.015), space = "Lab", 
  name="Fst") +
 theme_minimal()+ xlab("")+ylab("")+
theme(axis.text.x = element_text(angle = 0, vjust = 0, 
   size = 12, hjust = 0.5))+
  theme(axis.text.y = element_text(size = 12))+
coord_fixed()+
geom_text(aes(Var2, Var1, label = value), color = "black", size = 5)


```

## Fst and PBS by chromosome (3 population comparison)
### BC, WA, and CA (2017)
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.width= 10, fig.height= 7}

# A population’s population branch statistic value at a given locus corresponds to the magnitude of allele frequency change relative to its divergence from the other two populations.

#  We implemented the population branch statistic (PBS) [27] to detect genomic regions that diverged rapidly on the Cape bee lineage compared to other African populations.

pop_names = c("PWS91","PWS96","PWS07","PWS17","TB91","TB96","TB06","TB17","SS96","SS06","SS17","BC17","WA17","CA17")

pop0 <- "TB17"
pop1 <- "PWS17"
pop2 <- "SS17"

fsts <- list.files(path="C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/fst/sliding_window/", full.names=T)

fsts <- fsts[1]
#f <- fsts[1]

for (f in fsts){  

pops <- strsplit(f, "_")
pops <- c(pops[[1]][10], pops[[1]][11],gsub("\\.txt","",pops[[1]][12]))

fst <-read.delim(paste("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/fst/sliding_window/fst_pbs_50kb_win_10kb_step_folded_",pops[1],"_",pops[2],"_",pops[3],".txt", sep = ""))
#head(fst)
fst <- fst %>% mutate(chr = as.numeric(gsub("chr", "", chr)))
#fst$chr_num <- factor(fst$chr, levels = c(1:26))

# Fst
p <- ggplot(fst, aes(x = midPos, y = Fst01)) + 
     geom_point(size = 1, color = gry,alpha = 0.4, shape = 1)+
     theme_minimal()+
     theme(axis.text.x=element_blank())+
     ylab("Fst\n")+ xlab("")+ 
     ggtitle(paste(pops[1],"vs. ", pops[2]))+
     geom_smooth(method = "loess", se = FALSE, span = 1/10, color = blu)+
     facet_wrap(~chr, ncol = 9)
print(p)

p <- ggplot(fst, aes(x = midPos, y = Fst02)) + 
     geom_point(size = 1, color = gry,alpha = 0.4, shape = 1)+
     theme_minimal()+
     theme(axis.text.x=element_blank())+
     ylab("Fst\n")+ xlab("")+ 
     ggtitle(paste(pops[1],"vs. ", pops[3]))+
     geom_smooth(method = "loess", se = FALSE, span = 1/10, color = blu)+
     facet_wrap(~chr, ncol = 9)
print(p)

p <- ggplot(fst, aes(x = midPos, y = Fst12)) + 
     geom_point(size = 1, color = gry,alpha = 0.4, shape = 1)+
     theme_minimal()+
     theme(axis.text.x=element_blank())+
     ylab("Fst\n")+ xlab("")+ 
     ggtitle(paste(pops[2],"vs. ", pops[3]))+
     geom_smooth(method = "loess", se = FALSE, span = 1/10, color = blu)+
     facet_wrap(~chr, ncol = 9)
print(p)

#PBS
p <- ggplot(fst, aes(x = midPos, y = PBS0)) + 
     geom_point(size = 1, color = gry,alpha = 0.4, shape = 1)+
     theme_minimal()+
     theme(axis.text.x=element_blank())+
     ylab("PBS\n")+ xlab("")+ 
     ggtitle(paste(pops[1]))+
     geom_smooth(method = "loess", se = FALSE, span = 1/10, color = red)+
     facet_wrap(~chr, ncol = 9)
print(p)

p <- ggplot(fst, aes(x = midPos, y = PBS1)) + 
     geom_point(size = 1, color = gry,alpha = 0.4, shape = 1)+
     theme_minimal()+
     theme(axis.text.x=element_blank())+
     ylab("PBS\n")+ xlab("")+ 
     ggtitle(paste(pops[2]))+
     geom_smooth(method = "loess", se = FALSE, span = 1/10, color = red)+
     facet_wrap(~chr, ncol = 9)
print(p)

p <- ggplot(fst, aes(x = midPos, y = PBS2)) + 
     geom_point(size = 1, color = gry,alpha = 0.4, shape = 1)+
     theme_minimal()+
     theme(axis.text.x=element_blank())+
     ylab("PBS\n")+ xlab("")+ 
     ggtitle(paste(pops[3]))+
     geom_smooth(method = "loess", se = FALSE, span = 1/10, color = red)+
     facet_wrap(~chr, ncol = 9)
print(p)

}
```


## Fst and PBS by chromosome (3 population comparison)
### PWS, SS, and CA (2017)
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.width= 10, fig.height= 7}

# A population’s population branch statistic value at a given locus corresponds to the magnitude of allele frequency change relative to its divergence from the other two populations.

#  We implemented the population branch statistic (PBS) [27] to detect genomic regions that diverged rapidly on the Cape bee lineage compared to other African populations.

pop_names = c("PWS91","PWS96","PWS07","PWS17","TB91","TB96","TB06","TB17","SS96","SS06","SS17","BC17","WA17","CA17")

pop0 <- "TB17"
pop1 <- "PWS17"
pop2 <- "SS17"

fsts <- list.files(path="C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/fst/sliding_window/", full.names=T)

fsts <- fsts[3]

for (f in fsts){  

pops <- strsplit(f, "_")
pops <- c(pops[[1]][10], pops[[1]][11],gsub("\\.txt","",pops[[1]][12]))

fst <-read.delim(paste("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/fst/sliding_window/fst_pbs_50kb_win_10kb_step_folded_",pops[1],"_",pops[2],"_",pops[3],".txt", sep = ""))
#head(fst)
fst <- fst %>% mutate(chr = as.numeric(gsub("chr", "", chr)))
#fst$chr_num <- factor(fst$chr, levels = c(1:26))

# Fst
p <- ggplot(fst, aes(x = midPos, y = Fst01)) + 
     geom_point(size = 1, color = gry,alpha = 0.4, shape = 1)+
     theme_minimal()+
     theme(axis.text.x=element_blank())+
     ylab("Fst\n")+ xlab("")+ 
     ggtitle(paste(pops[1],"vs. ", pops[2]))+
     geom_smooth(method = "loess", se = FALSE, span = 1/10, color = blu)+
     facet_wrap(~chr, ncol = 9)
print(p)

p <- ggplot(fst, aes(x = midPos, y = Fst02)) + 
     geom_point(size = 1, color = gry,alpha = 0.4, shape = 1)+
     theme_minimal()+
     theme(axis.text.x=element_blank())+
     ylab("Fst\n")+ xlab("")+ 
     ggtitle(paste(pops[1],"vs. ", pops[3]))+
     geom_smooth(method = "loess", se = FALSE, span = 1/10, color = blu)+
     facet_wrap(~chr, ncol = 9)
print(p)

p <- ggplot(fst, aes(x = midPos, y = Fst12)) + 
     geom_point(size = 1, color = gry,alpha = 0.4, shape = 1)+
     theme_minimal()+
     theme(axis.text.x=element_blank())+
     ylab("Fst\n")+ xlab("")+ 
     ggtitle(paste(pops[2],"vs. ", pops[3]))+
     geom_smooth(method = "loess", se = FALSE, span = 1/10, color = blu)+
     facet_wrap(~chr, ncol = 9)
print(p)

#PBS
p <- ggplot(fst, aes(x = midPos, y = PBS0)) + 
     geom_point(size = 1, color = gry,alpha = 0.4, shape = 1)+
     theme_minimal()+
     theme(axis.text.x=element_blank())+
     ylab("PBS\n")+ xlab("")+ 
     ggtitle(paste(pops[1]))+
     geom_smooth(method = "loess", se = FALSE, span = 1/10, color = red)+
     facet_wrap(~chr, ncol = 9)
print(p)

p <- ggplot(fst, aes(x = midPos, y = PBS1)) + 
     geom_point(size = 1, color = gry,alpha = 0.4, shape = 1)+
     theme_minimal()+
     theme(axis.text.x=element_blank())+
     ylab("PBS\n")+ xlab("")+ 
     ggtitle(paste(pops[2]))+
     geom_smooth(method = "loess", se = FALSE, span = 1/10, color = red)+
     facet_wrap(~chr, ncol = 9)
print(p)

p <- ggplot(fst, aes(x = midPos, y = PBS2)) + 
     geom_point(size = 1, color = gry,alpha = 0.4, shape = 1)+
     theme_minimal()+
     theme(axis.text.x=element_blank())+
     ylab("PBS\n")+ xlab("")+ 
     ggtitle(paste(pops[3]))+
     geom_smooth(method = "loess", se = FALSE, span = 1/10, color = red)+
     facet_wrap(~chr, ncol = 9)
print(p)

}
```


## Fst and PBS by chromosome (3 population comparison)
### PWS temporal comparisons
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.width= 10, fig.height= 7}

# A population’s population branch statistic value at a given locus corresponds to the magnitude of allele frequency change relative to its divergence from the other two populations.

#  We implemented the population branch statistic (PBS) [27] to detect genomic regions that diverged rapidly on the Cape bee lineage compared to other African populations.

pop_names = c("PWS91","PWS96","PWS07","PWS17","TB91","TB96","TB06","TB17","SS96","SS06","SS17","BC17","WA17","CA17")

fsts <- list.files(path="C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/fst/sliding_window/", full.names=T)

fsts <- fsts[6]

for (f in fsts){  

pops <- strsplit(f, "_")
pops <- c(pops[[1]][10], pops[[1]][11],gsub("\\.txt","",pops[[1]][12]))

fst <-read.delim(paste("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/fst/sliding_window/fst_pbs_50kb_win_10kb_step_folded_",pops[1],"_",pops[2],"_",pops[3],".txt", sep = ""))
#head(fst)
fst <- fst %>% mutate(chr = as.numeric(gsub("chr", "", chr)))
#fst$chr_num <- factor(fst$chr, levels = c(1:26))

# Fst
p <- ggplot(fst, aes(x = midPos, y = Fst01)) + 
     geom_point(size = 1, color = gry,alpha = 0.4, shape = 1)+
     theme_minimal()+
     theme(axis.text.x=element_blank())+
     ylab("Fst\n")+ xlab("")+ 
     ggtitle(paste(pops[1],"vs. ", pops[2]))+
     geom_smooth(method = "loess", se = FALSE, span = 1/10, color = blu)+
     facet_wrap(~chr, ncol = 9)
print(p)

p <- ggplot(fst, aes(x = midPos, y = Fst02)) + 
     geom_point(size = 1, color = gry,alpha = 0.4, shape = 1)+
     theme_minimal()+
     theme(axis.text.x=element_blank())+
     ylab("Fst\n")+ xlab("")+ 
     ggtitle(paste(pops[1],"vs. ", pops[3]))+
     geom_smooth(method = "loess", se = FALSE, span = 1/10, color = blu)+
     facet_wrap(~chr, ncol = 9)
print(p)

p <- ggplot(fst, aes(x = midPos, y = Fst12)) + 
     geom_point(size = 1, color = gry,alpha = 0.4, shape = 1)+
     theme_minimal()+
     theme(axis.text.x=element_blank())+
     ylab("Fst\n")+ xlab("")+ 
     ggtitle(paste(pops[2],"vs. ", pops[3]))+
     geom_smooth(method = "loess", se = FALSE, span = 1/10, color = blu)+
     facet_wrap(~chr, ncol = 9)
print(p)

#PBS
p <- ggplot(fst, aes(x = midPos, y = PBS0)) + 
     geom_point(size = 1, color = gry,alpha = 0.4, shape = 1)+
     theme_minimal()+
     theme(axis.text.x=element_blank())+
     ylab("PBS\n")+ xlab("")+ 
     ggtitle(paste(pops[1]))+
     geom_smooth(method = "loess", se = FALSE, span = 1/10, color = red)+
     facet_wrap(~chr, ncol = 9)
print(p)

p <- ggplot(fst, aes(x = midPos, y = PBS1)) + 
     geom_point(size = 1, color = gry,alpha = 0.4, shape = 1)+
     theme_minimal()+
     theme(axis.text.x=element_blank())+
     ylab("PBS\n")+ xlab("")+ 
     ggtitle(paste(pops[2]))+
     geom_smooth(method = "loess", se = FALSE, span = 1/10, color = red)+
     facet_wrap(~chr, ncol = 9)
print(p)

p <- ggplot(fst, aes(x = midPos, y = PBS2)) + 
     geom_point(size = 1, color = gry,alpha = 0.4, shape = 1)+
     theme_minimal()+
     theme(axis.text.x=element_blank())+
     ylab("PBS\n")+ xlab("")+ 
     ggtitle(paste(pops[3]))+
     geom_smooth(method = "loess", se = FALSE, span = 1/10, color = red)+
     facet_wrap(~chr, ncol = 9)
print(p)

}
```


## Fst and PBS by chromosome (3 population comparison)
### TB temporal comparisons
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.width= 10, fig.height= 7}

# A population’s population branch statistic value at a given locus corresponds to the magnitude of allele frequency change relative to its divergence from the other two populations.

#  We implemented the population branch statistic (PBS) [27] to detect genomic regions that diverged rapidly on the Cape bee lineage compared to other African populations.

pop_names = c("PWS91","PWS96","PWS07","PWS17","TB91","TB96","TB06","TB17","SS96","SS06","SS17","BC17","WA17","CA17")

fsts <- list.files(path="C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/fst/sliding_window/", full.names=T)

fsts <- fsts[15]

for (f in fsts){  

pops <- strsplit(f, "_")
pops <- c(pops[[1]][10], pops[[1]][11],gsub("\\.txt","",pops[[1]][12]))

fst <-read.delim(paste("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/fst/sliding_window/fst_pbs_50kb_win_10kb_step_folded_",pops[1],"_",pops[2],"_",pops[3],".txt", sep = ""))
#head(fst)
fst <- fst %>% mutate(chr = as.numeric(gsub("chr", "", chr)))
#fst$chr_num <- factor(fst$chr, levels = c(1:26))

# Fst
p <- ggplot(fst, aes(x = midPos, y = Fst01)) + 
     geom_point(size = 1, color = gry,alpha = 0.4, shape = 1)+
     theme_minimal()+
     theme(axis.text.x=element_blank())+
     ylab("Fst\n")+ xlab("")+ 
     ggtitle(paste(pops[1],"vs. ", pops[2]))+
     geom_smooth(method = "loess", se = FALSE, span = 1/10, color = blu)+
     facet_wrap(~chr, ncol = 9)
print(p)

p <- ggplot(fst, aes(x = midPos, y = Fst02)) + 
     geom_point(size = 1, color = gry,alpha = 0.4, shape = 1)+
     theme_minimal()+
     theme(axis.text.x=element_blank())+
     ylab("Fst\n")+ xlab("")+ 
     ggtitle(paste(pops[1],"vs. ", pops[3]))+
     geom_smooth(method = "loess", se = FALSE, span = 1/10, color = blu)+
     facet_wrap(~chr, ncol = 9)
print(p)

p <- ggplot(fst, aes(x = midPos, y = Fst12)) + 
     geom_point(size = 1, color = gry,alpha = 0.4, shape = 1)+
     theme_minimal()+
     theme(axis.text.x=element_blank())+
     ylab("Fst\n")+ xlab("")+ 
     ggtitle(paste(pops[2],"vs. ", pops[3]))+
     geom_smooth(method = "loess", se = FALSE, span = 1/10, color = blu)+
     facet_wrap(~chr, ncol = 9)
print(p)

#PBS
p <- ggplot(fst, aes(x = midPos, y = PBS0)) + 
     geom_point(size = 1, color = gry,alpha = 0.4, shape = 1)+
     theme_minimal()+
     theme(axis.text.x=element_blank())+
     ylab("PBS\n")+ xlab("")+ 
     ggtitle(paste(pops[1]))+
     geom_smooth(method = "loess", se = FALSE, span = 1/10, color = red)+
     facet_wrap(~chr, ncol = 9)
print(p)

p <- ggplot(fst, aes(x = midPos, y = PBS1)) + 
     geom_point(size = 1, color = gry,alpha = 0.4, shape = 1)+
     theme_minimal()+
     theme(axis.text.x=element_blank())+
     ylab("PBS\n")+ xlab("")+ 
     ggtitle(paste(pops[2]))+
     geom_smooth(method = "loess", se = FALSE, span = 1/10, color = red)+
     facet_wrap(~chr, ncol = 9)
print(p)

p <- ggplot(fst, aes(x = midPos, y = PBS2)) + 
     geom_point(size = 1, color = gry,alpha = 0.4, shape = 1)+
     theme_minimal()+
     theme(axis.text.x=element_blank())+
     ylab("PBS\n")+ xlab("")+ 
     ggtitle(paste(pops[3]))+
     geom_smooth(method = "loess", se = FALSE, span = 1/10, color = red)+
     facet_wrap(~chr, ncol = 9)
print(p)

}
```


## Fst and PBS by chromosome (3 population comparison)
### TB, PWS, SS temporal comparisons (1996)
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.width= 10, fig.height= 7}

# A population’s population branch statistic value at a given locus corresponds to the magnitude of allele frequency change relative to its divergence from the other two populations.

#  We implemented the population branch statistic (PBS) [27] to detect genomic regions that diverged rapidly on the Cape bee lineage compared to other African populations.

pop_names = c("PWS91","PWS96","PWS07","PWS17","TB91","TB96","TB06","TB17","SS96","SS06","SS17","BC17","WA17","CA17")

fsts <- list.files(path="C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/fst/sliding_window/", full.names=T)

fsts <- fsts[17]

for (f in fsts){  

pops <- strsplit(f, "_")
pops <- c(pops[[1]][10], pops[[1]][11],gsub("\\.txt","",pops[[1]][12]))

fst <-read.delim(paste("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/fst/sliding_window/fst_pbs_50kb_win_10kb_step_folded_",pops[1],"_",pops[2],"_",pops[3],".txt", sep = ""))
#head(fst)
fst <- fst %>% mutate(chr = as.numeric(gsub("chr", "", chr)))
#fst$chr_num <- factor(fst$chr, levels = c(1:26))

# Fst
p <- ggplot(fst, aes(x = midPos, y = Fst01)) + 
     geom_point(size = 1, color = gry,alpha = 0.4, shape = 1)+
     theme_minimal()+
     theme(axis.text.x=element_blank())+
     ylab("Fst\n")+ xlab("")+ 
     ggtitle(paste(pops[1],"vs. ", pops[2]))+
     geom_smooth(method = "loess", se = FALSE, span = 1/10, color = blu)+
     facet_wrap(~chr, ncol = 9)
print(p)

p <- ggplot(fst, aes(x = midPos, y = Fst02)) + 
     geom_point(size = 1, color = gry,alpha = 0.4, shape = 1)+
     theme_minimal()+
     theme(axis.text.x=element_blank())+
     ylab("Fst\n")+ xlab("")+ 
     ggtitle(paste(pops[1],"vs. ", pops[3]))+
     geom_smooth(method = "loess", se = FALSE, span = 1/10, color = blu)+
     facet_wrap(~chr, ncol = 9)
print(p)

p <- ggplot(fst, aes(x = midPos, y = Fst12)) + 
     geom_point(size = 1, color = gry,alpha = 0.4, shape = 1)+
     theme_minimal()+
     theme(axis.text.x=element_blank())+
     ylab("Fst\n")+ xlab("")+ 
     ggtitle(paste(pops[2],"vs. ", pops[3]))+
     geom_smooth(method = "loess", se = FALSE, span = 1/10, color = blu)+
     facet_wrap(~chr, ncol = 9)
print(p)

#PBS
p <- ggplot(fst, aes(x = midPos, y = PBS0)) + 
     geom_point(size = 1, color = gry,alpha = 0.4, shape = 1)+
     theme_minimal()+
     theme(axis.text.x=element_blank())+
     ylab("PBS\n")+ xlab("")+ 
     ggtitle(paste(pops[1]))+
     geom_smooth(method = "loess", se = FALSE, span = 1/10, color = red)+
     facet_wrap(~chr, ncol = 9)
print(p)

p <- ggplot(fst, aes(x = midPos, y = PBS1)) + 
     geom_point(size = 1, color = gry,alpha = 0.4, shape = 1)+
     theme_minimal()+
     theme(axis.text.x=element_blank())+
     ylab("PBS\n")+ xlab("")+ 
     ggtitle(paste(pops[2]))+
     geom_smooth(method = "loess", se = FALSE, span = 1/10, color = red)+
     facet_wrap(~chr, ncol = 9)
print(p)

p <- ggplot(fst, aes(x = midPos, y = PBS2)) + 
     geom_point(size = 1, color = gry,alpha = 0.4, shape = 1)+
     theme_minimal()+
     theme(axis.text.x=element_blank())+
     ylab("PBS\n")+ xlab("")+ 
     ggtitle(paste(pops[3]))+
     geom_smooth(method = "loess", se = FALSE, span = 1/10, color = red)+
     facet_wrap(~chr, ncol = 9)
print(p)

}
```





## Fst outliers and GO enrichment for Chr 6

```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.width= 10, fig.height= 7}

gff <- read.delim("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/annotations/Clupea_harengus.Ch_v2.0.2.100.gff3",header = FALSE, sep = "\t",comment.char = '#')
#head(gff)
colnames(gff) <- c("seqnames","source","type","start","end","score","strand","phase","attributes")
genes <- gff[gff$type=="gene",]
genes <- genes %>% separate(attributes, c("at1","at2"), remove = FALSE,sep = "Name=") %>% separate(at2, c("gene_name","at3"), extra = "drop",sep = ";")
genes <- select(genes, -at1,-at3)
g <- genes[genes$strand == "\\.",]
#head(genes)
genes <- droplevels(genes)

genes <- genes %>% as_granges()
fsts <- list.files(path="C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/fst/sliding_window/", full.names=T)


fst <- read.delim("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/fst/sliding_window/fst_pbs_50kb_win_10kb_step_folded_PWS91_PWS96_PWS07.txt")
fst <- fst %>% mutate(chr = gsub("[A-z]", "", chr))
fst$start <- fst$midPos-9999
fst$end <- fst$midPos+9999
colnames(fst)[colnames(fst) == 'chr'] <- 'seqnames'

#quantile(fst$Fst01,.999)
fst_outlier <- fst[fst$Fst01 >= quantile(fst$Fst01,.99)[[1]],]


focal_chr <- fst[fst$seqnames == 6,]
focal_chr_outlier <- fst_outlier[fst_outlier$seqnames == 6,]


focal_chr_outlier <- focal_chr_outlier %>% as_granges()
focal_chr_genes <- join_overlap_intersect(focal_chr_outlier, genes) %>% as.data.frame()
print("54 genes (Danio orthologs) within 50kb Fst outlier window (genome-wide mean > 99%ile)")
unique(na.omit(focal_chr_genes$gene_name))
gene_loc <- unique(focal_chr_genes[c("gene_name","start","end")])

plot(focal_chr$midPos, focal_chr$Fst01, col = gry, xlab = "position", ylab = "Fst", main = "chr 6: PWS91 v. PWS96")
points(focal_chr_genes$midPos, focal_chr_genes$Fst01, ylim = c(0,1), col = blu)
# for (i in c(1:nrow(gene_loc))){
# rect(gene_loc$start[i], -0, gene_loc$end[i], 1, border = NA, col = col2alpha(gre,0.2))
# }
#write.table(as.data.frame(unique(na.omit(chr6$gene_name))), "C:/Users/jmcgirr/Desktop/genes_delete.txt", quote= FALSE, row.names = FALSE)



```

## GO enrichment for 54 genes
![](C:/Users/jmcgirr/Desktop/Capture.PNG)


## Fst outliers and GO enrichment for CA v PWS chr 7

```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.width= 10, fig.height= 7}

gff <- read.delim("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/annotations/Clupea_harengus.Ch_v2.0.2.100.gff3",header = FALSE, sep = "\t",comment.char = '#')
#head(gff)
colnames(gff) <- c("seqnames","source","type","start","end","score","strand","phase","attributes")
genes <- gff[gff$type=="gene",]
genes <- genes %>% separate(attributes, c("at1","at2"), remove = FALSE,sep = "Name=") %>% separate(at2, c("gene_name","at3"), extra = "drop",sep = ";")
genes <- select(genes, -at1,-at3)
g <- genes[genes$strand == "\\.",]
#head(genes)
genes <- droplevels(genes)

genes <- genes %>% as_granges()
fsts <- list.files(path="C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/fst/sliding_window/", full.names=T)


fst <- read.delim("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/fst/sliding_window/fst_pbs_50kb_win_10kb_step_folded_BC17_WA17_CA17.txt")
fst <- fst %>% mutate(chr = gsub("[A-z]", "", chr))
fst$start <- fst$midPos-9999
fst$end <- fst$midPos+9999
colnames(fst)[colnames(fst) == 'chr'] <- 'seqnames'

#quantile(fst$Fst01,.999)
fst_outlier <- fst[fst$Fst02 >= quantile(fst$Fst02,.99)[[1]],]


focal_chr <- fst[fst$seqnames == 7,]
focal_chr_outlier <- fst_outlier[fst_outlier$seqnames == 7,]


focal_chr_outlier <- focal_chr_outlier %>% as_granges()
focal_chr_genes <- join_overlap_intersect(focal_chr_outlier, genes) %>% as.data.frame()
print("66 genes (Danio orthologs) within 50kb Fst outlier window (genome-wide mean > 99%ile)")
unique(na.omit(focal_chr_genes$gene_name))
gene_loc <- unique(focal_chr_genes[c("gene_name","start","end")])

plot(focal_chr$midPos, focal_chr$Fst02, col = gry, xlab = "position", ylab = "Fst", main = "chr 7: BC17 v. CA17")
points(focal_chr_genes$midPos, focal_chr_genes$Fst02, ylim = c(0,1), col = blu)
# for (i in c(1:nrow(gene_loc))){
# rect(gene_loc$start[i], -0, gene_loc$end[i], 1, border = NA, col = col2alpha(gre,0.2))
# }
#write.table(as.data.frame(unique(na.omit(focal_chr_genes$gene_name))), "C:/Users/jmcgirr/Desktop/genes_delete.txt", quote= FALSE, row.names = FALSE)


```

## GO enrichment for 66 genes
![](C:/Users/jmcgirr/Desktop/Capture1.PNG)

