{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import io\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "\n",
    "def sbatch_header(job,mem,tasks,hours):\n",
    "    #sbatch submission script header\n",
    "    script = 'script_' + job + '.sh'\n",
    "    outfile = io.open(script,'w', newline='\\n')    \n",
    "    outfile.write('#!/bin/bash\\n\\n#SBATCH --job-name='+job+'\\n')\n",
    "    outfile.write('#SBATCH --mem='+mem+'G \\n')\n",
    "    outfile.write('#SBATCH --ntasks='+tasks+' \\n')\n",
    "    outfile.write('#SBATCH -e '+job+'_%A_%a.err \\n')\n",
    "    outfile.write('#SBATCH --time='+hours+':00:00  \\n')\n",
    "    outfile.write('#SBATCH --mail-user=jamcgirr@ucdavis.edu ##email you when job starts,ends,etc\\n#SBATCH --mail-type=ALL\\n')\n",
    "    outfile.write('#SBATCH -p high \\n\\n')\n",
    "    outfile.close()\n",
    "    \n",
    "def sbatch_header_loop(job,mem,tasks,hours,infile):\n",
    "    #sbatch submission script header\n",
    "    script = 'script_' + infile + job + '.sh'\n",
    "    outfile = io.open(script,'w', newline='\\n') \n",
    "    jobname= infile + job   \n",
    "    outfile.write('#!/bin/bash\\n\\n#SBATCH --job-name='+jobname+'\\n')\n",
    "    outfile.write('#SBATCH --mem='+mem+'G \\n')\n",
    "    outfile.write('#SBATCH --ntasks='+tasks+' \\n')\n",
    "    outfile.write('#SBATCH -e '+jobname+'_%A_%a.err \\n')\n",
    "    outfile.write('#SBATCH --time='+hours+':00:00 \\n')\n",
    "    outfile.write('#SBATCH --mail-user=jamcgirr@ucdavis.edu ##email you when job starts,ends,etc\\n#SBATCH --mail-type=ALL\\n')\n",
    "    outfile.write('#SBATCH -p high \\n\\n')\n",
    "    outfile.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Fst persite PWS96 + PWS07 + PWS17 vs. PWS 91\n",
    "# Fst persite PWS96 + PWS07 + PWS17 vs. NEPac (SS96,SS06,SS17,BC17,WA17)\n",
    "# Fst persite PWS17 vs. NEPac (SS17,BC17,WA17)\n",
    "\n",
    "job_name = 'fst_pops_PWS96_07_17_v_PWS91'\n",
    "script_dir = '/home/jamcgirr/ph/scripts/angsd/SFS/fst_fixed_PWS/'\n",
    "fst_dir = '/home/jamcgirr/ph/data/angsd/SFS/fst_fixed_PWS/'\n",
    "\n",
    "sbatch_header(job_name,'16','4','144')\n",
    "script = 'script_' + job_name + '.sh'\n",
    "o = io.open(script,'a+', newline='\\n')\n",
    "\n",
    "o.write('module load samtools \\n')\n",
    "o.write('module load bcftools \\n')\n",
    "o.write('cat /home/jamcgirr/ph/data/angsd/SFS/thetas/PWS96_bams_p_1_5_rm.txt /home/jamcgirr/ph/data/angsd/SFS/thetas/PWS07_bams_p_1_5_rm.txt /home/jamcgirr/ph/data/angsd/SFS/thetas/PWS17_bams_p_1_5_rm.txt > '+fst_dir+'pops_PWS96_07_17.txt \\n')\n",
    "o.write('sed -i \\'s/\\/home\\/eoziolor\\/phpopg\\/data\\/align\\///g\\' '+fst_dir+'pops_PWS96_07_17.txt \\n')\n",
    "o.write('sed -i \\'s/.bam//g\\' '+fst_dir+'pops_PWS96_07_17.txt \\n\\n')\n",
    "\n",
    "o.write('cp /home/jamcgirr/ph/data/angsd/SFS/thetas/PWS91_bams_p_1_5_rm.txt '+fst_dir+'pops_PWS91.txt \\n')\n",
    "o.write('sed -i \\'s/\\/home\\/eoziolor\\/phpopg\\/data\\/align\\///g\\' '+fst_dir+'pops_PWS91.txt \\n')\n",
    "o.write('sed -i \\'s/.bam//g\\' '+fst_dir+'pops_PWS91.txt \\n\\n')\n",
    "\n",
    "#subset pop vcfs\n",
    "o.write('bcftools view -S '+fst_dir+'pops_PWS96_07_17.txt -Ov /home/jamcgirr/ph/data/vcfs/vince/ph_filtered_snps_minDP600_maxDP2000_maf0.05_minQ20_minMQ30_maxmiss0.5_outliers_rm.vcf.gz --threads 4 > '+fst_dir+'pops_PWS96_07_17.vcf \\n')\n",
    "o.write('bcftools view -S '+fst_dir+'pops_PWS91.txt -Ov /home/jamcgirr/ph/data/vcfs/vince/ph_filtered_snps_minDP600_maxDP2000_maf0.05_minQ20_minMQ30_maxmiss0.5_outliers_rm.vcf.gz --threads 4 > '+fst_dir+'pops_PWS91.vcf \\n\\n')\n",
    "\n",
    "# make saf from vcfs\n",
    "o.write('/home/jamcgirr/apps/angsd_sep_20/angsd/angsd -doSaf 1 -vcf-pl '+fst_dir+'pops_PWS96_07_17.vcf -out '+fst_dir+'pops_PWS96_07_17 -anc /home/jamcgirr/ph/data/c_harengus/c.harengus.fa \\n')\n",
    "o.write('/home/jamcgirr/apps/angsd_sep_20/angsd/angsd -doSaf 1 -vcf-pl '+fst_dir+'pops_PWS91.vcf -out '+fst_dir+'pops_PWS91 -anc /home/jamcgirr/ph/data/c_harengus/c.harengus.fa \\n\\n')\n",
    "o.write('rm '+fst_dir+'pops_PWS96_07_17.vcf')\n",
    "o.write('rm '+fst_dir+'pops_PWS91.vcf')\n",
    "\n",
    "# make folded sfs\n",
    "o.write('/home/jamcgirr/apps/angsd_sep_20/angsd/misc/realSFS '+fst_dir+'pops_PWS96_07_17.saf.idx -P 4 -fold 1 > '+fst_dir+'pops_PWS96_07_17.sfs \\n')\n",
    "o.write('/home/jamcgirr/apps/angsd_sep_20/angsd/misc/realSFS '+fst_dir+'pops_PWS91.saf.idx -P 4 -fold 1 > '+fst_dir+'pops_PWS91.sfs \\n\\n')\n",
    "\n",
    "# make 2dsfs\n",
    "o.write('/home/jamcgirr/apps/angsd_sep_20/angsd/misc/realSFS '+fst_dir+'pops_PWS96_07_17.saf.idx '+fst_dir+'pops_PWS91.saf.idx -fold 1 -P 4 > '+fst_dir+'pops_PWS96_07_17_v_PWS91.sfs \\n\\n')\n",
    "\n",
    "# estimate fst\n",
    "o.write('/home/jamcgirr/apps/angsd_sep_20/angsd/misc/realSFS fst index '+fst_dir+'pops_PWS96_07_17.saf.idx '+fst_dir+'pops_PWS91.saf.idx -sfs '+fst_dir+'pops_PWS96_07_17_v_PWS91.sfs -fold 1 -fstout '+fst_dir+'pops_PWS96_07_17_v_PWS91 \\n')\n",
    "o.write('/home/jamcgirr/apps/angsd_sep_20/angsd/misc/realSFS fst stats2 '+fst_dir+'pops_PWS96_07_17_v_PWS91.fst.idx -win 50000 -step 10000 > '+fst_dir+'fst_50kb_win_10kb_step_folded_pops_PWS96_07_17_v_PWS91.txt \\n')\n",
    "o.write('/home/jamcgirr/apps/angsd_sep_20/angsd/misc/realSFS fst print '+fst_dir+'pops_PWS96_07_17_v_PWS91.fst.idx > '+fst_dir+'fst_persite_folded_pops_PWS96_07_17_v_PWS91.txt')\n",
    "    \n",
    "o.write('\\n\\n#run: sbatch '+script)\n",
    "o.close()\n",
    "    \n",
    "    \n",
    "    \n",
    "###################################################################\n",
    "job_name = 'fst_pops_PWS96_07_17_v_NEPac'\n",
    "\n",
    "sbatch_header(job_name,'16','4','144')\n",
    "script = 'script_' + job_name + '.sh'\n",
    "o = io.open(script,'a+', newline='\\n')\n",
    "\n",
    "o.write('module load samtools \\n')\n",
    "o.write('module load bcftools \\n')  \n",
    "o.write('cat /home/jamcgirr/ph/data/angsd/SFS/thetas/SS96_bams_p_1_5_rm.txt /home/jamcgirr/ph/data/angsd/SFS/thetas/SS06_bams_p_1_5_rm.txt /home/jamcgirr/ph/data/angsd/SFS/thetas/SS17_bams_p_1_5_rm.txt /home/jamcgirr/ph/data/angsd/SFS/thetas/BC17_bams_p_1_5_rm.txt /home/jamcgirr/ph/data/angsd/SFS/thetas/WA17_bams_p_1_5_rm.txt > '+fst_dir+'pops_NEPac.txt \\n')\n",
    "o.write('sed -i \\'s/\\/home\\/eoziolor\\/phpopg\\/data\\/align\\///g\\' '+fst_dir+'pops_NEPac.txt \\n')\n",
    "o.write('sed -i \\'s/.bam//g\\' '+fst_dir+'pops_NEPac.txt \\n\\n')\n",
    "\n",
    "o.write('# cat /home/jamcgirr/ph/data/angsd/SFS/thetas/PWS96_bams_p_1_5_rm.txt /home/jamcgirr/ph/data/angsd/SFS/thetas/PWS07_bams_p_1_5_rm.txt /home/jamcgirr/ph/data/angsd/SFS/thetas/PWS17_bams_p_1_5_rm.txt > '+fst_dir+'pops_PWS96_07_17.txt \\n')\n",
    "o.write('# sed -i \\'s/\\/home\\/eoziolor\\/phpopg\\/data\\/align\\///g\\' '+fst_dir+'pops_PWS96_07_17.txt \\n')\n",
    "o.write('# sed -i \\'s/.bam//g\\' '+fst_dir+'pops_PWS96_07_17.txt \\n\\n')\n",
    "\n",
    "#subset pop vcfs\n",
    "o.write('#bcftools view -S '+fst_dir+'pops_PWS96_07_17.txt -Ov /home/jamcgirr/ph/data/vcfs/vince/ph_filtered_snps_minDP600_maxDP2000_maf0.05_minQ20_minMQ30_maxmiss0.5_outliers_rm.vcf.gz --threads 4 > '+fst_dir+'pops_PWS96_07_17.vcf \\n')\n",
    "o.write('bcftools view -S '+fst_dir+'pops_NEPac.txt -Ov /home/jamcgirr/ph/data/vcfs/vince/ph_filtered_snps_minDP600_maxDP2000_maf0.05_minQ20_minMQ30_maxmiss0.5_outliers_rm.vcf.gz --threads 4 > '+fst_dir+'pops_NEPac.vcf \\n\\n')\n",
    "\n",
    "# make saf from vcfs\n",
    "o.write('#/home/jamcgirr/apps/angsd_sep_20/angsd/angsd -doSaf 1 -vcf-pl '+fst_dir+'pops_PWS96_07_17.vcf -out '+fst_dir+'pops_PWS96_07_17 -anc /home/jamcgirr/ph/data/c_harengus/c.harengus.fa \\n')\n",
    "o.write('/home/jamcgirr/apps/angsd_sep_20/angsd/angsd -doSaf 1 -vcf-pl '+fst_dir+'pops_NEPac.vcf -out '+fst_dir+'pops_NEPac -anc /home/jamcgirr/ph/data/c_harengus/c.harengus.fa \\n\\n')\n",
    "o.write('#rm '+fst_dir+'pops_PWS96_07_17.vcf')\n",
    "o.write('rm '+fst_dir+'pops_NEPac.vcf')\n",
    "\n",
    "# make folded sfs\n",
    "o.write('#/home/jamcgirr/apps/angsd_sep_20/angsd/misc/realSFS '+fst_dir+'pops_PWS96_07_17.saf.idx -P 4 -fold 1 > '+fst_dir+'pops_PWS96_07_17.sfs \\n')\n",
    "o.write('/home/jamcgirr/apps/angsd_sep_20/angsd/misc/realSFS '+fst_dir+'pops_NEPac.saf.idx -P 4 -fold 1 > '+fst_dir+'pops_NEPac.sfs \\n\\n')\n",
    "\n",
    "# make 2dsfs\n",
    "o.write('/home/jamcgirr/apps/angsd_sep_20/angsd/misc/realSFS '+fst_dir+'pops_PWS96_07_17.saf.idx '+fst_dir+'pops_NEPac.saf.idx -fold 1 -P 4 > '+fst_dir+'pops_PWS96_07_17_v_NEPac.sfs \\n\\n')\n",
    "\n",
    "# estimate fst\n",
    "o.write('/home/jamcgirr/apps/angsd_sep_20/angsd/misc/realSFS fst index '+fst_dir+'pops_PWS96_07_17.saf.idx '+fst_dir+'pops_NEPac.saf.idx -sfs '+fst_dir+'pops_PWS96_07_17_v_NEPac.sfs -fold 1 -fstout '+fst_dir+'pops_PWS96_07_17_v_NEPac \\n')\n",
    "o.write('/home/jamcgirr/apps/angsd_sep_20/angsd/misc/realSFS fst stats2 '+fst_dir+'pops_PWS96_07_17_v_NEPac.fst.idx -win 50000 -step 10000 > '+fst_dir+'fst_50kb_win_10kb_step_folded_pops_PWS96_07_17_v_NEPac.txt \\n')\n",
    "o.write('/home/jamcgirr/apps/angsd_sep_20/angsd/misc/realSFS fst print '+fst_dir+'pops_PWS96_07_17_v_NEPac.fst.idx > '+fst_dir+'fst_persite_folded_pops_PWS96_07_17_v_NEPac.txt')\n",
    "    \n",
    "o.write('\\n\\n#run: sbatch '+script)\n",
    "o.close()\n",
    "   \n",
    "\n",
    "    \n",
    "###################################################################\n",
    "job_name = 'fst_pops_PWS17_v_NEPac17'\n",
    "\n",
    "sbatch_header(job_name,'16','4','144')\n",
    "script = 'script_' + job_name + '.sh'\n",
    "o = io.open(script,'a+', newline='\\n')\n",
    "\n",
    "o.write('module load samtools \\n')\n",
    "o.write('module load bcftools \\n')  \n",
    "o.write('cat /home/jamcgirr/ph/data/angsd/SFS/thetas/SS17_bams_p_1_5_rm.txt /home/jamcgirr/ph/data/angsd/SFS/thetas/BC17_bams_p_1_5_rm.txt /home/jamcgirr/ph/data/angsd/SFS/thetas/WA17_bams_p_1_5_rm.txt  > '+fst_dir+'pops_NEPac17.txt \\n')\n",
    "o.write('sed -i \\'s/\\/home\\/eoziolor\\/phpopg\\/data\\/align\\///g\\' '+fst_dir+'pops_NEPac17.txt \\n')\n",
    "o.write('sed -i \\'s/.bam//g\\' '+fst_dir+'pops_NEPac17.txt \\n\\n')\n",
    "\n",
    "o.write('cp /home/jamcgirr/ph/data/angsd/SFS/thetas/PWS17_bams_p_1_5_rm.txt '+fst_dir+'pops_PWS17.txt \\n')\n",
    "o.write('sed -i \\'s/\\/home\\/eoziolor\\/phpopg\\/data\\/align\\///g\\' '+fst_dir+'pops_PWS17.txt \\n')\n",
    "o.write('sed -i \\'s/.bam//g\\' '+fst_dir+'pops_PWS17.txt \\n\\n')\n",
    "\n",
    "#subset pop vcfs\n",
    "o.write('bcftools view -S '+fst_dir+'pops_PWS17.txt -Ov /home/jamcgirr/ph/data/vcfs/vince/ph_filtered_snps_minDP600_maxDP2000_maf0.05_minQ20_minMQ30_maxmiss0.5_outliers_rm.vcf.gz --threads 4 > '+fst_dir+'pops_PWS17.vcf \\n')\n",
    "o.write('bcftools view -S '+fst_dir+'pops_NEPac17.txt -Ov /home/jamcgirr/ph/data/vcfs/vince/ph_filtered_snps_minDP600_maxDP2000_maf0.05_minQ20_minMQ30_maxmiss0.5_outliers_rm.vcf.gz --threads 4 > '+fst_dir+'pops_NEPac17.vcf \\n\\n')\n",
    "\n",
    "# make saf from vcfs\n",
    "o.write('/home/jamcgirr/apps/angsd_sep_20/angsd/angsd -doSaf 1 -vcf-pl '+fst_dir+'pops_PWS17.vcf -out '+fst_dir+'pops_PWS17 -anc /home/jamcgirr/ph/data/c_harengus/c.harengus.fa \\n')\n",
    "o.write('/home/jamcgirr/apps/angsd_sep_20/angsd/angsd -doSaf 1 -vcf-pl '+fst_dir+'pops_NEPac17.vcf -out '+fst_dir+'pops_NEPac17 -anc /home/jamcgirr/ph/data/c_harengus/c.harengus.fa \\n\\n')\n",
    "o.write('rm '+fst_dir+'pops_PWS17.vcf')\n",
    "o.write('rm '+fst_dir+'pops_NEPac17.vcf')\n",
    "\n",
    "# make folded sfs\n",
    "o.write('/home/jamcgirr/apps/angsd_sep_20/angsd/misc/realSFS '+fst_dir+'pops_PWS17.saf.idx -P 4 -fold 1 > '+fst_dir+'pops_PWS17.sfs \\n')\n",
    "o.write('/home/jamcgirr/apps/angsd_sep_20/angsd/misc/realSFS '+fst_dir+'pops_NEPac17.saf.idx -P 4 -fold 1 > '+fst_dir+'pops_NEPac17.sfs \\n\\n')\n",
    "\n",
    "# make 2dsfs\n",
    "o.write('/home/jamcgirr/apps/angsd_sep_20/angsd/misc/realSFS '+fst_dir+'pops_PWS17.saf.idx '+fst_dir+'pops_NEPac17.saf.idx -fold 1 -P 4 > '+fst_dir+'pops_PWS17_v_NEPac17.sfs \\n\\n')\n",
    "\n",
    "# estimate fst\n",
    "o.write('/home/jamcgirr/apps/angsd_sep_20/angsd/misc/realSFS fst index  '+fst_dir+'pops_PWS17.saf.idx '+fst_dir+'pops_NEPac17.saf.idx -sfs '+fst_dir+'pops_PWS17_v_NEPac17.sfs -fold 1 -fstout '+fst_dir+'pops_PWS17_v_NEPac17 \\n')\n",
    "o.write('/home/jamcgirr/apps/angsd_sep_20/angsd/misc/realSFS fst stats2 '+fst_dir+'pops_PWS17_v_NEPac17.fst.idx -win 50000 -step 10000 > '+fst_dir+'fst_50kb_win_10kb_step_folded_pops_PWS17_v_NEPac17.txt \\n')\n",
    "o.write('/home/jamcgirr/apps/angsd_sep_20/angsd/misc/realSFS fst print '+fst_dir+'pops_PWS17_v_NEPac17.fst.idx > '+fst_dir+'fst_persite_folded_pops_PWS17_v_NEPac17.txt')\n",
    "    \n",
    "o.write('\\n\\n#run: sbatch '+script)\n",
    "o.close()\n",
    "    \n",
    "#EST run time ~ 3 days\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# output SNP table\n",
    "\n",
    "job_name = 'snp_table'\n",
    "\n",
    "sbatch_header(job_name,'8','4','24')\n",
    "script = 'script_' + job_name + '.sh'\n",
    "o = io.open(script,'a+', newline='\\n')\n",
    "\n",
    "o.write('module load R \\n')\n",
    "o.write('module load maven \\n')\n",
    "o.write('module load java \\n')\n",
    "o.write('module load GATK/4.1.4.1 \\n\\n')\n",
    "\n",
    "o.write('gatk VariantsToTable -V /home/jamcgirr/ph/data/vcfs/vince/ph_filtered_snps_minDP600_maxDP2000_maf0.05_minQ20_minMQ30_maxmiss0.5_outliers_rm.vcf.gz -F CHROM -F POS -F REF -F ALT -O /home/jamcgirr/ph/data/vcfs/ph_filtered_snps_minDP600_maxDP2000_maf0.05_minQ20_minMQ30_maxmiss0.5_outliers_rm_table.txt \\n\\n')\n",
    "\n",
    "#sed 's/chr//g' /home/jamcgirr/ph/data/vcfs/ph_filtered_snps_minDP600_maxDP2000_maf0.05_minQ20_minMQ30_maxmiss0.5_outliers_rm_table.txt > /home/jamcgirr/ph/data/LoF/snp_table_VEP.txt\n",
    "#sed -i 's/\\t/:/g' /home/jamcgirr/ph/data/LoF/snp_table_VEP.txt\n",
    "#sed -i '1d' /home/jamcgirr/ph/data/LoF/snp_table_VEP.txt\n",
    "\n",
    "o.write('\\n\\n#run: sbatch '+script)\n",
    "o.close()\n",
    "#EST run time ~ 1 hr"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "# parse SNP table with R script (LoF.rmd)\n",
    "# run variant effect predictor (https://uswest.ensembl.org/info/docs/tools/vep/script/index.html)\n",
    "\n",
    "job_name = 'vep'\n",
    "\n",
    "sbatch_header(job_name,'8','4','24')\n",
    "script = 'script_' + job_name + '.sh'\n",
    "o = io.open(script,'a+', newline='\\n')\n",
    "\n",
    "o.write('/home/jamcgirr/ph/data/LoF/ensembl-vep/vep --input_file /home/jamcgirr/ph/data/LoF/snp_table_VEP_standard_format.txt --output_file /home/jamcgirr/ph/data/LoF/test_output.txt --cache --species clupea_harengus \\n')\n",
    "      \n",
    "o.write('\\n\\n#run: sbatch '+script)\n",
    "o.close()       \n",
    "        \n",
    "#EST run time ~ 1 hr 20 min      "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "# create maf file using -ref to polarize.\n",
    "# this will allow us to assign frequencies to REF and ALT rather than major and minor\n",
    "# the latter will vary at some sites by population\n",
    "\n",
    "job_name = '_maf_for_vep'\n",
    "vcf_name = 'ph_filtered_snps_minDP600_maxDP2000_maf0.05_minQ20_minMQ30_maxmiss0.5'\n",
    "infiles = [\"BC17\",\"CA17\",\"PWS07\",\"PWS17\",\"PWS91\",\"PWS96\",\"SS06\",\"SS17\",\"SS96\",\"TB06\",\"TB17\",\"TB91\",\"TB96\",\"WA17\"]\n",
    "\n",
    "for infile in infiles:\n",
    "    script = 'script_' + infile + job_name + '.sh'\n",
    "    sbatch_header_loop(job_name,'8','4','1', infile)\n",
    "    o = io.open(script,'a+', newline='\\n')\n",
    "    \n",
    "    \n",
    "    # make mafs from vcf\n",
    "    o.write('/home/jamcgirr/apps/angsd_sep_20/angsd/angsd -out /home/jamcgirr/ph/data/LoF/mafs_REF_ALT/'+infile+'_vcf -fai /home/jamcgirr/ph/data/c_harengus/c.harengus.fa.fai -ref /home/jamcgirr/ph/data/c_harengus/c.harengus.fa -doMajorMinor 4 -doMaf 1 -vcf-pl /home/jamcgirr/ph/data/vcfs/split_pops/population_'+infile+'_ph_filtered_snps_minDP600_maxDP2000_maf0.05_minQ20_minMQ30_maxmiss0.5_plates_1_5_rm.vcf -P 4 \\n')\n",
    "    o.write('gzip -d /home/jamcgirr/ph/data/LoF/mafs_REF_ALT/'+infile+'_vcf.mafs.gz \\n')\n",
    "    \n",
    "    \n",
    "    o.write('\\n\\n#command to run: sbatch '+script)\n",
    "    o.close()\n",
    "\n",
    "# EST run time ~ 5 minutes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>2045</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>2079</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>2085</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>2583</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>2674</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>1</td>\n",
       "      <td>7452</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>1</td>\n",
       "      <td>7496</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>1</td>\n",
       "      <td>8834</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>1</td>\n",
       "      <td>8964</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>1</td>\n",
       "      <td>9009</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   0     1  2  3\n",
       "0  1  2045  G  A\n",
       "1  1  2079  T  C\n",
       "2  1  2085  T  C\n",
       "3  1  2583  C  T\n",
       "4  1  2674  G  A\n",
       "5  1  7452  G  T\n",
       "6  1  7496  G  C\n",
       "7  1  8834  A  T\n",
       "8  1  8964  T  C\n",
       "9  1  9009  G  C"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "snps = pd.read_csv('C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/fst/fixed_pws_persite/snp_table_VEP.txt', sep = ':', header = None)\n",
    "snps.head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromo</th>\n",
       "      <th>position</th>\n",
       "      <th>major</th>\n",
       "      <th>minor</th>\n",
       "      <th>knownEM</th>\n",
       "      <th>nInd</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>2079</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.022064</td>\n",
       "      <td>36</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>2085</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>35</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>2583</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.176155</td>\n",
       "      <td>34</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>7452</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>0.182796</td>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>7496</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>0.168173</td>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr1</td>\n",
       "      <td>9070</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>0.029781</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr1</td>\n",
       "      <td>9098</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.032971</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr1</td>\n",
       "      <td>13381</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr1</td>\n",
       "      <td>13904</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>0.504820</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14280</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>0.631408</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  chromo  position major minor   knownEM  nInd\n",
       "0   chr1      2079     T     C  0.022064    36\n",
       "1   chr1      2085     T     C  0.000005    35\n",
       "2   chr1      2583     C     T  0.176155    34\n",
       "3   chr1      7452     G     T  0.182796    37\n",
       "4   chr1      7496     G     C  0.168173    37\n",
       "5   chr1      9070     T     A  0.029781    25\n",
       "6   chr1      9098     A     G  0.032971    26\n",
       "7   chr1     13381     A     C  0.000003    32\n",
       "8   chr1     13904     C     A  0.504820    26\n",
       "9   chr1     14280     C     A  0.631408    30"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "maf = pd.read_csv('C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/dxy/mafs/PWS91_vcf.mafs', sep = '\\t', header = 0)\n",
    "maf.head(10)\n",
    "maf = maf.rename(columns={0:'chrom', 3:'start', 4:'stop'})\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "ename": "ParserError",
     "evalue": "Error tokenizing data. C error: Expected 1 fields in line 31, saw 14\n",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mParserError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[1;32m<ipython-input-8-ff4335c7030e>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m\u001b[0m\n\u001b[1;32m----> 1\u001b[1;33m \u001b[0mvep\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mpd\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mread_csv\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m'C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/fst/fixed_pws_persite/VEP_output_all_snps.txt'\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0msep\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;34m'\\t'\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m      2\u001b[0m \u001b[0mvep\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mhead\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;36m10\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\Anaconda3\\lib\\site-packages\\pandas\\io\\parsers.py\u001b[0m in \u001b[0;36mparser_f\u001b[1;34m(filepath_or_buffer, sep, delimiter, header, names, index_col, usecols, squeeze, prefix, mangle_dupe_cols, dtype, engine, converters, true_values, false_values, skipinitialspace, skiprows, skipfooter, nrows, na_values, keep_default_na, na_filter, verbose, skip_blank_lines, parse_dates, infer_datetime_format, keep_date_col, date_parser, dayfirst, iterator, chunksize, compression, thousands, decimal, lineterminator, quotechar, quoting, doublequote, escapechar, comment, encoding, dialect, tupleize_cols, error_bad_lines, warn_bad_lines, delim_whitespace, low_memory, memory_map, float_precision)\u001b[0m\n\u001b[0;32m    700\u001b[0m                     skip_blank_lines=skip_blank_lines)\n\u001b[0;32m    701\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 702\u001b[1;33m         \u001b[1;32mreturn\u001b[0m \u001b[0m_read\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mfilepath_or_buffer\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mkwds\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    703\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    704\u001b[0m     \u001b[0mparser_f\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m__name__\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mname\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\Anaconda3\\lib\\site-packages\\pandas\\io\\parsers.py\u001b[0m in \u001b[0;36m_read\u001b[1;34m(filepath_or_buffer, kwds)\u001b[0m\n\u001b[0;32m    433\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    434\u001b[0m     \u001b[1;32mtry\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 435\u001b[1;33m         \u001b[0mdata\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mparser\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mread\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mnrows\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    436\u001b[0m     \u001b[1;32mfinally\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    437\u001b[0m         \u001b[0mparser\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mclose\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\Anaconda3\\lib\\site-packages\\pandas\\io\\parsers.py\u001b[0m in \u001b[0;36mread\u001b[1;34m(self, nrows)\u001b[0m\n\u001b[0;32m   1137\u001b[0m     \u001b[1;32mdef\u001b[0m \u001b[0mread\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mnrows\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;32mNone\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1138\u001b[0m         \u001b[0mnrows\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0m_validate_integer\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m'nrows'\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mnrows\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m-> 1139\u001b[1;33m         \u001b[0mret\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_engine\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mread\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mnrows\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m   1140\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1141\u001b[0m         \u001b[1;31m# May alter columns / col_dict\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\Anaconda3\\lib\\site-packages\\pandas\\io\\parsers.py\u001b[0m in \u001b[0;36mread\u001b[1;34m(self, nrows)\u001b[0m\n\u001b[0;32m   1993\u001b[0m     \u001b[1;32mdef\u001b[0m \u001b[0mread\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mnrows\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;32mNone\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1994\u001b[0m         \u001b[1;32mtry\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m-> 1995\u001b[1;33m             \u001b[0mdata\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_reader\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mread\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mnrows\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m   1996\u001b[0m         \u001b[1;32mexcept\u001b[0m \u001b[0mStopIteration\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1997\u001b[0m             \u001b[1;32mif\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_first_chunk\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32mpandas/_libs/parsers.pyx\u001b[0m in \u001b[0;36mpandas._libs.parsers.TextReader.read\u001b[1;34m()\u001b[0m\n",
      "\u001b[1;32mpandas/_libs/parsers.pyx\u001b[0m in \u001b[0;36mpandas._libs.parsers.TextReader._read_low_memory\u001b[1;34m()\u001b[0m\n",
      "\u001b[1;32mpandas/_libs/parsers.pyx\u001b[0m in \u001b[0;36mpandas._libs.parsers.TextReader._read_rows\u001b[1;34m()\u001b[0m\n",
      "\u001b[1;32mpandas/_libs/parsers.pyx\u001b[0m in \u001b[0;36mpandas._libs.parsers.TextReader._tokenize_rows\u001b[1;34m()\u001b[0m\n",
      "\u001b[1;32mpandas/_libs/parsers.pyx\u001b[0m in \u001b[0;36mpandas._libs.parsers.raise_parser_error\u001b[1;34m()\u001b[0m\n",
      "\u001b[1;31mParserError\u001b[0m: Error tokenizing data. C error: Expected 1 fields in line 31, saw 14\n"
     ]
    }
   ],
   "source": [
    "vep = pd.read_csv('C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/fst/fixed_pws_persite/VEP_output_all_snps.txt', sep = '\\t')\n",
    "vep.head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "all_dna_snps = pd.read_table(snp_dir + \"58_dna_maf_0.5_maxmiss_0.9_snp_table.txt\")\n",
    "all_dna_snps['snpIndex'] = all_dna_snps['CHROM'].astype(str) + ':' + all_dna_snps['POS'].astype(str)\n",
    "#all_dna_snps.head(10)\n",
    "features = pd.read_table(cts_dir +\"ref_C_variegatus-1.0_scaffolds_no_header.gff3\", header = None)\n",
    "gene = features[features[2] == 'gene']\n",
    "gene = gene.join(gene[8].str.split('Name=', 1, expand=True).rename(columns={0:'junk1', 1:'junk2'}))\n",
    "gene = gene.join(gene['junk2'].str.split(';', 1, expand=True).rename(columns={0:'geneName', 1:'junk3'}))\n",
    "gene = gene.join(gene['junk1'].str.split('ID=', 1, expand=True).rename(columns={0:'junk4', 1:'junk5'}))\n",
    "gene = gene.join(gene['junk5'].str.split(';', 1, expand=True).rename(columns={0:'geneID', 1:'junk6'}))\n",
    "gene['geneID'] = gene['geneID'].astype(str) + ';' + gene['geneName'].astype(str)\n",
    "gene = gene.rename(columns={0:'chrom', 3:'start', 4:'stop'})\n",
    "gene = gene[['chrom','start','stop','geneID']]\n",
    "mrna = features[features[2] == 'mRNA']\n",
    "mrna = mrna.join(mrna[8].str.split('gene=', 1, expand=True).rename(columns={0:'junk1', 1:'junk2'}))\n",
    "mrna = mrna.join(mrna['junk2'].str.split(';', 1, expand=True).rename(columns={0:'geneName', 1:'junk3'}))\n",
    "mrna = mrna.join(mrna['junk3'].str.split('transcript_id=', 1, expand=True).rename(columns={0:'junk7', 1:'transcriptID'}))\n",
    "mrna = mrna.join(mrna['junk1'].str.split('ID=', 1, expand=True).rename(columns={0:'junk4', 1:'junk5'}))\n",
    "mrna = mrna.join(mrna['junk5'].str.split(';', 1, expand=True).rename(columns={0:'rnaID', 1:'junk6'}))\n",
    "mrna['mrnaID'] = mrna['transcriptID'].astype(str) + ';' + mrna['geneName'].astype(str)\n",
    "mrna = mrna.rename(columns={0:'chrom', 3:'start', 4:'stop'})\n",
    "mrna = mrna[['chrom','start','stop','mrnaID']]\n",
    "exon = features[features[2] == 'exon']\n",
    "exon = exon.join(exon[8].str.split('gene=', 1, expand=True).rename(columns={0:'junk1', 1:'junk2'}))\n",
    "exon = exon.join(exon['junk2'].str.split(';', 1, expand=True).rename(columns={0:'geneName', 1:'junk3'}))\n",
    "exon = exon.join(exon['junk1'].str.split('ID=', 1, expand=True).rename(columns={0:'junk4', 1:'junk5'}))\n",
    "exon = exon.join(exon['junk5'].str.split(';', 1, expand=True).rename(columns={0:'rnaID', 1:'junk6'}))\n",
    "exon['exonID'] = exon['rnaID'].astype(str) + ';' + exon['geneName'].astype(str)\n",
    "exon = exon.rename(columns={0:'chrom', 3:'start', 4:'stop'})\n",
    "exon = exon[['chrom','start','stop','exonID']]\n",
    "\n",
    "cts = cts_dir +infile + \"_counts.csv\"\n",
    "cts = pd.read_table(cts)\n",
    "cts['snpIndex'] = cts['contig'].astype(str) + ':'+ cts['position'].astype(str) \n",
    "cts = cts[['snpIndex','refAllele','altAllele','refCount', 'altCount', 'totalCount']]\n",
    "mom = mom_dict[infile] +\".GT\"\n",
    "snps = all_dna_snps[[mom,'snpIndex']]\n",
    "\n",
    "mom_kid = cts.merge(snps, on='snpIndex')\n",
    "mom_kid = mom_kid.join(mom_kid[mom].str.split('/', 1, expand=True).rename(columns={0:'momAllele', 1:'a2'}))\n",
    "#only analyze homozygous alleles in mom that are het in offspring\n",
    "mom_kid = mom_kid[mom_kid['momAllele'] == mom_kid['a2']]\n",
    "mom_kid = mom_kid[(mom_kid['momAllele'] == mom_kid['refAllele']) | (mom_kid['momAllele'] == mom_kid['altAllele'])]\n",
    "mom_kid = mom_kid[(mom_kid['refCount'] >= 10) & (mom_kid['altCount'] >= 10)]\n",
    "mom_kid.loc[mom_kid['refAllele'] == mom_kid['momAllele'], 'momCount'] = mom_kid['refCount']\n",
    "mom_kid.loc[mom_kid['altAllele'] == mom_kid['momAllele'], 'momCount'] = mom_kid['altCount']\n",
    "mom_kid = mom_kid.join(mom_kid['snpIndex'].str.split(':', 1, expand=True).rename(columns={0:'chrom', 1:'position'}))\n",
    "mom_kid = mom_kid[['chrom','position','snpIndex','refAllele','altAllele','refCount','altCount','totalCount','momAllele', 'momCount']]\n",
    "\n",
    "all_feature_snps = pd.DataFrame()\n",
    "all_feature_snps['chrom'] = []\n",
    "all_feature_snps['position'] = []\n",
    "#feature_dfs = [gene, mrna, exon]\n",
    "feature_dfs = [gene, mrna]\n",
    "for feature_df in feature_dfs:\n",
    "    \n",
    "\n",
    "    #scaff = 'NW_015150454.1'\n",
    "    #feature_df = gene\n",
    "    \n",
    "    feature = feature_df\n",
    "    scaffs = mom_kid['chrom']\n",
    "    scaffs1 = list(scaffs)\n",
    "    scaffs_unique = set(scaffs1)\n",
    "    scaffs = list(scaffs_unique)\n",
    "    scaffolds = []  \n",
    "    start_s = []    \n",
    "    stop_s = []     \n",
    "    feature_names = [] \n",
    "    snp_pss = [] \n",
    "    \n",
    "    for scaff in scaffs:\n",
    "        snp_table   = mom_kid[mom_kid['chrom'] == scaff]  \n",
    "        ftable = feature[feature['chrom'] == scaff]\n",
    "        fstart =   list(ftable[\"start\"])\n",
    "        fstop  =   list(ftable[\"stop\"])  \n",
    "        fname =   list(ftable[ftable.columns[3]]) \n",
    "        positions =   snp_table['position']\n",
    "        \n",
    "        run = list(range(1, len(fstart)))\n",
    "        if len(feature_table) > 0:    \n",
    "            for ps in positions:\n",
    "                for i in run:\n",
    "                     if (int(ps) >= int(fstart[i])) & (int(ps) <= int(fstop[i])):\n",
    "                        scaffolds.append(scaff) \n",
    "                        start_s.append(fstart[i])   \n",
    "                        stop_s.append(fstop[i])    \n",
    "                        feature_names.append(fname[i])\n",
    "                        snp_pss.append(ps)\n",
    "    feature_df_name = ftable.columns[3]\n",
    "    feature_snps = pd.DataFrame(\n",
    "        {'chrom': scaffolds,\n",
    "        'position': snp_pss,\n",
    "        feature_df_name: feature_names,\n",
    "        })\n",
    "    \n",
    "    all_feature_snps = feature_snps.merge(all_feature_snps,how='left', on = ['chrom','position'])\n",
    "final = all_feature_snps.merge(mom_kid, on = ['chrom', 'position'])\n",
    "final.to_csv(out_dir +infile+'maternal_counts_features.txt',index=False, sep = \"\\t\")\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
