{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import io\n",
    "import pandas as pd\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "job = 'vcf_to_plink_and_beagle'\n",
    "\n",
    "vcf_name = 'ph_filtered_snps_minDP600_maxDP2000_maf0.05_minQ20_minMQ30_maxmiss0.5'\n",
    "\n",
    "#sbatch submission script header\n",
    "script = 'script_' + job + '.sh'\n",
    "outfile = io.open(script,'w', newline='\\n')    \n",
    "outfile.write('#!/bin/bash\\n\\n#SBATCH --job-name='+job+'\\n')\n",
    "outfile.write('#SBATCH --mem=16G \\n')\n",
    "outfile.write('#SBATCH --ntasks=4 \\n')\n",
    "outfile.write('#SBATCH -e '+job+'_%A_%a.err \\n')\n",
    "outfile.write('#SBATCH --time=24:00:00  \\n')\n",
    "outfile.write('#SBATCH --mail-user=jamcgirr@ucdavis.edu ##email you when job starts,ends,etc\\n#SBATCH --mail-type=ALL\\n')\n",
    "outfile.write('#SBATCH -p high \\n\\n')\n",
    "\n",
    "outfile.write('module load vcftools \\n')\n",
    "outfile.write('vcftools --vcf /home/jamcgirr/ph/data/vcfs/'+vcf_name+'.vcf --BEAGLE-PL --chr chr1 --out /home/jamcgirr/ph/data/vcfs/chr1_'+vcf_name+' \\n')\n",
    "outfile.write('#vcftools --vcf /home/jamcgirr/ph/data/vcfs/'+vcf_name+'.vcf --BEAGLE-PL --chr chr1,chr2,chr3,chr4,chr5,chr6,chr7,chr8,chr9,chr10,chr11,chr12,chr13,chr14,chr15,chr16,chr17,chr18,chr19,chr20,chr21,chr22,chr23,chr24,chr25,chr26  --out /home/jamcgirr/ph/data/vcfs/'+vcf_name+' \\n')\n",
    "outfile.write('#vcftools --vcf /home/jamcgirr/ph/data/vcfs/'+vcf_name+'.vcf --plink --out /home/jamcgirr/ph/data/vcfs/'+vcf_name+' \\n')\n",
    "\n",
    "\n",
    "#run sbatch submission \n",
    "outfile.write('\\n\\n#command to run: sbatch '+script)\n",
    "outfile.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "job = 'PCAngsd'\n",
    "\n",
    "#git clone https://github.com/Rosemeis/pcangsd.git\n",
    "#cd pcangsd/\\n')\n",
    "#python setup.py build_ext --inplace\n",
    "\n",
    "vcf_name = 'ph_filtered_snps_minDP600_maxDP2000_maf0.05_minQ20_minMQ30_maxmiss0.5'\n",
    "\n",
    "#sbatch submission script header\n",
    "script = 'script_' + job + '.sh'\n",
    "outfile = io.open(script,'w', newline='\\n')    \n",
    "outfile.write('#!/bin/bash\\n\\n#SBATCH --job-name='+job+'\\n')\n",
    "outfile.write('#SBATCH --mem=16G \\n')\n",
    "outfile.write('#SBATCH --ntasks=4 \\n')\n",
    "outfile.write('#SBATCH -e '+job+'_%A_%a.err \\n')\n",
    "outfile.write('#SBATCH --time=24:00:00  \\n')\n",
    "outfile.write('#SBATCH --mail-user=jamcgirr@ucdavis.edu ##email you when job starts,ends,etc\\n#SBATCH --mail-type=ALL\\n')\n",
    "outfile.write('#SBATCH -p high \\n\\n')\n",
    "\n",
    "outfile.write('module load pcangsd \\n')\n",
    "outfile.write('#gzip /home/jamcgirr/ph/data/vcfs/chr1_'+vcf_name+'.BEAGLE.PL \\n')\n",
    "\n",
    "outfile.write('python /home/jamcgirr/apps/pcangsd/pcangsd.py -beagle /home/jamcgirr/ph/data/vcfs/chr1_'+vcf_name+'_plate4_outliers_rm_no_TB.vcf.BEAGLE.PL.gz -o /home/jamcgirr/ph/data/angsd/PCAngsd/chr1_'+vcf_name+'_plate4_outliers_rm_no_TB.vcf -threads 4 \\n')\n",
    "outfile.write('#python /home/jamcgirr/apps/pcangsd/pcangsd.py -beagle /home/jamcgirr/ph/data/vcfs/chr2_'+vcf_name+'.BEAGLE.PL.gz -o /home/jamcgirr/ph/data/angsd/PCAngsd/chr2_'+vcf_name+' -threads 4 \\n')\n",
    "outfile.write('#python /home/jamcgirr/apps/pcangsd/pcangsd.py -beagle /home/jamcgirr/ph/data/vcfs/chr3_'+vcf_name+'.BEAGLE.PL.gz -o /home/jamcgirr/ph/data/angsd/PCAngsd/chr3_'+vcf_name+' -threads 4 \\n')\n",
    "\n",
    "\n",
    "\n",
    "#run sbatch submission \n",
    "outfile.write('\\n\\n#command to run: sbatch '+script)\n",
    "outfile.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "job = 'plink_PCA'\n",
    "\n",
    "vcf_name = 'ph_filtered_snps_minDP600_maxDP2000_maf0.05_minQ20_minMQ30_maxmiss0.5'\n",
    "#vcf_name = 'ph_filtered_snps_minDP600_maxDP2000_maf0.05_minQ30_maxmiss0.5'\n",
    "\n",
    "#sbatch submission script header\n",
    "script = 'script_' + job + '.sh'\n",
    "outfile = io.open(script,'w', newline='\\n')    \n",
    "outfile.write('#!/bin/bash\\n\\n#SBATCH --job-name='+job+'\\n')\n",
    "outfile.write('#SBATCH --mem=16G \\n')\n",
    "outfile.write('#SBATCH --ntasks=4 \\n')\n",
    "outfile.write('#SBATCH -e '+job+'_%A_%a.err \\n')\n",
    "outfile.write('#SBATCH --time=24:00:00  \\n')\n",
    "outfile.write('#SBATCH --mail-user=jamcgirr@ucdavis.edu ##email you when job starts,ends,etc\\n#SBATCH --mail-type=ALL\\n')\n",
    "outfile.write('#SBATCH -p high \\n\\n')\n",
    "\n",
    "\n",
    "outfile.write('module load plink \\n')\n",
    "outfile.write('plink --file /home/jamcgirr/ph/data/vcfs/'+vcf_name+' --pca --out /home/jamcgirr/ph/data/plink/'+vcf_name+'\\n') \n",
    "\n",
    "\n",
    "#run sbatch submission \n",
    "outfile.write('\\n\\n#command to run: sbatch '+script)\n",
    "outfile.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "job = 'MDS'\n",
    "\n",
    "vcf_name = 'ph_filtered_snps_minDP600_maxDP2000_maf0.05_minQ20_minMQ30_maxmiss0.5'\n",
    "\n",
    "#sbatch submission script header\n",
    "script = 'script_' + job + '.sh'\n",
    "outfile = io.open(script,'w', newline='\\n')    \n",
    "outfile.write('#!/bin/bash\\n\\n#SBATCH --job-name='+job+'\\n')\n",
    "outfile.write('#SBATCH --mem=8G \\n')\n",
    "outfile.write('#SBATCH --ntasks=1 \\n')\n",
    "outfile.write('#SBATCH -e '+job+'_%A_%a.err \\n')\n",
    "outfile.write('#SBATCH --time=24:00:00  \\n')\n",
    "outfile.write('#SBATCH --mail-user=jamcgirr@ucdavis.edu ##email you when job starts,ends,etc\\n#SBATCH --mail-type=ALL\\n')\n",
    "outfile.write('#SBATCH -p high \\n\\n')\n",
    "\n",
    "outfile.write('module load plink \\n')\n",
    "outfile.write('#plink --file /home/jamcgirr/ph/data/vcfs/'+vcf_name+' --genome --chr chr1 --out /home/jamcgirr/ph/data/plink/chr_1'+vcf_name+'\\n')\n",
    "outfile.write('#plink --file /home/jamcgirr/ph/data/vcfs/'+vcf_name+' --cluster --chr chr1 --out /home/jamcgirr/ph/data/plink/chr_1'+vcf_name+'\\n') \n",
    "outfile.write('plink --file /home/jamcgirr/ph/data/vcfs/'+vcf_name+' --read-genome /home/jamcgirr/ph/data/plink/chr_1'+vcf_name+'.genome --mds-plot 4 --cluster --chr chr1 --out /home/jamcgirr/ph/data/plink/chr_1'+vcf_name+'\\n') \n",
    "\n",
    "#run sbatch submission \n",
    "outfile.write('\\n\\n#command to run: sbatch '+script)\n",
    "outfile.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "job = 'NGSadmix'\n",
    "\n",
    "vcf_name = 'ph_filtered_snps_minDP600_maxDP2000_maf0.05_minQ20_minMQ30_maxmiss0.5'\n",
    "\n",
    "#sbatch submission script header\n",
    "script = 'script_' + job + '.sh'\n",
    "outfile = io.open(script,'w', newline='\\n')    \n",
    "outfile.write('#!/bin/bash\\n\\n#SBATCH --job-name='+job+'\\n')\n",
    "outfile.write('#SBATCH --mem=16G \\n')\n",
    "outfile.write('#SBATCH --ntasks=4 \\n')\n",
    "outfile.write('#SBATCH -e '+job+'_%A_%a.err \\n')\n",
    "outfile.write('#SBATCH --time=24:00:00  \\n')\n",
    "outfile.write('#SBATCH --mail-user=jamcgirr@ucdavis.edu ##email you when job starts,ends,etc\\n#SBATCH --mail-type=ALL\\n')\n",
    "outfile.write('#SBATCH -p high \\n\\n')\n",
    "\n",
    "outfile.write('#git clone https://github.com/Rosemeis/pcangsd.git\\n')\n",
    "outfile.write('#cd pcangsd/\\n')\n",
    "outfile.write('#python setup.py build_ext --inplace\\n\\n')\n",
    "\n",
    "outfile.write('module load angsd \\n')\n",
    "outfile.write('NGSadmix -likes /home/jamcgirr/ph/data/vcfs/chr1_'+vcf_name+'.BEAGLE.PL.gz -K 2 -o /home/jamcgirr/ph/data/angsd/NGSadmix/k2_chr1_'+vcf_name+' -P 4 \\n')\n",
    "outfile.write('NGSadmix -likes /home/jamcgirr/ph/data/vcfs/chr1_'+vcf_name+'.BEAGLE.PL.gz -K 3 -o /home/jamcgirr/ph/data/angsd/NGSadmix/k3_chr1_'+vcf_name+' -P 4 \\n')\n",
    "outfile.write('NGSadmix -likes /home/jamcgirr/ph/data/vcfs/chr1_'+vcf_name+'.BEAGLE.PL.gz -K 4 -o /home/jamcgirr/ph/data/angsd/NGSadmix/k4_chr1_'+vcf_name+' -P 4 \\n')\n",
    "\n",
    "\n",
    "#run sbatch submission \n",
    "outfile.write('\\n\\n#command to run: sbatch '+script)\n",
    "outfile.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "job = 'evalAdmix'\n",
    "\n",
    "vcf_name = 'ph_filtered_snps_minDP600_maxDP2000_maf0.05_minQ20_minMQ30_maxmiss0.5'\n",
    "infiles = list(range(2,21))\n",
    "\n",
    "for infile in infiles:\n",
    "    \n",
    "    #sbatch submission script header\n",
    "    script = 'script_' + str(infile) + job + '.sh'\n",
    "    outfile = io.open(script,'w', newline='\\n') \n",
    "    jobname= str(infile) + job   \n",
    "    outfile.write('#!/bin/bash\\n\\n#SBATCH --job-name='+jobname+'\\n')\n",
    "    outfile.write('#SBATCH --mem=16G \\n')\n",
    "    outfile.write('#SBATCH --ntasks=4 \\n')\n",
    "    outfile.write('#SBATCH -e '+jobname+'_%A_%a.err \\n')\n",
    "    outfile.write('#SBATCH --time=24:00:00 \\n')\n",
    "    outfile.write('#SBATCH --mail-user=jamcgirr@ucdavis.edu ##email you when job starts,ends,etc\\n#SBATCH --mail-type=ALL\\n')\n",
    "    outfile.write('#SBATCH -p high \\n\\n')\n",
    "    \n",
    "    outfile.write('module load angsd \\n')\n",
    "    outfile.write('/home/jamcgirr/apps/evalAdmix/evalAdmix -beagle /home/jamcgirr/ph/data/vcfs/chr1_'+vcf_name+'.BEAGLE.PL.gz -fname /home/jamcgirr/ph/data/angsd/NGSadmix/k'+str(infile)+'_chr1_'+vcf_name+'.fopt.gz -qname /home/jamcgirr/ph/data/angsd/NGSadmix/k'+str(infile)+'_chr1_'+vcf_name+'.qopt -o /home/jamcgirr/ph/data/angsd/NGSadmix/evalAdmix/eval_k'+str(infile)+'_chr1_'+vcf_name+' -P 4 \\n')\n",
    "\n",
    "#run sbatch submission \n",
    "outfile.write('\\n\\n#command to run: sbatch '+script)\n",
    "outfile.close()\n",
    "\n",
    "# EST run time ~1hr 30min"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "job = '_subset_pops_vcfs'\n",
    "\n",
    "vcf_name = 'ph_filtered_snps_minDP600_maxDP2000_maf0.05_minQ20_minMQ30_maxmiss0.5'\n",
    "infiles = [\"BC17\",\"CA17\",\"PWS07\",\"PWS17\",\"PWS91\",\"PWS96\",\"SS06\",\"SS17\",\"SS96\",\"TB06\",\"TB17\",\"TB91\",\"TB96\",\"WA17\"]\n",
    "\n",
    "for infile in infiles:\n",
    "    \n",
    "    #sbatch submission script header\n",
    "    script = 'script_' + infile + job + '.sh'\n",
    "    outfile = io.open(script,'w', newline='\\n') \n",
    "    jobname= infile + job   \n",
    "    outfile.write('#!/bin/bash\\n\\n#SBATCH --job-name='+jobname+'\\n')\n",
    "    outfile.write('#SBATCH --mem=8G \\n')\n",
    "    outfile.write('#SBATCH --ntasks=1 \\n')\n",
    "    outfile.write('#SBATCH -e '+jobname+'_%A_%a.err \\n')\n",
    "    outfile.write('#SBATCH --time=24:00:00 \\n')\n",
    "    outfile.write('#SBATCH --mail-user=jamcgirr@ucdavis.edu ##email you when job starts,ends,etc\\n#SBATCH --mail-type=ALL\\n')\n",
    "    outfile.write('#SBATCH -p high \\n\\n')\n",
    "        \n",
    "    outfile.write('module load samtools \\n')\n",
    "    outfile.write('module load bcftools \\n')\n",
    "\n",
    "    outfile.write('bcftools view -S /home/jamcgirr/ph/scripts/angsd/SFS/SFS_by_pop/'+infile+'_plates_1_through_5_rm.txt -Ov /home/jamcgirr/ph/data/vcfs/'+vcf_name+'.vcf > /home/jamcgirr/ph/data/vcfs/split_pops/population_'+infile+'_'+vcf_name+'_plates_1_5_rm.vcf \\n')\n",
    "\n",
    "    #run sbatch submission \n",
    "    outfile.write('\\n\\n#command to run: sbatch '+script)\n",
    "    outfile.close()\n",
    "    \n",
    "#EST run time ~30min"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# install angsd locally to use most recent version and avoid issues with vcf parsing\n",
    "# git clone https://github.com/samtools/htslib.git;\n",
    "# git clone https://github.com/angsd/angsd.git;\n",
    "# cd htslib;make;cd ../angsd;make HTSSRC=../htslib\n",
    "\n",
    "job = '_SFS'\n",
    "vcf_name = 'ph_filtered_snps_minDP600_maxDP2000_maf0.05_minQ20_minMQ30_maxmiss0.5'\n",
    "infiles = [\"BC17\",\"CA17\",\"PWS07\",\"PWS17\",\"PWS91\",\"PWS96\",\"SS06\",\"SS17\",\"SS96\",\"TB06\",\"TB17\",\"TB91\",\"TB96\",\"WA17\"]\n",
    "\n",
    "for infile in infiles:\n",
    "    \n",
    "    #sbatch submission script header\n",
    "    script = 'script_' + infile + job + '.sh'\n",
    "    outfile = io.open(script,'w', newline='\\n') \n",
    "    jobname= infile + job   \n",
    "    outfile.write('#!/bin/bash\\n\\n#SBATCH --job-name='+jobname+'\\n')\n",
    "    outfile.write('#SBATCH --mem=8G \\n')\n",
    "    outfile.write('#SBATCH --ntasks=4 \\n')\n",
    "    outfile.write('#SBATCH -e '+jobname+'_%A_%a.err \\n')\n",
    "    outfile.write('#SBATCH --time=24:00:00 \\n')\n",
    "    outfile.write('#SBATCH --mail-user=jamcgirr@ucdavis.edu ##email you when job starts,ends,etc\\n#SBATCH --mail-type=ALL\\n')\n",
    "    outfile.write('#SBATCH -p high \\n\\n')\n",
    "    \n",
    "    # make saf from vcf\n",
    "    outfile.write('#/home/jamcgirr/apps/angsd_sep_20/angsd/angsd -doSaf 1 -vcf-pl /home/jamcgirr/ph/data/vcfs/split_pops/population_'+infile+'_'+vcf_name+'_plates_1_5_rm.vcf -out /home/jamcgirr/ph/data/angsd/SFS/split_pops/population_'+infile+'_'+vcf_name+' -anc /home/jamcgirr/ph/data/c_harengus/c.harengus.fa \\n')\n",
    "    # make folded sfs\n",
    "    outfile.write('#/home/jamcgirr/apps/angsd_sep_20/angsd/misc/realSFS /home/jamcgirr/ph/data/angsd/SFS/split_pops/population_'+infile+'_'+vcf_name+'.saf.idx -P 4 -fold 1 > /home/jamcgirr/ph/data/angsd/SFS/split_pops/folded/population_'+infile+'_'+vcf_name+'.sfs \\n')\n",
    "    # make unfolded sfs\n",
    "    outfile.write('/home/jamcgirr/apps/angsd_sep_20/angsd/misc/realSFS /home/jamcgirr/ph/data/angsd/SFS/split_pops/population_'+infile+'_'+vcf_name+'.saf.idx -P 4 > /home/jamcgirr/ph/data/angsd/SFS/split_pops/unfolded/population_'+infile+'_'+vcf_name+'.sfs \\n')\n",
    "\n",
    "    #run sbatch submission \n",
    "    outfile.write('\\n\\n#command to run: sbatch '+script)\n",
    "    outfile.close()\n",
    "\n",
    "# EST run time ~15min"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "# theta estimates using saf from vcf are way too high\n",
    "# using angsd to estimate saf from bam files generates reasonable theta\n",
    "# likely has to do with invariant site corrections\n",
    "\n",
    "# if using vcf sites, need to index sites.txt \n",
    "# /home/jamcgirr/apps/angsd/angsd sites index sites.txt\n",
    "\n",
    "\n",
    "job = '_thetas'\n",
    "vcf_name = 'ph_filtered_snps_minDP600_maxDP2000_maf0.05_minQ20_minMQ30_maxmiss0.5'\n",
    "infiles = [\"BC17\",\"CA17\",\"PWS07\",\"PWS17\",\"PWS91\",\"PWS96\",\"SS06\",\"SS17\",\"SS96\",\"TB06\",\"TB17\",\"TB91\",\"TB96\",\"WA17\"]\n",
    "\n",
    "for infile in infiles:\n",
    "    \n",
    "    #sbatch submission script header\n",
    "    script = 'script_' + infile + job + '.sh'\n",
    "    outfile = io.open(script,'w', newline='\\n') \n",
    "    jobname= infile + job   \n",
    "    outfile.write('#!/bin/bash\\n\\n#SBATCH --job-name='+jobname+'\\n')\n",
    "    outfile.write('#SBATCH --mem=8G \\n')\n",
    "    outfile.write('#SBATCH --ntasks=4 \\n')\n",
    "    outfile.write('#SBATCH -e '+jobname+'_%A_%a.err \\n')\n",
    "    outfile.write('#SBATCH --time=24:00:00 \\n')\n",
    "    outfile.write('#SBATCH --mail-user=jamcgirr@ucdavis.edu ##email you when job starts,ends,etc\\n#SBATCH --mail-type=ALL\\n')\n",
    "    outfile.write('#SBATCH -p high \\n\\n')\n",
    "    \n",
    "    # make saf from bams\n",
    "    outfile.write('/home/jamcgirr/apps/angsd_sep_20/angsd/angsd -bam /home/jamcgirr/ph/data/angsd/SFS/thetas/'+infile+'_bams_p_1_5_rm.txt -doSaf 1 -anc /home/jamcgirr/ph/data/c_harengus/c.harengus.fa -minMapQ 30 -minQ 20 -GL 1 -P 4 -out /home/jamcgirr/ph/data/angsd/SFS/thetas/population_'+infile+'_minQ20_minMQ30 \\n')\n",
    "    # make unfolded sfs\n",
    "    outfile.write('/home/jamcgirr/apps/angsd_sep_20/angsd/misc/realSFS /home/jamcgirr/ph/data/angsd/SFS/thetas/population_'+infile+'_minQ20_minMQ30.saf.idx -P 4 > /home/jamcgirr/ph/data/angsd/SFS/thetas/unfolded/population_'+infile+'_minQ20_minMQ30_unfolded.sfs \\n')\n",
    "    outfile.write('/home/jamcgirr/apps/angsd_sep_20/angsd/misc/realSFS saf2theta /home/jamcgirr/ph/data/angsd/SFS/thetas/population_'+infile+'_minQ20_minMQ30.saf.idx -outname /home/jamcgirr/ph/data/angsd/SFS/thetas/unfolded/population_'+infile+'_minQ20_minMQ30 -sfs /home/jamcgirr/ph/data/angsd/SFS/thetas/unfolded/population_'+infile+'_minQ20_minMQ30_unfolded.sfs \\n')\n",
    "    outfile.write('/home/jamcgirr/apps/angsd_sep_20/angsd/misc/thetaStat do_stat /home/jamcgirr/ph/data/angsd/SFS/thetas/unfolded/population_'+infile+'_minQ20_minMQ30.thetas.idx \\n')\n",
    "    outfile.write('/home/jamcgirr/apps/angsd_sep_20/angsd/misc/thetaStat do_stat /home/jamcgirr/ph/data/angsd/SFS/thetas/unfolded/population_'+infile+'_minQ20_minMQ30.thetas.idx -win 50000 -step 10000 -outnames /home/jamcgirr/ph/data/angsd/SFS/thetas/unfolded/population_'+infile+'_minQ20_minMQ30_50kb_win_10kb_step \\n')\n",
    "\n",
    "    \n",
    "    #run sbatch submission \n",
    "    outfile.write('\\n\\n#command to run: sbatch '+script)\n",
    "    outfile.close()\n",
    "\n",
    "# EST run time ~"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Steps for Fst and PBS for andgsd folding update\n",
    "\n",
    "# 1. Create 2d sfs from saf\n",
    "\n",
    "job = '_2d_sfs'\n",
    "\n",
    "vcf_name = 'ph_filtered_snps_minDP600_maxDP2000_maf0.05_minQ20_minMQ30_maxmiss0.5'\n",
    "realSFS = '/home/jamcgirr/apps/angsd/misc/realSFS'\n",
    "wk_dir = '/home/jamcgirr/ph/data/angsd/SFS/fst_pbs/'\n",
    "saf_dir = '/home/jamcgirr/ph/data/angsd/SFS/split_pops/folded/'\n",
    "\n",
    "comps = [\"BC17_CA17\",\"BC17_WA17\",\"PWS07_PWS17\",\"PWS07_SS06\",\"PWS17_BC17\",\"PWS17_CA17\",\"PWS17_SS17\",\"PWS17_WA17\",\"PWS91_PWS07\",\"PWS91_PWS17\",\"PWS91_PWS96\",\"PWS96_PWS07\",\"PWS96_PWS17\",\"PWS96_SS96\",\"SS06_SS17\",\"SS17_BC17\",\"SS17_CA17\",\"SS17_WA17\",\"SS96_SS06\",\"SS96_SS17\",\"TB06_PWS07\",\"TB06_SS06\",\"TB06_TB17\",\"TB17_BC17\",\"TB17_CA17\",\"TB17_PWS17\",\"TB17_SS17\",\"TB17_WA17\",\"TB91_TB06\",\"TB91_TB17\",\"TB91_TB96\",\"TB96_PWS96\",\"TB96_SS96\",\"TB96_TB06\",\"TB96_TB17\",\"WA17_CA17\"]\n",
    "\n",
    "for comp in comps:\n",
    "    \n",
    "    #sbatch submission script header\n",
    "    script = 'script_' + comp + job + '.sh'\n",
    "    outfile = io.open(script,'w', newline='\\n') \n",
    "    jobname= comp + job   \n",
    "    outfile.write('#!/bin/bash\\n\\n#SBATCH --job-name='+jobname+'\\n')\n",
    "    outfile.write('#SBATCH --mem=8G \\n')\n",
    "    outfile.write('#SBATCH --ntasks=4 \\n')\n",
    "    outfile.write('#SBATCH -e '+jobname+'_%A_%a.err \\n')\n",
    "    outfile.write('#SBATCH --time=24:00:00 \\n')\n",
    "    outfile.write('#SBATCH --mail-user=jamcgirr@ucdavis.edu ##email you when job starts,ends,etc\\n#SBATCH --mail-type=ALL\\n')\n",
    "    outfile.write('#SBATCH -p high \\n\\n')\n",
    "    \n",
    "    pops = ''.join(comp).split(\"_\")\n",
    "    outfile.write(realSFS+' '+saf_dir+'population_'+pops[0]+'_'+vcf_name+'.saf.idx '+saf_dir+'population_'+pops[1]+'_'+vcf_name+'.saf.idx -fold 1 -P 4 > '+wk_dir+'folded_'+comp+'.sfs \\n')\n",
    "    \n",
    "    #run sbatch submission \n",
    "    outfile.write('\\n\\n#command to run: sbatch '+script)\n",
    "    outfile.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "# 2. Estimate Fst and PBS\n",
    "\n",
    "job = '_fst_pbs_folded'\n",
    "\n",
    "vcf_name = 'ph_filtered_snps_minDP600_maxDP2000_maf0.05_minQ20_minMQ30_maxmiss0.5'\n",
    "realSFS = '/home/jamcgirr/apps/angsd/misc/realSFS'\n",
    "wk_dir = '/home/jamcgirr/ph/data/angsd/SFS/fst_pbs/'\n",
    "saf_dir = '/home/jamcgirr/ph/data/angsd/SFS/split_pops/'\n",
    "\n",
    "comps = [\"BC17_WA17_CA17\",\"PWS17_SS17_BC17\",\"PWS17_SS17_CA17\",\"PWS17_SS17_WA17\",\"TB17_PWS17_BC17\",\"TB17_PWS17_CA17\",\"TB17_PWS17_SS17\",\"TB17_PWS17_WA17\",\"TB06_PWS07_SS06\",\"TB96_PWS96_SS96\",\"PWS91_PWS07_PWS17\",\"PWS91_PWS96_PWS07\",\"PWS91_PWS96_PWS17\",\"TB91_TB06_TB17\",\"TB91_TB96_TB06\",\"TB91_TB96_TB17\",\"SS96_SS06_SS17\"]\n",
    "\n",
    "\n",
    "for comp in comps:\n",
    "    \n",
    "    #sbatch submission script header\n",
    "    script = 'script_' + comp + job + '.sh'\n",
    "    outfile = io.open(script,'w', newline='\\n') \n",
    "    jobname= comp + job   \n",
    "    outfile.write('#!/bin/bash\\n\\n#SBATCH --job-name='+jobname+'\\n')\n",
    "    outfile.write('#SBATCH --mem=8G \\n')\n",
    "    outfile.write('#SBATCH --ntasks=1 \\n')\n",
    "    outfile.write('#SBATCH -e '+jobname+'_%A_%a.err \\n')\n",
    "    outfile.write('#SBATCH --time=24:00:00 \\n')\n",
    "    outfile.write('#SBATCH --mail-user=jamcgirr@ucdavis.edu ##email you when job starts,ends,etc\\n#SBATCH --mail-type=ALL\\n')\n",
    "    outfile.write('#SBATCH -p high \\n\\n')\n",
    "    \n",
    "    pops = ''.join(comp).split(\"_\")\n",
    "    #outfile.write(realSFS+' fst index '+saf_dir+'population_'+pops[0]+'_'+vcf_name+'.saf.idx '+saf_dir+'population_'+pops[1]+'_'+vcf_name+'.saf.idx -sfs '+wk_dir+'folded_'+pops[0]+'_'+pops[1]+'.sfs -fold 1 -fstout '+wk_dir+'folded_'+pops[0]+'_'+pops[1]+'persite \\n')\n",
    "    #outfile.write(realSFS+' fst index '+saf_dir+'population_'+pops[0]+'_'+vcf_name+'.saf.idx '+saf_dir+'population_'+pops[2]+'_'+vcf_name+'.saf.idx -sfs '+wk_dir+'folded_'+pops[0]+'_'+pops[2]+'.sfs -fold 1 -fstout '+wk_dir+'folded_'+pops[0]+'_'+pops[2]+'persite \\n')\n",
    "    #outfile.write(realSFS+' fst index '+saf_dir+'population_'+pops[1]+'_'+vcf_name+'.saf.idx '+saf_dir+'population_'+pops[2]+'_'+vcf_name+'.saf.idx -sfs '+wk_dir+'folded_'+pops[1]+'_'+pops[2]+'.sfs -fold 1 -fstout '+wk_dir+'folded_'+pops[1]+'_'+pops[2]+'persite \\n\\n')\n",
    "\n",
    "    outfile.write(realSFS+' fst index '+saf_dir+'population_'+pops[0]+'_'+vcf_name+'.saf.idx '+saf_dir+'population_'+pops[1]+'_'+vcf_name+'.saf.idx '+saf_dir+'population_'+pops[2]+'_'+vcf_name+'.saf.idx -sfs '+wk_dir+'folded_'+pops[0]+'_'+pops[1]+'.sfs -sfs '+wk_dir+'folded_'+pops[0]+'_'+pops[2]+'.sfs -sfs '+wk_dir+'folded_'+pops[1]+'_'+pops[2]+'.sfs -fold 1 -fstout '+wk_dir+'folded_'+pops[0]+'_'+pops[1]+'_'+pops[2]+'_persite \\n')\n",
    "    outfile.write(realSFS+' fst stats2 '+wk_dir+'folded_'+pops[0]+'_'+pops[1]+'_'+pops[2]+'_persite.fst.idx -win 50000 -step 10000 > '+wk_dir+'fst_pbs_50kb_win_10kb_step_folded_'+pops[0]+'_'+pops[1]+'_'+pops[2]+'.txt \\n')\n",
    "\n",
    "    #run sbatch submission \n",
    "    outfile.write('\\n\\n#command to run: sbatch '+script)\n",
    "    outfile.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "# LD with ngsLD\n",
    "\n",
    "job = 'prune_chr1'\n",
    "\n",
    "vcf_name = 'ph_filtered_snps_minDP600_maxDP2000_maf0.05_minQ20_minMQ30_maxmiss0.5_outliers_rm'\n",
    "vcf_dir = '/home/jamcgirr/ph/data/vcfs/'\n",
    "\n",
    "#sbatch submission script header\n",
    "script = 'script_' + job + '.sh'\n",
    "outfile = io.open(script,'w', newline='\\n')    \n",
    "outfile.write('#!/bin/bash\\n\\n#SBATCH --job-name='+job+'\\n')\n",
    "outfile.write('#SBATCH --mem=16G \\n')\n",
    "outfile.write('#SBATCH --ntasks=1 \\n')\n",
    "outfile.write('#SBATCH -e '+job+'_%A_%a.err \\n')\n",
    "outfile.write('#SBATCH --time=48:00:00  \\n')\n",
    "outfile.write('#SBATCH --mail-user=jamcgirr@ucdavis.edu ##email you when job starts,ends,etc\\n#SBATCH --mail-type=ALL\\n')\n",
    "outfile.write('#SBATCH -p high \\n\\n')\n",
    "\n",
    "outfile.write('module load bcftools \\n')\n",
    "outfile.write('bcftools view '+vcf_dir+'vince/'+vcf_name+'.vcf.gz -Ov --regions chr1 > '+vcf_dir+'chr1_'+vcf_name+'.vcf \\n')\n",
    "outfile.write('java -jar /home/jamcgirr/apps/jvarkit/dist/downsamplevcf.jar -n 20000 '+vcf_dir+'chr1_'+vcf_name+'.vcf > '+vcf_dir+'chr1_thin20k_'+vcf_name+'.vcf \\n')\n",
    "outfile.write('module load vcftools \\n')\n",
    "outfile.write('vcftools --vcf '+vcf_dir+'chr1_thin20k_'+vcf_name+'.vcf --BEAGLE-PL --chr chr1 --out '+vcf_dir+'chr1_thin20k_'+vcf_name+' \\n')\n",
    "\n",
    "\n",
    "outfile.write('\\n\\n#command to run: sbatch '+script)\n",
    "outfile.close()\n",
    "\n",
    "# Run to create position file for --pos flag for ngsLD:\n",
    "\n",
    "# awk '{ print $1 }' chr1_thin20k_ph_filtered_snps_minDP600_maxDP2000_maf0.05_minQ20_minMQ30_maxmiss0.5_outliers_rm.BEAGLE.PL > /home/jamcgirr/ph/data/vcfs/split_pops/chr1_thin20k_sites.txt\n",
    "# sed 's/:/\\t/g' /home/jamcgirr/ph/data/vcfs/split_pops/chr1_thin20k_sites.txt > /home/jamcgirr/ph/data/vcfs/split_pops/test_pos_file2\n",
    "# tail -n +2 /home/jamcgirr/ph/data/vcfs/split_pops/test_pos_file2 > /home/jamcgirr/ph/data/vcfs/split_pops/chr1_thin20k_sites.txt\n",
    "# rm /home/jamcgirr/ph/data/vcfs/split_pops/test_pos_file2\n",
    "\n",
    "job = '_LD_chr1_20k_snps'\n",
    "infiles = [\"BC17_64\",\"CA17_70\",\"PWS07_46\",\"PWS17_56\",\"PWS91_58\",\"PWS96_72\",\"SS06_41\",\"SS17_64\",\"SS96_78\",\"TB06_52\",\"TB17_72\",\"TB91_74\",\"TB96_73\",\"WA17_72\"]\n",
    "\n",
    "for infile in infiles:\n",
    "    \n",
    "    #sbatch submission script header\n",
    "    script = 'script_' + infile + job + '.sh'\n",
    "    outfile = io.open(script,'w', newline='\\n') \n",
    "    jobname= infile + job   \n",
    "    outfile.write('#!/bin/bash\\n\\n#SBATCH --job-name='+jobname+'\\n')\n",
    "    outfile.write('#SBATCH --mem=8G \\n')\n",
    "    outfile.write('#SBATCH --ntasks=1 \\n')\n",
    "    outfile.write('#SBATCH -e '+jobname+'_%A_%a.err \\n')\n",
    "    outfile.write('#SBATCH --time=24:00:00 \\n')\n",
    "    outfile.write('#SBATCH --mail-user=jamcgirr@ucdavis.edu ##email you when job starts,ends,etc\\n#SBATCH --mail-type=ALL\\n')\n",
    "    outfile.write('#SBATCH -p high \\n\\n')\n",
    "    \n",
    "    pops = ''.join(infile).split(\"_\")\n",
    "\n",
    "    outfile.write('module load vcftools \\n')\n",
    "    outfile.write('vcftools --vcf '+vcf_dir+'chr1_thin20k_'+vcf_name+'.vcf --keep /home/jamcgirr/ph/scripts/angsd/SFS/SFS_by_pop/'+pops[0]+'_plates_1_through_5_rm.txt --BEAGLE-PL --chr chr1 --out '+vcf_dir+'split_pops/chr1_thin20k_'+pops[0]+'_'+vcf_name+' \\n')\n",
    "    outfile.write('gzip '+vcf_dir+'split_pops/chr1_thin20k_'+pops[0]+'_'+vcf_name+'.BEAGLE.PL \\n')\n",
    "    outfile.write('/home/jamcgirr/apps/ngsLD/ngsLD --geno '+vcf_dir+'split_pops/chr1_thin20k_'+pops[0]+'_'+vcf_name+'.BEAGLE.PL.gz --pos '+vcf_dir+'split_pops/chr1_thin20k_sites.txt --n_ind '+pops[1]+' --n_sites 20000 --out /home/jamcgirr/ph/data/angsd/ngsLD/chr1_thin20k_'+pops[0]+'_'+vcf_name+'_ld.txt \\n')\n",
    "\n",
    "    #run sbatch submission \n",
    "    outfile.write('\\n\\n#command to run: sbatch '+script)\n",
    "    outfile.close()\n",
    "    \n",
    "# Rscript --vanilla --slave /home/jamcgirr/apps/ngsLD/scripts/fit_LDdecay.R --ld_files /home/jamcgirr/ph/data/angsd/ngsLD/ld_files.list.txt --fit_level 3 --out test_plot.pdf\n",
    "# Rscript --vanilla --slave /home/jamcgirr/apps/ngsLD/scripts/fit_LDdecay.R --ld_files /home/jamcgirr/ph/data/angsd/ngsLD/ld_files.list_lg.txt --max_kb_dist 100 --fit_level 3 --out test_plot_lg.pdf\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
