{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "import io\n",
    "import pandas as pd\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "def sbatch_header(job,mem,tasks,hours):\n",
    "    #sbatch submission script header\n",
    "    script = 'script_' + job + '.sh'\n",
    "    outfile = io.open(script,'w', newline='\\n')    \n",
    "    outfile.write('#!/bin/bash\\n\\n#SBATCH --job-name='+job+'\\n')\n",
    "    outfile.write('#SBATCH --mem='+mem+'G \\n')\n",
    "    outfile.write('#SBATCH --ntasks='+tasks+' \\n')\n",
    "    outfile.write('#SBATCH -e '+job+'_%A_%a.err \\n')\n",
    "    outfile.write('#SBATCH --time='+hours+':00:00  \\n')\n",
    "    outfile.write('#SBATCH --mail-user=jamcgirr@ucdavis.edu ##email you when job starts,ends,etc\\n#SBATCH --mail-type=ALL\\n')\n",
    "    outfile.write('#SBATCH -p high \\n\\n')\n",
    "    outfile.close()\n",
    "    \n",
    "def sbatch_header_loop(job,mem,tasks,hours,infile):\n",
    "    #sbatch submission script header\n",
    "    script = 'script_' + infile + job + '.sh'\n",
    "    outfile = io.open(script,'w', newline='\\n') \n",
    "    jobname= infile + job   \n",
    "    outfile.write('#!/bin/bash\\n\\n#SBATCH --job-name='+jobname+'\\n')\n",
    "    outfile.write('#SBATCH --mem='+mem+'G \\n')\n",
    "    outfile.write('#SBATCH --ntasks='+tasks+' \\n')\n",
    "    outfile.write('#SBATCH -e '+jobname+'_%A_%a.err \\n')\n",
    "    outfile.write('#SBATCH --time='+hours+':00:00 \\n')\n",
    "    outfile.write('#SBATCH --mail-user=jamcgirr@ucdavis.edu ##email you when job starts,ends,etc\\n#SBATCH --mail-type=ALL\\n')\n",
    "    outfile.write('#SBATCH -p high \\n\\n')\n",
    "    outfile.close()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [],
   "source": [
    "job_name = 'vcf_to_plink_and_beagle'\n",
    "\n",
    "vcf_name = 'ph_filtered_snps_minDP600_maxDP2000_maf0.05_minQ20_minMQ30_maxmiss0.5'\n",
    "\n",
    "sbatch_header(job_name,'16','4','24')\n",
    "script = 'script_' + job_name + '.sh'\n",
    "o = io.open(script,'a+', newline='\\n')\n",
    "\n",
    "o.write('module load vcftools \\n')\n",
    "o.write('vcftools --vcf /home/jamcgirr/ph/data/vcfs/'+vcf_name+'.vcf --BEAGLE-PL --chr chr1 --out /home/jamcgirr/ph/data/vcfs/chr1_'+vcf_name+' \\n')\n",
    "o.write('#vcftools --vcf /home/jamcgirr/ph/data/vcfs/'+vcf_name+'.vcf --BEAGLE-PL --chr chr1,chr2,chr3,chr4,chr5,chr6,chr7,chr8,chr9,chr10,chr11,chr12,chr13,chr14,chr15,chr16,chr17,chr18,chr19,chr20,chr21,chr22,chr23,chr24,chr25,chr26  --out /home/jamcgirr/ph/data/vcfs/'+vcf_name+' \\n')\n",
    "o.write('#vcftools --vcf /home/jamcgirr/ph/data/vcfs/'+vcf_name+'.vcf --plink --out /home/jamcgirr/ph/data/vcfs/'+vcf_name+' \\n')\n",
    "\n",
    "o.write('\\n\\n#command to run: sbatch '+script)\n",
    "o.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "job_name = 'PCAngsd'\n",
    "\n",
    "#git clone https://github.com/Rosemeis/pcangsd.git\n",
    "#cd pcangsd/\\n')\n",
    "#python setup.py build_ext --inplace\n",
    "\n",
    "vcf_name = 'ph_filtered_snps_minDP600_maxDP2000_maf0.05_minQ20_minMQ30_maxmiss0.5'\n",
    "\n",
    "sbatch_header(job_name,'16','4','24')\n",
    "script = 'script_' + job_name + '.sh'\n",
    "o = io.open(script,'a+', newline='\\n')\n",
    "\n",
    "\n",
    "o.write('module load pcangsd \\n')\n",
    "o.write('#gzip /home/jamcgirr/ph/data/vcfs/chr1_'+vcf_name+'.BEAGLE.PL \\n')\n",
    "\n",
    "o.write('python /home/jamcgirr/apps/pcangsd/pcangsd.py -beagle /home/jamcgirr/ph/data/vcfs/chr1_'+vcf_name+'_plate4_outliers_rm_no_TB.vcf.BEAGLE.PL.gz -o /home/jamcgirr/ph/data/angsd/PCAngsd/chr1_'+vcf_name+'_plate4_outliers_rm_no_TB.vcf -threads 4 \\n')\n",
    "o.write('#python /home/jamcgirr/apps/pcangsd/pcangsd.py -beagle /home/jamcgirr/ph/data/vcfs/chr2_'+vcf_name+'.BEAGLE.PL.gz -o /home/jamcgirr/ph/data/angsd/PCAngsd/chr2_'+vcf_name+' -threads 4 \\n')\n",
    "o.write('#python /home/jamcgirr/apps/pcangsd/pcangsd.py -beagle /home/jamcgirr/ph/data/vcfs/chr3_'+vcf_name+'.BEAGLE.PL.gz -o /home/jamcgirr/ph/data/angsd/PCAngsd/chr3_'+vcf_name+' -threads 4 \\n')\n",
    "\n",
    "\n",
    "o.write('\\n\\n#command to run: sbatch '+script)\n",
    "o.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "job_name = 'plink_PCA'\n",
    "vcf_name = 'ph_filtered_snps_minDP600_maxDP2000_maf0.05_minQ20_minMQ30_maxmiss0.5'\n",
    "#vcf_name = 'ph_filtered_snps_minDP600_maxDP2000_maf0.05_minQ30_maxmiss0.5'\n",
    "\n",
    "sbatch_header(job_name,'8','4','24')\n",
    "script = 'script_' + job_name + '.sh'\n",
    "o = io.open(script,'a+', newline='\\n')\n",
    "\n",
    "\n",
    "o.write('module load plink \\n')\n",
    "o.write('plink --file /home/jamcgirr/ph/data/vcfs/'+vcf_name+' --pca --out /home/jamcgirr/ph/data/plink/'+vcf_name+'\\n') \n",
    "\n",
    "\n",
    "outfile.write('\\n\\n#command to run: sbatch '+script)\n",
    "outfile.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "job_name = 'MDS'\n",
    "vcf_name = 'ph_filtered_snps_minDP600_maxDP2000_maf0.05_minQ20_minMQ30_maxmiss0.5'\n",
    "\n",
    "sbatch_header(job_name,'8','4','24')\n",
    "script = 'script_' + job_name + '.sh'\n",
    "o = io.open(script,'a+', newline='\\n')\n",
    "\n",
    "\n",
    "o.write('module load plink \\n')\n",
    "o.write('#plink --file /home/jamcgirr/ph/data/vcfs/'+vcf_name+' --genome --chr chr1 --out /home/jamcgirr/ph/data/plink/chr_1'+vcf_name+'\\n')\n",
    "o.write('#plink --file /home/jamcgirr/ph/data/vcfs/'+vcf_name+' --cluster --chr chr1 --out /home/jamcgirr/ph/data/plink/chr_1'+vcf_name+'\\n') \n",
    "o.write('plink --file /home/jamcgirr/ph/data/vcfs/'+vcf_name+' --read-genome /home/jamcgirr/ph/data/plink/chr_1'+vcf_name+'.genome --mds-plot 4 --cluster --chr chr1 --out /home/jamcgirr/ph/data/plink/chr_1'+vcf_name+'\\n') \n",
    "\n",
    "\n",
    "o.write('\\n\\n#command to run: sbatch '+script)\n",
    "o.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "job_name = 'NGSadmix'\n",
    "vcf_name = 'ph_filtered_snps_minDP600_maxDP2000_maf0.05_minQ20_minMQ30_maxmiss0.5'\n",
    "\n",
    "sbatch_header(job_name,'8','1','24')\n",
    "script = 'script_' + job_name + '.sh'\n",
    "o = io.open(script,'a+', newline='\\n')\n",
    "\n",
    "\n",
    "o.write('#git clone https://github.com/Rosemeis/pcangsd.git\\n')\n",
    "o.write('#cd pcangsd/\\n')\n",
    "o.write('#python setup.py build_ext --inplace\\n\\n')\n",
    "\n",
    "o.write('module load angsd \\n')\n",
    "o.write('NGSadmix -likes /home/jamcgirr/ph/data/vcfs/chr1_'+vcf_name+'.BEAGLE.PL.gz -K 2 -o /home/jamcgirr/ph/data/angsd/NGSadmix/k2_chr1_'+vcf_name+' -P 4 \\n')\n",
    "o.write('NGSadmix -likes /home/jamcgirr/ph/data/vcfs/chr1_'+vcf_name+'.BEAGLE.PL.gz -K 3 -o /home/jamcgirr/ph/data/angsd/NGSadmix/k3_chr1_'+vcf_name+' -P 4 \\n')\n",
    "o.write('NGSadmix -likes /home/jamcgirr/ph/data/vcfs/chr1_'+vcf_name+'.BEAGLE.PL.gz -K 4 -o /home/jamcgirr/ph/data/angsd/NGSadmix/k4_chr1_'+vcf_name+' -P 4 \\n')\n",
    "\n",
    "\n",
    "o.write('\\n\\n#command to run: sbatch '+script)\n",
    "o.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "job_name = 'evalAdmix'\n",
    "vcf_name = 'ph_filtered_snps_minDP600_maxDP2000_maf0.05_minQ20_minMQ30_maxmiss0.5'\n",
    "infiles = list(range(2,21))\n",
    "\n",
    "for infile in infiles:\n",
    "    script = 'script_' + infile + job_name + '.sh'\n",
    "    sbatch_header_loop(job_name,'16','4','24', infile)\n",
    "    o = io.open(script,'a+', newline='\\n')\n",
    "    \n",
    "    \n",
    "    o.write('module load angsd \\n')\n",
    "    o.write('/home/jamcgirr/apps/evalAdmix/evalAdmix -beagle /home/jamcgirr/ph/data/vcfs/chr1_'+vcf_name+'.BEAGLE.PL.gz -fname /home/jamcgirr/ph/data/angsd/NGSadmix/k'+str(infile)+'_chr1_'+vcf_name+'.fopt.gz -qname /home/jamcgirr/ph/data/angsd/NGSadmix/k'+str(infile)+'_chr1_'+vcf_name+'.qopt -o /home/jamcgirr/ph/data/angsd/NGSadmix/evalAdmix/eval_k'+str(infile)+'_chr1_'+vcf_name+' -P 4 \\n')\n",
    "\n",
    "    \n",
    "    o.write('\\n\\n#command to run: sbatch '+script)\n",
    "    o.close()\n",
    "\n",
    "# EST run time ~1hr 30min"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "job_name = '_subset_pops_vcfs'\n",
    "vcf_name = 'ph_filtered_snps_minDP600_maxDP2000_maf0.05_minQ20_minMQ30_maxmiss0.5'\n",
    "infiles = [\"BC17\",\"CA17\",\"PWS07\",\"PWS17\",\"PWS91\",\"PWS96\",\"SS06\",\"SS17\",\"SS96\",\"TB06\",\"TB17\",\"TB91\",\"TB96\",\"WA17\"]\n",
    "\n",
    "for infile in infiles:\n",
    "    script = 'script_' + infile + job_name + '.sh'\n",
    "    sbatch_header_loop(job_name,'8','4','24', infile)\n",
    "    o = io.open(script,'a+', newline='\\n')\n",
    "    \n",
    "    \n",
    "    o.write('module load samtools \\n')\n",
    "    o.write('module load bcftools \\\\n')\n",
    "    o.write('bcftools view -S /home/jamcgirr/ph/scripts/angsd/SFS/SFS_by_pop/'+infile+'_plates_1_through_5_rm.txt -Ov /home/jamcgirr/ph/data/vcfs/'+vcf_name+'.vcf > /home/jamcgirr/ph/data/vcfs/split_pops/population_'+infile+'_'+vcf_name+'_plates_1_5_rm.vcf \\n')\n",
    "\n",
    "    \n",
    "    o.write('\\n\\n#command to run: sbatch '+script)\n",
    "    o.close()\n",
    "    \n",
    "#EST run time ~30min"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# install angsd locally to use most recent version and avoid issues with vcf parsing\n",
    "# git clone https://github.com/samtools/htslib.git;\n",
    "# git clone https://github.com/angsd/angsd.git;\n",
    "# cd htslib;make;cd ../angsd;make HTSSRC=../htslib\n",
    "\n",
    "job_name = '_SFS'\n",
    "vcf_name = 'ph_filtered_snps_minDP600_maxDP2000_maf0.05_minQ20_minMQ30_maxmiss0.5'\n",
    "infiles = [\"BC17\",\"CA17\",\"PWS07\",\"PWS17\",\"PWS91\",\"PWS96\",\"SS06\",\"SS17\",\"SS96\",\"TB06\",\"TB17\",\"TB91\",\"TB96\",\"WA17\"]\n",
    "\n",
    "for infile in infiles:\n",
    "    script = 'script_' + infile + job_name + '.sh'\n",
    "    sbatch_header_loop(job_name,'8','4','21', infile)\n",
    "    o = io.open(script,'a+', newline='\\n')\n",
    "    \n",
    "    \n",
    "    # make saf from vcf\n",
    "    o.write('#/home/jamcgirr/apps/angsd_sep_20/angsd/angsd -doSaf 1 -vcf-pl /home/jamcgirr/ph/data/vcfs/split_pops/population_'+infile+'_'+vcf_name+'_plates_1_5_rm.vcf -out /home/jamcgirr/ph/data/angsd/SFS/split_pops/population_'+infile+'_'+vcf_name+' -anc /home/jamcgirr/ph/data/c_harengus/c.harengus.fa \\n')\n",
    "    # make folded sfs\n",
    "    o.write('#/home/jamcgirr/apps/angsd_sep_20/angsd/misc/realSFS /home/jamcgirr/ph/data/angsd/SFS/split_pops/population_'+infile+'_'+vcf_name+'.saf.idx -P 4 -fold 1 > /home/jamcgirr/ph/data/angsd/SFS/split_pops/folded/population_'+infile+'_'+vcf_name+'.sfs \\n')\n",
    "    # make unfolded sfs\n",
    "    o.write('/home/jamcgirr/apps/angsd_sep_20/angsd/misc/realSFS /home/jamcgirr/ph/data/angsd/SFS/split_pops/population_'+infile+'_'+vcf_name+'.saf.idx -P 4 > /home/jamcgirr/ph/data/angsd/SFS/split_pops/unfolded/population_'+infile+'_'+vcf_name+'.sfs \\n')\n",
    "\n",
    "    \n",
    "    o.write('\\n\\n#command to run: sbatch '+script)\n",
    "    o.close()\n",
    "\n",
    "# EST run time ~15min"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "# theta estimates using saf from vcf are way too high\n",
    "# using angsd to estimate saf from bam files generates reasonable theta\n",
    "# likely has to do with invariant site corrections and MAF cutoff\n",
    "\n",
    "# if using vcf sites, need to index sites.txt \n",
    "# /home/jamcgirr/apps/angsd/angsd sites index sites.txt\n",
    "\n",
    "job_name = '_thetas'\n",
    "vcf_name = 'ph_filtered_snps_minDP600_maxDP2000_maf0.05_minQ20_minMQ30_maxmiss0.5'\n",
    "infiles = [\"BC17\",\"CA17\",\"PWS07\",\"PWS17\",\"PWS91\",\"PWS96\",\"SS06\",\"SS17\",\"SS96\",\"TB06\",\"TB17\",\"TB91\",\"TB96\",\"WA17\"]\n",
    "\n",
    "for infile in infiles:\n",
    "    script = 'script_' + infile + job_name + '.sh'\n",
    "    sbatch_header_loop(job_name,'60','8','120', infile)\n",
    "    o = io.open(script,'a+', newline='\\n')\n",
    "    \n",
    "    \n",
    "    # make saf from bams\n",
    "    o.write('#/home/jamcgirr/apps/angsd_sep_20/angsd/angsd -bam /home/jamcgirr/ph/data/angsd/SFS/thetas/'+infile+'_bams_p_1_5_rm.txt -doSaf 1 -doMajorMinor 1 -doMaf 3 -anc /home/jamcgirr/ph/data/c_harengus/c.harengus.fa -minMapQ 30 -minQ 20 -GL 1 -P 8 -out /home/jamcgirr/ph/data/angsd/SFS/thetas/population_'+infile+'_minQ20_minMQ30 \\n')\n",
    "    # make unfolded sfs\n",
    "    o.write('#/home/jamcgirr/apps/angsd_sep_20/angsd/misc/realSFS /home/jamcgirr/ph/data/angsd/SFS/thetas/population_'+infile+'_minQ20_minMQ30.saf.idx -P 8 -nSites 100000 > /home/jamcgirr/ph/data/angsd/SFS/thetas/unfolded/population_'+infile+'_minQ20_minMQ30_unfolded.sfs \\n')\n",
    "    o.write('#/home/jamcgirr/apps/angsd_sep_20/angsd/misc/realSFS saf2theta /home/jamcgirr/ph/data/angsd/SFS/thetas/population_'+infile+'_minQ20_minMQ30.saf.idx -outname /home/jamcgirr/ph/data/angsd/SFS/thetas/unfolded/population_'+infile+'_minQ20_minMQ30 -sfs /home/jamcgirr/ph/data/angsd/SFS/thetas/unfolded/population_'+infile+'_minQ20_minMQ30_unfolded.sfs \\n')\n",
    "    o.write('#/home/jamcgirr/apps/angsd_sep_20/angsd/misc/thetaStat do_stat /home/jamcgirr/ph/data/angsd/SFS/thetas/unfolded/population_'+infile+'_minQ20_minMQ30.thetas.idx \\n')\n",
    "    o.write('#/home/jamcgirr/apps/angsd_sep_20/angsd/misc/thetaStat do_stat /home/jamcgirr/ph/data/angsd/SFS/thetas/unfolded/population_'+infile+'_minQ20_minMQ30.thetas.idx -win 50000 -step 10000 -outnames /home/jamcgirr/ph/data/angsd/SFS/thetas/unfolded/population_'+infile+'_minQ20_minMQ30_50kb_win_10kb_step \\n')\n",
    "    # make folded sfs\n",
    "    o.write('/home/jamcgirr/apps/angsd_sep_20/angsd/misc/realSFS /home/jamcgirr/ph/data/angsd/SFS/thetas/population_'+infile+'_minQ20_minMQ30.saf.idx -P 8 -nSites 100000 -fold 1 > /home/jamcgirr/ph/data/angsd/SFS/thetas/folded/population_'+infile+'_minQ20_minMQ30_folded.sfs \\n')\n",
    "    o.write('/home/jamcgirr/apps/angsd_sep_20/angsd/misc/realSFS saf2theta /home/jamcgirr/ph/data/angsd/SFS/thetas/population_'+infile+'_minQ20_minMQ30.saf.idx -outname /home/jamcgirr/ph/data/angsd/SFS/thetas/folded/population_'+infile+'_minQ20_minMQ30 -fold 1 -sfs /home/jamcgirr/ph/data/angsd/SFS/thetas/folded/population_'+infile+'_minQ20_minMQ30_folded.sfs \\n')\n",
    "    o.write('/home/jamcgirr/apps/angsd_sep_20/angsd/misc/thetaStat do_stat /home/jamcgirr/ph/data/angsd/SFS/thetas/folded/population_'+infile+'_minQ20_minMQ30.thetas.idx \\n')\n",
    "    o.write('/home/jamcgirr/apps/angsd_sep_20/angsd/misc/thetaStat do_stat /home/jamcgirr/ph/data/angsd/SFS/thetas/folded/population_'+infile+'_minQ20_minMQ30.thetas.idx -win 50000 -step 10000 -outnames /home/jamcgirr/ph/data/angsd/SFS/thetas/folded/population_'+infile+'_minQ20_minMQ30_50kb_win_10kb_step \\n')\n",
    "    o.write('#rm /home/jamcgirr/ph/data/angsd/SFS/thetas/population_'+infile+'_minQ20_minMQ30.saf.gz')\n",
    "    \n",
    "    \n",
    "    o.write('\\n\\n#run: sbatch '+script)\n",
    "    o.close()\n",
    "\n",
    "# EST run time ~"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Steps for Fst and PBS for andgsd folding update\n",
    "\n",
    "# 1. Create 2d sfs from saf\n",
    "\n",
    "job_name = '_2d_sfs'\n",
    "\n",
    "vcf_name = 'ph_filtered_snps_minDP600_maxDP2000_maf0.05_minQ20_minMQ30_maxmiss0.5'\n",
    "realSFS = '/home/jamcgirr/apps/angsd/misc/realSFS'\n",
    "wk_dir = '/home/jamcgirr/ph/data/angsd/SFS/fst_pbs/'\n",
    "saf_dir = '/home/jamcgirr/ph/data/angsd/SFS/split_pops/'\n",
    "\n",
    "infiles = [\"BC17_CA17\",\"BC17_WA17\",\"PWS07_PWS17\",\"PWS07_SS06\",\"PWS17_BC17\",\"PWS17_CA17\",\"PWS17_SS17\",\"PWS17_WA17\",\"PWS91_PWS07\",\"PWS91_PWS17\",\"PWS91_PWS96\",\"PWS96_PWS07\",\"PWS96_PWS17\",\"PWS96_SS96\",\"SS06_SS17\",\"SS17_BC17\",\"SS17_CA17\",\"SS17_WA17\",\"SS96_SS06\",\"SS96_SS17\",\"TB06_PWS07\",\"TB06_SS06\",\"TB06_TB17\",\"TB17_BC17\",\"TB17_CA17\",\"TB17_PWS17\",\"TB17_SS17\",\"TB17_WA17\",\"TB91_TB06\",\"TB91_TB17\",\"TB91_TB96\",\"TB96_PWS96\",\"TB96_SS96\",\"TB96_TB06\",\"TB96_TB17\",\"WA17_CA17\"]\n",
    "\n",
    "for infile in infiles:\n",
    "    script = 'script_' + infile + job_name + '.sh'\n",
    "    sbatch_header_loop(job_name,'8','1','24', infile)\n",
    "    o = io.open(script,'a+', newline='\\n')\n",
    "    \n",
    "    \n",
    "    pops = ''.join(infile).split(\"_\")\n",
    "    o.write(realSFS+' '+saf_dir+'population_'+pops[0]+'_'+vcf_name+'.saf.idx '+saf_dir+'population_'+pops[1]+'_'+vcf_name+'.saf.idx -fold 1 -P 4 > '+wk_dir+'folded_'+infile+'.sfs \\n')\n",
    "    o.write(realSFS+' '+saf_dir+'population_'+pops[0]+'_'+vcf_name+'.saf.idx '+saf_dir+'population_'+pops[1]+'_'+vcf_name+'.saf.idx -P 4 > '+wk_dir+'unfolded_'+infile+'.sfs \\n')\n",
    "\n",
    "    \n",
    "    o.write('\\n\\n#command to run: sbatch '+script)\n",
    "    o.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# 2. Estimate Fst and PBS\n",
    "\n",
    "job_name = '_fst_pbs_folded'\n",
    "\n",
    "vcf_name = 'ph_filtered_snps_minDP600_maxDP2000_maf0.05_minQ20_minMQ30_maxmiss0.5'\n",
    "realSFS = '/home/jamcgirr/apps/angsd/misc/realSFS'\n",
    "wk_dir = '/home/jamcgirr/ph/data/angsd/SFS/fst_pbs/'\n",
    "saf_dir = '/home/jamcgirr/ph/data/angsd/SFS/split_pops/'\n",
    "infiles = [\"BC17_WA17_CA17\",\"PWS17_SS17_BC17\",\"PWS17_SS17_CA17\",\"PWS17_SS17_WA17\",\"TB17_PWS17_BC17\",\"TB17_PWS17_CA17\",\"TB17_PWS17_SS17\",\"TB17_PWS17_WA17\",\"TB06_PWS07_SS06\",\"TB96_PWS96_SS96\",\"PWS91_PWS07_PWS17\",\"PWS91_PWS96_PWS07\",\"PWS91_PWS96_PWS17\",\"TB91_TB06_TB17\",\"TB91_TB96_TB06\",\"TB91_TB96_TB17\",\"SS96_SS06_SS17\"]\n",
    "\n",
    "for infile in infiles:\n",
    "    script = 'script_' + infile + job_name + '.sh'\n",
    "    sbatch_header_loop(job_name,'8','1','24', infile)\n",
    "    o = io.open(script,'a+', newline='\\n')\n",
    "    \n",
    "    \n",
    "    pops = ''.join(infile).split(\"_\")\n",
    "    #o.write(realSFS+' fst index '+saf_dir+'population_'+pops[0]+'_'+vcf_name+'.saf.idx '+saf_dir+'population_'+pops[1]+'_'+vcf_name+'.saf.idx -sfs '+wk_dir+'folded_'+pops[0]+'_'+pops[1]+'.sfs -fold 1 -fstout '+wk_dir+'folded_'+pops[0]+'_'+pops[1]+'persite \\n')\n",
    "    #o.write(realSFS+' fst index '+saf_dir+'population_'+pops[0]+'_'+vcf_name+'.saf.idx '+saf_dir+'population_'+pops[2]+'_'+vcf_name+'.saf.idx -sfs '+wk_dir+'folded_'+pops[0]+'_'+pops[2]+'.sfs -fold 1 -fstout '+wk_dir+'folded_'+pops[0]+'_'+pops[2]+'persite \\n')\n",
    "    #o.write(realSFS+' fst index '+saf_dir+'population_'+pops[1]+'_'+vcf_name+'.saf.idx '+saf_dir+'population_'+pops[2]+'_'+vcf_name+'.saf.idx -sfs '+wk_dir+'folded_'+pops[1]+'_'+pops[2]+'.sfs -fold 1 -fstout '+wk_dir+'folded_'+pops[1]+'_'+pops[2]+'persite \\n\\n')\n",
    "\n",
    "    o.write(realSFS+' fst index '+saf_dir+'population_'+pops[0]+'_'+vcf_name+'.saf.idx '+saf_dir+'population_'+pops[1]+'_'+vcf_name+'.saf.idx '+saf_dir+'population_'+pops[2]+'_'+vcf_name+'.saf.idx -sfs '+wk_dir+'folded_'+pops[0]+'_'+pops[1]+'.sfs -sfs '+wk_dir+'folded_'+pops[0]+'_'+pops[2]+'.sfs -sfs '+wk_dir+'folded_'+pops[1]+'_'+pops[2]+'.sfs -fold 1 -fstout '+wk_dir+'folded_'+pops[0]+'_'+pops[1]+'_'+pops[2]+'_persite \\n')\n",
    "    o.write(realSFS+' fst stats2 '+wk_dir+'folded_'+pops[0]+'_'+pops[1]+'_'+pops[2]+'_persite.fst.idx -win 50000 -step 10000 > '+wk_dir+'fst_pbs_50kb_win_10kb_step_folded_'+pops[0]+'_'+pops[1]+'_'+pops[2]+'.txt \\n')\n",
    "\n",
    "    \n",
    "    o.write('\\n\\n#command to run: sbatch '+script)\n",
    "    o.close()\n",
    "    \n",
    "# vcf_name = 'ph_filtered_snps_minDP600_maxDP2000_maf0.05_minQ20_minMQ30_maxmiss0.5'\n",
    "# realSFS = '/home/jamcgirr/apps/angsd/misc/realSFS'\n",
    "# wk_dir = '/home/jamcgirr/ph/data/angsd/SFS/fst_pbs/'\n",
    "# saf_dir = '/home/jamcgirr/ph/data/angsd/SFS/split_pops/'\n",
    "# \n",
    "# infiles = [\"BC17_WA17_CA17\",\"PWS17_SS17_BC17\",\"PWS17_SS17_CA17\",\"PWS17_SS17_WA17\",\"TB17_PWS17_BC17\",\"TB17_PWS17_CA17\",\"TB17_PWS17_SS17\",\"TB17_PWS17_WA17\",\"TB06_PWS07_SS06\",\"TB96_PWS96_SS96\",\"PWS91_PWS07_PWS17\",\"PWS91_PWS96_PWS07\",\"PWS91_PWS96_PWS17\",\"TB91_TB06_TB17\",\"TB91_TB96_TB06\",\"TB91_TB96_TB17\",\"SS96_SS06_SS17\"]\n",
    "# \n",
    "# job_name = '_fst_pbs_unfolded'\n",
    "# \n",
    "# for infile in infiles:\n",
    "#     script = 'script_' + infile + job_name + '.sh'\n",
    "#     sbatch_header_loop(job_name,'8','1','24', infile)\n",
    "#     o = io.open(script,'a+', newline='\\n')\n",
    "#     \n",
    "#     \n",
    "#     pops = ''.join(infile).split(\"_\")\n",
    "#     #o.write(realSFS+' fst index '+saf_dir+'population_'+pops[0]+'_'+vcf_name+'.saf.idx '+saf_dir+'population_'+pops[1]+'_'+vcf_name+'.saf.idx -sfs '+wk_dir+'folded_'+pops[0]+'_'+pops[1]+'.sfs -fold 1 -fstout '+wk_dir+'folded_'+pops[0]+'_'+pops[1]+'persite \\n')\n",
    "#     #o.write(realSFS+' fst index '+saf_dir+'population_'+pops[0]+'_'+vcf_name+'.saf.idx '+saf_dir+'population_'+pops[2]+'_'+vcf_name+'.saf.idx -sfs '+wk_dir+'folded_'+pops[0]+'_'+pops[2]+'.sfs -fold 1 -fstout '+wk_dir+'folded_'+pops[0]+'_'+pops[2]+'persite \\n')\n",
    "#     #o.write(realSFS+' fst index '+saf_dir+'population_'+pops[1]+'_'+vcf_name+'.saf.idx '+saf_dir+'population_'+pops[2]+'_'+vcf_name+'.saf.idx -sfs '+wk_dir+'folded_'+pops[1]+'_'+pops[2]+'.sfs -fold 1 -fstout '+wk_dir+'folded_'+pops[1]+'_'+pops[2]+'persite \\n\\n')\n",
    "# \n",
    "#     o.write(realSFS+' fst index '+saf_dir+'population_'+pops[0]+'_'+vcf_name+'.saf.idx '+saf_dir+'population_'+pops[1]+'_'+vcf_name+'.saf.idx '+saf_dir+'population_'+pops[2]+'_'+vcf_name+'.saf.idx -sfs '+wk_dir+'unfolded_'+pops[0]+'_'+pops[1]+'.sfs -sfs '+wk_dir+'unfolded_'+pops[0]+'_'+pops[2]+'.sfs -sfs '+wk_dir+'unfolded_'+pops[1]+'_'+pops[2]+'.sfs -fstout '+wk_dir+'unfolded_'+pops[0]+'_'+pops[1]+'_'+pops[2]+'_persite \\n')\n",
    "#     o.write(realSFS+' fst stats2 '+wk_dir+'unfolded_'+pops[0]+'_'+pops[1]+'_'+pops[2]+'_persite.fst.idx -win 50000 -step 10000 > '+wk_dir+'fst_pbs_50kb_win_10kb_step_unfolded_'+pops[0]+'_'+pops[1]+'_'+pops[2]+'.txt \\n')\n",
    "# \n",
    "#     \n",
    "#     o.write('\\n\\n#command to run: sbatch '+script)\n",
    "#     o.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "job_name = '_maf_for_dxy'\n",
    "vcf_name = 'ph_filtered_snps_minDP600_maxDP2000_maf0.05_minQ20_minMQ30_maxmiss0.5'\n",
    "infiles = [\"BC17\",\"CA17\",\"PWS07\",\"PWS17\",\"PWS91\",\"PWS96\",\"SS06\",\"SS17\",\"SS96\",\"TB06\",\"TB17\",\"TB91\",\"TB96\",\"WA17\"]\n",
    "\n",
    "for infile in infiles:\n",
    "    script = 'script_' + infile + job_name + '.sh'\n",
    "    sbatch_header_loop(job_name,'8','4','1', infile)\n",
    "    o = io.open(script,'a+', newline='\\n')\n",
    "    \n",
    "    \n",
    "    # make mafs from vcf\n",
    "    o.write('/home/jamcgirr/apps/angsd_sep_20/angsd/angsd -out /home/jamcgirr/ph/data/angsd/SFS/dxy/mafs/'+infile+'_vcf -fai /home/jamcgirr/ph/data/c_harengus/c.harengus.fa.fai -doMajorMinor 1 -doMaf 3 -vcf-pl /home/jamcgirr/ph/data/vcfs/split_pops/population_'+infile+'_ph_filtered_snps_minDP600_maxDP2000_maf0.05_minQ20_minMQ30_maxmiss0.5_plates_1_5_rm.vcf -P 4 \\n')\n",
    "    o.write('gzip -d /home/jamcgirr/ph/data/angsd/SFS/dxy/mafs/'+infile+'_vcf.mafs.gz \\n')\n",
    "    \n",
    "    \n",
    "    o.write('\\n\\n#command to run: sbatch '+script)\n",
    "    o.close()\n",
    "\n",
    "# EST run time ~ 5 minutes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "# LD with ngsLD\n",
    "chroms = [\"chr1\",\"chr2\",\"chr3\",\"chr4\",\"chr5\",\"chr6\",\"chr7\",\"chr8\",\"chr9\",\"chr10\",\"chr11\",\"chr12\",\"chr13\",\"chr14\",\"chr15\",\"chr16\",\"chr17\",\"chr18\",\"chr19\",\"chr20\",\"chr21\",\"chr22\",\"chr23\",\"chr24\",\"chr25\",\"chr26\"]\n",
    "chroms = [\"chr2\",\"chr3\",\"chr4\",\"chr5\",\"chr6\",\"chr7\",\"chr8\",\"chr9\",\"chr10\",\"chr11\",\"chr12\",\"chr13\",\"chr14\",\"chr15\",\"chr16\",\"chr17\",\"chr18\",\"chr19\",\"chr20\",\"chr21\",\"chr22\",\"chr23\",\"chr24\",\"chr25\",\"chr26\"]\n",
    "\n",
    "for chrom in chroms:\n",
    "    job_name = 'prune_'+chrom\n",
    "    \n",
    "    vcf_name = 'ph_filtered_snps_minDP600_maxDP2000_maf0.05_minQ20_minMQ30_maxmiss0.5_outliers_rm'\n",
    "    vcf_dir = '/home/jamcgirr/ph/data/vcfs/'\n",
    "    \n",
    "    sbatch_header(job_name,'8','1','24')\n",
    "    script = 'script_' + job_name + '.sh'\n",
    "    o = io.open(script,'a+', newline='\\n')    \n",
    "    \n",
    "    \n",
    "    o.write('module load bcftools \\n')\n",
    "    o.write('bcftools view '+vcf_dir+'vince/'+vcf_name+'.vcf.gz -Ov --regions '+chrom+' > '+vcf_dir+chrom+'_'+vcf_name+'.vcf \\n')\n",
    "    o.write('java -jar /home/jamcgirr/apps/jvarkit/dist/downsamplevcf.jar -n 20000 '+vcf_dir+chrom+'_'+vcf_name+'.vcf > '+vcf_dir+chrom+'_thin20k_'+vcf_name+'.vcf \\n')\n",
    "    o.write('module load vcftools \\n')\n",
    "    o.write('vcftools --vcf '+vcf_dir+chrom+'_thin20k_'+vcf_name+'.vcf --BEAGLE-PL --chr '+chrom+' --out '+vcf_dir+chrom+'_thin20k_'+vcf_name+' \\n')\n",
    "    \n",
    "    \n",
    "    o.write('\\n\\n#command to run: sbatch '+script)\n",
    "    o.close()\n",
    "    \n",
    "    # ~1 hr\n",
    "    \n",
    "    # Run to create position file for --pos flag for ngsLD:\n",
    "    \n",
    "    # awk '{ print $1 }' chr1_thin20k_ph_filtered_snps_minDP600_maxDP2000_maf0.05_minQ20_minMQ30_maxmiss0.5_outliers_rm.BEAGLE.PL > /home/jamcgirr/ph/data/vcfs/split_pops/chr1_thin20k_sites.txt\n",
    "    # sed 's/:/\\t/g' /home/jamcgirr/ph/data/vcfs/split_pops/chr1_thin20k_sites.txt > /home/jamcgirr/ph/data/vcfs/split_pops/test_pos_file2\n",
    "    # tail -n +2 /home/jamcgirr/ph/data/vcfs/split_pops/test_pos_file2 > /home/jamcgirr/ph/data/vcfs/split_pops/chr1_thin20k_sites.txt\n",
    "    # rm /home/jamcgirr/ph/data/vcfs/split_pops/test_pos_file2\n",
    "    \n",
    "    job_name = '_LD_'+chrom+'_20k_snps'\n",
    "    infiles = [\"BC17_64\",\"CA17_70\",\"PWS07_46\",\"PWS17_56\",\"PWS91_58\",\"PWS96_72\",\"SS06_41\",\"SS17_64\",\"SS96_78\",\"TB06_52\",\"TB17_72\",\"TB91_74\",\"TB96_73\",\"WA17_72\"]\n",
    "    infiles = [\"BC17_64\"]    \n",
    "    for infile in infiles:\n",
    "        script = 'script_' + infile + job_name + '.sh'\n",
    "        sbatch_header_loop(job_name,'8','1','24', infile)\n",
    "        o = io.open(script,'a+', newline='\\n')\n",
    "        \n",
    "        \n",
    "        pops = ''.join(infile).split(\"_\")\n",
    "        o.write('module load vcftools \\n')\n",
    "        o.write('vcftools --vcf '+vcf_dir+chrom+'_thin20k_'+vcf_name+'.vcf --keep /home/jamcgirr/ph/scripts/angsd/SFS/SFS_by_pop/'+pops[0]+'_plates_1_through_5_rm.txt --BEAGLE-PL --chr '+chrom+' --out '+vcf_dir+'split_pops/'+chrom+'_thin20k_'+pops[0]+'_'+vcf_name+' \\n')\n",
    "        o.write('gzip '+vcf_dir+'split_pops/'+chrom+'_thin20k_'+pops[0]+'_'+vcf_name+'.BEAGLE.PL \\n')\n",
    "        o.write('/home/jamcgirr/apps/ngsLD/ngsLD --geno '+vcf_dir+'split_pops/'+chrom+'_thin20k_'+pops[0]+'_'+vcf_name+'.BEAGLE.PL.gz --pos '+vcf_dir+'split_pops/'+chrom+'_thin20k_sites.txt --n_ind '+pops[1]+' --n_sites 20000 --out /home/jamcgirr/ph/data/angsd/ngsLD/'+chrom+'_thin20k_'+pops[0]+'_'+vcf_name+'_ld.txt \\n')\n",
    "    \n",
    "        \n",
    "        o.write('\\n\\n#run: sbatch '+script)\n",
    "        o.close()\n",
    "    \n",
    "# Rscript --vanilla --slave /home/jamcgirr/apps/ngsLD/scripts/fit_LDdecay.R --ld_files /home/jamcgirr/ph/data/angsd/ngsLD/ld_files.list.txt --fit_level 3 --out test_plot.pdf\n",
    "# Rscript --vanilla --slave /home/jamcgirr/apps/ngsLD/scripts/fit_LDdecay.R --ld_files /home/jamcgirr/ph/data/angsd/ngsLD/ld_files.list_lg.txt --max_kb_dist 100 --fit_level 3 --out test_plot_lg.pdf\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "# LD with plink\n",
    "\n",
    "job_name = 'LD'\n",
    "vcf_name = 'ph_filtered_snps_minDP600_maxDP2000_maf0.05_minQ20_minMQ30_maxmiss0.5_outliers_rm'\n",
    "\n",
    "sbatch_header(job_name,'8','8','24')\n",
    "script = 'script_' + job_name + '.sh'\n",
    "o = io.open(script,'a+', newline='\\n')\n",
    "\n",
    "\n",
    "o.write('module load plink \\n')\n",
    "o.write('plink --file /home/jamcgirr/ph/data/vcfs/'+vcf_name+' --indep-pairwise 100 10 0.8 --r2 --out /home/jamcgirr/ph/data/plink/'+vcf_name+' --threads 8 \\n') \n",
    "\n",
    "\n",
    "#run sbatch submission \n",
    "o.write('\\n\\n#command to run: sbatch '+script)\n",
    "o.close()\n",
    "\n",
    "# ~2 min\n",
    "\n",
    "# LD for each population\n",
    "job_name = '_LD'\n",
    "infiles = [\"BC17\",\"CA17\",\"PWS07\",\"PWS17\",\"PWS91\",\"PWS96\",\"SS06\",\"SS17\",\"SS96\",\"TB06\",\"TB17\",\"TB91\",\"TB96\",\"WA17\"]\n",
    "vcf_name = 'ph_filtered_snps_minDP600_maxDP2000_maf0.05_minQ20_minMQ30_maxmiss0.5_plates_1_5_rm'\n",
    "for infile in infiles:\n",
    "    script = 'script_' + infile + job_name + '.sh'\n",
    "    sbatch_header_loop(job_name,'8','4','1', infile)\n",
    "    o = io.open(script,'a+', newline='\\n')\n",
    "\n",
    "    \n",
    "    o.write('module load vcftools \\n')\n",
    "    o.write('vcftools --vcf /home/jamcgirr/ph/data/vcfs/split_pops/population_'+infile+'_'+vcf_name+'.vcf --plink --out /home/jamcgirr/ph/data/plink/population_'+infile+'_'+vcf_name+' \\n\\n')\n",
    "\n",
    "    o.write('module load plink \\n')\n",
    "    o.write('plink --file /home/jamcgirr/ph/data/plink/population_'+infile+'_'+vcf_name+' --indep-pairwise 100 10 0.8 --r2 --out /home/jamcgirr/ph/data/plink/population_'+infile+'_'+vcf_name+' --threads 4 \\n') \n",
    "\n",
    "    o.write('\\n\\n#command to run: sbatch '+script)\n",
    "    o.close()\n",
    "# ~ 5 min"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "'''\n",
    "job_name = 'popvae_PWS'\n",
    "\n",
    "sbatch_header(job_name,'32','4','24')\n",
    "script = 'script_' + job_name + '.sh'\n",
    "o = io.open(script,'a+', newline='\\n')  \n",
    "   \n",
    "    \n",
    "o.write('module load samtools \\n')\n",
    "o.write('module load bcftools \\n')\n",
    "o.write('cat /home/jamcgirr/ph/data/angsd/SFS/thetas/PWS91_bams_p_1_5_rm.txt /home/jamcgirr/ph/data/angsd/SFS/thetas/PWS96_bams_p_1_5_rm.txt /home/jamcgirr/ph/data/angsd/SFS/thetas/PWS07_bams_p_1_5_rm.txt /home/jamcgirr/ph/data/angsd/SFS/thetas/PWS17_bams_p_1_5_rm.txt > /home/jamcgirr/ph/data/popvae/pops_PWS.txt \\n')\n",
    "o.write('sed -i \\'s/\\/home\\/eoziolor\\/phpopg\\/data\\/align\\///g\\' /home/jamcgirr/ph/data/popvae/pops_PWS.txt \\n')\n",
    "o.write('sed -i \\'s/.bam//g\\' /home/jamcgirr/ph/data/popvae/pops_PWS.txt \\n')\n",
    "\n",
    "o.write('bcftools view -S /home/jamcgirr/ph/data/popvae/pops_PWS.txt -Oz /home/jamcgirr/ph/data/vcfs/vince/ph_filtered_snps_minDP600_maxDP2000_maf0.05_minQ20_minMQ30_maxmiss0.5_outliers_rm.vcf.gz -R /home/jamcgirr/ph/data/moments/ld_prune/random_thin_keep.txt --threads 4 > /home/jamcgirr/ph/data/popvae/PWS_random_thin.vcf.gz \\n')\n",
    "o.write('source /home/jamcgirr/apps/my_python3.7/bin/activate \\n')\n",
    "o.write('popvae.py --infile /home/jamcgirr/ph/data/popvae/PWS_random_thin.vcf.gz --out /home/jamcgirr/ph/data/popvae/PWS_random_thin --seed 42 \\n')\n",
    "\n",
    "\n",
    "o.write('\\n\\n#run: sbatch '+script)\n",
    "o.close()\n",
    "\n",
    "###################################\n",
    "job_name = 'popvae_TB'\n",
    "\n",
    "sbatch_header(job_name,'32','4','24')\n",
    "script = 'script_' + job_name + '.sh'\n",
    "o = io.open(script,'a+', newline='\\n')  \n",
    "   \n",
    "    \n",
    "o.write('module load samtools \\n')\n",
    "o.write('module load bcftools \\n')\n",
    "o.write('cat /home/jamcgirr/ph/data/angsd/SFS/thetas/TB91_bams_p_1_5_rm.txt /home/jamcgirr/ph/data/angsd/SFS/thetas/TB96_bams_p_1_5_rm.txt /home/jamcgirr/ph/data/angsd/SFS/thetas/TB06_bams_p_1_5_rm.txt /home/jamcgirr/ph/data/angsd/SFS/thetas/TB17_bams_p_1_5_rm.txt > /home/jamcgirr/ph/data/popvae/pops_TB.txt \\n')\n",
    "o.write('sed -i \\'s/\\/home\\/eoziolor\\/phpopg\\/data\\/align\\///g\\' /home/jamcgirr/ph/data/popvae/pops_TB.txt \\n')\n",
    "o.write('sed -i \\'s/.bam//g\\' /home/jamcgirr/ph/data/popvae/pops_TB.txt \\n')\n",
    "\n",
    "o.write('bcftools view -S /home/jamcgirr/ph/data/popvae/pops_TB.txt -Oz /home/jamcgirr/ph/data/vcfs/vince/ph_filtered_snps_minDP600_maxDP2000_maf0.05_minQ20_minMQ30_maxmiss0.5_outliers_rm.vcf.gz -R /home/jamcgirr/ph/data/moments/ld_prune/random_thin_keep.txt --threads 4 > /home/jamcgirr/ph/data/popvae/TB_random_thin.vcf.gz \\n')\n",
    "o.write('source /home/jamcgirr/apps/my_python3.7/bin/activate \\n')\n",
    "o.write('popvae.py --infile /home/jamcgirr/ph/data/popvae/TB_random_thin.vcf.gz --out /home/jamcgirr/ph/data/popvae/TB_random_thin --seed 42 \\n')\n",
    "\n",
    "\n",
    "o.write('\\n\\n#run: sbatch '+script)\n",
    "o.close()\n",
    "\n",
    "###################################\n",
    "job_name = 'popvae_SS'\n",
    "\n",
    "sbatch_header(job_name,'32','4','24')\n",
    "script = 'script_' + job_name + '.sh'\n",
    "o = io.open(script,'a+', newline='\\n')  \n",
    "   \n",
    "    \n",
    "o.write('module load samtools \\n')\n",
    "o.write('module load bcftools \\n')\n",
    "o.write('cat /home/jamcgirr/ph/data/angsd/SFS/thetas/SS96_bams_p_1_5_rm.txt /home/jamcgirr/ph/data/angsd/SFS/thetas/SS06_bams_p_1_5_rm.txt /home/jamcgirr/ph/data/angsd/SFS/thetas/SS17_bams_p_1_5_rm.txt > /home/jamcgirr/ph/data/popvae/pops_SS.txt \\n')\n",
    "o.write('sed -i \\'s/\\/home\\/eoziolor\\/phpopg\\/data\\/align\\///g\\' /home/jamcgirr/ph/data/popvae/pops_SS.txt \\n')\n",
    "o.write('sed -i \\'s/.bam//g\\' /home/jamcgirr/ph/data/popvae/pops_SS.txt \\n')\n",
    "\n",
    "o.write('bcftools view -S /home/jamcgirr/ph/data/popvae/pops_SS.txt -Oz /home/jamcgirr/ph/data/vcfs/vince/ph_filtered_snps_minDP600_maxDP2000_maf0.05_minQ20_minMQ30_maxmiss0.5_outliers_rm.vcf.gz -R /home/jamcgirr/ph/data/moments/ld_prune/random_thin_keep.txt --threads 4 > /home/jamcgirr/ph/data/popvae/SS_random_thin.vcf.gz \\n')\n",
    "o.write('source /home/jamcgirr/apps/my_python3.7/bin/activate \\n')\n",
    "o.write('popvae.py --infile /home/jamcgirr/ph/data/popvae/SS_random_thin.vcf.gz --out /home/jamcgirr/ph/data/popvae/SS_random_thin --seed 42 \\n')\n",
    "\n",
    "\n",
    "o.write('\\n\\n#run: sbatch '+script)\n",
    "o.close()\n",
    "###################################\n",
    "job_name = 'popvae_NEPac17'\n",
    "\n",
    "sbatch_header(job_name,'32','4','24')\n",
    "script = 'script_' + job_name + '.sh'\n",
    "o = io.open(script,'a+', newline='\\n')  \n",
    "   \n",
    "    \n",
    "o.write('module load samtools \\n')\n",
    "o.write('module load bcftools \\n')\n",
    "o.write('cat /home/jamcgirr/ph/data/angsd/SFS/thetas/PWS17_bams_p_1_5_rm.txt /home/jamcgirr/ph/data/angsd/SFS/thetas/SS17_bams_p_1_5_rm.txt /home/jamcgirr/ph/data/angsd/SFS/thetas/BC17_bams_p_1_5_rm.txt /home/jamcgirr/ph/data/angsd/SFS/thetas/WA17_bams_p_1_5_rm.txt > /home/jamcgirr/ph/data/popvae/pops_NEPac17.txt \\n')\n",
    "o.write('sed -i \\'s/\\/home\\/eoziolor\\/phpopg\\/data\\/align\\///g\\' /home/jamcgirr/ph/data/popvae/pops_NEPac17.txt \\n')\n",
    "o.write('sed -i \\'s/.bam//g\\' /home/jamcgirr/ph/data/popvae/pops_NEPac17.txt \\n')\n",
    "\n",
    "o.write('bcftools view -S /home/jamcgirr/ph/data/popvae/pops_NEPac17.txt -Oz /home/jamcgirr/ph/data/vcfs/vince/ph_filtered_snps_minDP600_maxDP2000_maf0.05_minQ20_minMQ30_maxmiss0.5_outliers_rm.vcf.gz -R /home/jamcgirr/ph/data/moments/ld_prune/random_thin_keep.txt --threads 4 > /home/jamcgirr/ph/data/popvae/NEPac17_random_thin.vcf.gz \\n')\n",
    "o.write('source /home/jamcgirr/apps/my_python3.7/bin/activate \\n')\n",
    "o.write('popvae.py --infile /home/jamcgirr/ph/data/popvae/NEPac17_random_thin.vcf.gz --out /home/jamcgirr/ph/data/popvae/NEPac17_random_thin --seed 42 \\n')\n",
    "\n",
    "\n",
    "o.write('\\n\\n#run: sbatch '+script)\n",
    "o.close()\n",
    "'''\n",
    "###################################\n",
    "job_name = 'popvae_PWS_SS'\n",
    "\n",
    "sbatch_header(job_name,'32','4','72')\n",
    "script = 'script_' + job_name + '.sh'\n",
    "o = io.open(script,'a+', newline='\\n')  \n",
    "   \n",
    "    \n",
    "o.write('module load samtools \\n')\n",
    "o.write('module load bcftools \\n')\n",
    "o.write('cat /home/jamcgirr/ph/data/angsd/SFS/thetas/PWS17_bams_p_1_5_rm.txt /home/jamcgirr/ph/data/angsd/SFS/thetas/SS17_bams_p_1_5_rm.txt /home/jamcgirr/ph/data/angsd/SFS/thetas/PWS91_bams_p_1_5_rm.txt /home/jamcgirr/ph/data/angsd/SFS/thetas/PWS96_bams_p_1_5_rm.txt /home/jamcgirr/ph/data/angsd/SFS/thetas/PWS07_bams_p_1_5_rm.txt /home/jamcgirr/ph/data/angsd/SFS/thetas/SS96_bams_p_1_5_rm.txt /home/jamcgirr/ph/data/angsd/SFS/thetas/SS06_bams_p_1_5_rm.txt > /home/jamcgirr/ph/data/popvae/pops_PWS_SS.txt \\n')\n",
    "o.write('sed -i \\'s/\\/home\\/eoziolor\\/phpopg\\/data\\/align\\///g\\' /home/jamcgirr/ph/data/popvae/pops_PWS_SS.txt \\n')\n",
    "o.write('sed -i \\'s/.bam//g\\' /home/jamcgirr/ph/data/popvae/pops_PWS_SS.txt \\n')\n",
    "\n",
    "o.write('bcftools view -S /home/jamcgirr/ph/data/popvae/pops_PWS_SS.txt -Oz /home/jamcgirr/ph/data/vcfs/vince/ph_filtered_snps_minDP600_maxDP2000_maf0.05_minQ20_minMQ30_maxmiss0.5_outliers_rm.vcf.gz -R /home/jamcgirr/ph/data/moments/ld_prune/random_thin_keep.txt --threads 4 > /home/jamcgirr/ph/data/popvae/PWS_SS_random_thin.vcf.gz \\n')\n",
    "o.write('source /home/jamcgirr/apps/my_python3.7/bin/activate \\n')\n",
    "o.write('popvae.py --infile /home/jamcgirr/ph/data/popvae/PWS_SS_random_thin.vcf.gz --out /home/jamcgirr/ph/data/popvae/PWS_SS_random_thin --seed 42 \\n')\n",
    "\n",
    "\n",
    "o.write('\\n\\n#run: sbatch '+script)\n",
    "o.close()\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "job_name = '_fst_perms'\n",
    "vcf_name = 'ph_filtered_snps_minDP600_maxDP2000_maf0.05_minQ20_minMQ30_maxmiss0.5_outliers_rm'\n",
    "realSFS = '/home/jamcgirr/apps/angsd/misc/realSFS'\n",
    "wk_dir = '/home/jamcgirr/ph/data/angsd/SFS/fst_pbs/perms/'\n",
    "pop_n = {\"BC17\":\"64\",\"CA17\":\"70\",\"PWS07\":\"46\",\"PWS17\":\"56\",\"PWS91\":\"58\",\"PWS96\":\"72\",\"SS06\":\"41\",\"SS17\":\"64\",\"SS96\":\"78\",\"TB06\":\"52\",\"TB17\":\"72\",\"TB91\":\"74\",\"TB96\":\"73\",\"WA17\":\"72\"}\n",
    "infiles = [\"TB17_PWS17\"]\n",
    "\n",
    "for infile in infiles:\n",
    "    script = 'script_' + infile + job_name + '.sh'\n",
    "    sbatch_header_loop(job_name,'8','4','72', infile)\n",
    "    o = io.open(script,'a+', newline='\\n')\n",
    "    \n",
    "    pops = ''.join(infile).split(\"_\")\n",
    "\n",
    "    o.write('cat /home/jamcgirr/ph/scripts/angsd/SFS/SFS_by_pop/'+pops[0]+'_plates_1_through_5_rm.txt /home/jamcgirr/ph/scripts/angsd/SFS/SFS_by_pop/'+pops[1]+'_plates_1_through_5_rm.txt > '+wk_dir+'pops'+pops[0]+pops[1]+'.txt \\n')\n",
    "    o.write('shuf '+wk_dir+'pops'+pops[0]+pops[1]+'.txt > '+wk_dir+'shufpops'+pops[0]+pops[1]+'.txt \\n')\n",
    "    o.write('head -'+pop_n[pops[0]]+' '+wk_dir+'shufpops'+pops[0]+pops[1]+'.txt > '+wk_dir+'shufpop'+pops[0]+'.txt \\n')\n",
    "    o.write('tail -'+pop_n[pops[1]]+' '+wk_dir+'shufpops'+pops[0]+pops[1]+'.txt > '+wk_dir+'shufpop'+pops[1]+'.txt \\n\\n')\n",
    "\n",
    "    o.write('module load samtools \\n')\n",
    "    o.write('module load bcftools \\n')\n",
    "    o.write('bcftools view -S '+wk_dir+'shufpop'+pops[0]+'.txt -Ov /home/jamcgirr/ph/data/vcfs/'+vcf_name+'.vcf.gz > '+wk_dir+'shufpop'+pops[0]+'.vcf \\n')\n",
    "    o.write('bcftools view -S '+wk_dir+'shufpop'+pops[1]+'.txt -Ov /home/jamcgirr/ph/data/vcfs/'+vcf_name+'.vcf.gz > '+wk_dir+'shufpop'+pops[1]+'.vcf \\n\\n')\n",
    "    \n",
    "    o.write('/home/jamcgirr/apps/angsd_sep_20/angsd/angsd -doSaf 1 -vcf-pl '+wk_dir+'shufpop'+pops[0]+'.vcf -out '+wk_dir+'shufpop'+pops[0]+' -anc /home/jamcgirr/ph/data/c_harengus/c.harengus.fa \\n')\n",
    "    o.write('/home/jamcgirr/apps/angsd_sep_20/angsd/angsd -doSaf 1 -vcf-pl '+wk_dir+'shufpop'+pops[1]+'.vcf -out '+wk_dir+'shufpop'+pops[1]+' -anc /home/jamcgirr/ph/data/c_harengus/c.harengus.fa \\n')\n",
    "\n",
    "    o.write(realSFS+' '+wk_dir+'shufpop'+pops[0]+'.saf.idx '+wk_dir+'shufpop'+pops[1]+'.saf.idx -P 4 -fold 1 > '+wk_dir+'shufpops'+pops[0]+pops[1]+'.sfs \\n')\n",
    "    o.write(realSFS+' fst index '+wk_dir+'shufpop'+pops[0]+'.saf.idx '+wk_dir+'shufpop'+pops[1]+'.saf.idx -sfs '+wk_dir+'shufpops'+pops[0]+pops[1]+'.sfs -fold 1 -fstout '+wk_dir+'shufpops'+pops[0]+pops[1]+'_persite \\n')\n",
    "    o.write(realSFS+' fst stats2 '+wk_dir+'shufpops'+pops[0]+pops[1]+'_persite.fst.idx -win 50000 -step 10000 > '+wk_dir+'shufpops'+pops[0]+pops[1]+'_fst_pbs_50kb_win_10kb_step_folded.txt \\n')\n",
    "\n",
    "    \n",
    "    o.write('\\n\\n#command to run: sbatch '+script)\n",
    "    o.close()\n",
    "    \n",
    "#EST run time ~8hr"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# using mpileup.bcf files with PL Format for angsd input\n",
    "# for some stupid fucking reason, angsd throws a seg fault halfway through chr1 during dosaf\n",
    "# so nevermind about this approach. \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# lets try downsampling populations to the smallest n (SS06 = 41)\n",
    "# run angsd direct on bam files\n",
    "\n",
    "# if using vcf sites, need to index sites.txt \n",
    "# /home/jamcgirr/apps/angsd/angsd sites index sites.txt\n",
    "\n",
    "job_name = '_angsd_downsample_sfs'\n",
    "wk_dir = \"/home/jamcgirr/ph/data/angsd/SFS/downsample/\"\n",
    "#vcf_name = 'ph_filtered_snps_minDP600_maxDP2000_maf0.05_minQ20_minMQ30_maxmiss0.5'\n",
    "infiles = [\"BC17\",\"CA17\",\"PWS07\",\"PWS17\",\"PWS91\",\"PWS96\",\"SS06\",\"SS17\",\"SS96\",\"TB06\",\"TB17\",\"TB91\",\"TB96\",\"WA17\"]\n",
    "infiles = [\"PWS07\",\"SS96\",\"TB06\",\"TB17\",\"TB91\",\"TB96\",\"WA17\"]\n",
    "#infiles = [\"BC17\"]\n",
    "\n",
    "for infile in infiles:\n",
    "    script = 'script_' + infile + job_name + '.sh'\n",
    "    sbatch_header_loop(job_name,'40','8','48', infile)\n",
    "    o = io.open(script,'a+', newline='\\n')\n",
    "    \n",
    "    \n",
    "    o.write('shuf /home/jamcgirr/ph/data/angsd/SFS/bamlist_test/'+infile+'_bams_p_1_5_rm.txt | head -41 > '+wk_dir+'downsample_bams_'+infile+'.txt \\n\\n')\n",
    "    # make saf from bams\n",
    "    #o.write('/home/jamcgirr/apps/angsd_sep_20/angsd/angsd -bam '+wk_dir+'downsample_bams_'+infile+'.txt -doSaf 1 -doMajorMinor 1 -doMaf 3 -anc /home/jamcgirr/ph/data/c_harengus/c.harengus.fa -minMapQ 30 -minQ 20 -GL 1 -P 8 -out '+wk_dir+infile+'_minQ20_minMQ30 \\n\\n')\n",
    "    # make saf from bams using filters\n",
    "    o.write('/home/jamcgirr/apps/angsd_sep_20/angsd/angsd -bam '+wk_dir+'downsample_bams_'+infile+'.txt -doSaf 1 -doMajorMinor 1 -doMaf 3 -doCounts 1 -doGlf 3 -anc /home/jamcgirr/ph/data/c_harengus/c.harengus.fa -ref /home/jamcgirr/ph/data/c_harengus/c.harengus.fa -minMapQ 30 -minQ 20 -GL 1 -P 8 -uniqueOnly 1 -remove_bads 1 -only_proper_pairs 1 -trim 0 -C 50 -minInd 10 -setMinDepth 10 -setMaxDepth 100 -out '+wk_dir+infile+'_minQ20_minMQ30 \\n\\n')\n",
    "    # make folded sfs\n",
    "    o.write('/home/jamcgirr/apps/angsd_sep_20/angsd/misc/realSFS '+wk_dir+infile+'_minQ20_minMQ30.saf.idx -P 8 -fold 1 -nSites 100000000 > '+wk_dir+infile+'_minQ20_minMQ30_folded.sfs \\n')\n",
    "    # make unfolded sfs\n",
    "    o.write('/home/jamcgirr/apps/angsd_sep_20/angsd/misc/realSFS '+wk_dir+infile+'_minQ20_minMQ30.saf.idx -P 8 -nSites 100000000 > '+wk_dir+infile+'_minQ20_minMQ30_unfolded.sfs \\n')\n",
    "\n",
    "    o.write('\\n\\n#run: sbatch '+script)\n",
    "    o.close()\n",
    "\n",
    "# EST run time ~ 1 day\n",
    "\n",
    "# use the -nSites 100000000 flag due to memory limits\n",
    "# this breaks up genome into about 7 chunks\n",
    "# this gives you seven lines in the output.sfs\n",
    "# need to sum columns to make final sfs for following step\n",
    "\n",
    "job_name = '_angsd_downsample_thetas'\n",
    "saf_dir = \"/home/jamcgirr/ph/data/angsd/SFS/downsample/saf/\"\n",
    "fold_dir = \"/home/jamcgirr/ph/data/angsd/SFS/downsample/sfs/folded/\"\n",
    "unfold_dir = \"/home/jamcgirr/ph/data/angsd/SFS/downsample/sfs/folded/\"\n",
    "thetas_dir = \"/home/jamcgirr/ph/data/angsd/SFS/downsample/thetas/\"\n",
    "#vcf_name = 'ph_filtered_snps_minDP600_maxDP2000_maf0.05_minQ20_minMQ30_maxmiss0.5'\n",
    "#infiles = [\"BC17\",\"CA17\",\"PWS07\",\"PWS17\",\"PWS91\",\"PWS96\",\"SS06\",\"SS17\",\"SS96\",\"TB06\",\"TB17\",\"TB91\",\"TB96\",\"WA17\"]\n",
    "#infiles = [\"CA17\"]\n",
    "\n",
    "for infile in infiles:\n",
    "    script = 'script_' + infile + job_name + '.sh'\n",
    "    sbatch_header_loop(job_name,'30','8','24', infile)\n",
    "    o = io.open(script,'a+', newline='\\n')\n",
    "    \n",
    "    # folded\n",
    "    o.write('/home/jamcgirr/apps/angsd_sep_20/angsd/misc/realSFS saf2theta '+saf_dir+infile+'_minQ20_minMQ30.saf.idx -outname '+thetas_dir+'folded/'+infile+'_minQ20_minMQ30 -fold 1 -sfs '+fold_dir+infile+'_minQ20_minMQ30_folded.sfs \\n')\n",
    "    o.write('/home/jamcgirr/apps/angsd_sep_20/angsd/misc/thetaStat do_stat '+thetas_dir+'folded/'+infile+'_minQ20_minMQ30.thetas.idx \\n')\n",
    "    o.write('/home/jamcgirr/apps/angsd_sep_20/angsd/misc/thetaStat do_stat '+thetas_dir+'folded/'+infile+'_minQ20_minMQ30.thetas.idx -win 50000 -step 10000 -outnames '+thetas_dir+'folded/'+infile+'_minQ20_minMQ30_50kb_win_10kb_step \\n\\n')\n",
    "    # unfolded\n",
    "    o.write('/home/jamcgirr/apps/angsd_sep_20/angsd/misc/realSFS saf2theta '+saf_dir+infile+'_minQ20_minMQ30.saf.idx -outname '+thetas_dir+'unfolded/'+infile+'_minQ20_minMQ30 -sfs '+fold_dir+infile+'_minQ20_minMQ30_folded.sfs \\n')\n",
    "    o.write('/home/jamcgirr/apps/angsd_sep_20/angsd/misc/thetaStat do_stat '+thetas_dir+'unfolded/'+infile+'_minQ20_minMQ30.thetas.idx \\n')\n",
    "    o.write('/home/jamcgirr/apps/angsd_sep_20/angsd/misc/thetaStat do_stat '+thetas_dir+'unfolded/'+infile+'_minQ20_minMQ30.thetas.idx -win 50000 -step 10000 -outnames '+thetas_dir+'unfolded/'+infile+'_minQ20_minMQ30_50kb_win_10kb_step \\n')\n",
    "    \n",
    "    \n",
    "    o.write('\\n\\n#run: sbatch '+script)\n",
    "    o.close()\n",
    "\n",
    "# EST run time ~4hr"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# sfs from downsample_41 saf using sites from plink pruned master vcf looks REAL dumb\n",
    "# try using genome-wide LD with genotype likelihoods\n",
    "# doGlf and ngsLD \n",
    "#https://github.com/fgvieira/ngsLD/issues/1\n",
    "\n",
    "# ngsLD doesn't like .glf input. try angsd doGeno to get .geno.gz\n",
    "\n",
    "# cool, geno works when you split by chr in a pop. Now LD prune in one pop across all chrs and use those sites to make sfs.\n",
    "\n",
    "job_name = '_ngsLD_downsample'\n",
    "glf_dir = \"/home/jamcgirr/ph/data/angsd/SFS/downsample/glf/\"\n",
    "#infiles = [\"BC17\",\"CA17\",\"PWS07\",\"PWS17\",\"PWS91\",\"PWS96\",\"SS06\",\"SS17\",\"SS96\",\"TB06\",\"TB17\",\"TB91\",\"TB96\",\"WA17\"]\n",
    "infiles = [\"PWS07\"]    \n",
    "for infile in infiles:\n",
    "    script = 'script_' + infile + job_name + '.sh'\n",
    "    sbatch_header_loop(job_name,'8','8','24', infile)\n",
    "    o = io.open(script,'a+', newline='\\n')\n",
    "    \n",
    "    o.write('#gzip -d '+saf_dir+infile+'_minQ20_minMQ30.glf.gz \\n')\n",
    "    o.write('/home/jamcgirr/apps/ngsLD/ngsLD --geno '+glf_dir+infile+'_minQ20_minMQ30.glf.gz --n_ind 41 --max_kb_dist 5 --n_threads 8 --out /home/jamcgirr/ph/data/angsd/ngsLD/downsample/'+infile+'_minQ20_minMQ30_ld.txt \\n')\n",
    "\n",
    "    \n",
    "    o.write('\\n\\n#run: sbatch '+script)\n",
    "    o.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Fst and PBS using downsample 41 samples\n",
    "\n",
    "# 1. Create 2d sfs from saf\n",
    "\n",
    "job_name = '_2d_sfs_downsample'\n",
    "\n",
    "realSFS = '/home/jamcgirr/apps/angsd_sep_20/angsd/misc/realSFS'\n",
    "wk_dir = '/home/jamcgirr/ph/data/angsd/SFS/downsample/fst/'\n",
    "saf_dir = '/home/jamcgirr/ph/data/angsd/SFS/downsample/saf/'\n",
    "\n",
    "infiles = [\"BC17_CA17\",\"BC17_WA17\",\"PWS07_PWS17\",\"PWS07_SS06\",\"PWS17_BC17\",\"PWS17_CA17\",\"PWS17_SS17\",\"PWS17_WA17\",\"PWS91_PWS07\",\"PWS91_PWS17\",\"PWS91_PWS96\",\"PWS96_PWS07\",\"PWS96_PWS17\",\"PWS96_SS96\",\"SS06_SS17\",\"SS17_BC17\",\"SS17_CA17\",\"SS17_WA17\",\"SS96_SS06\",\"SS96_SS17\",\"TB06_PWS07\",\"TB06_SS06\",\"TB06_TB17\",\"TB17_BC17\",\"TB17_CA17\",\"TB17_PWS17\",\"TB17_SS17\",\"TB17_WA17\",\"TB91_TB06\",\"TB91_TB17\",\"TB91_TB96\",\"TB96_PWS96\",\"TB96_SS96\",\"TB96_TB06\",\"TB96_TB17\",\"WA17_CA17\"]\n",
    "infiles = [\"BC17_CA17\"]\n",
    "\n",
    "for infile in infiles:\n",
    "    script = 'script_' + infile + job_name + '.sh'\n",
    "    sbatch_header_loop(job_name,'8','8','24', infile)\n",
    "    o = io.open(script,'a+', newline='\\n')\n",
    "    \n",
    "    \n",
    "    pops = ''.join(infile).split(\"_\")\n",
    "    o.write(realSFS+' '+saf_dir+pops[0]+'_minQ20_minMQ30.saf.idx '+saf_dir+pops[1]+'_minQ20_minMQ30.saf.idx -fold 1 -P 8 -nSites 100000000 > '+wk_dir+'folded/'+infile+'_folded_chunks.sfs \\n')\n",
    "    o.write(realSFS+' '+saf_dir+pops[0]+'_minQ20_minMQ30.saf.idx '+saf_dir+pops[1]+'_minQ20_minMQ30.saf.idx -fold 1 -P 8 > '+wk_dir+'folded/'+infile+'_folded_full.sfs \\n')\n",
    "\n",
    "    #o.write(realSFS+' '+saf_dir+pops[0]+'_minQ20_minMQ30.saf.idx '+saf_dir+pops[1]+'_minQ20_minMQ30.saf.idx -P 8 > '+wk_dir+'/unfolded/'+infile+'unfolded.sfs \\n')\n",
    "\n",
    "    \n",
    "    o.write('\\n\\n#command to run: sbatch '+script)\n",
    "    o.close()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
