---
title: "moments"
author: "Joe McGirr"
date: "9/23/2020"
output: html_document
editor_options: 
  chunk_output_type: console
---

## 1D SFS 
```{python, echo = FALSE,results='hide'}

import moments
import math
fs = moments.Spectrum([88113,1222,132,445,285,176,41,20,73,276,192,68,50,23,3,2,47,50,0,0,362])

with open('C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/unfolded/population_PWS07_ph_filtered_snps_minDP600_maxDP2000_maf0.05_minQ20_minMQ30_maxmiss0.5.sfs') as data:
    sfs = data.read().split()
    sfs = sfs[math.ceil(((len(sfs)/2)*0.05)+2):]
    sfs = [float(i) for i in sfs] 
    sfs = [round(num) for num in sfs]
    sfs = [0] + sfs
    fs = moments.Spectrum(sfs)
    thetaW = fs.Watterson_theta()
    print(thetaW)
    #print(math.ceil(((len(sfs)/2)*0.05)+1))
    nSites = sum(sfs[1:-1])
    ne = thetaW / 2.0e-9 / nSites / 4
    print(ne)
    #print(nSites)
#f = open("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/unfolded/population_PWS91_ph_filtered_snps_minDP600_maxDP2000_maf0.05_minQ20_minMQ30_maxmiss0.5.sfs", "r")
#print(f)
#from numpy import loadtxt
#lines = loadtxt("filename.dat", comments="#", delimiter=",", unpack=False)
#lines = text_file.read("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/unfolded/population_PWS91_ph_filtered_snps_minDP600_maxDP2000_maf0.05_minQ20_minMQ30_maxmiss0.5.sfs").split(' ')
#D = fs . Tajima_D ()
#pi = fs.pi()
#thetaW = fs.Watterson_theta()

```

```{r}
sfs<-(scan(  'C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/folded/population_BC17_ph_filtered_snps_minDP600_maxDP2000_maf0.05_minQ20_minMQ30_maxmiss0.5.sfs'))

#png("C:/Users/jmcgirr/Desktop/Fig_3.png", height = 5, width = 8, units = 'in', res = 300)

barplot(sfs[-c(1,length(sfs))]) #plot variable sites 
abline(v=(((length(sfs)/2)*0.05)+1), col = "red")

#dev.off()

```

## 2D SFS
```{python, echo = FALSE,results='hide'}

import moments
import matplotlib
dd = moments.Misc.make_data_dict('C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/moments/2d_sfs_dadi/BC17_CA17_dadi_snp.data')
fs = moments.Spectrum.from_data_dict(dd,pop_ids=[ 'BC17' , 'CA17'],projections=[16 , 16],polarized = True)
S = fs.S()
Fst = fs.Fst()
moments.Plotting.plot_single_2d_sfs(fs, vmin =1)
matplotlib.pyplot.savefig("my_plot") 


```











# Rough Ne estimate using waterson's theta
```{r}

# http://evomics.org/learning/population-and-speciation-genomics/2018-population-and-speciation-genomics/angsd-activity-sfs-fst-pbs/

pop_names = c("PWS91","PWS96","PWS07","PWS17","TB91","TB96","TB06","TB17","SS96","SS06","SS17","BC17","WA17","CA17")
pop <- "PWS07"
ne_s <- c()
for (pop in pop_names){
x<-scan(paste("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/unfolded/population_",pop,"_ph_filtered_snps_minDP600_maxDP2000_maf0.05_minQ20_minMQ30_maxmiss0.5.sfs", sep = ""))

x <- x[c(round((((length(x)/2)*0.05)+2)):length(x))]

nSites<-sum(x)   #Number of sites where we have data
#nSeg<-sum(x[c(-1)])    #Number of segregating sites
nSeg <- nSites
an <- function(n) sum(1/1:(n-1))
thetaW <- nSeg/an(length(x[c(-1)])/2) # Wattersons Theta
#print(pop)
#print("effective population size")
ne <- thetaW / 2.0e-9 / nSites / 4 # effective population size
#print(ne)
ne_s <- c(ne_s, ne)
}

```





https://github.com/z0on/AFS-analysis-with-moments
http://computationalgenomics.bioinformatics.ucla.edu/portfolio/simon-gravel-inferring-demographic-history-using-moments-cgsi-2019/

https://bitbucket.org/simongravel/moments/src/master/moments/

https://dadi.readthedocs.io/en/latest/user-guide/importing-data/

https://github.com/isaacovercast/easySFS

```{python, echo = FALSE,results='hide'}
import moments
dd = moments.Misc.make_data_dict('C:/Users/jmcgirr/Desktop/2pops_dadi.data')
fs = moments.Spectrum.from_data_dict(dd,pop_ids=[ 'pop0' , 'pop1'],projections=[16 , 16],polarized = True)
S = fs.S()
Fst = fs.Fst()
moments.Plotting.plot_single_2d_sfs(fs, vmin =1)
import matplotlib
matplotlib.pyplot.savefig("C:/Users/jmcgirr/Desktop/my_plot2") 


```



```{r}
#https://github.com/z0on/AFS-analysis-with-moments
#http://computationalgenomics.bioinformatics.ucla.edu/portfolio/simon-gravel-inferring-demographic-history-using-moments-cgsi-2019/

#https://bitbucket.org/simongravel/moments/src/master/moments/

#https://dadi.readthedocs.io/en/latest/user-guide/importing-data/

#https://github.com/isaacovercast/easySFS

#https://github.com/dportik/moments_pipeline

#https://notebook.community/claudiuskerth/PhDthesis/Data_analysis/SNP-indel-calling/dadi/dadiExercises/.ipynb_checkpoints/First_Steps_with_dadi-checkpoint

```


#!/bin/bash

#SBATCH --job-name=moments_test
#SBATCH --mem=8G 
#SBATCH --ntasks=4 
#SBATCH -e moments_test_%A_%a.err 
#SBATCH --time=1:00:00 
#SBATCH --mail-user=jamcgirr@ucdavis.edu ##email you when job starts,ends,etc
#SBATCH --mail-type=ALL
#SBATCH -p high 

#/home/jamcgirr/apps/angsd_sep_20/angsd/angsd -bam p1.bams -doSaf 1 -anc /home/jamcgirr/ph/data/c_harengus/c.harengus.fa -GL 1 -P 8 -out p1 -minMapQ 1 -minQ 20 
#/home/jamcgirr/apps/angsd_sep_20/angsd/angsd -bam p2.bams -doSaf 1 -anc /home/jamcgirr/ph/data/c_harengus/c.harengus.fa -GL 1 -P 8 -out p2 -minMapQ 1 -minQ 20                                                                                                                                                       
 
#/home/jamcgirr/apps/angsd_sep_20/angsd/misc/realSFS p1.saf.idx p2.saf.idx -bootstrap 10 -P 4 -r chr1:1-100000 > p12_boots.sfs
/home/jamcgirr/apps/angsd_sep_20/angsd/misc/realSFS p1.saf.idx  -P 4 -r chr1:1-100000 > p1.sfs
/home/jamcgirr/apps/angsd_sep_20/angsd/misc/realSFS p2.saf.idx  -P 4 -r chr1:1-100000 > p2.sfs

#split -l 1 -d --additional-suffix=.sfs p12_boots.sfs p12_

# mv p12_00.sfs p12_10.sfs
# mv p12_01.sfs p12_1.sfs
# mv p12_02.sfs p12_2.sfs
# mv p12_03.sfs p12_3.sfs
# mv p12_04.sfs p12_4.sfs
# mv p12_05.sfs p12_5.sfs
# mv p12_06.sfs p12_6.sfs
# mv p12_07.sfs p12_7.sfs
# mv p12_08.sfs p12_8.sfs
# mv p12_09.sfs p12_9.sfs

module load R
Rscript /home/jamcgirr/apps/moments/AFS-analysis-with-moments/modSel_write.R contrast=p12 args="p1 p2 16 16 0.02 0.005"
source /home/jamcgirr/apps/my_python3.7/bin/activate

sleep 1 && /home/jamcgirr/apps/moments/AFS-analysis-with-moments/multimodel_inference/py3/IM.py p12_1.sfs p1 p2 16 16 0.02 0.005 >>p12.modsel

#/home/jamcgirr/apps/angsd_sep_20/angsd/misc/realSFS dadi p1.saf.idx p2.saf.idx -sfs p1.sfs -sfs p2.sfs -r chr1:1-100000 -ref /home/jamcgirr/ph/data/c_harengus/c.harengus.fa -anc /home/jamcgirr/ph/data/c_harengus/c.harengus.fa > dadiout
#
#/home/jamcgirr/apps/moments/AFS-analysis-with-moments/multimodel_inference/realsfs2dadi.pl dadiout 10 10 > 2pops_dadi.data
#
#/home/jamcgirr/apps/moments/AFS-analysis-with-moments/multimodel_inference/py3/2dAFS.py 2pops_dadi.data p1 p2 16 16


# get Fst for 2d SFS and plot SFS

python
import moments
dd = moments.Misc.make_data_dict('2pops_dadi.data')
fs = moments.Spectrum.from_data_dict(dd,pop_ids=[ 'pop0' , 'pop1'],projections=[16 , 16],polarized = True)
S = fs.S()
Fst = fs.Fst()
moments.Plotting.plot_single_2d_sfs(fs, vmin =1)
import matplotlib
matplotlib.pyplot.savefig("my_plot") 


# get pi and tajD for 1d SFS

python
import moments
dd = moments.Misc.make_data_dict('2pops_dadi.data')
#fs = moments.Spectrum([88113,1222,132,445,285,176,41,20,73,276,192,68,50,23,3,2,47,50,0,0,362])
D = fs . Tajima_D ()
pi = fs.pi()
thetaW = fs.Watterson_theta()

#command to run: sbatch script_moments_test.sh