---
title: "Time Series AF Shifts"
author: "Joe McGirr"
date: '`r Sys.Date()`'
output:
  html_document:
    code_folding: hide
    collapsed: no
    df_print: paged
    number_sections: yes
    theme: paper
    toc: yes
    toc_depth: 5
    toc_float: yes
  html_notebook:
    toc: yes
    toc_depth: 5
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(ggplot2)
library(tidyverse)
library(reticulate)
library(reshape2)
library(plyranges)
library(seqinr)
library(windowscanr)
#BiocManager::install("seqinr")

  # Get upper triangle of the correlation matrix
  get_upper_tri <- function(cormat){
    cormat[lower.tri(cormat)]<- NA
    return(cormat)
  }

    get_lower_tri<-function(cormat){
    cormat[upper.tri(cormat)] <- NA
    return(cormat)
  }

# color-blind friendly 
# Wong, B. Points of view: Color blindness. Nat Methods (2011).
bla <- "#000000"
blu <- "#0072b2"
grb <- "#56b4e9"
lir <- "#cc79a7"
gre <- "#009e73"
red <- "#d55e00"
org <- "#e69f00"
yel <- "#f0e442"
gry<-  '#BBBBBB'

```

# Summary

This notebook analyzes allele frequency shifts between Pacific Herring samples collected in 1991, 1996, 2006-7, and 2017. The summary statistic âˆ†z is the difference in arcsine square-root transformed minor allele frequencies (Following methods used for the [Longshanks experiment](https://elifesciences.org/articles/42014#s4)). 

The data set is a filtered `.vcf` with 200k SNPs across 892 Pacific herring low coverage genomes:

minimum depth = 600  
maximum depth = 2000  
minimum base quality = 20  
minimum mapping quality = 30  
minor allele frequency > 5%  
genotyping rate > 50% for each population (grouped by samping location and year)

`ph_filtered_snps_minDP600_maxDP2000_minQ20_minMQ30_NS0.5_maf0.05.vcf`

Written by Joe McGirr, postdoc in Andrew Whitehead's lab.

# Output allele frequncies from vcf


example script
```{bash,eval=FALSE,class.source = 'fold-show'}
#!/bin/bash

#SBATCH --job-name=BC17_freqs
#SBATCH --mem=8G 
#SBATCH --ntasks=4 
#SBATCH -e BC17_freqs_%A_%a.err 
#SBATCH --time=1:00:00 
#SBATCH --mail-user=jamcgirr@ucdavis.edu ##email you when job starts,ends,etc
#SBATCH --mail-type=ALL
#SBATCH -p high 

module load bcftools
bcftools query -f '%CHROM %POS %AF\n' /home/jamcgirr/ph/data/vcfs/split_pops/maf05/population_BC17_ph_filtered_snps_minDP600_maxDP2000_minQ20_minMQ30_NS0.5_maf0.05.vcf.gz > /home/jamcgirr/ph/data/freqs/population_BC17_ph_filtered_snps_minDP600_maxDP2000_minQ20_minMQ30_NS0.5_maf0.05_freqs.txt 


#command to run: sbatch script_BC17_freqs.sh
```

## Generate scripts

```{python,results='hide', eval = FALSE}
# get allele frequencies for arcsine transformed shifts
job_name = '_freqs'

infiles = ["BC17","CA17","PWS07","PWS17","PWS91","PWS96","SS06","SS17","SS96","TB06","TB17","TB91","TB96","WA17"]
vcf_name = 'ph_filtered_snps_minDP600_maxDP2000_minQ20_minMQ30_NS0.5_maf0.05'
for infile in infiles:
    script = 'script_' + infile + job_name + '.sh'
    sbatch_header_loop(job_name,'8','4','1', infile)
    o = io.open(script,'a+', newline='\n') 

    o.write('module load bcftools\n')

    o.write('bcftools query -f \'%CHROM %POS %AF\\n\' /home/jamcgirr/ph/data/vcfs/split_pops/maf05/population_'+infile+'_'+vcf_name+'.vcf.gz > /home/jamcgirr/ph/data/freqs/population_'+infile+'_'+vcf_name+'_freqs.txt \n')

    o.write('\n\n#command to run: sbatch '+script)
    o.close()
            
# calculate direct from AC and AN        
#bcftools query -f '%CHROM %POS %AN %AC{0}\n' /home/jamcgirr/ph/data/vcfs/split_pops/population_PWS07_ph_filtered_snps_minDP600_maxDP2000_maf0.05_minQ20_minMQ30_maxmiss0.5_plates_1_5_rm.vcf > /home/jamcgirr/ph/data/freqs/PWS07_freqs_test.txt

```

# Transform AFs and average across sliding windows

```{r, message=FALSE, warning=FALSE, fig.width= 10, fig.height= 7,class.source = 'fold-show'}

#AF is calculated as AC/AN, where AC is total ALT allele count and AN is total allele called in genotypes

t0 <- read.table("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/freqs/population_PWS91_ph_filtered_snps_minDP600_maxDP2000_minQ20_minMQ30_NS0.5_maf0.05_freqs.txt", header = FALSE, stringsAsFactors = FALSE, col.names = c("chr","pos","t0_AF"))
t1 <- read.table("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/freqs/population_PWS96_ph_filtered_snps_minDP600_maxDP2000_minQ20_minMQ30_NS0.5_maf0.05_freqs.txt", header = FALSE, stringsAsFactors = FALSE, col.names = c("chr","pos","t1_AF"))
t2 <- read.table("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/freqs/population_PWS07_ph_filtered_snps_minDP600_maxDP2000_minQ20_minMQ30_NS0.5_maf0.05_freqs.txt", header = FALSE, stringsAsFactors = FALSE, col.names = c("chr","pos","t2_AF"))
t3 <- read.table("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/freqs/population_PWS17_ph_filtered_snps_minDP600_maxDP2000_minQ20_minMQ30_NS0.5_maf0.05_freqs.txt", header = FALSE, stringsAsFactors = FALSE, col.names = c("chr","pos","t3_AF"))


t0 <- read.table("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/freqs/population_TB91_ph_filtered_snps_minDP600_maxDP2000_minQ20_minMQ30_NS0.5_maf0.05_freqs.txt", header = FALSE, stringsAsFactors = FALSE, col.names = c("chr","pos","t0_AF"))
t1 <- read.table("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/freqs/population_TB96_ph_filtered_snps_minDP600_maxDP2000_minQ20_minMQ30_NS0.5_maf0.05_freqs.txt", header = FALSE, stringsAsFactors = FALSE, col.names = c("chr","pos","t1_AF"))
t2 <- read.table("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/freqs/population_TB06_ph_filtered_snps_minDP600_maxDP2000_minQ20_minMQ30_NS0.5_maf0.05_freqs.txt", header = FALSE, stringsAsFactors = FALSE, col.names = c("chr","pos","t2_AF"))
t3 <- read.table("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/freqs/population_TB17_ph_filtered_snps_minDP600_maxDP2000_minQ20_minMQ30_NS0.5_maf0.05_freqs.txt", header = FALSE, stringsAsFactors = FALSE, col.names = c("chr","pos","t3_AF"))


freqs <- data.frame(chr = t0$chr,
                    pos = t0$pos,
                    t0_AF = t0$t0_AF,
                    t1_AF = t1$t1_AF,
                    t2_AF = t2$t2_AF,
                    t3_AF = t3$t3_AF)

freqs$t0_transformed_freq <- asin(sqrt(freqs$t0_AF))
freqs$t1_transformed_freq <- asin(sqrt(freqs$t1_AF))
freqs$t2_transformed_freq <- asin(sqrt(freqs$t2_AF))
freqs$t3_transformed_freq <- asin(sqrt(freqs$t3_AF))

freqs$zt01 <- freqs$t1_transformed_freq - freqs$t0_transformed_freq
freqs$zt02 <- freqs$t2_transformed_freq - freqs$t0_transformed_freq
freqs$zt03 <- freqs$t3_transformed_freq - freqs$t0_transformed_freq
freqs$zt12 <- freqs$t2_transformed_freq - freqs$t1_transformed_freq
freqs$zt13 <- freqs$t3_transformed_freq - freqs$t1_transformed_freq
freqs$zt23 <- freqs$t3_transformed_freq - freqs$t2_transformed_freq


head(freqs)
nrow(freqs)
#write.table(freqs, "C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/freqs/persite/TB_shifts_persite.txt")


freqs_win <- winScan(x = freqs, 
                   groups = "chr", 
                   position = "pos", 
                   values = c("zt01","zt02","zt03","zt12","zt13","zt23"), 
                   win_size = 50000,
                   win_step = 10000,
                   funs = c("mean"))
freqs_win <- na.omit(freqs_win)
freqs_win <- freqs_win %>% mutate(chr = as.numeric(gsub("chr", "", chr)))
freqs_win$chr_num <- factor(freqs_win$chr, levels = c(1:26))

#write.table(freqs_win, "C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/freqs/window/TB_shifts_50kb_win_10kb_step.txt")

```

# Compare magnitude of AF shifts between populations

```{r, message=FALSE, warning=FALSE, fig.width= 10, fig.height= 7,class.source = 'fold-show'}
freqs_win_p <- read.table("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/freqs/window/PWS_shifts_50kb_win_10kb_step.txt")
freqs_win_t <- read.table("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/freqs/window/TB_shifts_50kb_win_10kb_step.txt")
freqs_win_s <- read.table("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/freqs/window/SS_shifts_50kb_win_10kb_step.txt")

all_pops <- data.frame(chr = freqs_win_p$chr,
                       win_mid = freqs_win_p$win_mid,
                       zt01_p = freqs_win_p$zt01_mean,
                       zt02_p = freqs_win_p$zt02_mean,
                       zt03_p = freqs_win_p$zt03_mean,
                       zt12_p = freqs_win_p$zt12_mean,
                       zt13_p = freqs_win_p$zt13_mean,
                       zt23_p = freqs_win_p$zt23_mean,
                       zt01_t = freqs_win_t$zt01_mean,
                       zt02_t = freqs_win_t$zt02_mean,
                       zt03_t = freqs_win_t$zt03_mean,
                       zt12_t = freqs_win_t$zt12_mean,
                       zt13_t = freqs_win_t$zt13_mean,
                       zt23_t = freqs_win_t$zt23_mean,
                       zt12_s = freqs_win_s$zt12_mean,
                       zt13_s = freqs_win_s$zt13_mean,
                       zt23_s = freqs_win_s$zt23_mean)



vio_plot <- all_pops[,-c(1:2)]
all_pops$chr_num <- factor(all_pops$chr, levels = c(1:26))




zp <- data.frame(z91_96 = all_pops$zt01_p,pop = "PWS")
zt <- data.frame(z91_96 = all_pops$zt01_t,pop = "TB")
z <- rbind(zp,zt)

p<-ggplot(z, aes(x=z91_96, fill=pop)) +
   geom_density(alpha=0.4)+
   scale_fill_manual(name = "Population",values=c(red,blu))+
   theme_minimal()
p


zp <- data.frame(z96_07 = all_pops$zt12_p,pop = "PWS")
zt <- data.frame(z96_07 = all_pops$zt12_t,pop = "TB")
zs <- data.frame(z96_07 = all_pops$zt12_s,pop = "SS")

z <- rbind(zp,zt,zs)

p<-ggplot(z, aes(x=z96_07, fill=pop)) +
   geom_density(alpha=0.4)+
   scale_fill_manual(name = "Population",values=c(red,yel,blu))+
   theme_minimal()
p

zp <- data.frame(z07_17 = all_pops$zt23_p,pop = "PWS")
zt <- data.frame(z07_17 = all_pops$zt23_t,pop = "TB")
zs <- data.frame(z07_17 = all_pops$zt23_s,pop = "SS")

z <- rbind(zp,zt,zs)

p<-ggplot(z, aes(x=z07_17, fill=pop)) +
   geom_density(alpha=0.4)+
   scale_fill_manual(name = "Population",values=c(red,yel,blu))+
   theme_minimal()
p




p1 <- ggplot(vio_plot,aes(factor(Year.Collected), MEAN_COVERAGE))+
      geom_violin()+ xlab("year collected") + ylab("mean coverage")+
      #geom_boxplot(width=0.1)
      theme_minimal()+
      geom_jitter(height = 0, width = 0.1, alpha = 0.1)

my_cols <- c(grb,org,red,red,red,red,yel,yel,yel,blu,blu,blu,blu,lir, "black")

p1 <- vio_plot %>% 
      gather(key="MesureType", value="Val") %>%
      ggplot( aes(x=MesureType, y=Val, fill=MesureType)) +
      geom_violin() + xlab("\n") + ylab(expression(paste(Delta, "z")))+
      scale_fill_manual(name = "Population",values=my_cols,guide=FALSE)+
      #stat_summary(fun=mean, geom="line")+
      theme_minimal()+
      geom_jitter(height = 0, width = 0.1, alpha = 0.1)

      #geom_jitter(height = 0, width = 0.1, alpha = 0.1)

#png("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/figs/population_structure/taj_d.png", height = 5, width = 7, units = 'in', res = 600)

p1

```

# Identify windows where allele frequencies:

  - increased between each sampling period
  - decreased between each sampling period
  - increased between 91 and 96, decreased between 96 and 07, and continued to decrease between 07 and 17
  - decreased between 91 and 96, increased between 96 and 07, and continued to increase between 07 and 17

```{r, message=FALSE, warning=FALSE, fig.width= 10, fig.height= 7,class.source = 'fold-show'}

freqs_win <- read.table("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/freqs/window/PWS_shifts_50kb_win_10kb_step.txt")

# identify windows where allele frequencies increased between each sampling period
increased_freq <- freqs_win[freqs_win$zt01_mean > 0 &
                            freqs_win$zt12_mean > 0 &
                            freqs_win$zt23_mean > 0,]
nrow(increased_freq)
nrow(increased_freq) / nrow(freqs_win)
# identify windows where allele frequencies increased between each sampling period
decreased_freq <- freqs_win[freqs_win$zt01_mean < 0 &
                            freqs_win$zt12_mean < 0 &
                            freqs_win$zt23_mean < 0,]
nrow(decreased_freq)
nrow(decreased_freq) / nrow(freqs_win)
# identify windows where allele frequencies increased between 91 and 96
# and decreased between 96 and 07
# and continued to decrease between 07 and 17
up_dn_dn_freq <- freqs_win[freqs_win$zt01_mean > 0 &
                            freqs_win$zt12_mean < 0 &
                            freqs_win$zt23_mean < 0,]
nrow(up_dn_dn_freq)
nrow(up_dn_dn_freq) / nrow(freqs_win)
# identify windows where allele frequencies decreased between 91 and 96
# and increased between 96 and 07
# and continued to increase between 07 and 17
dn_up_up_freq <- freqs_win[freqs_win$zt01_mean < 0 &
                            freqs_win$zt12_mean > 0 &
                            freqs_win$zt23_mean > 0,]

nrow(dn_up_up_freq)
nrow(dn_up_up_freq) / nrow(freqs_win)


```

# Most extreme shifts

```{r, message=FALSE, warning=FALSE, fig.width= 10, fig.height= 7,class.source = 'fold-show'}

freqs_win <- read.table("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/freqs/window/PWS_shifts_50kb_win_10kb_step.txt")
freqs_win <- read.table("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/freqs/window/TB_shifts_50kb_win_10kb_step.txt")


# identify windows where allele frequencies increased between each sampling period
top_increased <- freqs_win[freqs_win$zt01_mean >= quantile(freqs_win$zt01_mean, .9)[[1]] &
                           freqs_win$zt12_mean >= quantile(freqs_win$zt12_mean, .9)[[1]] &
                           freqs_win$zt23_mean >= quantile(freqs_win$zt23_mean, .9)[[1]],]

# identify windows where allele frequencies decreased between each sampling period
top_decreased <- freqs_win[freqs_win$zt01_mean <= quantile(freqs_win$zt01_mean, .1)[[1]] &
                           freqs_win$zt12_mean <= quantile(freqs_win$zt12_mean, .1)[[1]] &
                           freqs_win$zt23_mean <= quantile(freqs_win$zt23_mean, .1)[[1]],]

# identify windows where allele frequencies increased between 91 and 96
# and decreased between 96 and 07
# and continued to decrease between 07 and 17
top_up_dn_dn_freq <- freqs_win[freqs_win$zt01_mean >= quantile(freqs_win$zt01_mean, .9)[[1]] &
                               freqs_win$zt12_mean <= quantile(freqs_win$zt12_mean, .1)[[1]] &
                               freqs_win$zt23_mean <= quantile(freqs_win$zt23_mean, .1)[[1]],]

# identify windows where allele frequencies decreased between 91 and 96
# and increased between 96 and 07
# and continued to increase between 07 and 17
top_dn_up_up_freq <- freqs_win[freqs_win$zt01_mean <= quantile(freqs_win$zt01_mean, .1)[[1]] &
                               freqs_win$zt12_mean >= quantile(freqs_win$zt12_mean, .9)[[1]] &
                               freqs_win$zt23_mean >= quantile(freqs_win$zt23_mean, .9)[[1]],]

nrow(top_increased)# / nrow(freqs_win)
nrow(top_decreased)# / nrow(freqs_win)
nrow(top_up_dn_dn_freq)# / nrow(freqs_win)
nrow(top_dn_up_up_freq)# / nrow(freqs_win)


print("number of 50kb windows with z > genome-wide 99th %ile in each comparison")
nrow(tops)
tops$chr_num <- factor(tops$chr, levels = c(1:26))

p <- ggplot(freqs_win, aes(x = win_mid, y = zt01_mean)) + 
     geom_point(size = 1, color = gry,alpha = 0.4, shape = 1)+
     theme_minimal()+
     theme(axis.text.x=element_blank())+
     ylab(expression(paste(Delta, "z"^"2")))+ xlab("")+ 
     #geom_smooth(method = "loess", se = FALSE, span = 1/10, color = blu)+
     facet_wrap(~chr_num, ncol = 9)

p     
#p + geom_vline(aes(xintercept = win_mid), tops)

# just chr 1

freqs_win1 <- freqs_win[freqs_win$chr == "1",]

p01 <- ggplot(freqs_win1, aes(x = win_mid, y = zt01_mean)) + 
     geom_point(size = 2, color = gry,alpha = 0.6, shape = 1)+
     theme_minimal()+
     theme(axis.text.x=element_blank())+
     ylab(expression(paste(Delta, "z"^"2")))+ xlab("")
p02 <- ggplot(freqs_win1, aes(x = win_mid, y = zt01_mean)) + 
     geom_point(size = 2, color = gry,alpha = 0.6, shape = 1)+
     theme_minimal()+
     theme(axis.text.x=element_blank())+
     ylab(expression(paste(Delta, "z"^"2")))+ xlab("")
p03 <- ggplot(freqs_win1, aes(x = win_mid, y = zt01_mean)) + 
     geom_point(size = 2, color = gry,alpha = 0.6, shape = 1)+
     theme_minimal()+
     theme(axis.text.x=element_blank())+
     ylab(expression(paste(Delta, "z"^"2")))+ xlab("")

 

ggarrange(bxp, dp, bp + rremove("x.text"), 
          labels = c("A", "B", "C"),
          ncol = 2, nrow = 2)

```

## PWS 1991 --> 1996

## What does the arcsine transformation do
Maybe use logit transformation instead?

http://strata.uga.edu/8370/rtips/proportions.html

```{r, message=FALSE, warning=FALSE, fig.width= 10, fig.height= 7}

t0 <- c(0,0.01,0.02,0.03,0.1,0.2,0.3,1)
t1 <- rev(t0)

#t0 <- c(0.01,0.05,0.06,0.07,0.10,0,1)
#t1 <- c(0.02,0.06,0.10,0.20,0.05,1,0)

af <- data.frame(t0 = t0, t1 = t1)
head(af)

af$t0_transformed_freq <- asin(sqrt(af$t0))
af$t1_transformed_freq <- asin(sqrt(af$t1))

af$zt01 <- (af$t1_transformed_freq - af$t0_transformed_freq)^2
af

plot(af$t0,af$t0_transformed_freq, xlab = "t0 allele frequency", ylab = "arcsin(sqrt(t0 af))")

```

# Fst matrix temporal samples (within population comparisons)

## PWS
```{r, message=FALSE, warning=FALSE, fig.width= 10, fig.height= 7}

fst <- read.delim("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/fst/sliding_window/folded/fst_pbs_50kb_win_10kb_step_folded_PWS91_PWS96_PWS07.txt")
p91xp96 <- round(mean(fst$Fst01),4)
p91xp07 <- round(mean(fst$Fst02),4)
p96xp07 <- round(mean(fst$Fst12),4)
fst <- read.delim("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/fst/sliding_window/folded/fst_pbs_50kb_win_10kb_step_folded_PWS91_PWS96_PWS17.txt")
p91xp17 <- round(mean(fst$Fst02),4)
p96xp17 <- round(mean(fst$Fst12),4)
fst <- read.delim("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/fst/sliding_window/folded/fst_pbs_50kb_win_10kb_step_folded_PWS91_PWS07_PWS17.txt")
p07xp17 <- round(mean(fst$Fst12),4)

fst_vec <- c(0,p91xp96,p91xp07,p91xp17,
             p91xp96,0,p96xp07,p96xp17,
             p91xp07,p96xp07,0,p07xp17,
             p91xp17,p96xp17,p07xp17,0)

fst_mat = matrix(fst_vec, nrow = 4, ncol = 4)
colnames(fst_mat) <- c("PWS91","PWS96","PWS07","PWS17")
rownames(fst_mat) <- c("PWS91","PWS96","PWS07","PWS17")
print(fst_mat)
upper_tri <- get_upper_tri(fst_mat)

# Melt the correlation matrix
melted_cormat <- melt(upper_tri, na.rm = TRUE)
# Heatmap
ggplot(data = melted_cormat, aes(Var2, Var1, fill = value))+
 geom_tile(color = "white")+
 scale_fill_gradient(low = blu, high = red, 
    limit = c(0,0.015), space = "Lab", 
   name="Fst") +
  theme_minimal()+ xlab("")+ylab("")+
 theme(axis.text.x = element_text(angle = 0, vjust = 0, 
    size = 12, hjust = 0.5))+
   theme(axis.text.y = element_text(size = 12))+
 coord_fixed()+
 geom_text(aes(Var2, Var1, label = value), color = "black", size = 5)
```

## TB

```{r, message=FALSE, warning=FALSE, fig.width= 10, fig.height= 7}

fst <- read.delim("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/fst/sliding_window/folded/fst_pbs_50kb_win_10kb_step_folded_TB91_TB96_TB06.txt")
p91xp96 <- round(mean(fst$Fst01),4)
p91xp07 <- round(mean(fst$Fst02),4)
p96xp07 <- round(mean(fst$Fst12),4)
fst <- read.delim("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/fst/sliding_window/folded/fst_pbs_50kb_win_10kb_step_folded_TB91_TB96_TB17.txt")
p91xp17 <- round(mean(fst$Fst02),4)
p96xp17 <- round(mean(fst$Fst12),4)
fst <- read.delim("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/fst/sliding_window/folded/fst_pbs_50kb_win_10kb_step_folded_TB91_TB06_TB17.txt")
p07xp17 <- round(mean(fst$Fst12),4)

fst_vec <- c(0,p91xp96,p91xp07,p91xp17,
            p91xp96,0,p96xp07,p96xp17,
            p91xp07,p96xp07,0,p07xp17,
            p91xp17,p96xp17,p07xp17,0)

fst_mat = matrix(fst_vec, nrow = 4, ncol = 4)
colnames(fst_mat) <- c("TB91","TB96","TB06","TB17")
rownames(fst_mat) <- c("TB91","TB96","TB06","TB17")
print(fst_mat)
upper_tri <- get_upper_tri(fst_mat)

# Melt the correlation matrix
melted_cormat <- melt(upper_tri, na.rm = TRUE)
# Heatmap
ggplot(data = melted_cormat, aes(Var2, Var1, fill = value))+
geom_tile(color = "white")+
scale_fill_gradient(low = blu, high = red, 
   limit = c(0,0.015), space = "Lab", 
  name="Fst") +
 theme_minimal()+ xlab("")+ylab("")+
theme(axis.text.x = element_text(angle = 0, vjust = 0, 
   size = 12, hjust = 0.5))+
  theme(axis.text.y = element_text(size = 12))+
coord_fixed()+
geom_text(aes(Var2, Var1, label = value), color = "black", size = 5)

```

## SS

```{r, message=FALSE, warning=FALSE, fig.width= 10, fig.height= 7}

fst <- read.delim("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/fst/sliding_window/folded/fst_pbs_50kb_win_10kb_step_folded_SS96_SS06_SS17.txt")
p96xp07 <- round(mean(fst$Fst01),4)
p96xp17 <- round(mean(fst$Fst02),4)
p07xp17 <- round(mean(fst$Fst12),4)


fst_vec <- c(0,p96xp07,p96xp17,
            p96xp07,0,p07xp17,
            p96xp17,p07xp17,0)

fst_mat = matrix(fst_vec, nrow = 3, ncol = 3)
colnames(fst_mat) <- c("SS96","SS06","SS17")
rownames(fst_mat) <- c("SS96","SS06","SS17")
print(fst_mat)
upper_tri <- get_upper_tri(fst_mat)

# Melt the correlation matrix
melted_cormat <- melt(upper_tri, na.rm = TRUE)
# Heatmap
ggplot(data = melted_cormat, aes(Var2, Var1, fill = value))+
geom_tile(color = "white")+
scale_fill_gradient(low = blu, high = red, 
   limit = c(0,0.015), space = "Lab", 
  name="Fst") +
 theme_minimal()+ xlab("")+ylab("")+
theme(axis.text.x = element_text(angle = 0, vjust = 0, 
   size = 12, hjust = 0.5))+
  theme(axis.text.y = element_text(size = 12))+
coord_fixed()+
geom_text(aes(Var2, Var1, label = value), color = "black", size = 5)


```

# PWS Fst windows
## pre-spill vs. post-spill

Which loci are consistently divergent in pre-spill vs. post-spill comparisons?

Find windows that show high Fst in all comparisons with 1991.

intersect((T0 vs. T1),(T0 vs. T2),(T0 vs. T3))
```{r, message=FALSE, warning=FALSE, fig.width= 10, fig.height= 7}
fsts <- list.files(path="C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/fst/sliding_window/folded/", full.names=T)
#fsts <- fsts[1]


fst1 <- read.table("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/fst/sliding_window/folded/fst_pbs_50kb_win_10kb_step_folded_PWS91_PWS96_PWS07.txt", header = TRUE, stringsAsFactors = FALSE)
fst2 <- read.table("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/fst/sliding_window/folded/fst_pbs_50kb_win_10kb_step_folded_PWS91_PWS07_PWS17.txt", header = TRUE, stringsAsFactors = FALSE)

#fst1 <- read.table("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/fst/sliding_window/folded/fst_pbs_50kb_win_10kb_step_folded_TB91_TB96_TB06.txt", header = TRUE, stringsAsFactors = FALSE)
#fst2 <- read.table("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/fst/sliding_window/folded/fst_pbs_50kb_win_10kb_step_folded_TB91_TB96_TB17.txt", header = TRUE, stringsAsFactors = FALSE)


fst1$Fst03 <- fst2$Fst02

fst1 <- fst1 %>% mutate(chr = as.numeric(gsub("chr", "", chr)))
fst1$chr_num <- factor(fst1$chr, levels = c(1:26))

t0102 <- fst1[fst1$Fst01 >= quantile(fst1$Fst01, .99)[[1]] & fst1$Fst02 >= quantile(fst1$Fst02, .99)[[1]],]
t03 <- fst1[fst1$Fst03 >= quantile(fst1$Fst03, .99)[[1]],]

test <- merge(t0102,t03, by = c("chr","midPos"))
print("number of 50kb windows with Fst > genome-wide 99th %ile in each comparison")
nrow(test)
test$chr_num <- factor(test$chr, levels = c(1:26))

p <- ggplot(fst1, aes(x = midPos, y = Fst01)) + 
     geom_point(size = 1, color = gry,alpha = 0.4, shape = 1)+
     theme_minimal()+
     theme(axis.text.x=element_blank())+
     ylab("Fst\n")+ xlab("")+ 
     geom_smooth(method = "loess", se = FALSE, span = 1/10, color = blu)+
     facet_wrap(~chr_num, ncol = 9)

#p     
p + geom_vline(aes(xintercept = midPos), test)

```

## oil exposed vs. not exposed
Which loci are consistently divergent in (oil exposed vs. not exposed) comparisons?

Find windows that show high Fst in all comparisons with 1996.

intersect((T1 vs. T0),(T1 vs. T2),(T1 vs. T3))
```{r, message=FALSE, warning=FALSE, fig.width= 10, fig.height= 7}
fsts <- list.files(path="C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/fst/sliding_window/folded/", full.names=T)
#fsts <- fsts[1]

fst1 <- read.table("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/fst/sliding_window/folded/fst_pbs_50kb_win_10kb_step_folded_PWS91_PWS96_PWS07.txt", header = TRUE, stringsAsFactors = FALSE)
fst2 <- read.table("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/fst/sliding_window/folded/fst_pbs_50kb_win_10kb_step_folded_PWS91_PWS96_PWS17.txt", header = TRUE, stringsAsFactors = FALSE)

#fst1 <- read.table("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/fst/sliding_window/folded/fst_pbs_50kb_win_10kb_step_folded_TB91_TB96_TB06.txt", header = TRUE, stringsAsFactors = FALSE)
#fst2 <- read.table("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/fst/sliding_window/folded/fst_pbs_50kb_win_10kb_step_folded_TB91_TB96_TB17.txt", header = TRUE, stringsAsFactors = FALSE)


fst1$Fst13 <- fst2$Fst12

fst1 <- fst1 %>% mutate(chr = as.numeric(gsub("chr", "", chr)))
fst1$chr_num <- factor(fst1$chr, levels = c(1:26))

t0112 <- fst1[fst1$Fst01 >= quantile(fst1$Fst01, .99)[[1]] & fst1$Fst12 >= quantile(fst1$Fst12, .99)[[1]],]
t13 <- fst1[fst1$Fst13 >= quantile(fst1$Fst13, .99)[[1]],]

test <- merge(t0112,t13, by = c("chr","midPos"))
print("number of 50kb windows with Fst > genome-wide 99th %ile in each comparison")
nrow(test)
test$chr_num <- factor(test$chr, levels = c(1:26))

p <- ggplot(fst1, aes(x = midPos, y = Fst01)) + 
     geom_point(size = 1, color = gry,alpha = 0.4, shape = 1)+
     theme_minimal()+
     theme(axis.text.x=element_blank())+
     ylab("Fst\n")+ xlab("")+ 
     geom_smooth(method = "loess", se = FALSE, span = 1/10, color = blu)+
     facet_wrap(~chr_num, ncol = 9)

#p     
p + geom_vline(aes(xintercept = midPos), test)

```
