---
title: "Time Series AF Shifts"
author: "Joe McGirr"
date: '`r Sys.Date()`'
output:
  html_document:
    code_folding: hide
    collapsed: no
    df_print: paged
    number_sections: yes
    theme: paper
    toc: yes
    toc_depth: 5
    toc_float: yes
  html_notebook:
    toc: yes
    toc_depth: 5
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(ggplot2)
library(tidyverse)
library(reticulate)
library(reshape2)
library(plyranges)
library(seqinr)
library(windowscanr)
#BiocManager::install("seqinr")

  # Get upper triangle of the correlation matrix
  get_upper_tri <- function(cormat){
    cormat[lower.tri(cormat)]<- NA
    return(cormat)
  }

    get_lower_tri<-function(cormat){
    cormat[upper.tri(cormat)] <- NA
    return(cormat)
  }

# color-blind friendly 
# Wong, B. Points of view: Color blindness. Nat Methods (2011).
bla <- "#000000"
blu <- "#0072b2"
grb <- "#56b4e9"
lir <- "#cc79a7"
gre <- "#009e73"
red <- "#d55e00"
org <- "#e69f00"
yel <- "#f0e442"
gry<-  '#BBBBBB'

```

# Allele Frequency Shifts

## PWS 1991 --> 1996
The summary statistic ∆z2 was the squared within-line difference in arcsine square-root transformed MAF q; it ranges from 0 to π2. 

Following [Longshanks](https://elifesciences.org/articles/42014#s4)

```{r, message=FALSE, warning=FALSE, fig.width= 10, fig.height= 7}

#AF is calculated as AC/AN, where AC is total ALT allele count and AN is total allele called in genotypes
t0 <- read.table("C:/Users/jmcgirr/Desktop/PWS91_freqs_test.txt", header = FALSE, stringsAsFactors = FALSE, col.names = c("chr","pos","t0_AN","t0_AC"))
t1 <- read.table("C:/Users/jmcgirr/Desktop/PWS96_freqs_test.txt", header = FALSE, stringsAsFactors = FALSE, col.names = c("chr","pos","t1_AN","t1_AC"))
t2 <- read.table("C:/Users/jmcgirr/Desktop/PWS07_freqs_test.txt", header = FALSE, stringsAsFactors = FALSE, col.names = c("chr","pos","t2_AN","t2_AC"))
t3 <- read.table("C:/Users/jmcgirr/Desktop/PWS17_freqs_test.txt", header = FALSE, stringsAsFactors = FALSE, col.names = c("chr","pos","t3_AN","t3_AC"))
head(t0)
freqs <- data.frame(chr = t0$chr,
                    pos = t0$pos,
                    t0_AN = t0$t0_AN,
                    t0_AC = t0$t0_AC,
                    t1_AN = t1$t1_AN,
                    t1_AC = t1$t1_AC,
                    t2_AN = t2$t2_AN,
                    t2_AC = t2$t2_AC,
                    t3_AN = t3$t3_AN,
                    t3_AC = t3$t3_AC)

freqs$t0_AF <- freqs$t0_AC / freqs$t0_AN
freqs$t1_AF <- freqs$t1_AC / freqs$t1_AN
freqs$t2_AF <- freqs$t2_AC / freqs$t2_AN
freqs$t3_AF <- freqs$t3_AC / freqs$t3_AN
freqs <- na.omit(freqs)

freqs$t0_transformed_freq <- asin(sqrt(freqs$t0_AF))
freqs$t1_transformed_freq <- asin(sqrt(freqs$t1_AF))
freqs$t2_transformed_freq <- asin(sqrt(freqs$t2_AF))
freqs$t3_transformed_freq <- asin(sqrt(freqs$t3_AF))

freqs$zt01 <- (freqs$t1_transformed_freq - freqs$t0_transformed_freq)^2
freqs$zt02 <- (freqs$t2_transformed_freq - freqs$t0_transformed_freq)^2
freqs$zt03 <- (freqs$t3_transformed_freq - freqs$t0_transformed_freq)^2

head(freqs)
nrow(freqs)

freqs_win <- winScan(x = freqs, 
                   groups = "chr", 
                   position = "pos", 
                   values = c("zt01","zt02","zt03"), 
                   win_size = 10000,
                   win_step = 5000,
                   funs = c("mean"))
freqs_win <- na.omit(freqs_win)
freqs_win <- freqs_win %>% mutate(chr = as.numeric(gsub("chr", "", chr)))
freqs_win$chr_num <- factor(freqs_win$chr, levels = c(1:26))

#write.table(freqs_win, "C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/")

tops <- freqs_win[freqs_win$zt01_mean >= quantile(freqs_win$zt01_mean, .99)[[1]] &
                  freqs_win$zt02_mean >= quantile(freqs_win$zt02_mean, .99)[[1]] &
                  freqs_win$zt03_mean >= quantile(freqs_win$zt03_mean, .99)[[1]],]


print("number of 10kb windows with z > genome-wide 99th %ile in each comparison")
nrow(tops)
tops$chr_num <- factor(tops$chr, levels = c(1:26))

p <- ggplot(freqs_win, aes(x = win_mid, y = zt01_mean)) + 
     geom_point(size = 1, color = gry,alpha = 0.4, shape = 1)+
     theme_minimal()+
     theme(axis.text.x=element_blank())+
     ylab(expression(paste(Delta, "z"^"2")))+ xlab("")+ 
     #geom_smooth(method = "loess", se = FALSE, span = 1/10, color = blu)+
     facet_wrap(~chr_num, ncol = 9)

#p     
p + geom_vline(aes(xintercept = win_mid), tops)

```

## What does the arcsine transformation do
Maybe use logit transformation instead?

http://strata.uga.edu/8370/rtips/proportions.html

```{r, message=FALSE, warning=FALSE, fig.width= 10, fig.height= 7}

t0 <- c(0,0.01,0.02,0.03,0.1,0.2,0.3,1)
t1 <- rev(t0)

#t0 <- c(0.01,0.05,0.06,0.07,0.10,0,1)
#t1 <- c(0.02,0.06,0.10,0.20,0.05,1,0)

af <- data.frame(t0 = t0, t1 = t1)
head(af)

af$t0_transformed_freq <- asin(sqrt(af$t0))
af$t1_transformed_freq <- asin(sqrt(af$t1))

af$zt01 <- (af$t1_transformed_freq - af$t0_transformed_freq)^2
af

plot(af$t0,af$t0_transformed_freq, xlab = "t0 allele frequency", ylab = "arcsin(sqrt(t0 af))")

```

# Fst matrix temporal samples (within population comparisons)

## PWS
```{r, message=FALSE, warning=FALSE, fig.width= 10, fig.height= 7}

fst <- read.delim("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/fst/sliding_window/folded/fst_pbs_50kb_win_10kb_step_folded_PWS91_PWS96_PWS07.txt")
p91xp96 <- round(mean(fst$Fst01),4)
p91xp07 <- round(mean(fst$Fst02),4)
p96xp07 <- round(mean(fst$Fst12),4)
fst <- read.delim("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/fst/sliding_window/folded/fst_pbs_50kb_win_10kb_step_folded_PWS91_PWS96_PWS17.txt")
p91xp17 <- round(mean(fst$Fst02),4)
p96xp17 <- round(mean(fst$Fst12),4)
fst <- read.delim("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/fst/sliding_window/folded/fst_pbs_50kb_win_10kb_step_folded_PWS91_PWS07_PWS17.txt")
p07xp17 <- round(mean(fst$Fst12),4)

fst_vec <- c(0,p91xp96,p91xp07,p91xp17,
             p91xp96,0,p96xp07,p96xp17,
             p91xp07,p96xp07,0,p07xp17,
             p91xp17,p96xp17,p07xp17,0)

fst_mat = matrix(fst_vec, nrow = 4, ncol = 4)
colnames(fst_mat) <- c("PWS91","PWS96","PWS07","PWS17")
rownames(fst_mat) <- c("PWS91","PWS96","PWS07","PWS17")
print(fst_mat)
upper_tri <- get_upper_tri(fst_mat)

# Melt the correlation matrix
melted_cormat <- melt(upper_tri, na.rm = TRUE)
# Heatmap
ggplot(data = melted_cormat, aes(Var2, Var1, fill = value))+
 geom_tile(color = "white")+
 scale_fill_gradient(low = blu, high = red, 
    limit = c(0,0.015), space = "Lab", 
   name="Fst") +
  theme_minimal()+ xlab("")+ylab("")+
 theme(axis.text.x = element_text(angle = 0, vjust = 0, 
    size = 12, hjust = 0.5))+
   theme(axis.text.y = element_text(size = 12))+
 coord_fixed()+
 geom_text(aes(Var2, Var1, label = value), color = "black", size = 5)
```

## TB

```{r, message=FALSE, warning=FALSE, fig.width= 10, fig.height= 7}

fst <- read.delim("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/fst/sliding_window/folded/fst_pbs_50kb_win_10kb_step_folded_TB91_TB96_TB06.txt")
p91xp96 <- round(mean(fst$Fst01),4)
p91xp07 <- round(mean(fst$Fst02),4)
p96xp07 <- round(mean(fst$Fst12),4)
fst <- read.delim("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/fst/sliding_window/folded/fst_pbs_50kb_win_10kb_step_folded_TB91_TB96_TB17.txt")
p91xp17 <- round(mean(fst$Fst02),4)
p96xp17 <- round(mean(fst$Fst12),4)
fst <- read.delim("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/fst/sliding_window/folded/fst_pbs_50kb_win_10kb_step_folded_TB91_TB06_TB17.txt")
p07xp17 <- round(mean(fst$Fst12),4)

fst_vec <- c(0,p91xp96,p91xp07,p91xp17,
            p91xp96,0,p96xp07,p96xp17,
            p91xp07,p96xp07,0,p07xp17,
            p91xp17,p96xp17,p07xp17,0)

fst_mat = matrix(fst_vec, nrow = 4, ncol = 4)
colnames(fst_mat) <- c("TB91","TB96","TB06","TB17")
rownames(fst_mat) <- c("TB91","TB96","TB06","TB17")
print(fst_mat)
upper_tri <- get_upper_tri(fst_mat)

# Melt the correlation matrix
melted_cormat <- melt(upper_tri, na.rm = TRUE)
# Heatmap
ggplot(data = melted_cormat, aes(Var2, Var1, fill = value))+
geom_tile(color = "white")+
scale_fill_gradient(low = blu, high = red, 
   limit = c(0,0.015), space = "Lab", 
  name="Fst") +
 theme_minimal()+ xlab("")+ylab("")+
theme(axis.text.x = element_text(angle = 0, vjust = 0, 
   size = 12, hjust = 0.5))+
  theme(axis.text.y = element_text(size = 12))+
coord_fixed()+
geom_text(aes(Var2, Var1, label = value), color = "black", size = 5)

```

## SS

```{r, message=FALSE, warning=FALSE, fig.width= 10, fig.height= 7}

fst <- read.delim("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/fst/sliding_window/folded/fst_pbs_50kb_win_10kb_step_folded_SS96_SS06_SS17.txt")
p96xp07 <- round(mean(fst$Fst01),4)
p96xp17 <- round(mean(fst$Fst02),4)
p07xp17 <- round(mean(fst$Fst12),4)


fst_vec <- c(0,p96xp07,p96xp17,
            p96xp07,0,p07xp17,
            p96xp17,p07xp17,0)

fst_mat = matrix(fst_vec, nrow = 3, ncol = 3)
colnames(fst_mat) <- c("SS96","SS06","SS17")
rownames(fst_mat) <- c("SS96","SS06","SS17")
print(fst_mat)
upper_tri <- get_upper_tri(fst_mat)

# Melt the correlation matrix
melted_cormat <- melt(upper_tri, na.rm = TRUE)
# Heatmap
ggplot(data = melted_cormat, aes(Var2, Var1, fill = value))+
geom_tile(color = "white")+
scale_fill_gradient(low = blu, high = red, 
   limit = c(0,0.015), space = "Lab", 
  name="Fst") +
 theme_minimal()+ xlab("")+ylab("")+
theme(axis.text.x = element_text(angle = 0, vjust = 0, 
   size = 12, hjust = 0.5))+
  theme(axis.text.y = element_text(size = 12))+
coord_fixed()+
geom_text(aes(Var2, Var1, label = value), color = "black", size = 5)


```

# PWS Fst windows
## pre-spill vs. post-spill

Which loci are consistently divergent in pre-spill vs. post-spill comparisons?

Find windows that show high Fst in all comparisons with 1991.

intersect((T0 vs. T1),(T0 vs. T2),(T0 vs. T3))
```{r, message=FALSE, warning=FALSE, fig.width= 10, fig.height= 7}
fsts <- list.files(path="C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/fst/sliding_window/folded/", full.names=T)
#fsts <- fsts[1]


fst1 <- read.table("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/fst/sliding_window/folded/fst_pbs_50kb_win_10kb_step_folded_PWS91_PWS96_PWS07.txt", header = TRUE, stringsAsFactors = FALSE)
fst2 <- read.table("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/fst/sliding_window/folded/fst_pbs_50kb_win_10kb_step_folded_PWS91_PWS07_PWS17.txt", header = TRUE, stringsAsFactors = FALSE)

#fst1 <- read.table("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/fst/sliding_window/folded/fst_pbs_50kb_win_10kb_step_folded_TB91_TB96_TB06.txt", header = TRUE, stringsAsFactors = FALSE)
#fst2 <- read.table("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/fst/sliding_window/folded/fst_pbs_50kb_win_10kb_step_folded_TB91_TB96_TB17.txt", header = TRUE, stringsAsFactors = FALSE)


fst1$Fst03 <- fst2$Fst02

fst1 <- fst1 %>% mutate(chr = as.numeric(gsub("chr", "", chr)))
fst1$chr_num <- factor(fst1$chr, levels = c(1:26))

t0102 <- fst1[fst1$Fst01 >= quantile(fst1$Fst01, .99)[[1]] & fst1$Fst02 >= quantile(fst1$Fst02, .99)[[1]],]
t03 <- fst1[fst1$Fst03 >= quantile(fst1$Fst03, .99)[[1]],]

test <- merge(t0102,t03, by = c("chr","midPos"))
print("number of 50kb windows with Fst > genome-wide 99th %ile in each comparison")
nrow(test)
test$chr_num <- factor(test$chr, levels = c(1:26))

p <- ggplot(fst1, aes(x = midPos, y = Fst01)) + 
     geom_point(size = 1, color = gry,alpha = 0.4, shape = 1)+
     theme_minimal()+
     theme(axis.text.x=element_blank())+
     ylab("Fst\n")+ xlab("")+ 
     geom_smooth(method = "loess", se = FALSE, span = 1/10, color = blu)+
     facet_wrap(~chr_num, ncol = 9)

#p     
p + geom_vline(aes(xintercept = midPos), test)

```

## oil exposed vs. not exposed
Which loci are consistently divergent in (oil exposed vs. not exposed) comparisons?

Find windows that show high Fst in all comparisons with 1996.

intersect((T1 vs. T0),(T1 vs. T2),(T1 vs. T3))
```{r, message=FALSE, warning=FALSE, fig.width= 10, fig.height= 7}
fsts <- list.files(path="C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/fst/sliding_window/folded/", full.names=T)
#fsts <- fsts[1]

fst1 <- read.table("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/fst/sliding_window/folded/fst_pbs_50kb_win_10kb_step_folded_PWS91_PWS96_PWS07.txt", header = TRUE, stringsAsFactors = FALSE)
fst2 <- read.table("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/fst/sliding_window/folded/fst_pbs_50kb_win_10kb_step_folded_PWS91_PWS96_PWS17.txt", header = TRUE, stringsAsFactors = FALSE)

#fst1 <- read.table("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/fst/sliding_window/folded/fst_pbs_50kb_win_10kb_step_folded_TB91_TB96_TB06.txt", header = TRUE, stringsAsFactors = FALSE)
#fst2 <- read.table("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/fst/sliding_window/folded/fst_pbs_50kb_win_10kb_step_folded_TB91_TB96_TB17.txt", header = TRUE, stringsAsFactors = FALSE)


fst1$Fst13 <- fst2$Fst12

fst1 <- fst1 %>% mutate(chr = as.numeric(gsub("chr", "", chr)))
fst1$chr_num <- factor(fst1$chr, levels = c(1:26))

t0112 <- fst1[fst1$Fst01 >= quantile(fst1$Fst01, .99)[[1]] & fst1$Fst12 >= quantile(fst1$Fst12, .99)[[1]],]
t13 <- fst1[fst1$Fst13 >= quantile(fst1$Fst13, .99)[[1]],]

test <- merge(t0112,t13, by = c("chr","midPos"))
print("number of 50kb windows with Fst > genome-wide 99th %ile in each comparison")
nrow(test)
test$chr_num <- factor(test$chr, levels = c(1:26))

p <- ggplot(fst1, aes(x = midPos, y = Fst01)) + 
     geom_point(size = 1, color = gry,alpha = 0.4, shape = 1)+
     theme_minimal()+
     theme(axis.text.x=element_blank())+
     ylab("Fst\n")+ xlab("")+ 
     geom_smooth(method = "loess", se = FALSE, span = 1/10, color = blu)+
     facet_wrap(~chr_num, ncol = 9)

#p     
p + geom_vline(aes(xintercept = midPos), test)

```
