---
title: "Sample_Validation_Outlier_Detection"
author: "Joe McGirr"
date: "6/29/2020"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(ggplot2)
library(tidyverse)
library(reticulate)
library(ggpubr)
library(sf)
library(gridExtra)
library(grid)
library(pophelper)
#BiocManager::install("visFuns.R")


# color-blind friendly 
# Wong, B. Points of view: Color blindness. Nat Methods (2011).
bla <- "#000000"
blu <- "#0072b2"
grb <- "#56b4e9"
lir <- "#cc79a7"
gre <- "#009e73"
red <- "#d55e00"
org <- "#e69f00"
yel <- "#f0e442"
gry<-  '#BBBBBB'

```



```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.width= 10, fig.height= 7, eval = FALSE}

## Original data set, includes all individuals



pop_info <- read.table("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/familiarize/EVOS_MasterSheet_JoeMcGirr_April2020_plate_rows.txt", header = TRUE, stringsAsFactors = FALSE, sep = "\t")
aligned <- read.table("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/familiarize/aligned_samples.txt", header = TRUE, stringsAsFactors = FALSE, sep = "\t")
pop_info <- pop_info[pop_info$Sample %in% aligned$sample,]
vcf_samples <- read.table("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/familiarize/samples_in_vcf.txt", header = FALSE, stringsAsFactors = FALSE, sep = "\t")
names(vcf_samples)[names(vcf_samples)=="V1"] <- "Sample"
vcf_sample_info <- inner_join(vcf_samples,pop_info, by = "Sample")

chr <- "1"
C <- as.matrix(read.table(paste("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/PCAngsd/chr",chr,"_ph_filtered_snps_minDP600_maxDP2000_maf0.05_minQ20_minMQ30_maxmiss0.5.cov", sep = "")))
e <- eigen(C)
pca <-data.frame(Sample=vcf_sample_info$Sample, 
                 pop=vcf_sample_info$Population.Year,
                 plate=vcf_sample_info$Sequence.Plate,
                 plate_row=vcf_sample_info$plate_row,
                 PC1=e$vectors[,1],PC2=e$vectors[,2],
                 PC3=e$vectors[,3],PC4=e$vectors[,4],
                 PC5=e$vectors[,5],PC6=e$vectors[,6],
                 PC7=e$vectors[,7],PC8=e$vectors[,8],
                 PC9=e$vectors[,9],PC10=e$vectors[,10],
                 stringsAsFactors=FALSE)

prop_explained <- c()
for (s in e$values[1:10]) {
  #print(s / sum(e$values))
  prop_explained <- c(prop_explained,round(((s / sum(e$values))*100),2))
}

pca <- pca %>% separate(pop, into = c("pop", "year"), sep = "(?<=[A-Za-z])(?=[0-9])")
pca1 <- pca

ggplot(data = pca1)+
  geom_point(data = pca1, aes(x = PC1, y = PC2, fill = pop, color = pop, shape = year), size = 3)+
  scale_fill_manual(values=c(grb,org,red,yel,blu,lir), guide=FALSE)+
  scale_color_manual(name = "population", values=c(grb,org,red,yel,blu,lir))+
  scale_shape_manual(name = "year", values=c(21,22,23,24,25))+
  xlab(paste("PC 1: ", prop_explained[1],"%\n",sep = ""))+
  ylab(paste("PC 2: ", prop_explained[2],"%\n",sep = ""))+
  theme_bw()+
  #xlim(0,0.17)+ylim(-0.05,0.1)+
  #xlim(-0.05,-0.042)+ylim(-0.035,-0.025)+
  #xlim(-0.055,-0.04)+ylim(-0.031,-0.025)+
  #geom_text(aes(x = PC1, y = PC2), label = pca1$Sample)+
  #geom_text(aes(x = PC1, y = PC2), label = pca1$plate)+
  ggtitle("samples labeled by sequencing plate\nchr 1")

#ggplot(data = pca1)+
#  geom_point(data = pca1, aes(x = PC1, y = PC2, fill = pop, color = pop, shape = year), size = 3)+
#  scale_fill_manual(values=c(grb,org,red,yel,blu,lir), guide=FALSE)+
#  scale_color_manual(name = "population", values=c(grb,org,red,yel,blu,lir))+
#  scale_shape_manual(name = "year", values=c(21,22,23,24,25))+
#  xlab(paste("PC 1: ", prop_explained[1],"%\n",sep = ""))+
#  ylab(paste("PC 2: ", prop_explained[2],"%\n",sep = ""))+
#  theme_bw()+
#  #xlim(-0.15,0)+#ylim(-0.05,0.1)+
#  #xlim(-0.05,-0.042)+ylim(-0.035,-0.025)+
#  #xlim(-0.055,-0.04)+ylim(-0.031,-0.025)+
#  geom_text(data=subset(pca1, pop == "TB" & plate_row == "5" & plate == "1"),aes(x = PC1, y = PC2,label #= Sample))+
#  #geom_text(aes(x = PC1, y = PC2), label = pca1$plate)+
#  ggtitle("samples labeled by sequencing plate\nchr 1")

```


```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.width= 10, fig.height= 7, eval = FALSE}


### For each plate, view individuals in sets of 8 corresponding to their row on the plate. See PCA_row_labels.html for all plates.  
### Example: Plate 4





#for (plate_num in c(1:20))
for (plate_num in c(4))
{
plate_num_str <- as.character(plate_num)
  for (plate_row in c(1:8))
  {
  plate_row_str <- as.character(plate_row)
  p1 <- ggplot(data = pca1)+
    geom_point(data = pca1, aes(x = PC1, y = PC2, fill = pop, color = pop, shape = year), size = 3)+
    scale_fill_manual(values=c(grb,org,red,yel,blu,lir), guide=FALSE)+
    scale_color_manual(name = "population", values=c(grb,org,red,yel,blu,lir))+
    scale_shape_manual(name = "year", values=c(21,22,23,24,25))+
    xlab(paste("PC 1: ", prop_explained[1],"%\n",sep = ""))+
    ylab(paste("PC 2: ", prop_explained[2],"%\n",sep = ""))+
    theme_bw()+
    geom_text(data=subset(pca1, plate_row == plate_row_str & plate == plate_num_str),aes(x = PC1, y = PC2,label = Sample))+
    ggtitle(paste("plate ",plate_num_str,"\nrow ",plate_row_str,sep = ""))
    print(p1)
}
}


#out_rm <- vcf_sample_info %>% filter(Sequence.Plate != 1 & 
#                                     Sequence.Plate != 2 &
#                                     Sequence.Plate != 3 &
#                                     Sequence.Plate != 4 &
#                                     Sequence.Plate != 5) %>%
#           separate(Population.Year, into = c("pop", "year"), sep = "(?<=[A-Za-z])(?=[0-9])")%>% 
#            filter(pop != "TB")
#count(out_rm, vars = Population.Year)

#write.table(as.data.frame(out_rm$Sample),"C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/plink/plates_1_through_5_rm_no_TB.txt",quote = FALSE,row.names = FALSE)

                                     
```

### For each plate, I labeled individuals from the same row on PC plots. If all individuals showed obviously wrong grouping, I labeled the row "bad". "strange" indicates that one or two samples in the row showed odd grouping. (See PCA_row_labels.html to see all 160 plots that I checked to create the table).

```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.width= 10, fig.height= 7}
bads <- read.table("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/familiarize/bad_plates_by_row.txt", header = TRUE, stringsAsFactors = FALSE, sep = "\t")
#head(t(bads))
grid.arrange(tableGrob(bads,theme=ttheme_default(base_size = 10),rows = NULL))

```

### <span style="color:red">**Conclusion: Ploting by row in groups of 8 suggested obvious contamination in plates 1-5.**</span>  

***

### Not all rows of plates 1-5 showed contamination, but I removed all individuals from plate 1-5 to be conservative and reran PCA  


```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.width= 10, fig.height= 7}

pop_info <- read.table("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/familiarize/EVOS_MasterSheet_JoeMcGirr_April2020_plate_rows.txt", header = TRUE, stringsAsFactors = FALSE, sep = "\t")
aligned <- read.table("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/familiarize/aligned_samples.txt", header = TRUE, stringsAsFactors = FALSE, sep = "\t")
pop_info <- pop_info[pop_info$Sample %in% aligned$sample,]
vcf_samples <- read.table("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/plink/plates_1_through_5_rm.txt", header = FALSE, stringsAsFactors = FALSE, sep = "\t")
names(vcf_samples)[names(vcf_samples)=="V1"] <- "Sample"
vcf_sample_info <- inner_join(vcf_samples,pop_info, by = "Sample")

chr <- "1"
C <- as.matrix(read.table(paste("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/PCAngsd/chr",chr,"_ph_filtered_snps_minDP600_maxDP2000_maf0.05_minQ20_minMQ30_maxmiss0.5_plates_1_through_5_rm.cov", sep = "")))
e <- eigen(C)
pca <-data.frame(Sample=vcf_sample_info$Sample, 
                 pop=vcf_sample_info$Population.Year,
                 plate=vcf_sample_info$Sequence.Plate,
                 plate_row=vcf_sample_info$plate_row,
                 PC1=e$vectors[,1],PC2=e$vectors[,2],
                 PC3=e$vectors[,3],PC4=e$vectors[,4],
                 PC5=e$vectors[,5],PC6=e$vectors[,6],
                 PC7=e$vectors[,7],PC8=e$vectors[,8],
                 PC9=e$vectors[,9],PC10=e$vectors[,10],
                 stringsAsFactors=FALSE)

prop_explained <- c()
for (s in e$values[1:10]) {
  #print(s / sum(e$values))
  prop_explained <- c(prop_explained,round(((s / sum(e$values))*100),2))
}

pca <- pca %>% separate(pop, into = c("pop", "year"), sep = "(?<=[A-Za-z])(?=[0-9])")
pca1 <- pca

ggplot(data = pca1)+
  geom_point(data = pca1, aes(x = PC1, y = PC2, fill = pop, color = pop, shape = year), size = 3)+
  scale_fill_manual(values=c(grb,org,red,yel,blu,lir), guide=FALSE)+
  scale_color_manual(name = "population", values=c(grb,org,red,yel,blu,lir))+
  scale_shape_manual(name = "year", values=c(21,22,23,24,25))+
  xlab(paste("PC 1: ", prop_explained[1],"%\n",sep = ""))+
  ylab(paste("PC 2: ", prop_explained[2],"%\n",sep = ""))+
  theme_bw()+
  #geom_text(aes(x = PC1, y = PC2), label = pca1$plate)+
  ggtitle("PC 1 and 2 chr 1\nplates 1-5 removed")

ggplot(data = pca1)+
  geom_point(data = pca1, aes(x = PC3, y = PC4, fill = pop, color = pop, shape = year), size = 3)+
  scale_fill_manual(values=c(grb,org,red,yel,blu,lir), guide=FALSE)+
  scale_color_manual(name = "population", values=c(grb,org,red,yel,blu,lir))+
  scale_shape_manual(name = "year", values=c(21,22,23,24,25))+
  xlab(paste("PC 3: ", prop_explained[3],"%\n",sep = ""))+
  ylab(paste("PC 4: ", prop_explained[4],"%\n",sep = ""))+
  theme_bw()+
  #geom_text(aes(x = PC1, y = PC2), label = pca1$plate)+
  ggtitle("PC 3 and 4 chr 1\nplates 1-5 removed")

# ggplot(data = pca1)+
#   geom_point(data = pca1, aes(x = PC5, y = PC6, fill = pop, color = pop, shape = year), size = 3)+
#   scale_fill_manual(values=c(grb,org,red,yel,blu,lir), guide=FALSE)+
#   scale_color_manual(name = "population", values=c(grb,org,red,yel,blu,lir))+
#   scale_shape_manual(name = "year", values=c(21,22,23,24,25))+
#   xlab(paste("PC 5: ", prop_explained[5],"%\n",sep = ""))+
#   ylab(paste("PC 6: ", prop_explained[6],"%\n",sep = ""))+
#   theme_bw()+
#   #geom_text(aes(x = PC1, y = PC2), label = pca1$plate)+
#   ggtitle("samples labeled by sequencing plate\nchr 1")



```

### <span style="color:red">**Conclusion: TB shows clear structure, but all other populations overlap**</span>  

***

### PCA all individuals from plate 1-5 removed and TB removed

```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.width= 10, fig.height= 7}

pop_info <- read.table("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/familiarize/EVOS_MasterSheet_JoeMcGirr_April2020_plate_rows.txt", header = TRUE, stringsAsFactors = FALSE, sep = "\t")
aligned <- read.table("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/familiarize/aligned_samples.txt", header = TRUE, stringsAsFactors = FALSE, sep = "\t")
pop_info <- pop_info[pop_info$Sample %in% aligned$sample,]
vcf_samples <- read.table("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/plink/plates_1_through_5_rm_no_TB.txt", header = FALSE, stringsAsFactors = FALSE, sep = "\t")
names(vcf_samples)[names(vcf_samples)=="V1"] <- "Sample"
vcf_sample_info <- inner_join(vcf_samples,pop_info, by = "Sample")

chr <- "1"
C <- as.matrix(read.table(paste("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/PCAngsd/chr",chr,"_ph_filtered_snps_minDP600_maxDP2000_maf0.05_minQ20_minMQ30_maxmiss0.5_outliers_rm.cov", sep = "")))
e <- eigen(C)
pca <-data.frame(Sample=vcf_sample_info$Sample, 
                 pop=vcf_sample_info$Population.Year,
                 plate=vcf_sample_info$Sequence.Plate,
                 plate_row=vcf_sample_info$plate_row,
                 PC1=e$vectors[,1],PC2=e$vectors[,2],
                 PC3=e$vectors[,3],PC4=e$vectors[,4],
                 PC5=e$vectors[,5],PC6=e$vectors[,6],
                 PC7=e$vectors[,7],PC8=e$vectors[,8],
                 PC9=e$vectors[,9],PC10=e$vectors[,10],
                 stringsAsFactors=FALSE)

prop_explained <- c()
for (s in e$values[1:10]) {
  #print(s / sum(e$values))
  prop_explained <- c(prop_explained,round(((s / sum(e$values))*100),2))
}

pca <- pca %>% separate(pop, into = c("pop", "year"), sep = "(?<=[A-Za-z])(?=[0-9])")
pca1 <- pca

ggplot(data = pca1)+
  geom_point(data = pca1, aes(x = PC1, y = PC2, fill = pop, color = pop, shape = year), size = 3)+
  scale_fill_manual(values=c(grb,org,red,yel,lir), guide=FALSE)+
  scale_color_manual(name = "population", values=c(grb,org,red,yel,lir))+
  scale_shape_manual(name = "year", values=c(21,22,23,24,25))+
  xlab(paste("PC 1: ", prop_explained[1],"%\n",sep = ""))+
  ylab(paste("PC 2: ", prop_explained[2],"%\n",sep = ""))+
  theme_bw()+
  #xlim(0,0.17)+ylim(-0.05,0.1)+
  #xlim(-0.05,-0.042)+ylim(-0.035,-0.025)+
  #xlim(-0.055,-0.04)+ylim(-0.031,-0.025)+
  #geom_text(aes(x = PC1, y = PC2), label = pca1$Sample)+
  #geom_text(aes(x = PC1, y = PC2), label = pca1$plate)+
  ggtitle("PC 1 and 2 chr 1\nplates 1-5 and TB removed")

ggplot(data = pca1)+
  geom_point(data = pca1, aes(x = PC3, y = PC4, fill = pop, color = pop, shape = year), size = 3)+
  scale_fill_manual(values=c(grb,org,red,yel,lir), guide=FALSE)+
  scale_color_manual(name = "population", values=c(grb,org,red,yel,lir))+
  scale_shape_manual(name = "year", values=c(21,22,23,24,25))+
  xlab(paste("PC 3: ", prop_explained[3],"%\n",sep = ""))+
  ylab(paste("PC 4: ", prop_explained[4],"%\n",sep = ""))+
  theme_bw()+
  #xlim(0,0.17)+ylim(-0.05,0.1)+
  #xlim(-0.05,-0.042)+ylim(-0.035,-0.025)+
  #xlim(-0.055,-0.04)+ylim(-0.031,-0.025)+
  #geom_text(aes(x = PC1, y = PC2), label = pca1$Sample)+
  #geom_text(aes(x = PC1, y = PC2), label = pca1$plate)+
  ggtitle("PC 3 and 4 chr 1\nplates 1-5 and TB removed")

#for (plate_num in c(6:20))
##for (plate_num in c(6))
#{
#plate_num_str <- as.character(plate_num)
#  for (plate_row in c(1:8))
#  {
#  plate_row_str <- as.character(plate_row)
#  p1 <- ggplot(data = pca1)+
#    geom_point(data = pca1, aes(x = PC1, y = PC2, fill = pop, color = pop, shape = year), size = 3)+
#    scale_fill_manual(values=c(grb,org,red,yel,lir), guide=FALSE)+
#    scale_color_manual(name = "population", values=c(grb,org,red,yel,lir))+
#    scale_shape_manual(name = "year", values=c(21,22,23,24,25))+
#    xlab(paste("PC 1: ", prop_explained[1],"%\n",sep = ""))+
#    ylab(paste("PC 2: ", prop_explained[2],"%\n",sep = ""))+
#    theme_bw()+
#    geom_text(data=subset(pca1, plate_row == plate_row_str & plate == plate_num_str),aes(x = PC1, y = #PC2,label = Sample))+
#    ggtitle(paste("plate ",plate_num_str,"\nrow ",plate_row_str,sep = ""))
#    print(p1)
#}
#}


```

### <span style="color:red">**Conclusion: Little structure across populations other than TB**</span>  

***


```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.width= 10, fig.height= 7, eval = FALSE}

### all individuals from plate 1-5 removed and TB removed, break down meta populations



pop_info <- read.table("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/familiarize/EVOS_MasterSheet_JoeMcGirr_April2020_plate_rows.txt", header = TRUE, stringsAsFactors = FALSE, sep = "\t")
aligned <- read.table("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/familiarize/aligned_samples.txt", header = TRUE, stringsAsFactors = FALSE, sep = "\t")
pop_info <- pop_info[pop_info$Sample %in% aligned$sample,]
vcf_samples <- read.table("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/plink/plates_1_through_5_rm_no_TB.txt", header = FALSE, stringsAsFactors = FALSE, sep = "\t")
names(vcf_samples)[names(vcf_samples)=="V1"] <- "Sample"
vcf_sample_info <- inner_join(vcf_samples,pop_info, by = "Sample")

chr <- "1"
C <- as.matrix(read.table(paste("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/PCAngsd/chr",chr,"_ph_filtered_snps_minDP600_maxDP2000_maf0.05_minQ20_minMQ30_maxmiss0.5_outliers_rm.cov", sep = "")))
e <- eigen(C)
pca <-data.frame(Sample=vcf_sample_info$Sample, 
                 pop=vcf_sample_info$Population.Year,
                 plate=vcf_sample_info$Sequence.Plate,
                 plate_row=vcf_sample_info$plate_row,
                 location=vcf_sample_info$Location,
                 PC1=e$vectors[,1],PC2=e$vectors[,2],
                 PC3=e$vectors[,3],PC4=e$vectors[,4],
                 PC5=e$vectors[,5],PC6=e$vectors[,6],
                 PC7=e$vectors[,7],PC8=e$vectors[,8],
                 PC9=e$vectors[,9],PC10=e$vectors[,10],
                 stringsAsFactors=FALSE)

prop_explained <- c()
for (s in e$values[1:10]) {
  #print(s / sum(e$values))
  prop_explained <- c(prop_explained,round(((s / sum(e$values))*100),2))
}

pca <- pca %>% separate(pop, into = c("pop", "year"), sep = "(?<=[A-Za-z])(?=[0-9])")
pca1 <- pca

p1 <- ggplot(data = pca1)+
  geom_point(data = pca1, aes(x = PC1, y = PC2, fill = pop, color = pop, shape = year), size = 3)+
  scale_fill_manual(values=c(grb,org,red,yel,lir), guide=FALSE)+
  scale_color_manual(name = "population", values=c(grb,org,red,yel,lir))+
  scale_shape_manual(name = "year", values=c(21,22,23,24,25))+
  xlab(paste("PC 1: ", prop_explained[1],"%\n",sep = ""))+
  ylab(paste("PC 2: ", prop_explained[2],"%\n",sep = ""))+
  theme_bw()+
  geom_text(data=subset(pca1, pop == "PWS"),aes(x = PC1, y = PC2,label = location))+
  #geom_text(aes(x = PC1, y = PC2), label = pca1$location)+
  ggtitle(paste("plate ",plate_num_str,"\nrow ",plate_row_str,sep = ""))
  print(p1)

#for (plate_num in c(6:20))
##for (plate_num in c(6))
#{
#plate_num_str <- as.character(plate_num)
#  for (plate_row in c(1:8))
#  {
#  plate_row_str <- as.character(plate_row)
#  p1 <- ggplot(data = pca1)+
#    geom_point(data = pca1, aes(x = PC1, y = PC2, fill = pop, color = pop, shape = year), size = 3)+
#    scale_fill_manual(values=c(grb,org,red,yel,lir), guide=FALSE)+
#    scale_color_manual(name = "population", values=c(grb,org,red,yel,lir))+
#    scale_shape_manual(name = "year", values=c(21,22,23,24,25))+
#    xlab(paste("PC 1: ", prop_explained[1],"%\n",sep = ""))+
#    ylab(paste("PC 2: ", prop_explained[2],"%\n",sep = ""))+
#    theme_bw()+
#    geom_text(data=subset(pca1, plate_row == plate_row_str & plate == plate_num_str),aes(x = PC1, y = PC2,label = Sample))+
#    ggtitle(paste("plate ",plate_num_str,"\nrow ",plate_row_str,sep = ""))
#    print(p1)
#}
#}


```


***

### Double check patterns are consistent genome-wide with PLINK

```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.width= 10, fig.height= 7}
# PLINK pca
pop_info <- read.table("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/familiarize/EVOS_MasterSheet_JoeMcGirr_April2020_plate_rows.txt", header = TRUE, stringsAsFactors = FALSE, sep = "\t")
aligned <- read.table("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/familiarize/aligned_samples.txt", header = TRUE, stringsAsFactors = FALSE, sep = "\t")
pop_info <- pop_info[pop_info$Sample %in% aligned$sample,]

eigenvec <- read.table("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/plink/ph_filtered_snps_minDP600_maxDP2000_maf0.05_minQ20_minMQ30_maxmiss0.5_outliers_rm.eigenvec", header=F, stringsAsFactors = F)
eigenval <- read.table("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/plink/ph_filtered_snps_minDP600_maxDP2000_maf0.05_minQ20_minMQ30_maxmiss0.5_outliers_rm.eigenval", header=F, stringsAsFactors = F)
names(eigenvec)[names(eigenvec)=="V1"] <- "Sample"
vcf_sample_info <- inner_join(eigenvec,pop_info, by = "Sample")
 
 pca <-data.frame(sample=vcf_sample_info$Sample, 
                  pop=vcf_sample_info$Population.Year,
                  plate=vcf_sample_info$Sequence.Plate,
                  plate_row=vcf_sample_info$plate_row,
                  PC1=vcf_sample_info$V3,
                  PC2=vcf_sample_info$V4,
                  PC3=vcf_sample_info$V5,
                  PC4=vcf_sample_info$V6,
                  stringsAsFactors=FALSE)
 
 prop_explained <- c()
 for (s in eigenval$V1) {
   #print(s / sum(eigenval$V1))
   prop_explained <- c(prop_explained,round(((s / sum(eigenval$V1))*100),2))
 }
 
 
 pca <- pca %>% separate(pop, into = c("pop", "year"), sep = "(?<=[A-Za-z])(?=[0-9])")
 pca1 <- pca
 
 ggplot(data = pca1)+
   geom_point(data = pca1, aes(x = PC1, y = PC2, fill = pop, color = pop, shape = year), size = 3)+
   scale_fill_manual(values=c(grb,org,red,yel,blu,lir), guide=FALSE)+
   scale_color_manual(name = "population", values=c(grb,org,red,yel,blu,lir))+
   scale_shape_manual(name = "year", values=c(21,22,23,24,25))+
   xlab(paste("PC1: ", prop_explained[1],"%",sep = ""))+
   ylab(paste("PC2: ", prop_explained[2],"%",sep = ""))+
   theme_bw()+
   #xlim(-0.038,-0.004)+
   #geom_text(aes(x = PC1, y = PC2), label = pca1$sample)+
   ggtitle("PC 1 and 2 genome-wide\nplates 1-5 removed")
 
  ggplot(data = pca1)+
   geom_point(data = pca1, aes(x = PC3, y = PC4, fill = pop, color = pop, shape = year), size = 3)+
   scale_fill_manual(values=c(grb,org,red,yel,blu,lir), guide=FALSE)+
   scale_color_manual(name = "population", values=c(grb,org,red,yel,blu,lir))+
   scale_shape_manual(name = "year", values=c(21,22,23,24,25))+
   xlab(paste("PC3: ", prop_explained[3],"%",sep = ""))+
   ylab(paste("PC4: ", prop_explained[4],"%",sep = ""))+
   theme_bw()+
   #xlim(-0.038,-0.004)+
   #geom_text(aes(x = PC1, y = PC2), label = pca1$sample)+
   ggtitle("PC 3 and 4 genome-wide\nplates 1-5 removed")


#ggplot(data = pca1)+
#  geom_point(data = pca1, aes(x = PC1, y = PC2, fill = pop, color = pop, shape = year), size = 3)+
#  scale_fill_manual(values=c(grb,org,red,yel,blu,lir), guide=FALSE)+
#  scale_color_manual(name = "population", values=c(grb,org,red,yel,blu,lir))+
#  scale_shape_manual(name = "year", values=c(21,22,23,24,25))+
#  xlab(paste("PC1: ", prop_explained[1],"%",sep = ""))+
#  ylab(paste("PC2: ", prop_explained[2],"%",sep = ""))+
#  theme_bw()+
#  xlim(-0.038,-0.004)+
#  #geom_text(aes(x = PC1, y = PC2), label = pca1$sample)+
#  ggtitle("all populations and sampling years\ngenome-wide") 


```

### <span style="color:red">**Conclusion: Similar patterns genome-wide, with more separation of CA, and some strucutre within TB considering PC 4**</span>  

***


### plates 1-5 removed. 2017 samples. NGSAdmix

```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.width= 10, fig.height= 7}

pop_info <- read.table("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/familiarize/EVOS_MasterSheet_JoeMcGirr_April2020.txt", header = TRUE, stringsAsFactors = FALSE, sep = "\t")
aligned <- read.table("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/familiarize/aligned_samples.txt", header = TRUE, stringsAsFactors = FALSE, sep = "\t")
pop_info <- pop_info[pop_info$Sample %in% aligned$sample,]

vcf_samples <- read.table("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/plink/plates_1_through_5_rm.txt", header = FALSE, stringsAsFactors = FALSE, sep = "\t")
names(vcf_samples)[names(vcf_samples)=="V1"] <- "Sample"
vcf_sample_info <- inner_join(vcf_samples,pop_info, by = "Sample")

sfiles <- list.files(path="C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/NGSadmix/plates_1_thru_5_removed/", full.names=T)
slist <- readQ(files=sfiles,indlabfromfile = FALSE)

onelabset <-vcf_sample_info[,3,drop=FALSE] 
rownames(slist[[3]]) <- vcf_sample_info$Sample
#if(length(unique(sapply(slist,nrow)))==1) slist <- lapply(slist,"rownames<-",labs$ind)

# for (i in c(1:6)){
# p1 <- plotQ(slist[i],returnplot=T,exportplot=F,basesize=17,showindlab=F,useindlab=T,
#             sortind="all",grplab = onelabset,
#             grplabsize=6,linesize=2,pointsize=8, grplabangle=0, grplabjust=0.3, grplabheight=7,
#             divcol="white",divtype=2, divsize=1,clustercol=c(grb,org,red,yel,blu,lir,gry))
# plot(p1$plot[[1]])
# }

#attributes(slist)  
subsetpops <- c("BC17","CA17","PWS17","SS17","TB17","WA17")
#subsetpops <- c("TB91","TB96","TB06","TB17")
#subsetpops <- c("PWS91","PWS96","PWS07","PWS17")
#subsetpops <- c("SS96","SS06","SS17")
#subsetpops <- c("BC17","CA17","WA17")

for (i in c(1:5)){
p1 <- plotQ(slist[i],returnplot=T,exportplot=F,basesize=17,showindlab=F,useindlab=T,
            sortind="all",grplab = onelabset,subsetgrp=subsetpops,
            grplabsize=6,linesize=2,pointsize=8, grplabangle=0, grplabjust=0.3, grplabheight=7,
            divcol="white",divtype=2, divsize=1,clustercol=c(grb,org,red,yel,blu,lir,gry))
plot(p1$plot[[1]])
}

```

### <span style="color:red">**Conclusion: We might expect clearer grouping at k=6 based on geography, but we see a split within TB17 at k=5 **</span>  

***

### Run PCA on each metapopulation separately to check for obvious outliers. Individuals from  plates 1-5 removed

```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.width= 10, fig.height= 7}

pop_names <- c("PWS","TB","SS","BC17","WA17","CA17")
pop_name <- "PWS"
for (pop_name in pop_names){
pop_info <- read.table("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/familiarize/EVOS_MasterSheet_JoeMcGirr_April2020_plate_rows.txt", header = TRUE, stringsAsFactors = FALSE, sep = "\t")
aligned <- read.table("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/familiarize/aligned_samples.txt", header = TRUE, stringsAsFactors = FALSE, sep = "\t")
pop_info <- pop_info[pop_info$Sample %in% aligned$sample,]
vcf_samples <- read.table(paste("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/plink/",pop_name,"_outliers_rm.txt",sep = ""), header = FALSE, stringsAsFactors = FALSE, sep = "\t")
names(vcf_samples)[names(vcf_samples)=="V1"] <- "Sample"
vcf_sample_info <- inner_join(vcf_samples,pop_info, by = "Sample")

chr <- "1"
C <- as.matrix(read.table(paste("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/PCAngsd/chr",chr,"_",pop_name,"_ph_filtered_snps_minDP600_maxDP2000_maf0.05_minQ20_minMQ30_maxmiss0.5_outliers_rm.cov", sep = "")))
e <- eigen(C)
pca <-data.frame(Sample=vcf_sample_info$Sample,
                 Location=vcf_sample_info$Location,
                 pop=vcf_sample_info$Population.Year,
                 plate=vcf_sample_info$Sequence.Plate,
                 plate_row=vcf_sample_info$plate_row,
                 PC1=e$vectors[,1],PC2=e$vectors[,2],
                 PC3=e$vectors[,3],PC4=e$vectors[,4],
                 PC5=e$vectors[,5],PC6=e$vectors[,6],
                 PC7=e$vectors[,7],PC8=e$vectors[,8],
                 PC9=e$vectors[,9],PC10=e$vectors[,10],
                 stringsAsFactors=FALSE)

prop_explained <- c()
for (s in e$values[1:10]) {
  #print(s / sum(e$values))
  prop_explained <- c(prop_explained,round(((s / sum(e$values))*100),2))
}

pca <- pca %>% separate(pop, into = c("pop", "year"), sep = "(?<=[A-Za-z])(?=[0-9])")
pca1 <- pca

# p1 <- ggplot(data = pca1)+
#   geom_point(data = pca1, aes(x = PC1, y = PC2, fill = pop, color = pop, shape = year), size = 3)+
#   scale_fill_manual(values=c(bla,grb,org,red,yel,lir), guide=FALSE)+
#   scale_color_manual(name = "population", values=c(bla,grb,org,red,yel,lir))+
#   scale_shape_manual(name = "year", values=c(21,22,23,24,25))+
#   xlab(paste("PC 1: ", prop_explained[1],"%\n",sep = ""))+
#   ylab(paste("PC 2: ", prop_explained[2],"%\n",sep = ""))+
#   theme_bw()+
#   #xlim(0,0.17)+ylim(-0.05,0.1)+
#   #xlim(-0.05,-0.042)+ylim(-0.035,-0.025)+
#   #xlim(-0.055,-0.04)+ylim(-0.031,-0.025)+
#   #geom_text(aes(x = PC1, y = PC2), label = pca1$Sample)+
#   #geom_text(aes(x = PC1, y = PC2), label = pca1$plate_row)+
#   ggtitle(paste(pop_name," PC 1 and 2 chr 1\nplates 1-5 removed", sep = ""))
# print(p1)
p1 <- ggplot(data = pca1)+
  geom_point(data = pca1, aes(x = PC1, y = PC2, fill = pop, color = pop, shape = year), size = 3)+
  scale_fill_manual(values=c(bla,grb,org,red,yel,lir), guide=FALSE)+
  scale_color_manual(name = "population", values=c("white","white","white","white","white","white","white"))+
  scale_shape_manual(name = "year", values=c(1,1,1,1,1))+
  xlab(paste("PC 1: ", prop_explained[1],"%\n",sep = ""))+
  ylab(paste("PC 2: ", prop_explained[2],"%\n",sep = ""))+
  theme_bw()+
  #xlim(0,0.17)+ylim(-0.05,0.1)+
  #xlim(-0.05,-0.042)+ylim(-0.035,-0.025)+
  #xlim(-0.055,-0.04)+ylim(-0.031,-0.025)+
  #geom_text(aes(x = PC1, y = PC2), label = pca1$Sample)+
  geom_text(aes(x = PC1, y = PC2), label = pca1$plate)+
  ggtitle(paste(pop_name," PC 1 and 2 chr 1\nplates 1-5 removed", sep = ""))
print(p1)
p1 <- ggplot(data = pca1)+
  geom_point(data = pca1, aes(x = PC1, y = PC2, fill = pop, color = pop, shape = year), size = 3)+
  scale_fill_manual(values=c(bla,grb,org,red,yel,lir), guide=FALSE)+
  scale_color_manual(name = "population", values=c("white","white","white","white","white","white","white"))+
  scale_shape_manual(name = "year", values=c(1,1,1,1,1))+
  xlab(paste("PC 1: ", prop_explained[1],"%\n",sep = ""))+
  ylab(paste("PC 2: ", prop_explained[2],"%\n",sep = ""))+
  theme_bw()+
  #xlim(0,0.17)+ylim(-0.05,0.1)+
  #xlim(-0.05,-0.042)+ylim(-0.035,-0.025)+
  #xlim(-0.055,-0.04)+ylim(-0.031,-0.025)+
  #geom_text(aes(x = PC1, y = PC2), label = pca1$Sample)+
  geom_text(aes(x = PC1, y = PC2), label = pca1$plate)+
  ggtitle(paste(pop_name," PC 1 and 2 chr 1\nplates 1-5 removed", sep = ""))
print(p1)
p1 <- ggplot(data = pca1)+
  geom_point(data = pca1, aes(x = PC3, y = PC4, fill = pop, color = pop, shape = year), size = 3)+
  scale_fill_manual(values=c(bla,grb,org,red,yel,lir), guide=FALSE)+
  scale_color_manual(name = "population", values=c(bla,grb,org,red,yel,lir))+
  scale_shape_manual(name = "year", values=c(21,22,23,24,25))+
  xlab(paste("PC 3: ", prop_explained[3],"%\n",sep = ""))+
  ylab(paste("PC 4: ", prop_explained[4],"%\n",sep = ""))+
  theme_bw()+
  #xlim(0,0.17)+ylim(-0.05,0.1)+
  ggtitle(paste(pop_name," PC 3 and 4 chr 1\nplates 1-5 removed", sep = ""))
#print(p1)
p1 <- ggplot(data = pca1)+
  geom_point(data = pca1, aes(x = PC5, y = PC6, fill = pop, color = pop, shape = year), size = 3)+
  scale_fill_manual(values=c(bla,grb,org,red,yel,lir), guide=FALSE)+
  scale_color_manual(name = "population", values=c(bla,grb,org,red,yel,lir))+
  scale_shape_manual(name = "year", values=c(21,22,23,24,25))+
  xlab(paste("PC 5: ", prop_explained[5],"%\n",sep = ""))+
  ylab(paste("PC 6: ", prop_explained[6],"%\n",sep = ""))+
  theme_bw()+
  #xlim(0,0.17)+ylim(-0.05,0.1)+
  ggtitle(paste(pop_name," PC 5 and 6 chr 1\nplates 1-5 removed", sep = ""))
#print(p1)
}


```

