---
title: "Pacific Herring Taj D and Fst"
author: "Joe McGirr"
date: "7/27/2020"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(ggplot2)
library(tidyverse)
library(reticulate)
library(reshape2)
#BiocManager::install("visFuns.R")


# color-blind friendly 
# Wong, B. Points of view: Color blindness. Nat Methods (2011).
bla <- "#000000"
blu <- "#0072b2"
grb <- "#56b4e9"
lir <- "#cc79a7"
gre <- "#009e73"
red <- "#d55e00"
org <- "#e69f00"
yel <- "#f0e442"
gry<-  '#BBBBBB'

```

## Tajima's D

```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.width= 10, fig.height= 7}

# thetas <-read.delim('C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/thetas/population_BC17_ph_filtered_snps_minD# P600_maxDP2000_maf0.05_minQ20_minMQ30_maxmiss0.5_theta.thetas10kbWindow.gz.pestPG')
# #pi = tP/nsites
# #https://doi.org/10.1186/s12862-018-1209-y
# thetas$pi <- thetas$tP/thetas$nSites 
# chr1 <- thetas %>% filter(Chr == "chr1")
# range(thetas$Tajima)
# range(na.omit(thetas$pi))
# 
# 
# plot(chr1$WinCenter, chr1$Tajima)
# plot(chr1$WinCenter, chr1$pi)

pop_names = c("PWS91","PWS96","PWS07","PWS17","TB91","TB96","TB06","TB17","SS96","SS06","SS17","BC17","WA17","CA17")
pop_names = c("PWS96","PWS07","PWS17","TB91","TB96","TB06","TB17","SS96","SS06","SS17","BC17","WA17","CA17")

my_cols <- c(red,red,red,red,blu,blu,blu,blu,yel,yel,yel,grb,lir,org)
pop <- "PWS91"
thetas <-read.delim(paste('C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/thetas/50kb/population_',pop,'_ph_filtered_snps_minDP600_maxDP2000_maf0.05_minQ20_minMQ30_maxmiss0.5_theta.thetas50kbWindow.gz.pestPG',sep = ""))
thetas <- thetas[c("Tajima")]
names(thetas)[names(thetas)=="Tajima"] <- pop

for (pop in pop_names){

thetas1 <-read.delim(paste('C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/thetas/50kb/population_',pop,'_ph_filtered_snps_minDP600_maxDP2000_maf0.05_minQ20_minMQ30_maxmiss0.5_theta.thetas50kbWindow.gz.pestPG',sep = ""))
thetas1[c("Chr","WinCenter","Tajima")]
thetas$Tajima <- thetas1$Tajima
names(thetas)[names(thetas)=="Tajima"] <- pop 
  
}

my_cols <- c(grb,org,red,red,red,red,yel,yel,yel,blu,blu,blu,blu,lir)
theta_test <- thetas %>% 
      gather(key="MesureType", value="Val")
p1 <- thetas %>% 
      gather(key="MesureType", value="Val") %>%
      ggplot( aes(x=reorder(MesureType,Val, FUN = mean), y=Val, fill=MesureType)) +
      geom_violin() + xlab("\npopulation/year") + ylab("Tajima's D\n")+
      scale_fill_manual(name = "Population",values=my_cols,guide=FALSE)+
      #stat_summary(fun=mean, geom="line")+
      theme_minimal()
      #geom_jitter(height = 0, width = 0.1, alpha = 0.1)

print(p1)

# by chromosome

fsts <- list.files(path="C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/fst/sliding_window/", full.names=T)


for (pop in pop_names){  

thetas <-read.delim(paste('C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/thetas/50kb/population_',pop,'_ph_filtered_snps_minDP600_maxDP2000_maf0.05_minQ20_minMQ30_maxmiss0.5_theta.thetas50kbWindow.gz.pestPG',sep = ""))

thetas <- thetas %>% mutate(chr = as.numeric(gsub("chr", "", Chr)))
thetas$chr_num <- factor(thetas$chr, levels = c(1:26))

# thetas
p <- ggplot(thetas, aes(x = WinCenter, y = Tajima)) + 
     geom_point(size = 1, color = gry,alpha = 0.4, shape = 1)+
     theme_minimal()+
     theme(axis.text.x=element_blank())+
     ylab("Tajima's D\n")+ xlab("")+ 
     ggtitle(pop)+
     geom_smooth(method = "loess", se = FALSE, span = 1/10, color = gre)+
     facet_wrap(~chr_num, ncol = 9)
print(p)

}





# pi

#pop <- "PWS91"
#thetas <-read.delim(paste('C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/thetas/population_',pop,'_ph_filtered_snps_minDP600_maxDP2000_maf0.05_minQ20_minMQ30_maxmiss0.5_theta.thetas10kbWindow.gz.pestPG',sep = ""))
#thetas$pi<- (thetas$tP*thetas$nSites)/10000
#thetas <- thetas[c("pi")]
#names(thetas)[names(thetas)=="pi"] <- pop

#for (pop in pop_names){

#thetas1 <-read.delim(paste('C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/thetas/population_',pop,'_ph_filtered_snps_minDP600_maxDP2000_maf0.05_minQ20_minMQ30_maxmiss0.5_theta.thetas10kbWindow.gz.pestPG',sep = ""))
#thetas1$pi<- (thetas1$tP*thetas1$nSites)/10000
#thetas1[c("Chr","WinCenter","pi")]
#thetas$pi <- thetas1$pi
#names(thetas)[names(thetas)=="pi"] <- pop 
  
#}

#p1 <- thetas %>% 
#      gather(key="MesureType", value="Val") %>%
#      ggplot( aes(x=reorder(MesureType,Val, FUN = median), y=Val, fill=MesureType)) +
#      geom_violin() + xlab("\npopulation/year") + ylab("pi\n")+
#      scale_fill_manual(name = "Population",values=my_cols,guide=FALSE)+
#      theme_minimal()+ ylim(0,0.15)
#      #geom_jitter(height = 0, width = 0.1, alpha = 0.1)

#p1




```

## Fst matrix

```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.width= 10, fig.height= 7}

fsts <- list.files(path="C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/fst/sliding_window/", full.names=T)


# 2017 samples
fst <- read.delim("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/fst/sliding_window/fst_pbs_50kb_win_10kb_step_TB17_PWS17_SS17.txt")
tbxpw <- round(mean(fst$Fst01),2)
tbxss <- round(mean(fst$Fst02),2)
pwxss <- round(mean(fst$Fst12),2)
fst <- read.delim("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/fst/sliding_window/fst_pbs_50kb_win_10kb_step_TB17_PWS17_BC17.txt")
tbxbc <- round(mean(fst$Fst02),2)
pwxbc <- round(mean(fst$Fst12),2)
fst <- read.delim("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/fst/sliding_window/fst_pbs_50kb_win_10kb_step_TB17_PWS17_WA17.txt")
tbxwa <- round(mean(fst$Fst02),2)
pwxwa <- round(mean(fst$Fst12),2)
fst <- read.delim("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/fst/sliding_window/fst_pbs_50kb_win_10kb_step_TB17_PWS17_CA17.txt")
tbxca <- round(mean(fst$Fst02),2)
pwxca <- round(mean(fst$Fst12),2)

fst <- read.delim("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/fst/sliding_window/fst_pbs_50kb_win_10kb_step_PWS17_SS17_BC17.txt")
ssxbc <- round(mean(fst$Fst12),2)
fst <- read.delim("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/fst/sliding_window/fst_pbs_50kb_win_10kb_step_PWS17_SS17_WA17.txt")
ssxwa <- round(mean(fst$Fst12),2)
fst <- read.delim("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/fst/sliding_window/fst_pbs_50kb_win_10kb_step_PWS17_SS17_CA17.txt")
ssxca <- round(mean(fst$Fst12),2)
fst <- read.delim("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/fst/sliding_window/fst_pbs_50kb_win_10kb_step_BC17_WA17_CA17.txt")
bcxwa <- round(mean(fst$Fst01),2)
bcxca <- round(mean(fst$Fst02),2)
waxca <- round(mean(fst$Fst12),2)

fst_vec <- c(0,tbxpw,tbxss,tbxbc,tbxwa,tbxca,
             tbxpw,0,pwxss,pwxbc,pwxwa,pwxca,
             tbxss,pwxss,0,ssxbc,ssxwa,ssxca,
             tbxbc,pwxbc,ssxbc,0,bcxwa,bcxca,
             tbxwa,pwxwa,ssxwa,bcxwa,0,waxca,
             tbxca,pwxca,ssxca,bcxca,waxca,0)

fst_mat = matrix(fst_vec, nrow = 6, ncol = 6)
colnames(fst_mat) <- c("TB17","PWS17","SS17","BC17","WA17","CA17")
rownames(fst_mat) <- c("TB17","PWS17","SS17","BC17","WA17","CA17")
print(fst_mat)

  # Get upper triangle of the correlation matrix
  get_upper_tri <- function(cormat){
    cormat[lower.tri(cormat)]<- NA
    return(cormat)
  }

    get_lower_tri<-function(cormat){
    cormat[upper.tri(cormat)] <- NA
    return(cormat)
  }
upper_tri <- get_upper_tri(fst_mat)

# Melt the correlation matrix
melted_cormat <- melt(upper_tri, na.rm = TRUE)
# Heatmap
ggplot(data = melted_cormat, aes(Var2, Var1, fill = value))+
 geom_tile(color = "white")+
 scale_fill_gradient(low = blu, high = red, 
    limit = c(0,max(fst_mat)), space = "Lab", 
   name="Fst") +
  theme_minimal()+ xlab("")+ylab("")+
 theme(axis.text.x = element_text(angle = 0, vjust = 0, 
    size = 12, hjust = 0.5))+
   theme(axis.text.y = element_text(size = 12))+
 coord_fixed()+
 geom_text(aes(Var2, Var1, label = value), color = "black", size = 5)



# temporal samples PWS
fst <- read.delim("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/fst/sliding_window/fst_pbs_50kb_win_10kb_step_PWS91_PWS96_PWS07.txt")
p91xp96 <- round(mean(fst$Fst01),2)
p91xp07 <- round(mean(fst$Fst02),2)
p96xp07 <- round(mean(fst$Fst12),2)
fst <- read.delim("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/fst/sliding_window/fst_pbs_50kb_win_10kb_step_PWS91_PWS96_PWS17.txt")
p91xp17 <- round(mean(fst$Fst02),2)
p96xp17 <- round(mean(fst$Fst12),2)
fst <- read.delim("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/fst/sliding_window/fst_pbs_50kb_win_10kb_step_PWS91_PWS07_PWS17.txt")
p07xp17 <- round(mean(fst$Fst12),2)

fst_vec <- c(0,p91xp96,p91xp07,p91xp17,
             p91xp96,0,p96xp07,p96xp17,
             p91xp07,p96xp07,0,p07xp17,
             p91xp17,p96xp17,p07xp17,0)

fst_mat = matrix(fst_vec, nrow = 4, ncol = 4)
colnames(fst_mat) <- c("PWS91","PWS96","PWS07","PWS17")
rownames(fst_mat) <- c("PWS91","PWS96","PWS07","PWS17")
print(fst_mat)
upper_tri <- get_upper_tri(fst_mat)

# Melt the correlation matrix
melted_cormat <- melt(upper_tri, na.rm = TRUE)
# Heatmap
ggplot(data = melted_cormat, aes(Var2, Var1, fill = value))+
 geom_tile(color = "white")+
 scale_fill_gradient(low = blu, high = red, 
    limit = c(0,0.301), space = "Lab", 
   name="Fst") +
  theme_minimal()+ xlab("")+ylab("")+
 theme(axis.text.x = element_text(angle = 0, vjust = 0, 
    size = 12, hjust = 0.5))+
   theme(axis.text.y = element_text(size = 12))+
 coord_fixed()+
 geom_text(aes(Var2, Var1, label = value), color = "black", size = 5)


# temporal samples TB
fst <- read.delim("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/fst/sliding_window/fst_pbs_50kb_win_10kb_step_TB91_TB96_TB06.txt")
p91xp96 <- round(mean(fst$Fst01),2)
p91xp07 <- round(mean(fst$Fst02),2)
p96xp07 <- round(mean(fst$Fst12),2)
fst <- read.delim("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/fst/sliding_window/fst_pbs_50kb_win_10kb_step_TB91_TB96_TB17.txt")
p91xp17 <- round(mean(fst$Fst02),2)
p96xp17 <- round(mean(fst$Fst12),2)
fst <- read.delim("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/fst/sliding_window/fst_pbs_50kb_win_10kb_step_TB91_TB06_TB17.txt")
p07xp17 <- round(mean(fst$Fst12),2)

fst_vec <- c(0,p91xp96,p91xp07,p91xp17,
            p91xp96,0,p96xp07,p96xp17,
            p91xp07,p96xp07,0,p07xp17,
            p91xp17,p96xp17,p07xp17,0)

fst_mat = matrix(fst_vec, nrow = 4, ncol = 4)
colnames(fst_mat) <- c("TB91","TB96","TB06","TB17")
rownames(fst_mat) <- c("TB91","TB96","TB06","TB17")
print(fst_mat)
upper_tri <- get_upper_tri(fst_mat)

# Melt the correlation matrix
melted_cormat <- melt(upper_tri, na.rm = TRUE)
# Heatmap
ggplot(data = melted_cormat, aes(Var2, Var1, fill = value))+
geom_tile(color = "white")+
scale_fill_gradient(low = blu, high = red, 
   limit = c(0,0.301), space = "Lab", 
  name="Fst") +
 theme_minimal()+ xlab("")+ylab("")+
theme(axis.text.x = element_text(angle = 0, vjust = 0, 
   size = 12, hjust = 0.5))+
  theme(axis.text.y = element_text(size = 12))+
coord_fixed()+
geom_text(aes(Var2, Var1, label = value), color = "black", size = 5)


# temporal samples SS
fst <- read.delim("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/fst/sliding_window/fst_pbs_50kb_win_10kb_step_SS96_SS06_SS17.txt")
p96xp07 <- round(mean(fst$Fst01),2)
p96xp17 <- round(mean(fst$Fst02),2)
p07xp17 <- round(mean(fst$Fst12),2)


fst_vec <- c(0,p96xp07,p96xp17,
            p96xp07,0,p07xp17,
            p96xp17,p07xp17,0)

fst_mat = matrix(fst_vec, nrow = 3, ncol = 3)
colnames(fst_mat) <- c("SS96","SS06","SS17")
rownames(fst_mat) <- c("SS96","SS06","SS17")
print(fst_mat)
upper_tri <- get_upper_tri(fst_mat)

# Melt the correlation matrix
melted_cormat <- melt(upper_tri, na.rm = TRUE)
# Heatmap
ggplot(data = melted_cormat, aes(Var2, Var1, fill = value))+
geom_tile(color = "white")+
scale_fill_gradient(low = blu, high = red, 
   limit = c(0,0.301), space = "Lab", 
  name="Fst") +
 theme_minimal()+ xlab("")+ylab("")+
theme(axis.text.x = element_text(angle = 0, vjust = 0, 
   size = 12, hjust = 0.5))+
  theme(axis.text.y = element_text(size = 12))+
coord_fixed()+
geom_text(aes(Var2, Var1, label = value), color = "black", size = 5)


```


## Fst and PBS by chromosome

```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.width= 10, fig.height= 7}

# A population’s population branch statistic value at a given locus corresponds to the magnitude of allele frequency change relative to its divergence from the other two populations.

#  We implemented the population branch statistic (PBS) [27] to detect genomic regions that diverged rapidly on the Cape bee lineage compared to other African populations.

pop_names = c("PWS91","PWS96","PWS07","PWS17","TB91","TB96","TB06","TB17","SS96","SS06","SS17","BC17","WA17","CA17")

pop0 <- "TB17"
pop1 <- "PWS17"
pop2 <- "SS17"

fsts <- list.files(path="C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/fst/sliding_window/", full.names=T)
#fsts <- fsts[1]

for (f in fsts){  

pops <- strsplit(f, "_")
pops <- c(pops[[1]][9], pops[[1]][10],gsub("\\.txt","",pops[[1]][11]))

fst <-read.delim(paste("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/angsd/SFS/fst/sliding_window/fst_pbs_50kb_win_10kb_step_",pops[1],"_",pops[2],"_",pops[3],".txt", sep = ""))
#head(fst)
fst <- fst %>% mutate(chr = as.numeric(gsub("chr", "", chr)))
fst$chr_num <- factor(fst$chr, levels = c(1:26))

# Fst
p <- ggplot(fst, aes(x = midPos, y = Fst01)) + 
     geom_point(size = 1, color = gry,alpha = 0.4, shape = 1)+
     theme_minimal()+
     theme(axis.text.x=element_blank())+
     ylab("Fst\n")+ xlab("")+ 
     ggtitle(paste(pops[1],"vs. ", pops[2]))+
     geom_smooth(method = "loess", se = FALSE, span = 1/10, color = blu)+
     facet_wrap(~chr_num, ncol = 9)
print(p)

p <- ggplot(fst, aes(x = midPos, y = Fst02)) + 
     geom_point(size = 1, color = gry,alpha = 0.4, shape = 1)+
     theme_minimal()+
     theme(axis.text.x=element_blank())+
     ylab("Fst\n")+ xlab("")+ 
     ggtitle(paste(pops[1],"vs. ", pops[3]))+
     geom_smooth(method = "loess", se = FALSE, span = 1/10, color = blu)+
     facet_wrap(~chr_num, ncol = 9)
print(p)

p <- ggplot(fst, aes(x = midPos, y = Fst12)) + 
     geom_point(size = 1, color = gry,alpha = 0.4, shape = 1)+
     theme_minimal()+
     theme(axis.text.x=element_blank())+
     ylab("Fst\n")+ xlab("")+ 
     ggtitle(paste(pops[2],"vs. ", pops[3]))+
     geom_smooth(method = "loess", se = FALSE, span = 1/10, color = blu)+
     facet_wrap(~chr_num, ncol = 9)
print(p)

#PBS
p <- ggplot(fst, aes(x = midPos, y = PBS0)) + 
     geom_point(size = 1, color = gry,alpha = 0.4, shape = 1)+
     theme_minimal()+
     theme(axis.text.x=element_blank())+
     ylab("PBS\n")+ xlab("")+ 
     ggtitle(paste(pops[1]))+
     geom_smooth(method = "loess", se = FALSE, span = 1/10, color = red)+
     facet_wrap(~chr_num, ncol = 9)
print(p)

p <- ggplot(fst, aes(x = midPos, y = PBS1)) + 
     geom_point(size = 1, color = gry,alpha = 0.4, shape = 1)+
     theme_minimal()+
     theme(axis.text.x=element_blank())+
     ylab("PBS\n")+ xlab("")+ 
     ggtitle(paste(pops[2]))+
     geom_smooth(method = "loess", se = FALSE, span = 1/10, color = red)+
     facet_wrap(~chr_num, ncol = 9)
print(p)

p <- ggplot(fst, aes(x = midPos, y = PBS2)) + 
     geom_point(size = 1, color = gry,alpha = 0.4, shape = 1)+
     theme_minimal()+
     theme(axis.text.x=element_blank())+
     ylab("PBS\n")+ xlab("")+ 
     ggtitle(paste(pops[3]))+
     geom_smooth(method = "loess", se = FALSE, span = 1/10, color = red)+
     facet_wrap(~chr_num, ncol = 9)
print(p)

}

# chr1 <- fst %>% filter(chr == "chr1")
# 
# plot(chr1$midPos, chr1$Fst01)
# plot(chr1$midPos, chr1$Fst02)
# plot(chr1$midPos, chr1$Fst12)
# 
# plot(chr1$midPos, chr1$PBS0)
# plot(chr1$midPos, chr1$PBS1)
# plot(chr1$midPos, chr1$PBS2)
# plot(fst$midPos, fst$Fst02)


```
























