---
title: "Pacific Herring fastq to vcf"
author: "Joe McGirr"
date: '`r Sys.Date()`'
output:
  html_document:
    code_folding: show
    collapsed: no
    df_print: paged
    number_sections: yes
    theme: paper
    toc: yes
    toc_depth: 5
    toc_float: yes
  html_notebook:
    toc: yes
    toc_depth: 5
editor_options: 
  chunk_output_type: console
---

# Summary
This notebook results from an R markdown file that generates scripts to create a `.vcf` file for 1000+ Pacific herring samples (low coverage whole genomes). The markdown file can be found [here](https://github.com/joemcgirr/pac_herring/blob/master/Rmarkdown/fastq_to_vcf/Pacific_Herring_fastq_to_vcf.Rmd)

This notebook uses the R package reticulate with python to automate the tedious process of writing bash scripts for each job run on the computing cluster. These scripts were made to run on the Farm cluster which uses SLURM scheduling. Each bash script has a SLURM header.

Written by Joe McGirr, postdoc in Andrew Whitehead's lab.

Data info: `fastq`, `bam`, and `.vcf` files are currently located on the Farm sever at UC Davis (/home/eoziolor/phpopg/data/align/). Raw g.vcfs were generated from `.bam`s aligned to c. harengus fasta.


# Load R libraries and set color palette 
```{r setup,message=FALSE, warning=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(ggplot2)
library(tidyverse)
library(reticulate)
library(ggpubr)

# color-blind friendly 
# Wong, B. Points of view: Color blindness. Nat Methods (2011).
bla <- "#000000"
blu <- "#0072b2"
grb <- "#56b4e9"
lir <- "#cc79a7"
gre <- "#009e73"
red <- "#d55e00"
org <- "#e69f00"
yel <- "#f0e442"
gry<-  '#BBBBBB'

```

# Load python libraries
```{python, results='hide', eval = FALSE}
import io
import pandas as pd
import numpy as np

def sbatch_header(job,mem,tasks,hours):
    #sbatch submission script header
    script = 'script_' + job + '.sh'
    outfile = io.open(script,'w', newline='\n')    
    outfile.write('#!/bin/bash\n\n#SBATCH --job-name='+job+'\n')
    outfile.write('#SBATCH --mem='+mem+'G \n')
    outfile.write('#SBATCH --ntasks='+tasks+' \n')
    outfile.write('#SBATCH -e '+job+'_%A_%a.err \n')
    outfile.write('#SBATCH --time='+hours+':00:00  \n')
    outfile.write('#SBATCH --mail-user=jamcgirr@ucdavis.edu ##email you when job starts,ends,etc\n#SBATCH --mail-type=ALL\n')
    outfile.write('#SBATCH -p high \n\n')
    outfile.close()
    
def sbatch_header_loop(job,mem,tasks,hours,infile):
    #sbatch submission script header
    script = 'script_' + infile + job + '.sh'
    outfile = io.open(script,'w', newline='\n') 
    jobname= infile + job   
    outfile.write('#!/bin/bash\n\n#SBATCH --job-name='+jobname+'\n')
    outfile.write('#SBATCH --mem='+mem+'G \n')
    outfile.write('#SBATCH --ntasks='+tasks+' \n')
    outfile.write('#SBATCH -e '+jobname+'_%A_%a.err \n')
    outfile.write('#SBATCH --time='+hours+':00:00 \n')
    outfile.write('#SBATCH --mail-user=jamcgirr@ucdavis.edu ##email you when job starts,ends,etc\n#SBATCH --mail-type=ALL\n')
    outfile.write('#SBATCH -p high \n\n')
    outfile.close()

```
# Trim, align, index
Trim `.fastq`, align with bwa mem, and index `.bam` with samtools

example script
```{bash,eval=FALSE}

#!/bin/bash

#SBATCH --job-name=trim_align
#SBATCH --mem=16G 
#SBATCH --ntasks=4 
#SBATCH -e trim_align_%A_%a.err 
#SBATCH --time=24:00:00  
#SBATCH --mail-user=jamcgirr@ucdavis.edu ##email you when job starts,ends,etc
#SBATCH --mail-type=ALL
#SBATCH -p high 


gzip -d /home/jamcgirr/ph/data/hi_cov/PH-Sitka-93_S1_L008_R1_001.fastq.gz 
gzip -d /home/jamcgirr/ph/data/hi_cov/PH-Sitka-93_S1_L008_R2_001.fastq.gz 

module load trim_galore 
trim_galore -q 20 --paired --illumina /home/jamcgirr/ph/data/hi_cov/PH-Sitka-93_S1_L008_R1_001.fastq /home/jamcgirr/ph/data/hi_cov/PH-Sitka-93_S1_L008_R2_001.fastq 

module load bwa
bwa mem -aM -p -t 4 -R "@RG\tID:group1\tSM:PH-Sitka-93_S1_L008\tPL:illumina\tLB:lib1" /home/eoziolor/phpopg/data/genome_chr/c.harengus.fa /home/jamcgirr/ph/data/hi_cov/PH-Sitka-93_S1_L008_R1_001_val_1.fq /home/jamcgirr/ph/data/hi_cov/PH-Sitka-93_S1_L008_R2_001_val_2.fq > /home/jamcgirr/ph/data/hi_cov/PH_Sitka_93.sam 
module load samtools
samtools view -Shu /home/jamcgirr/ph/data/hi_cov/PH_Sitka_93.sam > /home/jamcgirr/ph/data/hi_cov/PH_Sitka_93.bam
samtools index /home/jamcgirr/ph/data/hi_cov/PH_Sitka_93.bam
rm /home/jamcgirr/ph/data/hi_cov/PH_Sitka_93.sam 
samtools sort /home/jamcgirr/ph/data/hi_cov/PH_Sitka_93.bam -o /home/jamcgirr/ph/data/hi_cov/PH_Sitka_93.sort.bam
samtools index /home/jamcgirr/ph/data/hi_cov/PH_Sitka_93.sort.bam


#run: sbatch script_trim_align.sh
```

## Generate trim and align scripts
```{python,results='hide', eval = FALSE}
job_name = 'trim_align_high_cov_ph'

sbatch_header(job_name,'16','4','24')
script = 'script_' + job_name + '.sh'
o = io.open(script,'a+', newline='\n')

o.write('# cp /home/eoziolor/phgenome/data/raw/PH-Sitka-93_S1_L008_R1_001.fastq.gz /home/jamcgirr/ph/data/hi_cov/ \n')
o.write('# cp /home/eoziolor/phgenome/data/raw/PH-Sitka-93_S1_L008_R2_001.fastq.gz /home/jamcgirr/ph/data/hi_cov/ \n')
o.write('#gzip -d /home/jamcgirr/ph/data/hi_cov/PH-Sitka-93_S1_L008_R1_001.fastq.gz \n')
o.write('#gzip -d /home/jamcgirr/ph/data/hi_cov/PH-Sitka-93_S1_L008_R2_001.fastq.gz \n\n')

o.write('# module load trim_galore \n')
o.write('# trim_galore -q 20 --paired --illumina /home/jamcgirr/ph/data/hi_cov/PH-Sitka-93_S1_L008_R1_001.fastq /home/jamcgirr/ph/data/hi_cov/PH-Sitka-93_S1_L008_R2_001.fastq \n\n') 
o.write('# mv PH-Sitka-93_S1_L008_R1_001_val_1.fq /home/jamcgirr/ph/data/hi_cov \n')
o.write('# mv PH-Sitka-93_S1_L008_R2_001_val_2.fq /home/jamcgirr/ph/data/hi_cov \n')

o.write('# module load bwa\n')
o.write('# bwa mem -aM -p -t 4 -R "@RG\\tID:group1\\tSM:PH-Sitka-93_S1_L008\\tPL:illumina\\tLB:lib1" /home/eoziolor/phpopg/data/genome_chr/c.harengus.fa /home/jamcgirr/ph/data/hi_cov/PH-Sitka-93_S1_L008_R1_001_val_1.fq /home/jamcgirr/ph/data/hi_cov/PH-Sitka-93_S1_L008_R2_001_val_2.fq > /home/jamcgirr/ph/data/hi_cov/PH_Sitka_93.sam \n')
o.write('module load samtools\n')
o.write('# samtools view -Shu /home/jamcgirr/ph/data/hi_cov/PH_Sitka_93.sam > /home/jamcgirr/ph/data/hi_cov/PH_Sitka_93.bam\n')
o.write('# samtools index /home/jamcgirr/ph/data/hi_cov/PH_Sitka_93.bam\n')
o.write('# rm /home/jamcgirr/ph/data/hi_cov/PH_Sitka_93.sam \n')
o.write('samtools sort /home/jamcgirr/ph/data/hi_cov/PH_Sitka_93.bam -o /home/jamcgirr/ph/data/hi_cov/PH_Sitka_93.sort.bam\n')
o.write('samtools index /home/jamcgirr/ph/data/hi_cov/PH_Sitka_93.sort.bam\n') 

o.write('\n\n#run: sbatch '+script)
o.close()

```



# Interrogate alignment files
use picard-tools to generate `.bam` metrics

example script
```{bash,eval=FALSE}
#!/bin/bash

#SBATCH --job-name=00_CollectWgsMetrics
#SBATCH --mem=8G 
#SBATCH --ntasks=1 
#SBATCH -e 00_CollectWgsMetrics_%A_%a.err 
#SBATCH --time=24:00:00 
#SBATCH --mail-user=jamcgirr@ucdavis.edu ##email you when job starts,ends,etc
#SBATCH --mail-type=ALL
#SBATCH -p high 

module load picardtools
for file in /home/eoziolor/phpopg/data/align/00*.bam
do
filename=$(basename $file .bam)
picard-tools CollectWgsMetrics I=$file R=/home/jamcgirr/ph/data/c_harengus/c.harengus.fa O=/home/jamcgirr/ph/familiarize/elias_qc_stats/wgsMetrics/$filename.collect_wgs_metrics.txt
done


#run: sbatch script_00_CollectWgsMetrics.sh

```

## Generate bam metric scripts

```{python,results='hide', eval = FALSE}

job_name = '_CollectWgsMetrics'

infiles = ["00","01","02","03","04","05","06","07","08","09","1"]

for infile in infiles:
    script = 'script_' + infile + job_name + '.sh'
    sbatch_header_loop(job_name,'8','4','48', infile)
    o = io.open(script,'a+', newline='\n')
    
    o.write('module load picardtools\n') 

    o.write('for file in /home/eoziolor/phpopg/data/align/'+infile+'*.bam\n')
    o.write('do\n')
    
    o.write('filename=$(basename $file .bam)\n')
    o.write('picard-tools CollectWgsMetrics I=$file R=/home/jamcgirr/ph/data/c_harengus/c.harengus.fa O=/home/jamcgirr/ph/familiarize/elias_qc_stats/wgsMetrics/$filename.collect_wgs_metrics.txt\n')
    
    o.write('done\n')

    
    o.write('\n\n#run: sbatch '+script)
    o.close()
```

## Plot Coverage
```{r, message=FALSE, warning=FALSE, fig.width= 10, fig.height= 7}

pop_info <- read.table("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/familiarize/EVOS_MasterSheet_JoeMcGirr_April2020.txt", header = TRUE, stringsAsFactors = FALSE, sep = "\t")

bam_met_files <- list.files("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/familiarize/multiqc_data/wgsMetrics/")
met_files_in_pop_info <- c()
for(met_file in bam_met_files){
  met_files_in_pop_info <- c(met_files_in_pop_info,strsplit(met_file,"\\.collect_wgs_metrics.txt")[[1]])
}

pop_info <- pop_info[pop_info$Sample %in% met_files_in_pop_info, ]
sample_name <- pop_info$Sample[1]
bam_met <- paste("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/familiarize/multiqc_data/wgsMetrics/",sample_name, ".collect_wgs_metrics.txt", sep = "")
bam_met <- read.table(bam_met, header = TRUE, stringsAsFactors = FALSE, nrow= 1)
bam_met$Sample <- sample_name
pop_info_bams <- merge(pop_info,bam_met, by = "Sample")
  
for (sample_name in pop_info$Sample[2:(length(pop_info$Sample))])
{
  
  bam_met <- paste("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/familiarize/multiqc_data/wgsMetrics/",sample_name, ".collect_wgs_metrics.txt", sep = "")
  bam_met <- read.table(bam_met, header = TRUE, stringsAsFactors = FALSE, nrow= 1)
  bam_met$Sample <- sample_name
  pop_info_bams1 <- merge(pop_info,bam_met, by = "Sample")
  pop_info_bams <- rbind(pop_info_bams,pop_info_bams1)
  
}
```
### Compare by year collected
```{r, message=FALSE, warning=FALSE, fig.width= 10, fig.height= 7}

p1 <- ggplot(pop_info_bams,aes(factor(Year.Collected), MEAN_COVERAGE))+
      geom_violin()+ xlab("year collected") + ylab("mean coverage")+
      #geom_boxplot(width=0.1)
      theme_minimal()+
      geom_jitter(height = 0, width = 0.1, alpha = 0.1)+
      stat_compare_means(method = "anova", label.y = max(pop_info_bams$MEAN_COVERAGE)+(max(pop_info_bams$MEAN_COVERAGE)*.05),label.x = 2.25)      # Add global p-value
p1
an <- aov(pop_info_bams$MEAN_COVERAGE~factor(pop_info_bams$Year.Collected))
#summary(an)
TukeyHSD(an)
p1 <- ggplot(pop_info_bams,aes(factor(Year.Collected), PCT_EXC_TOTAL))+
      geom_violin()+ xlab("\nyear collected") + ylab("% bases excluded due to all filters\n")+
      #geom_boxplot(width=0.1)
      theme_minimal()+
      geom_jitter(height = 0, width = 0.1, alpha = 0.1)+
      stat_compare_means(method = "anova", label.y = max(pop_info_bams$PCT_EXC_TOTAL)+(max(pop_info_bams$PCT_EXC_TOTAL)*.05),label.x = 2.25)      # Add global p-value
p1
an <- aov(pop_info_bams$PCT_EXC_TOTAL~factor(pop_info_bams$Year.Collected))
#summary(an)
TukeyHSD(an)
p1 <- ggplot(pop_info_bams,aes(factor(Year.Collected), PCT_1X))+
      geom_violin()+ xlab("\nyear collected") + ylab("% bases with >= 1X coverage post-filtering\n")+
      #geom_boxplot(width=0.1)
      theme_minimal()+
      geom_jitter(height = 0, width = 0.1, alpha = 0.1)+
      stat_compare_means(method = "anova", label.y = max(pop_info_bams$PCT_1X)+(max(pop_info_bams$PCT_1X)*.05),label.x = 2.25)      # Add global p-value
p1
an <- aov(pop_info_bams$PCT_1X~factor(pop_info_bams$Year.Collected))
#summary(an)
TukeyHSD(an)

```

### Compare by location
```{r, message=FALSE, warning=FALSE, fig.width= 10, fig.height= 7}

p1 <- ggplot(pop_info_bams,aes(factor(Location), MEAN_COVERAGE))+
      geom_violin()+ xlab("population location") + ylab("mean coverage")+
      #geom_boxplot(width=0.1)
      theme_minimal()+
      geom_jitter(height = 0, width = 0.1, alpha = 0.1)#+
      #stat_compare_means(method = "anova", label.y = max(pop_info_bams$MEAN_COVERAGE)+(max(pop_info_bams$MEAN_COVERAGE)*.05),label.x = 2.25)      # Add global p-value
p1
an <- aov(pop_info_bams$MEAN_COVERAGE~factor(pop_info_bams$Location))
#summary(an)
#TukeyHSD(an)
p1 <- ggplot(pop_info_bams,aes(factor(Location), PCT_EXC_TOTAL))+
      geom_violin()+ xlab("\npopulation location") + ylab("% bases excluded due to all filters\n")+
      #geom_boxplot(width=0.1)
      theme_minimal()+
      geom_jitter(height = 0, width = 0.1, alpha = 0.1)#+
      #stat_compare_means(method = "anova", label.y = max(pop_info_bams$PCT_EXC_TOTAL)+(max(pop_info_bams$PCT_EXC_TOTAL)*.05),label.x = 2.25)      # Add global p-value
p1
an <- aov(pop_info_bams$PCT_EXC_TOTAL~factor(pop_info_bams$Location))
#summary(an)
#TukeyHSD(an)
p1 <- ggplot(pop_info_bams,aes(factor(Location), PCT_1X))+
      geom_violin()+ xlab("\npopulation location") + ylab("% bases with >= 1X coverage post-filtering\n")+
      #geom_boxplot(width=0.1)
      theme_minimal()+
      geom_jitter(height = 0, width = 0.1, alpha = 0.1)#+
      #stat_compare_means(method = "anova", label.y = max(pop_info_bams$PCT_1X)+(max(pop_info_bams$PCT_1X)*.05),label.x = 2.25)      # Add global p-value
p1
an <- aov(pop_info_bams$PCT_1X~factor(pop_info_bams$Location))
#summary(an)
#TukeyHSD(an)


```

# Combine `g.vcfs` to create SNP vcf

## Make a database for each of the 26 chromosomes
use gatk GenomicsDBImport

example script
```{bash,eval=FALSE}
#!/bin/bash

#SBATCH --job-name=chr2_make_db
#SBATCH --mem=16G 
#SBATCH --ntasks=4 
#SBATCH -e chr2_make_db_%A_%a.err 
#SBATCH --time=72:00:00 
#SBATCH --mail-user=jamcgirr@ucdavis.edu ##email you when job starts,ends,etc
#SBATCH --mail-type=ALL
#SBATCH -p high 

# This must be run in directory containing final DB (gendb://) (/home/jamcgirr/ph/data/combine_gvcfs)

scratch_tmp=/scratch/jamcgirr/combine_gvcfs_chr2_tmp 
scratch_cp_to_home=/scratch/jamcgirr/combine_gvcfs_chr2 
home_cp=/home/jamcgirr/ph/data/combine_gvcfs/chr2 

module load R 
module load maven 
module load java 
module load GATK/4.1.4.1 

mkdir -vp $scratch_tmp 

gatk --java-options "-Xmx12g -Xms8g" GenomicsDBImport --genomicsdb-workspace-path $scratch_cp_to_home --batch-size 100 -L chr2_interval.list --sample-name-map /home/jamcgirr/ph/scripts/fastq_to_vcf/combine_gvcfs/ph.gvcfs_map --tmp-dir=$scratch_tmp --reader-threads 4 

cp -r $scratch_cp_to_home/* $home_cp 
##rm SCRATCH ONLY!! 
rm -r $scratch_tmp 
rm -r $scratch_cp_to_home 

# run next chr
sbatch script_chr2_1_genotypegvcf.sh 
sbatch script_chr2_2_genotypegvcf.sh 


#run: sbatch script_chr2_make_db.sh
```
### Generate database scripts
```{python,results='hide', eval = FALSE}
job_name = '_make_db'

infiles =["chr1","chr2","chr3","chr4","chr5","chr6","chr7","chr8","chr9","chr10","chr11","chr12","chr13","chr14","chr15","chr16","chr17","chr18","chr19","chr20","chr21","chr22","chr23","chr24","chr25","chr26"]
#infiles =["chr2","chr3","chr4","chr5","chr6","chr7","chr8","chr9","chr10","chr11","chr12","chr13","chr14","chr15","chr16","chr17","chr18","chr19","chr20","chr21","chr22","chr23","chr24","chr25","chr26"]

for infile in infiles:
    script = 'script_' + infile + job_name + '.sh'
    sbatch_header_loop(job_name,'16','4','72', infile)
    o = io.open(script,'a+', newline='\n')
    
    o.write('# This must be run in directory containing final DB (gendb://) (/home/jamcgirr/ph/data/combine_gvcfs)\n\n')
    
    o.write('scratch_tmp=/scratch/jamcgirr/combine_gvcfs_'+infile+'_tmp \n')
    o.write('scratch_cp_to_home=/scratch/jamcgirr/combine_gvcfs_'+infile+' \n')
    o.write('home_cp=/home/jamcgirr/ph/data/combine_gvcfs/'+infile+' \n\n')
    
    o.write('module load R \n')
    o.write('module load maven \n')
    o.write('module load java \n')
    o.write('module load GATK/4.1.4.1 \n\n')

    o.write('mkdir -vp $scratch_tmp \n\n')

    o.write('gatk --java-options "-Xmx12g -Xms8g" GenomicsDBImport --genomicsdb-workspace-path $scratch_cp_to_home --batch-size 100 -L '+infile+'_interval.list --sample-name-map /home/jamcgirr/ph/scripts/fastq_to_vcf/combine_gvcfs/ph.gvcfs_map --tmp-dir=$scratch_tmp --reader-threads 4 \n\n')
    
    o.write('cp -r $scratch_cp_to_home/* $home_cp \n')

    o.write('##rm SCRATCH ONLY!! \n')
    o.write('rm -r $scratch_tmp \n')
    o.write('rm -r $scratch_cp_to_home \n\n')

    o.write('sbatch script_'+infile+'_1_genotypegvcf.sh \n')
    o.write('sbatch script_'+infile+'_2_genotypegvcf.sh \n')

    
    o.write('\n\n#run: sbatch '+script)
    o.close()
```

## Create `.vcf` for each chromosome split into two intervals 
use gatk GenotypeGVCFs

example script
```{bash,eval=FALSE}
#!/bin/bash

#SBATCH --job-name=chr1_1_genotypegvcf
#SBATCH --mem=16G 
#SBATCH --ntasks=1 
#SBATCH -e chr1_1_genotypegvcf_%A_%a.err 
#SBATCH --time=06-00:00 
#SBATCH --mail-user=jamcgirr@ucdavis.edu ##email you when job starts,ends,etc
#SBATCH --mail-type=ALL
#SBATCH -p high 

# This must be run in directory containing final DB (gendb://) (/home/jamcgirr/ph/data/combine_gvcfs)

module load R 
module load maven 
module load java 
module load GATK/4.1.4.1 

gatk GenotypeGVCFs -R /home/jamcgirr/ph/data/c_harengus/c.harengus.fa -V gendb://chr1 -L chr1:1-16542129 -O raw_variants_chr1_1.vcf 


#run: sbatch script_chr1_1_genotypegvcf.sh
```
### Generate database scripts

```{python,results='hide', eval = FALSE}

job_name = '_genotypegvcf'
infiles = ["chr1_1","chr2_1","chr3_1","chr4_1","chr5_1","chr6_1","chr7_1","chr8_1","chr9_1","chr10_1","chr11_1","chr12_1","chr13_1","chr14_1","chr15_1","chr16_1","chr17_1","chr18_1","chr19_1","chr20_1","chr21_1","chr22_1","chr23_1","chr24_1","chr25_1","chr26_1","chr1_2","chr2_2","chr3_2","chr4_2","chr5_2","chr6_2","chr7_2","chr8_2","chr9_2","chr10_2","chr11_2","chr12_2","chr13_2","chr14_2","chr15_2","chr16_2","chr17_2","chr18_2","chr19_2","chr20_2","chr21_2","chr22_2","chr23_2","chr24_2","chr25_2","chr26_2"]
chr_dict = {"chr1_1":"1-16542129","chr2_1":"1-16505160","chr3_1":"1-16263781","chr4_1":"1-16133824","chr5_1":"1-15793431","chr6_1":"1-15730777","chr7_1":"1-15495311","chr8_1":"1-15364778","chr9_1":"1-15238691","chr10_1":"1-15113866","chr11_1":"1-15048164","chr12_1":"1-15011240","chr13_1":"1-14922870","chr14_1":"1-14666386","chr15_1":"1-14356761","chr16_1":"1-13886911","chr17_1":"1-13784255","chr18_1":"1-13623647","chr19_1":"1-13565322","chr20_1":"1-13347081","chr21_1":"1-13232991","chr22_1":"1-12832026","chr23_1":"1-12646449","chr24_1":"1-10045549","chr25_1":"1-7462096","chr26_1":"1-6221605","chr1_2":"16542129-33084258","chr2_2":"16505160-33010319","chr3_2":"16263781-32527562","chr4_2":"16133824-32267647","chr5_2":"15793431-31586861","chr6_2":"15730777-31461554","chr7_2":"15495311-30990621","chr8_2":"15364778-30729556","chr9_2":"15238691-30477381","chr10_2":"15113866-30227731","chr11_2":"15048164-30096327","chr12_2":"15011240-30022480","chr13_2":"14922870-29845739","chr14_2":"14666386-29332771","chr15_2":"14356761-28713521","chr16_2":"13886911-27773822","chr17_2":"13784255-27568510","chr18_2":"13623647-27247294","chr19_2":"13565322-27130643","chr20_2":"13347081-26694162","chr21_2":"13232991-26465981","chr22_2":"12832026-25664052","chr23_2":"12646449-25292897","chr24_2":"10045549-20091098","chr25_2":"7462096-14924191","chr26_2":"6221605-12443209"}

for infile in infiles:
    chr_name = infile.split("_")[0]
    script = 'script_' + infile + job_name + '.sh'
    sbatch_header_loop(job_name,'16','4','144', infile)
    o = io.open(script,'a+', newline='\n')
    
    
    o.write('# This must be run in directory containing final DB (gendb://) (/home/jamcgirr/ph/data/combine_gvcfs)\n\n')
    
    o.write('module load R \n')
    o.write('module load maven \n')
    o.write('module load java \n')
    o.write('module load GATK/4.1.4.1 \n\n')

    o.write('gatk GenotypeGVCFs -R /home/jamcgirr/ph/data/c_harengus/c.harengus.fa -V gendb://'+chr_name+' -L '+chr_name+':'+chr_dict[infile]+' -O raw_variants_'+infile+'.vcf \n')

    
    o.write('\n\n#run: sbatch '+script)
    o.close()

job_name = '_genotypegvcf_allsites'
infiles = ["chr1_1","chr2_1","chr3_1","chr4_1","chr5_1","chr6_1","chr7_1","chr8_1","chr9_1","chr10_1","chr11_1","chr12_1","chr13_1","chr14_1","chr15_1","chr16_1","chr17_1","chr18_1","chr19_1","chr20_1","chr21_1","chr22_1","chr23_1","chr24_1","chr25_1","chr26_1","chr1_2","chr2_2","chr3_2","chr4_2","chr5_2","chr6_2","chr7_2","chr8_2","chr9_2","chr10_2","chr11_2","chr12_2","chr13_2","chr14_2","chr15_2","chr16_2","chr17_2","chr18_2","chr19_2","chr20_2","chr21_2","chr22_2","chr23_2","chr24_2","chr25_2","chr26_2"]
chr_dict = {"chr1_1":"1-16542129","chr2_1":"1-16505160","chr3_1":"1-16263781","chr4_1":"1-16133824","chr5_1":"1-15793431","chr6_1":"1-15730777","chr7_1":"1-15495311","chr8_1":"1-15364778","chr9_1":"1-15238691","chr10_1":"1-15113866","chr11_1":"1-15048164","chr12_1":"1-15011240","chr13_1":"1-14922870","chr14_1":"1-14666386","chr15_1":"1-14356761","chr16_1":"1-13886911","chr17_1":"1-13784255","chr18_1":"1-13623647","chr19_1":"1-13565322","chr20_1":"1-13347081","chr21_1":"1-13232991","chr22_1":"1-12832026","chr23_1":"1-12646449","chr24_1":"1-10045549","chr25_1":"1-7462096","chr26_1":"1-6221605","chr1_2":"16542129-33084258","chr2_2":"16505160-33010319","chr3_2":"16263781-32527562","chr4_2":"16133824-32267647","chr5_2":"15793431-31586861","chr6_2":"15730777-31461554","chr7_2":"15495311-30990621","chr8_2":"15364778-30729556","chr9_2":"15238691-30477381","chr10_2":"15113866-30227731","chr11_2":"15048164-30096327","chr12_2":"15011240-30022480","chr13_2":"14922870-29845739","chr14_2":"14666386-29332771","chr15_2":"14356761-28713521","chr16_2":"13886911-27773822","chr17_2":"13784255-27568510","chr18_2":"13623647-27247294","chr19_2":"13565322-27130643","chr20_2":"13347081-26694162","chr21_2":"13232991-26465981","chr22_2":"12832026-25664052","chr23_2":"12646449-25292897","chr24_2":"10045549-20091098","chr25_2":"7462096-14924191","chr26_2":"6221605-12443209"}

for infile in infiles:
    chr_name = infile.split("_")[0]
    script = 'script_' + infile + job_name + '.sh'
    sbatch_header_loop(job_name,'60','4','144', infile)
    o = io.open(script,'a+', newline='\n')
    
    
    o.write('# This must be run in directory containing final DB (gendb://) (/home/jamcgirr/ph/data/combine_gvcfs)\n\n')
    
    o.write('module load R \n')
    o.write('module load maven \n')
    o.write('module load java \n')
    o.write('module load GATK/4.1.4.1 \n\n')

    o.write('gatk GenotypeGVCFs -R /home/jamcgirr/ph/data/c_harengus/c.harengus.fa -V gendb://'+chr_name+' -L '+chr_name+':'+chr_dict[infile]+' -all-sites -O raw_variants_allsites_'+infile+'.vcf \n')

    
    o.write('\n\n#run: sbatch '+script)
    o.close()
```


## Create `.vcf` with SNPs only and filter SNPs
use gatk SelectVariants and bcftools filter

minimum DP = 600
maximum DP = 2000
maf = 5%
minimum mapping quality = 30 

example script
```{bash,eval=FALSE}
#!/bin/bash

#SBATCH --job-name=chr1_filter_gvcf
#SBATCH --mem=8G 
#SBATCH --ntasks=1 
#SBATCH -e chr1_filter_gvcf_%A_%a.err 
#SBATCH --time=24:00:00 
#SBATCH --mail-user=jamcgirr@ucdavis.edu ##email you when job starts,ends,etc
#SBATCH --mail-type=ALL
#SBATCH -p high 

module load R 
module load maven 
module load java 
module load GATK/4.1.4.1
module load vcftools 
 
gatk SelectVariants -R /home/jamcgirr/ph/data/c_harengus/c.harengus.fa -V /home/jamcgirr/ph/data/combine_gvcfs/raw_variants_chr1_1.vcf --select-type-to-include SNP -O /home/jamcgirr/ph/data/combine_gvcfs/raw_snps_chr1_1.vcf 
vcftools --vcf /home/jamcgirr/ph/data/combine_gvcfs/raw_snps_chr1_1.vcf --remove-indels --maf 0.05 --min-alleles 2 --max-alleles 2 --minQ 30 --minDP 600 --maxDP 2000 --recode --recode-INFO-all --out /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr1_1 

gatk SelectVariants -R /home/jamcgirr/ph/data/c_harengus/c.harengus.fa -V /home/jamcgirr/ph/data/combine_gvcfs/raw_variants_chr1_2.vcf --select-type-to-include SNP -O /home/jamcgirr/ph/data/combine_gvcfs/raw_snps_chr1_2.vcf 
vcftools --vcf /home/jamcgirr/ph/data/combine_gvcfs/raw_snps_chr1_2.vcf --remove-indels --maf 0.05 --min-alleles 2 --max-alleles 2 --minQ 30 --minDP 600 --maxDP 2000 --recode --recode-INFO-all --out /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr1_2 

module load samtools 
module load bcftools 

bgzip -c /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr1_1.recode.vcf > /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr1_1.recode.vcf.gz 
bgzip -c /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr1_2.recode.vcf > /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr1_2.recode.vcf.gz 
bcftools index /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr1_1.recode.vcf.gz 
bcftools index /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr1_2.recode.vcf.gz 


#run: sbatch script_chr1_filter_gvcf.sh
```
### Generate SelectVariants scripts
```{python,results='hide', eval = FALSE}
job_name = '_filter_gvcf'
infiles = ["chr1","chr2","chr3","chr4","chr5","chr6","chr7","chr8","chr9","chr10","chr11","chr12","chr13","chr14","chr15","chr16","chr17","chr18","chr19","chr20","chr21","chr22","chr23","chr24","chr25","chr26"]

for infile in infiles:
    script = 'script_' + infile + job_name + '.sh'
    sbatch_header_loop(job_name,'8','4','24', infile)
    o = io.open(script,'a+', newline='\n')
        
        
    o.write('module load R \n')
    o.write('module load maven \n')
    o.write('module load java \n')
    o.write('module load GATK/4.1.4.1 \n\n')

    o.write('#gatk SelectVariants -R /home/jamcgirr/ph/data/c_harengus/c.harengus.fa -V /home/jamcgirr/ph/data/combine_gvcfs/raw_variants_'+infile+'_1.vcf --select-type-to-include SNP -O /home/jamcgirr/ph/data/combine_gvcfs/raw_snps_'+infile+'_1.vcf \n\n')
    
    o.write('module load vcftools \n')
    o.write('vcftools --vcf /home/jamcgirr/ph/data/combine_gvcfs/raw_snps_'+infile+'_1.vcf --remove-indels --maf 0.05 --min-alleles 2 --max-alleles 2 --minQ 20 --minDP 600 --maxDP 2000 --recode --recode-INFO-all --out /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_'+infile+'_1_delete \n\n')
    
    o.write('#gatk SelectVariants -R /home/jamcgirr/ph/data/c_harengus/c.harengus.fa -V /home/jamcgirr/ph/data/combine_gvcfs/raw_variants_'+infile+'_2.vcf --select-type-to-include SNP -O /home/jamcgirr/ph/data/combine_gvcfs/raw_snps_'+infile+'_2.vcf \n\n')
    
    o.write('vcftools --vcf /home/jamcgirr/ph/data/combine_gvcfs/raw_snps_'+infile+'_2.vcf --remove-indels --maf 0.05 --min-alleles 2 --max-alleles 2 --minQ 20 --minDP 600 --maxDP 2000 --recode --recode-INFO-all --out /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_'+infile+'_2_delete \n\n')
     
    o.write('module load samtools \n')
    o.write('module load bcftools \n')
    
    o.write('bcftools filter -Oz -i \'MQ>30\' /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_'+infile+'_1_delete.recode.vcf -o /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_'+infile+'_1.recode.vcf.gz \n')
    o.write('bcftools filter -Oz -i \'MQ>30\' /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_'+infile+'_2_delete.recode.vcf -o /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_'+infile+'_2.recode.vcf.gz \n')

    #o.write('bgzip -c /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_'+infile+'_1_delete.recode.vcf > /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_'+infile+'_1_delete.recode.vcf.gz \n')
    #o.write('bgzip -c /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_'+infile+'_2_delete.recode.vcf > /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_'+infile+'_2_delete.recode.vcf.gz \n')
    
    o.write('bcftools index /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_'+infile+'_1.recode.vcf.gz \n')
    o.write('bcftools index /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_'+infile+'_2.recode.vcf.gz \n\n')
   
    o.write('rm /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_'+infile+'_1_delete.recode.vcf \n')
    o.write('rm /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_'+infile+'_2_delete.recode.vcf \n')

    
    o.write('\n\n#run: sbatch '+script)
    o.close()

    #EST time ~ 4 hr
```

## Plot SNP stats for first half of Chr 1
1,073,808 SNPs across 16,542,120 bp

```{r, message=FALSE, warning=FALSE, fig.width= 10, fig.height= 7}
snps <- read.table("C:/Users/jmcgirr/Documents/GitHub/pac_herring/slurm_scripts/fastq_to_vcf/combine_gvcfs/chr1_1_out.INFO", header = TRUE, stringsAsFactors = FALSE, sep = "\t")

# GATK Hard filters

#  qd_x <-  "Variant Confidence/Quality by Depth"
#  qd <- 2
#  sor_x <-  "Symmetric Odds Ratio of 2x2 contingency table to detect strand bias"
#  sor <- 3
#  fs_x <- "Phred-scaled p-value using Fisher's exact test to detect strand bias"
#  fs <- 60
#  mq_x <-  "RMS Mapping Quality"
#  mq <- 40
#  mqrs_x <-  "Z-score From Wilcoxon rank sum test of Alt vs. Ref read mapping qualities"
#  mqrs <- -12.5
#  rprs_x <-  "Z-score from Wilcoxon rank sum test of Alt vs. Ref read position bias"
#  rprs <- -8

#filtered <- snps %>% filter(QD < qd) %>%
#                     filter(SOR > sor) %>%
#                     filter(FS > fs) %>%
#                     filter(MQ < qd) 
#
#nrow(snps)
#nrow(filtered)
#head(snps)

#  {hist(as.numeric(snps$QD), xlab = qd_x, main = paste("< ",qd, sep = ""))
#  abline(v = qd, col = red, lty = 2)}
#  
#  {hist(as.numeric(snps$SOR), xlab = sor_x, main = paste("> ",sor, sep = ""))
#  abline(v = sor, col = red, lty = 2)}
#  
#  {hist(as.numeric(snps$FS), xlab = fs_x, main = paste("> ",fs, sep = ""))
#  abline(v = fs, col = red, lty = 2)}
#  
#  {hist(as.numeric(snps$MQ), xlab = mq_x, main = paste("< ",mq, sep = ""))
#  abline(v = mq, col = red, lty = 2)}
#  
#  
#  # filter high coverage regions
#  # doi.org/10.1093/bioinformatics/btu356
#  # max depth between between d+3sqrt(d) and d+4sqrt(d) where d is mean read depth
#  #"These false positives are mostly caused by CNVs 
#  # or paralogous sequences not present in the reference genome"
#  
  {hist(as.numeric(snps$DP), xlab = "Depth",breaks = 1000,xlim = c(0,20000), main = paste("600 > Depth ","< ",round((mean(snps$DP)+(4*sqrt(mean(snps$DP)))),0), sep = ""))
  abline(v = 600, col = red, lty = 2)
  abline(v = (mean(snps$DP)+(4*sqrt(mean(snps$DP)))), col = red, lty = 2)}
  filter_DP <- snps %>% filter(DP < (mean(snps$DP)+(4*sqrt(mean(snps$DP))))) %>% filter(DP >600)
  # % filtered
  #((nrow(snps) - nrow(filter_DP))/ nrow(snps))*100

  
# define populations for SNP filtering
# put these txt files in directory for step 4

pop_info <- read.table("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/familiarize/EVOS_MasterSheet_JoeMcGirr_April2020.txt", header = TRUE, stringsAsFactors = FALSE, sep = "\t")
aligned <- read.table("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/familiarize/aligned_samples.txt", header = TRUE, stringsAsFactors = FALSE, sep = "\t")
pop_info <- pop_info[pop_info$Sample %in% aligned$sample,]

for (pop in unique(pop_info$Population.Year)){
pop1 <- pop_info %>% filter(Population.Year == pop) 
#(unique(pop1$Location))
#print(unique(pop1$Year.Collected))
#print(nrow(pop1))
outfile_name <- paste("population_",pop1$Population.Year[1],".txt", sep = "")
#write.table(pop1$Sample, outfile_name, col.names = FALSE, row.names = FALSE, quote = FALSE)
}


```


# Concatenate chromosomes into one master `.vcf`
use bcftools concat

example script
```{bash,eval=FALSE}
#!/bin/bash

#SBATCH --job-name=merge_snp_vcfs
#SBATCH --mem=16G 
#SBATCH --ntasks=4 
#SBATCH -e merge_snp_vcfs_%A_%a.err 
#SBATCH --time=06-00:00  
#SBATCH --mail-user=jamcgirr@ucdavis.edu ##email you when job starts,ends,etc
#SBATCH --mail-type=ALL
#SBATCH -p high 

module load samtools 
module load bcftools 
bcftools concat -o /home/jamcgirr/ph/data/combine_gvcfs/merged_filtered_snps.bcf -O b --threads 4 /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr1_1.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr1_2.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr2_1.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr2_2.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr3_1.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr3_2.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr4_1.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr4_2.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr5_1.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr5_2.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr6_1.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr6_2.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr7_1.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr7_2.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr8_1.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr8_2.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr9_1.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr9_2.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr10_1.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr10_2.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr11_1.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr11_2.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr12_1.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr12_2.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr13_1.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr13_2.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr14_1.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr14_2.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr15_1.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr15_2.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr16_1.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr16_2.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr17_1.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr17_2.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr18_1.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr18_2.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr19_1.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr19_2.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr20_1.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr20_2.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr21_1.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr21_2.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr22_1.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr22_2.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr23_1.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr23_2.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr24_1.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr24_2.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr25_1.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr25_2.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr26_1.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr26_2.recode.vcf.gz 


#run: sbatch script_merge_snp_vcfs.sh
```
### Generate merge script
```{python,results='hide', eval = FALSE}
job = 'merge_snp_vcfs'

#sbatch submission script header
script = 'script_' + job + '.sh'
outfile = io.open(script,'w', newline='\n')    
outfile.write('#!/bin/bash\n\n#SBATCH --job-name='+job+'\n')
outfile.write('#SBATCH --mem=16G \n')
outfile.write('#SBATCH --ntasks=4 \n')
outfile.write('#SBATCH -e '+job+'_%A_%a.err \n')
outfile.write('#SBATCH --time=06-00:00  \n')
outfile.write('#SBATCH --mail-user=jamcgirr@ucdavis.edu ##email you when job starts,ends,etc\n#SBATCH --mail-type=ALL\n')
outfile.write('#SBATCH -p high \n\n')


outfile.write('module load samtools \n')
outfile.write('module load bcftools \n')
outfile.write('bcftools concat -o /home/jamcgirr/ph/data/combine_gvcfs/merged_filtered_snps.bcf -O b --threads 4 /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr1_1.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr1_2.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr2_1.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr2_2.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr3_1.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr3_2.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr4_1.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr4_2.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr5_1.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr5_2.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr6_1.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr6_2.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr7_1.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr7_2.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr8_1.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr8_2.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr9_1.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr9_2.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr10_1.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr10_2.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr11_1.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr11_2.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr12_1.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr12_2.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr13_1.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr13_2.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr14_1.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr14_2.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr15_1.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr15_2.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr16_1.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr16_2.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr17_1.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr17_2.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr18_1.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr18_2.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr19_1.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr19_2.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr20_1.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr20_2.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr21_1.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr21_2.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr22_1.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr22_2.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr23_1.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr23_2.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr24_1.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr24_2.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr25_1.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr25_2.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr26_1.recode.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/filtered_snps_chr26_2.recode.vcf.gz \n') 
              
#run sbatch submission 
outfile.write('\n\n#run: sbatch '+script)
outfile.close()

```

# Subset .`vcf` into 14 populations (grouped by unique(location:year)) and filter for 50% genotyping rate
use vcftools and bcftools

example script
```{bash,eval=FALSE}
#!/bin/bash

#SBATCH --job-name=BC17_subset_snps_by_pop
#SBATCH --mem=8G 
#SBATCH --ntasks=1 
#SBATCH -e BC17_subset_snps_by_pop_%A_%a.err 
#SBATCH --time=24:00:00 
#SBATCH --mail-user=jamcgirr@ucdavis.edu ##email you when job starts,ends,etc
#SBATCH --mail-type=ALL
#SBATCH -p high 

module load samtools 
module load bcftools 
bcftools view -S population_BC17.txt -O z /home/jamcgirr/ph/data/combine_gvcfs/merged_filtered_snps.bcf > /home/jamcgirr/ph/data/combine_gvcfs/population_BC17_snps.vcf.gz 
module load vcftools 
vcftools --gzvcf /home/jamcgirr/ph/data/combine_gvcfs/population_BC17_snps.vcf.gz --max-missing 0.5 --recode-INFO-all --recode --stdout | bgzip -c > /home/jamcgirr/ph/data/combine_gvcfs/population_BC17_filtered_snps.vcf.gz 
bcftools index /home/jamcgirr/ph/data/combine_gvcfs/population_BC17_filtered_snps.vcf.gz 



#run: sbatch script_BC17_subset_snps_by_pop.sh

```
### Generate subset/filter scripts
```{python,results='hide', eval = FALSE}

job_name = '_subset_snps_by_pop'

infiles = ["BC17","CA17","PWS07","PWS17","PWS91","PWS96","SS06","SS17","SS96","TB06","TB17","TB91","TB96","WA17"]

for infile in infiles:
    script = 'script_' + infile + job_name + '.sh'
    sbatch_header_loop(job_name,'8','4','24', infile)
    o = io.open(script,'a+', newline='\n')
        
        
    o.write('module load samtools \n')
    o.write('module load bcftools \n')

    o.write('bcftools view -S population_'+infile+'.txt -O z /home/jamcgirr/ph/data/combine_gvcfs/merged_filtered_snps.bcf > /home/jamcgirr/ph/data/combine_gvcfs/population_'+infile+'_snps.vcf.gz \n')
   
    o.write('module load vcftools \n')
    #o.write('vcftools --gzvcf /home/jamcgirr/ph/data/combine_gvcfs/population_'+infile+'_snps.vcf.gz --minDP 25 --max-missing 0.5 --recode-INFO-all --recode --stdout | bgzip -c > /home/jamcgirr/ph/data/combine_gvcfs/population_'+infile+'_filtered_snps.vcf.gz \n')
    # min DP of 25 across 14 population:year groups = min DP of 350 
    # minDP 350, maxDP 2000, maf 5%, minQ30 across all smaples and 50% genotyping rate across each population:year group)

    o.write('vcftools --gzvcf /home/jamcgirr/ph/data/combine_gvcfs/population_'+infile+'_snps.vcf.gz --max-missing 0.8 --recode-INFO-all --recode --stdout | bgzip -c > /home/jamcgirr/ph/data/combine_gvcfs/population_'+infile+'_filtered_snps.vcf.gz \n')
    # minDP 600, maxDP 2000, maf 5%, minQ20 across all smaples and 80% genotyping rate across each population:year group)

    o.write('bcftools index /home/jamcgirr/ph/data/combine_gvcfs/population_'+infile+'_filtered_snps.vcf.gz \n\n')


    o.write('\n\n#run: sbatch '+script)
    o.close()

# EST time ~2hr
```

# Merge final filtered vcfs into master vcf  
ph_filtered_snps_minDP600_maxDP2000_maf0.05_minQ20_maxmiss0.5.vcf  
output plink and beagle format

example script
```{bash,eval=FALSE}
#!/bin/bash

#SBATCH --job-name=merge_filtered_pops
#SBATCH --mem=16G 
#SBATCH --ntasks=4 
#SBATCH -e merge_filtered_pops_%A_%a.err 
#SBATCH --time=06-00:00  
#SBATCH --mail-user=jamcgirr@ucdavis.edu ##email you when job starts,ends,etc
#SBATCH --mail-type=ALL
#SBATCH -p high 

module load samtools 
module load bcftools 
bcftools merge -O v --threads 4 -o /home/jamcgirr/ph/data/vcfs/ph_filtered_snps_minDP600_maxDP2000_maf0.05_minQ30_maxmiss0.5.vcf /home/jamcgirr/ph/data/combine_gvcfs/population_BC17_filtered_snps.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/population_CA17_filtered_snps.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/population_PWS07_filtered_snps.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/population_PWS17_filtered_snps.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/population_PWS91_filtered_snps.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/population_PWS96_filtered_snps.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/population_SS06_filtered_snps.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/population_SS17_filtered_snps.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/population_SS96_filtered_snps.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/population_TB06_filtered_snps.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/population_TB17_filtered_snps.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/population_TB91_filtered_snps.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/population_TB96_filtered_snps.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/population_WA17_filtered_snps.vcf.gz 

module load vcftools 
vcftools --vcf /home/jamcgirr/ph/data/vcfs/ph_filtered_snps_minDP600_maxDP2000_maf0.05_minQ30_maxmiss0.5.vcf --BEAGLE-PL --chr chr1 --out /home/jamcgirr/ph/data/vcfs/chr1_ph_filtered_snps_minDP600_maxDP2000_maf0.05_minQ30_maxmiss0.5 
#vcftools --vcf /home/jamcgirr/ph/data/vcfs/ph_filtered_snps_minDP600_maxDP2000_maf0.05_minQ30_maxmiss0.5.vcf --BEAGLE-PL --chr chr1,chr2,chr3,chr4,chr5,chr6,chr7,chr8,chr9,chr10,chr11,chr12,chr13,chr14,chr15,chr16,chr17,chr18,chr19,chr20,chr21,chr22,chr23,chr24,chr25,chr26  --out /home/jamcgirr/ph/data/vcfs/ph_filtered_snps_minDP600_maxDP2000_maf0.05_minQ30_maxmiss0.5 
vcftools --vcf /home/jamcgirr/ph/data/vcfs/ph_filtered_snps_minDP600_maxDP2000_maf0.05_minQ30_maxmiss0.5.vcf --plink --out /home/jamcgirr/ph/data/vcfs/ph_filtered_snps_minDP600_maxDP2000_maf0.05_minQ30_maxmiss0.5 


#run: sbatch script_merge_filtered_pops.sh

```
### Generate merge scripts
```{python,results='hide', eval = FALSE}
job_name = 'merge_filtered_pops'

vcf_name = 'ph_filtered_snps_minDP600_maxDP2000_maf0.05_minQ20_minMQ30_maxmiss0.8' 


sbatch_header(job_name,'16','8','144')
script = 'script_' + job_name + '.sh'
o = io.open(script,'a+', newline='\n')

o.write('module load samtools \n')
o.write('module load bcftools \n')
o.write('bcftools merge -O v --threads 8 -o /home/jamcgirr/ph/data/vcfs/'+vcf_name+'.vcf /home/jamcgirr/ph/data/combine_gvcfs/population_BC17_filtered_snps.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/population_CA17_filtered_snps.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/population_PWS07_filtered_snps.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/population_PWS17_filtered_snps.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/population_PWS91_filtered_snps.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/population_PWS96_filtered_snps.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/population_SS06_filtered_snps.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/population_SS17_filtered_snps.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/population_SS96_filtered_snps.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/population_TB06_filtered_snps.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/population_TB17_filtered_snps.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/population_TB91_filtered_snps.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/population_TB96_filtered_snps.vcf.gz /home/jamcgirr/ph/data/combine_gvcfs/population_WA17_filtered_snps.vcf.gz \n\n')
o.write('wc -l /home/jamcgirr/ph/data/vcfs/'+vcf_name+'.vcf > lines_'+vcf_name+'.txt \n')
o.write('module load vcftools \n')
o.write('vcftools --vcf /home/jamcgirr/ph/data/vcfs/'+vcf_name+'.vcf --BEAGLE-PL --chr chr1 --out /home/jamcgirr/ph/data/vcfs/chr1_'+vcf_name+' \n')
o.write('#vcftools --vcf /home/jamcgirr/ph/data/vcfs/'+vcf_name+'.vcf --BEAGLE-PL --chr chr1,chr2,chr3,chr4,chr5,chr6,chr7,chr8,chr9,chr10,chr11,chr12,chr13,chr14,chr15,chr16,chr17,chr18,chr19,chr20,chr21,chr22,chr23,chr24,chr25,chr26  --out /home/jamcgirr/ph/data/vcfs/'+vcf_name+' \n')
o.write('vcftools --vcf /home/jamcgirr/ph/data/vcfs/'+vcf_name+'.vcf --plink --out /home/jamcgirr/ph/data/vcfs/'+vcf_name+' \n')
o.write('module load plink \n')
o.write('plink --file /home/jamcgirr/ph/data/vcfs/'+vcf_name+' --pca --out /home/jamcgirr/ph/data/plink/'+vcf_name+'\n') 


o.write('\n\n#run: sbatch '+script)
o.close()
```






