---
title: "Plot_VCF"
author: "Joe McGirr"
date: '`r Sys.Date()`'
output:
  html_document:
    code_folding: hide
    collapsed: no
    df_print: paged
    number_sections: no
    theme: paper
    toc: yes
    toc_depth: 5
    toc_float: yes
  html_notebook:
    toc: yes
    toc_depth: 5
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(ggplot2)
library(tidyverse)
library(vcfR)
#BiocManager::install("ape")
#remotes::install_github("JimWhiting91/genotype_plot")
library(GenotypePlot)

# color-blind friendly 
# Wong, B. Points of view: Color blindness. Nat Methods (2011).
bla <- "#000000"
blu <- "#0072b2"
grb <- "#56b4e9"
lir <- "#cc79a7"
gre <- "#009e73"
red <- "#d55e00"
org <- "#e69f00"
yel <- "#f0e442"
gry<-  '#BBBBBB'

```

```{css, echo=FALSE}
pre, code {white-space:pre !important; overflow-x:auto}
```

# Summary

This notebook uses the [GenotypePlot package](https://github.com/JimWhiting91/genotype_plot) to visualize variation on chromosomes.

# Subset main `.vcf` to include chromosome of interest

example script
```{bash,eval=FALSE,class.source = 'fold-show'}

#!/bin/bash

#SBATCH --job-name=subset_chr7
#SBATCH --mem=16G 
#SBATCH --ntasks=8 
#SBATCH -e subset_chr7_%A_%a.err 
#SBATCH --time=24:00:00 
#SBATCH --mail-user=jamcgirr@ucdavis.edu ##email you when job starts,ends,etc
#SBATCH --mail-type=ALL
#SBATCH -p high 

module load bcftools
bcftools view /home/jamcgirr/ph/data/vcfs/ph_filtered_snps_minDP600_maxDP2000_minQ20_minMQ30_NS0.5_maf0.05.vcf.gz -r chr7 -Oz --threads 8 > ph_filtered_snps_minDP600_maxDP2000_minQ20_minMQ30_NS0.5_maf0.05_chr7.vcf.gz


#command to run: sbatch script_subset_chr7.sh

```

# Visualize variation on chromosome 

```{r, message=FALSE, warning=FALSE, fig.width= 10, fig.height= 7}

chr <- "chr12"
my_vcf <- read.vcfR(paste("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/vcfs/chrs/ph_filtered_snps_minDP600_maxDP2000_minQ20_minMQ30_NS0.5_maf0.05_",chr,".vcf.gz", sep = ""))

meta <- read.csv('C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/vcfs/sample_metadata.txt', comment.char = '#', sep = "\t")
my_popmap <- meta[c("Sample","Population_Year")]

# plot variation for individuals grouped by population
new_plot <- genotype_plot(vcf_object  =  my_vcf,
                          popmap = my_popmap,                              
                          cluster        = FALSE,                           
                          snp_label_size = 5000000,                          
                          colour_scheme=c(yel,org,red))

#pdf(paste("C:/Users/jmcgirr/Documents/GitHub/pac_herring/Rmarkdown/inversions/",chr,"_snps.pdf",sep = ""),width=10,height=8)

combine_genotype_plot(new_plot)

#dev.off()


# plot allele frequencies for populations
new_plot <- genotype_plot(vcf_object  =  my_vcf,   
                          popmap = my_popmap,                           
                          snp_label_size = 5000000,                     
                          colour_scheme=c(yel,org,red),
                          plot_allele_frequency=TRUE,                    
                          invariant_filter = TRUE)                      

#pdf(paste("C:/Users/jmcgirr/Documents/GitHub/pac_herring/Rmarkdown/inversions/",chr,"_AF.pdf",sep = ""),width=10,height=8)

combine_genotype_plot(new_plot)

#dev.off()



```

# Visualize allele frequency on chromosome 

```{r, message=FALSE, warning=FALSE, fig.width= 10, fig.height= 7}

chr <- "chr12"
my_vcf <- read.vcfR(paste("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/vcfs/chrs/ph_filtered_snps_minDP600_maxDP2000_minQ20_minMQ30_NS0.5_maf0.05_",chr,".vcf.gz", sep = ""))

meta <- read.csv('C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/vcfs/sample_metadata.txt', comment.char = '#', sep = "\t")
my_popmap <- meta[c("Sample","Population_Year")]

#my_popmap <- head(my_popmap,100)
pops <- c("PWS17","CA17")
my_popmap <-  my_popmap[my_popmap$Population_Year %in% pops,]
my_popmap <-  sample_n(my_popmap[my_popmap$Population_Year %in% pops,],50)


# plot subset of chromosome
new_plot <- genotype_plot(vcf_object  =  my_vcf,   
                          #chr    = 12,                                 
                          #start  = 11700000,                           
                          #end    = 11800000,                           
                          popmap = my_popmap,                           
                          #cluster        = TRUE,                        
                          snp_label_size = 5000000,                     
                          colour_scheme=c(yel,org,red),
                          plot_allele_frequency=TRUE,                    
                          invariant_filter = TRUE)                      

#dendro_with_tips <- new_plot$dendrogram +
#                    geom_text(aes(x=1:length(new_plot$dendro_labels),y=-2.5,label=new_plot$dendro_labels))
#dendro_with_tips

#pdf("C:/Users/jmcgirr/Documents/GitHub/pac_herring/Rmarkdown/inversions/test_plot.pdf",width=10,height=8)

combine_genotype_plot(new_plot)

#dev.off()

```

