---
title: "Plot_VCF"
author: "Joe McGirr"
date: '`r Sys.Date()`'
output:
  html_document:
    code_folding: hide
    collapsed: no
    df_print: paged
    number_sections: no
    theme: paper
    toc: yes
    toc_depth: 5
    toc_float: yes
  html_notebook:
    toc: yes
    toc_depth: 5
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(ggplot2)
library(tidyverse)
library(vcfR)
#BiocManager::install("ape")
#remotes::install_github("JimWhiting91/genotype_plot")
library(GenotypePlot)

# color-blind friendly 
# Wong, B. Points of view: Color blindness. Nat Methods (2011).
bla <- "#000000"
blu <- "#0072b2"
grb <- "#56b4e9"
lir <- "#cc79a7"
gre <- "#009e73"
red <- "#d55e00"
org <- "#e69f00"
yel <- "#f0e442"
gry<-  '#BBBBBB'

```

# Visualize variation on chromosomes


```{r, message=FALSE, warning=FALSE, fig.width= 10, fig.height= 7}

my_vcf <- read.vcfR("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/vcfs/chrs/ph_filtered_snps_minDP600_maxDP2000_minQ20_minMQ30_NS0.5_maf0.05_chr7.vcf.gz")

meta <- read.csv('C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/vcfs/sample_metadata.txt', comment.char = '#', sep = "\t")
my_popmap <- meta[c("Sample","Population_Year")]

#my_popmap <- head(my_popmap,100)
pops <- c("PWS17","CA17")
my_popmap <-  my_popmap[my_popmap$Population_Year %in% pops,]
my_popmap <-  sample_n(my_popmap[my_popmap$Population_Year %in% pops,],50)


# plot entire chromosome
# new_plot <- genotype_plot(vcf_object  =  my_vcf,
#                           popmap = my_popmap,                              
#                           cluster        = FALSE,                           
#                           snp_label_size = 5000000,                          
#                           colour_scheme=c("#d4b9da","#e7298a","#980043"))
# 

# plot subset of chromosome
new_plot <- genotype_plot(vcf_object  =  my_vcf,   
                          #chr    = 12,                                      # chr or scaffold ID
                          #start  = 11700000,                                # start of region
                          #end    = 11800000,                                # end = end of region
                          popmap = my_popmap,                              # population membership
                          cluster        = TRUE,                           # whether to organise haplotypes by PCA clustering
                          snp_label_size = 5000000,                           # breaks for position labels, eg. plot a position every 100,000 bp
                          colour_scheme=c(yel,org,red),   # character vector of colour values
                          invariant_filter = TRUE)                          # Filter any invariant sites before plotting

dendro_with_tips <- new_plot$dendrogram +
                    geom_text(aes(x=1:length(new_plot$dendro_labels),
                    y=-2.5,
                    label=new_plot$dendro_labels))
dendro_with_tips

pdf("C:/Users/jmcgirr/Documents/Whitehead_Lab/ph/vcfs/chrs/test_plot.pdf",width=10,height=8)

combine_genotype_plot(new_plot)

dev.off()

```


